---
title: "NTP72 - 16S - taxa compostiion - chicken draft "
author: "Florentin Constancias"
date: "`r format(Sys.time(), '%B %d, %Y')`"
output: 
  html_document: 
    toc: yes
    keep_md: yes
---

```{r setup, include=FALSE}
rm(list = ls())
gc()
options(java.parameters = "-Xmx80000m")
```

```{r setup2, include=FALSE}

##knitr::opts_chunk$set(cache=TRUE)
##knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(dev = "ragg_png") # https://ragg.r-lib.org/index.html
##knitr::opts_chunk$set(fig.width = 6) #https://r4ds.had.co.nz/graphics-for-communication.html#figure-sizing
##knitr::opts_chunk$set(fig.asp = 0.618)
##knitr::opts_chunk$set(fig.show = "hold")
##knitr::opts_chunk$set(fig.show = "70%")
##knitr::opts_chunk$set(fig.align = "center")
```

```{r packages, results=FALSE, warning=FALSE}
#install.packages("tidyverse")
require(tidyverse); packageVersion("tidyverse")
require(phyloseq); packageVersion("phyloseq")
```

```{r sourcing, warning=FALSE, results=FALSE , message= FALSE}
# source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_taxa_tests.R") 
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_alpha.R") 
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_beta.R") 
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_heatmap.R")
# source("https://raw.githubusercontent.com/fconstancias/metabaRpipe-source/master/Rscripts/functions.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_normalisation.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_varia.R")


'%!in%' <- function(x,y)!('%in%'(x,y))
```


```{r}
out_pptx = "~/Desktop/16S-chick.pptx"
out_xlsx = "~/Desktop/16S-chick.xlsx"
```

```{r}
load(url("https://github.com/fconstancias/NRP72-FBT/blob/master/Figures/Rastetics.Rdata?raw=true"))
```

## load data:

```{r}
"data/processed/16S/16S_working_phyloseq.RDS" %>% 
  here::here() %>% 
  readRDS()  %>% 
  subset_samples(Enrichment == "NotEnriched") %>%
  subset_samples(Experiment %in% c("Continuous", "Cecum")) %>%
  subset_samples(Day_of_Treatment > -4 & Day_of_Treatment <= 30 | Reactor == "DONOR") %>% 
  subset_samples(Model == "Chicken") %>% 
  subset_samples(Reactor != "CR" & Model2 == "Chicken1" | Model2 == "Chicken2" & Reactor %in% c("TR2", "TR3", "TR4", "TR5", "TR6", "TR7", "CR", "DONOR")) %>% 
  subset_samples(Reactor != "IR") -> ps_filtered
```

```{r}
ps_filtered %>% 
  phyloseq_check_lib_size(data_color = "Reactor", data_facet = "Reactor", first_n = 50, nreads_display = 1000) -> out

out$df %>% 
  # rownames_to_column('sample_id') %>% 
  # filter(Treatment == "DONOR")  %>% 
  select(SampleID, Day_of_Treatment,Treatment, Reactor, Model2, Model, Period, LibrarySize) 
```

## rarefraction

```{r}
min_sample_size = 3738

ps_filtered %>%
  rarefy_even_depth(sample.size = min_sample_size, rngseed = 123)  -> ps_rare

ps_filtered %>%
  prune_samples(sample_sums(.)>= min_sample_size, .) -> ps_fil
```

```{r}
ps_rare %>%
  subset_samples(! Reactor %in% c("DONOR")) %>% 
  subset_samples(Model2 == "Chicken2") %>% 
  filter_taxa(function(x) sum(x > 0) > 0, TRUE) -> chick2 # %>% 
# subset_samples(Reactor %!in%c("IR")) %>% 
# subset_samples() -> before_treat

# ps_rare %>%
#   subset_samples(! Reactor %in% c("DONOR", "CR")) %>% 
#   subset_samples(Model2 = "Chicken1") %>% 
#   filter_taxa(function(x) sum(x > 0) > 0, TRUE) -> chick1 # %>% 


# mat.train.svd.corr <- svd(mat.train.clr.corr)


ps_rare %>%
  subset_samples(! Reactor %in% c("DONOR", "CR")) %>% 
  subset_samples(Model2 %in% c("Chicken1","Chicken2")) %>% 
  filter_taxa(function(x) sum(x > 0) > 0, TRUE) -> chicks # %>% 
```

# linear models: CLR taxa - ASV:


```{r}
# require(microViz)
# 
# # The code for `taxatree_models` is quite similar to tax_model. 
# # However, you might need to run `tax_prepend_ranks` to ensure that each taxon at each rank is always unique. 
# lms <- chick2 %>% 
#   phyloseq_validate() %>%
#   tax_fix() %>%
#   tax_prepend_ranks() %>%
#   tax_filter(min_prevalence = 0.25, undetected = 0, use_counts = TRUE) %>% 
#   tax_transform("clr", rank = "Strain", keep_counts = TRUE) %>% 
#   # tax_transform("compositional", rank = "unique", keep_counts = TRUE) %>% 
#   # tax_transform(trans = "log2", chain = TRUE, zero_replace = "halfmin") %>% 
#   taxatree_models(
#     type = lm, 
#     ranks = "Strain",
#     variables = c('Day_of_Treatment')
#   )
# 
# lms
```

```{r}
# lms %>% 
#   taxatree_models2stats(.) %>% 
#   taxatree_stats_p_adjust(method = "BH", grouping = "rank") %>% 
#   .$taxatree_stats %>% 
#   arrange(p.adj.BH.rank) %>% 
#   select(taxon,p.adj.BH.rank, everything())
```


```{r}
# require(microViz)
# 
# # The code for `taxatree_models` is quite similar to tax_model. 
# # However, you might need to run `tax_prepend_ranks` to ensure that each taxon at each rank is always unique. 
# lms <- chick2 %>% 
#   phyloseq_validate() %>%
#   tax_fix() %>%
#   tax_prepend_ranks() %>%
#   tax_filter(min_prevalence = 0.25, undetected = 0, use_counts = TRUE) %>% 
#   # tax_transform("clr", rank = "Strain", keep_counts = TRUE) %>% 
#   # tax_transform("compositional", rank = "unique", keep_counts = TRUE) %>% 
#   # tax_transform(trans = "log2", chain = TRUE, zero_replace = "halfmin") %>% 
#   taxatree_models(
#     type = lm, 
#     ranks = NULL,
#     variables = c('Day_of_Treatment')
#   )
# 
# lms
```

```{r}
# lms %>% 
#   taxatree_models2stats(.) %>% 
#   taxatree_stats_p_adjust(method = "BH", grouping = "rank") %>% 
#   .$taxatree_stats %>% 
#   arrange(p.adj.BH.rank) %>% 
#   select(taxon,p.adj.BH.rank, everything()) -> tmp
# 
# tmp
```

```{r}
# tmp %>% 
#   filter(p.adj.BH.rank <= 0.05) %>% 
#   filter(rank == "Genus") %>% 
#   mutate(taxon =  stringr::str_remove(taxon, "[PFGO]: ")) %>% 
#   pull(taxon) %>% 
#   as.character() -> genus
```

```{r}
# chick2 %>% 
#   speedyseq::tax_glom("Genus") %>% 
#   microbiome::transform(., "compositional") %>% 
#   subset_taxa(Genus %in% genus) %>% 
#   microbiomeutilities::plot_area(., xvar="Day_of_Treatment", 
#                                  level = "Genus",
#                                  facet.by = "Reactor_Treatment_Dose",
#                                  abund.thres = 0.1,
#                                  prev.thres=0.1,
#                                  # fill.colors=RColorBrewer::brewer.pal(12,"Paired"),
#                                  ncol=3,
#                                  nrow=4) -> p
# 
# p
```
Per reactor:

```{r}
function_lm_per_reactor <- function(chick2, var = "Reactor_Treatment_Dose", group = "TR6_VAN90", transform = "log", rank_glom = "Genus", ranks = "Genus",  min_prev = 0.25, adj_pval_cut = 0.05, padj_group = "rank", sub_test = TRUE, time_split = 4,  use_cluster = FALSE, max_clust = 9)
{
  #######------------------
  
  require(microViz); require(tidyverse)
  
  #######------------------
  
  # The code for `taxatree_models` is quite similar to tax_model. 
  # However, you might need to run `tax_prepend_ranks` to ensure that each taxon at each rank is always unique. 
  chick2 %>% 
    prune_samples(get_variable(chick2, var) %in% group,
                  .)  %>% 
    filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>% 
    # speedyseq::tax_glom(taxrank = rank_glom) %>%
    physeq_glom_rename(taxrank = rank_glom, rename_ASV = rank_glom) %>% 
    microbiome::transform(transform = transform) %>% 
    # subset_samples(eval(as.name(var)) %in% as.name(group)) %>% 
    # subset_samples(Reactor_Treatment_Dose == "TR6_VAN90") %>% 
    microViz::phyloseq_validate() %>%
    # tax_fix() %>%
    # tax_prepend_ranks() %>%
    # tax_transform("clr", rank = "Strain", keep_counts = TRUE) %>% 
    # tax_transform(transform = transform, rank = rank_glom, keep_counts = TRUE) %>%
    microViz::tax_filter(min_prevalence = min_prev, undetected = 0, use_counts = FALSE) -> ps_fil
  
  #######------------------
  
  lms <- ps_fil %>% 
    # tax_transform(trans = "log2", chain = TRUE, zero_replace = "halfmin") %>% 
    microViz::taxatree_models(
      type = lm, 
      ranks = ranks,
      variables = c('Day_of_Treatment')
    )
  
  #######------------------
  
  lms %>% 
    taxatree_models2stats(.) %>% 
    taxatree_stats_p_adjust(method = "BH", grouping = padj_group) %>% 
    .$taxatree_stats %>% 
    # mutate(taxon =  stringr::str_remove(taxon, "[PFGCOS]: ")) %>%  #-> tmp #%>% 
    # arrange(starts_with("p.adj")) %>%
    select(taxon,p.adj.BH.rank, everything()) -> tmp
  
  #######------------------
  
  tmp %>% 
    # filter(p.adj.BH.rank <= adj_pval_cut) %>% 
    # filter(rank == "Genus") %>% 
    filter(estimate > 0) %>% 
    mutate(comp = "pos") -> pos_taxa
  
  tmp %>% 
    # filter(p.adj.BH.rank <= adj_pval_cut) %>% 
    # filter(rank == "Genus") %>% 
    filter(estimate < 0) %>% 
    mutate(comp = "neg") -> neg_taxa
  
  #######------------------
  
  rbind(pos_taxa,
        neg_taxa) %>% 
    filter(p.adj.BH.rank <= adj_pval_cut) %>%
    select(comp, everything())  -> overall_taxa_long
  
  overall_taxa_long %>% 
    select(term, rank, taxon, comp, p.adj.BH.rank) %>% 
    pivot_wider(   names_from = comp,
                   values_from = p.adj.BH.rank) %>% 
    rowwise() %>%
    mutate(pos = ifelse("pos" %in% names(.), pos, NA),
           neg = ifelse("neg" %in% names(.), neg, NA)) %>% 
    #        p2_pos = ifelse("p2_pos" %in% names(.), p2_neg, NA),
    #        p2_neg = ifelse("p2_neg" %in% names(.), p2_neg, NA)) %>% 
    mutate(class = case_when(!is.na(pos) ~ "inc",
                             !is.na(neg) ~ "dec",
                             TRUE ~ "NA")) %>% 
    mutate(Group = group)-> overall_taxa
  # select(-pos) %>% 
  # add_column(., !!!c("pos", "neg")[setdiff(c("pos", "neg"), names(overall_taxa))])
  
  
  #######------------------
  
  if ( use_cluster ==TRUE) {
    
    
    ps_fil$ps %>% 
      t() %>% 
      otu_table() -> t_otu
    
    t_otu %>% 
      Hmisc::rcorr() -> cor_tax
    
    #######------------------
    
    rows.cor <- cor((t_otu),  method = "spearman")
    
    hclust.row <- hclust(as.dist(1-rows.cor),"ward.D2" )
    
    # factoextra::fviz_nbclust(rows.cor, kmeans, method= 'gap_stat') -> outt
    # 
    # n_clust<-outt$data
    
    #######------------------
    
    # library(fpc)
    # pamk.best <- pamk(rows.cor)
    # cat("number of clusters estimated by optimum average silhouette width:", pamk.best$nc, "\n")
    # plot(cluster::pam(rows.cor, pamk.best$nc))
    # 
    #######------------------
    
    library(mclust)
    # Run the function to see how many clusters
    # it finds to be optimal, set it to search for
    # at least 1 model and up 20.
    d_clust <- Mclust(as.matrix(rows.cor), G=1:max_clust)
    m.best <- dim(d_clust$z)[2]
    cat("model-based optimal number of clusters:", m.best, "\n")
    
    cutree(hclust.row, k = m.best) %>% 
      data.frame() %>% 
      rownames_to_column('taxon') -> taxa_gpes
    # mutate(taxon =  stringr::str_remove(taxon, "[PFCGOS]: ")) -> taxa_gpes
    
    colnames(taxa_gpes)  <- c("taxon", "cluster")
    
    
    out = list(); out_heat = list()
    
    for (cluster_nb in taxa_gpes$cluster %>%  unique()){
      
      print(cluster_nb)
      
      taxa_gpes %>% 
        filter(cluster == cluster_nb) %>% 
        pull(taxon) %>% 
        as.character()-> tax
      
      if(tax %>%  length() > 1 )
      {
        chick2 %>% 
          prune_samples(get_variable(chick2, var) %in% group,
                        .)  %>% 
          filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>% 
          microbiome::transform(., "compositional") %>% 
          subset_taxa(.,  eval(as.name(ranks)) %in% tax) %>% 
          
          # subset_taxa( Genus %in% tax) %>% 
          # subset_taxa(Genus %in% c("Proteus", "Ligilactobacillus")) %>% 
          microbiomeutilities::plot_area(., xvar="Day_of_Treatment", 
                                         level = ranks,
                                         facet.by =ranks,
                                         abund.thres = 0,
                                         prev.thres=0,
                                         fill.colors = microViz::distinct_palette(20, pal = "kelly"),
                                         ncol=6,
                                         nrow=4) -> out[[cluster_nb]]
        
      }
      
      # chick2 %>% 
      #   prune_samples(get_variable(chick2, var) %in% group,
      #                 .)  %>% 
      #   filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>% 
      #   microbiome::transform(., "compositional") %>% 
      #     subset_taxa(.,  eval(as.name(ranks)) %in% tax) %>% 
      #   phyloseq_ampvis_heatmap(physeq = ., tax_aggregate = ifelse(ranks == "Strains", "Species",ranks), 
      #                           transform = FALSE,
      #                           plot_values = FALSE,
      #                           facet_by = c("Model2", "Model", "Fermentation","Reactor_Treatment_Dose"),
      #                           group_by = "Day_of_Treatment",#"sample_name",
      #                           ntax = Inf
      #   ) -> out_heat[[cluster_nb]]
      # 
      
    }
    
    out <- list("ps_fil" = ps_fil,
                "model" = tmp,
                "overall_taxa" = overall_taxa,
                "overall_taxa_long" = overall_taxa_long,
                "taxa_cluster" = taxa_gpes,
                "out_heat" = out_heat,
                "out" = out)
    
    
  }else{
    out <- list("ps_fil" = ps_fil,
                "model" = tmp,
                "overall_taxa" = overall_taxa,
                "overall_taxa_long" = overall_taxa_long)
    
  }
  #######------------------
  # then for the stationary ones, we can test on -4 to 5 and 5 to end.
  
  if(sub_test == TRUE){
    
    #######------------------
    
    ps_fil %>% 
      ps_filter(Day_of_Treatment <= time_split) %>% 
      # tax_transform(trans = "log2", chain = TRUE, zero_replace = "halfmin") %>% 
      taxatree_models(
        type = lm, 
        ranks = ranks,
        variables = c('Day_of_Treatment')
      ) %>% 
      taxatree_models2stats(.) %>% 
      taxatree_stats_p_adjust(method = "BH", grouping = padj_group) %>% 
      .$taxatree_stats %>% 
      # mutate(taxon =  stringr::str_remove(taxon, "[PFCGOS]: ")) %>% 
      arrange(p.adj.BH.rank) %>% 
      select(taxon,p.adj.BH.rank, everything()) -> psfil_1
    
    psfil_1 %>% 
      # filter(p.adj.BH.rank <= adj_pval_cut) %>% 
      # filter(rank == "Genus") %>% 
      filter(estimate > 0) %>% 
      mutate(comp = "p1_pos") -> fil1_pos_taxa
    
    psfil_1 %>% 
      # filter(p.adj.BH.rank <= adj_pval_cut) %>% 
      # filter(rank == "Genus") %>% 
      filter(estimate < 0) %>% 
      mutate(comp = "p1_neg") -> fil1_neg_taxa
    
    #######------------------
    
    ps_fil %>% 
      ps_filter(Day_of_Treatment > time_split) %>% 
      # tax_transform(trans = "log2", chain = TRUE, zero_replace = "halfmin") %>% 
      taxatree_models(
        type = lm, 
        ranks = ranks,
        variables = c('Day_of_Treatment')
      ) %>% 
      taxatree_models2stats(.) %>% 
      taxatree_stats_p_adjust(method = "BH", grouping = padj_group) %>% 
      .$taxatree_stats %>% 
      # mutate(taxon =  stringr::str_remove(taxon, "[PFCGOS]: ")) %>% 
      arrange(p.adj.BH.rank) %>% 
      select(taxon,p.adj.BH.rank, everything()) -> psfil_2
    
    psfil_2 %>% 
      # filter(p.adj.BH.rank <= adj_pval_cut) %>% 
      # filter(rank == "Genus") %>% 
      filter(estimate > 0) %>% 
      mutate(comp = "p2_pos") -> fil2_pos_taxa
    
    psfil_2 %>% 
      # filter(p.adj.BH.rank <= adj_pval_cut) %>% 
      # filter(rank == "Genus") %>% 
      filter(estimate < 0) %>% 
      mutate(comp = "p2_neg") -> fil2_neg_taxa
    
    #######------------------
    
    
    rbind(pos_taxa,
          neg_taxa,
          fil1_pos_taxa,
          fil1_neg_taxa,
          fil2_pos_taxa,
          fil2_neg_taxa) %>% 
      filter(p.adj.BH.rank <= adj_pval_cut) -> overall_taxa_long
    
    overall_taxa_long %>% 
      select(term, rank, taxon, comp, p.adj.BH.rank) %>% 
      pivot_wider(   names_from = comp,
                     values_from = p.adj.BH.rank) %>% 
      rowwise() %>%
      mutate(pos = ifelse("pos" %in% names(.), pos, NA),
             neg = ifelse("neg" %in% names(.), neg, NA)) %>% 
      mutate(p1_pos = ifelse("p1_pos" %in% names(.), p1_pos, NA),
             p1_neg = ifelse("p1_neg" %in% names(.), p1_neg, NA),
             p2_pos = ifelse("p2_pos" %in% names(.), p2_pos, NA),
             p2_neg = ifelse("p2_neg" %in% names(.), p2_neg, NA)) %>% 
      mutate(class = case_when(!is.na(pos) | !is.na(p1_pos) && !is.na(p2_pos) ~ "inc",
                               !is.na(neg) & !is.na(p1_pos) | !is.na(p2_neg) & !is.na(p1_pos) ~ "inV",
                               !is.na(neg) && is.na(pos) && is.na(p1_pos) && is.na(p2_pos) | !is.na(p1_neg) && !is.na(p1_neg) ~ "dec",
                               !is.na(p1_neg) && !is.na(p2_pos) | !is.na(neg) && !is.na(p2_pos) | is.na(p2_pos) ~ "V",
                               TRUE ~ "NA"))  %>% 
      mutate(Group = group) -> overall_taxa
    
    
    
    
    if ( use_cluster ==TRUE) {
      
      out <- list("ps_fil" = ps_fil,
                  "model" = tmp,
                  "model_split1" = psfil_1,
                  "model_split2" = psfil_2,
                  "overall_taxa_long" = overall_taxa_long,
                  "overall_taxa" = overall_taxa,
                  "taxa_cluster" = taxa_gpes)
    }
    else{
      
      out <- list("ps_fil" = ps_fil,
                  "model" = tmp,
                  "model_split1" = psfil_1,
                  "model_split2" = psfil_2,
                  "overall_taxa_long" = overall_taxa_long,
                  "overall_taxa" = overall_taxa)
    }
  }
  
  #######------------------
  
  
  
  
  #######------------------
  
  return(out)
}
```


```{r}
chick2 %>%
  function_lm_per_reactor(transform = "Z", #compositional, 
                          time_split = 3, rank_glom = "Genus",
                          ranks = "Genus") -> out

out$overall_taxa
```

```{r}
setNames(vector("list", 
                length(physeq_get_unique(chick2, "Reactor_Treatment_Dose"))), 
         physeq_get_unique(chick2, "Reactor_Treatment_Dose")) -> lis


```

```{r}
for (grp in names(lis)){
  print(grp)
  
  function_lm_per_reactor(chick2, time_split = 4, rank_glom = "Genus",  
                          transform = "log",
                          ranks = "Genus", group = grp) -> lis[[grp]]
}


do.call(rbind, lapply(lis, "[[", "overall_taxa")) %>% 
  ungroup() %>% 
  select(taxon, Group, class) %>% 
  rename(OTU = taxon,
         Reactor_Treatment_Dose = Group,
         category = class)-> map_class

map_class %>% 
  head()
```

```{r}
chick2 %>% 
  # prune_samples(get_variable(chick2, var) %in% group,
  #               .)  %>%
  filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>% 
  physeq_glom_rename(taxrank = "Genus", rename_ASV = "Genus") -> ps_filt
```


```{r}
ps_filt %>% 
  speedyseq::psmelt() %>%
  # filter(OTU %in% c("Escherichia-Shigella") ) %>%
  filter(Day_of_Treatment < 10) %>% 
  select(OTU, Abundance, Day_of_Treatment, Reactor_Treatment_Dose) %>% #!!rank_glom,
  left_join(map_class,
            by = c("Reactor_Treatment_Dose", "OTU")) %>% 
  arrange(Reactor_Treatment_Dose) %>% 
  ggplot(data = ., aes_string(color = "OTU", fill = "OTU")) +
  geom_area(aes_string(x = "Day_of_Treatment", y = "Abundance")) + 
  facet_grid(category ~  Reactor_Treatment_Dose, scales = "free") + theme(legend.position = "none")
```



```{r}
var = "Reactor_Treatment_Dose"; group = "TR6_VAN90"
out$overall_taxa %>%  filter(class == "V") %>% pull(taxon) %>%  as.character() -> to_fil

chick2 %>% 
  prune_samples(get_variable(chick2, var) %in% group,
                .)  %>% 
  filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  # subset_samples(eval(as.name(var)) %in% as.name(group)) %>% 
  # subset_samples(Reactor_Treatment_Dose == "TR6_VAN90") %>% 
  # phyloseq_validate() %>%
  # tax_fix() %>%
  # tax_prepend_ranks() %>%
  speedyseq::tax_glom("Genus") %>% 
  subset_samples(Reactor_Treatment_Dose %in% "TR6_VAN90") %>% 
  subset_taxa(Genus %in%  to_fil) %>% # extract only the taxa to display - after percentage normalisation
  phyloseq_ampvis_heatmap(physeq = ., tax_aggregate = "Genus",
                          transform = FALSE,
                          plot_values = FALSE,
                          facet_by = c("Model2", "Model", "Fermentation","Reactor_Treatment_Dose"),
                          group_by = "Day_of_Treatment",#"sample_name",
                          ntax = Inf
  ) -> pre_gen_grp

pre_gen_grp
```

pre_gen_grp

```{r}
chick2 %>% 
  speedyseq::tax_glom("Genus") %>% 
  subset_samples(Reactor_Treatment_Dose == "TR6_VAN90") %>% 
  microbiome::transform(., "compositional") %>% 
  subset_taxa(Genus %in% c("Proteus", "Ligilactobacillus")) %>% 
  microbiomeutilities::plot_area(., xvar="Day_of_Treatment", 
                                 level = "Genus",
                                 facet.by = "Genus",
                                 abund.thres = 0,
                                 prev.thres=0,
                                 # fill.colors=RColorBrewer::brewer.pal(12,"Paired"),
                                 ncol=3,
                                 nrow=4) 
```


```{r}
out$taxa_cluster %>% 
  # mutate(taxon =  stringr::str_remove(taxon, "[PFGCOS]: ")) %>% 
  filter(. == 7)
```


```{r}
chick2 %>% 
  speedyseq::tax_glom("Genus") %>% 
  subset_samples(Reactor_Treatment_Dose == "TR6_VAN90") %>% 
  microbiome::transform(., "compositional") %>% 
  subset_taxa(Genus %in% c("Lactococcus", "Defluviitaleaceae UCG-011")) %>% 
  microbiomeutilities::plot_area(., xvar="Day_of_Treatment", 
                                 level = "Genus",
                                 facet.by = "Genus",
                                 abund.thres = 0,
                                 prev.thres=0,
                                 fill.colors=RColorBrewer::brewer.pal(12,"Paired"),
                                 ncol=4,
                                 nrow=4) 
```

```{r}
chick2 %>% 
  speedyseq::tax_glom("Genus") %>% 
  subset_samples(Reactor_Treatment_Dose == "TR6_VAN90") %>% 
  microbiome::transform(., "compositional") %>% 
  subset_taxa(Genus %in% c("Intestinimonas", "Prevotella_9", "Defluviitaleaceae UCG-011")) %>% 
  microbiomeutilities::plot_area(., xvar="Day_of_Treatment", 
                                 level = "Genus",
                                 facet.by = "Genus",
                                 abund.thres = 0,
                                 prev.thres=0,
                                 fill.colors=RColorBrewer::brewer.pal(12,"Paired"),
                                 ncol=4,
                                 nrow=4) 
```

```{r}
chick2 %>% 
  speedyseq::tax_glom("Genus") %>% 
  subset_samples(Reactor_Treatment_Dose == "TR6_VAN90") %>% 
  microbiome::transform(., "compositional") %>% 
  subset_taxa(Genus %in% c("Bacteroides", "Eisenbergiella", "Bacillus", "Lachnospira" , "Planococcaceae","Negativibacillus", "Blautia" , "Subdoligranulum",
                           "Oscillibacter", "UC5-1-2E3")) %>% 
  microbiomeutilities::plot_area(., xvar="Day_of_Treatment", 
                                 level = "Genus",
                                 facet.by = "Genus",
                                 abund.thres = 0,
                                 prev.thres=0,
                                 fill.colors=RColorBrewer::brewer.pal(12,"Paired"),
                                 ncol=4,
                                 nrow=4) 
```

```{r}
chick2 %>% 
  speedyseq::tax_glom("Genus") %>% 
  subset_samples(Reactor_Treatment_Dose == "TR6_VAN90") %>% 
  microbiome::transform(., "compositional") %>% 
  subset_taxa(Genus %in% c("Bacteroides", "Escherichia-Shigella")) %>% 
  microbiomeutilities::plot_area(., xvar="Day_of_Treatment", 
                                 level = "Genus",
                                 facet.by = "Genus",
                                 abund.thres = 0,
                                 prev.thres=0,
                                 fill.colors=RColorBrewer::brewer.pal(12,"Paired"),
                                 ncol=4,
                                 nrow=4) 
```

```{r}
chick2 %>% 
  speedyseq::tax_glom("Genus") %>% 
  subset_samples(Reactor_Treatment_Dose == "TR6_VAN90") %>% 
  microbiome::transform(., "compositional") %>% 
  subset_taxa(Genus %in% c("Enterococcus", "Pseudomonas")) %>% 
  microbiomeutilities::plot_area(., xvar="Day_of_Treatment", 
                                 level = "Genus",
                                 facet.by = "Genus",
                                 abund.thres = 0,
                                 prev.thres=0,
                                 fill.colors=RColorBrewer::brewer.pal(12,"Paired"),
                                 ncol=4,
                                 nrow=4) 
```

```{r}
chick2 %>% 
  speedyseq::tax_glom("Genus") %>% 
  subset_samples(Reactor_Treatment_Dose == "TR6_VAN90") %>% 
  microbiome::transform(., "compositional") %>% 
  subset_taxa(Genus %in% c("Tuzzerella", "Lacticaseibacillus", "Parasutterella")) %>% 
  microbiomeutilities::plot_area(., xvar="Day_of_Treatment", 
                                 level = "Genus",
                                 facet.by = "Genus",
                                 abund.thres = 0,
                                 prev.thres=0,
                                 fill.colors=RColorBrewer::brewer.pal(12,"Paired"),
                                 ncol=4,
                                 nrow=4) 
```


same has Hanna but with long df with wide time abundance:
```{r}
chick2
var = "Reactor_Treatment_Dose"
group = "TR4_VAN90"
transform = "log"
rank_glom = "Genus"
# ranks = "Genus"
min_prev = 0.25
adj_pval_cut = 0.05
padj_group = "rank"
sub_test = TRUE

#######------------------

require(tidyverse)

#######------------------

chick2 %>% 
  # prune_samples(get_variable(chick2, var) %in% group,
  #               .)  %>%
  filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>% 
  physeq_glom_rename(taxrank = rank_glom, rename_ASV = rank_glom) -> ps_filt
```



```{r}
#15 for f1 13 for f2:

ps_filt %>% 
  speedyseq::psmelt() %>% 
  select(OTU, Abundance, Reactor_Treatment_Dose ,Day_of_Treatment) %>% 
  arrange(Day_of_Treatment) %>% 
  filter(Day_of_Treatment <= 15) -> ps_filt2


ps_filt2 %>% 
  # filter(Abundance > 0) %>% 
  group_by(OTU, Reactor_Treatment_Dose) %>% 
  add_count() %>% 
  select(-Abundance) %>% 
  distinct(OTU, Reactor_Treatment_Dose, n) %>% 
  mutate(OTU_Reactor_Treatment_Dose = paste0(OTU,"_",Reactor_Treatment_Dose)) -> count_OTU_reactors


ps_filt2 %>% 
  # filter(Abundance > 0) %>% 
  group_by(OTU, Reactor_Treatment_Dose) %>% 
  summarise(mean_ab = mean(Abundance)) %>% 
  mutate(OTU_Reactor_Treatment_Dose = paste0(OTU,"_",Reactor_Treatment_Dose)) -> mean_ab_OTU_reactors


ps_filt2 %>% 
  mutate(OTU_Reactor_Treatment_Dose = paste0(OTU,"_",Reactor_Treatment_Dose)) %>% 
  left_join(count_OTU_reactors %>%  ungroup() %>%  select(-OTU, -Reactor_Treatment_Dose),
            by = c("OTU_Reactor_Treatment_Dose" = "OTU_Reactor_Treatment_Dose"),
            keep = FALSE) %>% 
  left_join(mean_ab_OTU_reactors %>%  ungroup() %>%  select(-OTU, -Reactor_Treatment_Dose),
            by = c("OTU_Reactor_Treatment_Dose" = "OTU_Reactor_Treatment_Dose"),
            keep = FALSE) %>% 
  # filter(mean_ab > 1 & n > 8)  %>% 
  #ilter(Abundance > 0.01) %>% 
  pivot_wider(names_from = Day_of_Treatment, 
              values_from = Abundance) %>% 
  # rowwise() %>%
  # mutate(`-3` = ifelse( -3 %in% names(.), `-3`, NA),
  #        `-2` = ifelse(-2 %in% names(.), `-2`, NA),
  #        `-1` = ifelse(-1  %in% names(.), `-1`, NA),
  #        `0` = ifelse(0 %in% names(.), `0`, NA),
  #        `1` = ifelse(1 %in% names(.), `1`, NA),
  #        `2` = ifelse(2 %in% names(.), `2`, NA),
  #        `3` = ifelse(3 %in% names(.), `3`, NA),
  #        `4` = ifelse(4 %in% names(.), `4`, NA),
  #        `5` = ifelse(5 %in% names(.), `5`, NA),
  #        `6` = ifelse(6 %in% names(.), `6`, NA),
#        `7` = ifelse(7 %in% names(.), `7`, NA),
#        `8` = ifelse(8 %in% names(.), `8`, NA))  %>% 
# select(OTU, Abundance, Reactor_Treatment_Dose ,Day_of_Treatment) %>% 
# group_by(Reactor_Treatment_Dose, OTU) %>% 
# pivot_wider(names_from = Day_of_Treatment, 
#             values_from = Abundance) %>% 
group_by(Reactor_Treatment_Dose, OTU) %>% 
  # summarise(category = case_when(-any_of(c("OTU", "Reactor_Treatment_Dose", "-1", "-2","-3","-4") > 0.001))  ~ "appearing")
  summarise(category = 
              case_when(
                max(`0`, `-1`, `-2`, `-3`, na.rm = TRUE) == 0 & min(`3` , `4`, `5`, `6`, `7`, `8`, `9`, `10`, `11`, `12`, `13`, `14`, `15`, na.rm = TRUE) > 0  ~ "new",
                max(`0`, `-1`, `-2`, `-3`, na.rm = TRUE) < 0.1 &
                  min(`0`, `-1`, `-2`, `-3`, na.rm = TRUE) > 1.1*max(`0`, `-1`, `-2`, na.rm = TRUE) &
                  min(`5`, `6`, `7`, `8`, na.rm = TRUE) > 1.1*max(`1`, `2`, `3`,`4`, na.rm = TRUE) &
                  min(`9`, `10`, `11`, `12`, `13`, `14`, `15`, na.rm = TRUE) > 1.1*max(`1`, `2`, `3`,`4`, na.rm = TRUE) ~ "prol",
                min(`0`, `-1`, `-2`, `-3`, na.rm = TRUE) > 0 &
                  min(`1`, `2`, `3` , `4`, na.rm =TRUE) >  max(`0`, `-1`, `-2`, na.rm = TRUE)  &
                  min(`5`,`6`, `7`, na.rm = TRUE) > max(`1`, `2` , `3` , `4`, na.rm = TRUE) ~ "prol",
                min(`0`, `-1`, `-2`, na.rm = TRUE) > 0 &
                  min(`1`, `2`, `3` , `4`, na.rm =TRUE) >=  max(`0`, `-1`, `-2`, na.rm = TRUE)  &
                  min(`5`,`6`, `7`, na.rm = TRUE) >= max(`1`, `2` , `3` , `4`, na.rm = TRUE) ~ "prol",
                # min(`0`, `-1`, `-2`, na.rm = TRUE) > 0.01 &
                #   (max(`1`, `2` , na.rm = TRUE) <= 0.5*min(`0`, `-1`, `-2`, na.rm = TRUE) &
                #      min( `3` , `4`, `5`, `6`, `7`, na.rm = TRUE) >=  min(`0`, `-1`, `-2`, na.rm = TRUE)) |
                #   min(`0`, `-1`, `-2`, na.rm = TRUE) > 0.01 &
                #   (max(`1`, `2`, `3` , na.rm = TRUE) <= 0.5*min(`0`, `-1`, `-2`, na.rm = TRUE) &
                #      min( `4`, `5`, `6`, `7`, na.rm = TRUE)  >=  min(`0`, `-1`, `-2`, na.rm = TRUE)) |
                #   min(`0`, `-1`, `-2`, na.rm = TRUE) > 0.1 &
                #   (max(`1`, `2`, `3`,  `4`, na.rm = TRUE) <= 0.5*min(`0`, `-1`, `-2`, na.rm = TRUE) &
                #      min( `5`, `6`, `7`, na.rm = TRUE)  >= min(`0`, `-1`, `-2`, na.rm = TRUE)) |
                min(`0`, `-1`, `-2`, na.rm = TRUE) > 0.1 & 
                  (max(`1`, `2`, `3` , `4`, `5`, na.rm = TRUE) < 0.50*min(`0`, `-1`, `-2`, na.rm = TRUE) &  
                     min( `6`, `7`, `8`,`9`, na.rm = TRUE)  >  max(`1`, `2`, `3` , `4`, `5`, na.rm = TRUE)) ~ "recovering-V",
                min(`0`, `-1`, `-2`, na.rm = TRUE) > 0 & 
                  max(`1`, `2` , na.rm = TRUE) <= min(`0`, `-1`, `-2`, na.rm = TRUE) &
                  max(`3`, `4`, na.rm = TRUE)  <=  min(`1`,`2`, na.rm = TRUE) ~ "dis",
                min(`0`, `-1`, `-2`, na.rm = TRUE) > 0 & 
                  max(`1`, `2` , na.rm = TRUE) <= min(`0`, `-1`, `-2`, na.rm = TRUE) &
                  max(`5`, `6`,`7` ,`8`, na.rm = TRUE) <=  min(`1`, `2`, `3`, `4`, na.rm = TRUE) ~ "dis",
                min(`0`, `-1`, `-2`, na.rm = TRUE) > 0 & 
                  (min(`1`, `2`, `3` , `4`, `5`, na.rm = TRUE) > 1.25*max(`0`, `-1`, `-2`, na.rm = TRUE) &  
                     max( `6`, `7`, `8`,`9`, na.rm = TRUE)  <  0.5*min(`1`, `2`, `3` , `4`, `5`, na.rm = TRUE)) ~ "inV",
                TRUE ~ "normal")) %>% 
  arrange(Reactor_Treatment_Dose,category) -> classif
```


same has Hanna but with long df with wide time abundance:
```{r}
chick2
var = "Reactor_Treatment_Dose"
transform = "log"
rank_glom = "Genus"
# ranks = "Genus"
min_prev = 0.25
adj_pval_cut = 0.05
padj_group = "rank"
sub_test = TRUE

#######------------------

require(tidyverse)

#######------------------

chick2 %>% 
  # prune_samples(get_variable(chick2, var) %in% group,
  #               .)  %>%
  filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>% 
  physeq_glom_rename(taxrank = rank_glom, rename_ASV = rank_glom) -> ps_filt
```


```{r}
ps_filt %>% 
  speedyseq::psmelt() %>% 
  select(OTU, Abundance, Reactor_Treatment_Dose ,Day_of_Treatment) %>% 
  arrange(Day_of_Treatment) %>% 
  filter(Day_of_Treatment <= 15) -> ps_filt2


ps_filt2 %>% 
  # filter(Abundance > 0) %>% 
  group_by(OTU, Reactor_Treatment_Dose) %>% 
  add_count() %>% 
  select(-Abundance) %>% 
  distinct(OTU, Reactor_Treatment_Dose, n) %>% 
  mutate(OTU_Reactor_Treatment_Dose = paste0(OTU,"_",Reactor_Treatment_Dose)) -> count_OTU_reactors


ps_filt2 %>% 
  # filter(Abundance > 0) %>% 
  group_by(OTU, Reactor_Treatment_Dose) %>% 
  summarise(mean_ab = mean(Abundance)) %>% 
  mutate(OTU_Reactor_Treatment_Dose = paste0(OTU,"_",Reactor_Treatment_Dose))  -> mean_ab_OTU_reactors


#define_groups:
ps_filt2 %>% 
  # select(Day_of_Treatment) %>% 
  distinct(Day_of_Treatment) %>% 
  mutate(period = c( "pret", "pret", "pret",
                    "t1","t1","t1","t1","t1","t1",
                    "t2","t2","t2","t2","t2",
                    "t3","t3","t3","t3")) -> day_period

ps_filt2 %>% 
  mutate(OTU_Reactor_Treatment_Dose = paste0(OTU,"_",Reactor_Treatment_Dose)) %>% 
  left_join(count_OTU_reactors %>%  ungroup() %>%  select(-OTU, -Reactor_Treatment_Dose),
            by = c("OTU_Reactor_Treatment_Dose" = "OTU_Reactor_Treatment_Dose"),
            keep = FALSE) %>% 
  left_join(mean_ab_OTU_reactors %>%  ungroup() %>%  select(-OTU, -Reactor_Treatment_Dose),
            by = c("OTU_Reactor_Treatment_Dose" = "OTU_Reactor_Treatment_Dose"),
            keep = FALSE) %>% 
  # filter(mean_ab > 1 & n > 8)  %>%
  mutate(Day_of_Treatment = as.double(Day_of_Treatment)) %>%
  left_join(day_period,
            by = c("Day_of_Treatment" = "Day_of_Treatment")) %>% 
  # mutate(period = ifelse(Day_of_Treatment <= 0, "pret",
  #                        ifelse(0 <  Day_of_treatment <= 5, "t1"
  #                        ))
  # mutate(period = case_when(Day_of_Treatment <= 0 ~ "pret",
  #                           0 <  Day_of_treatment <= 5 ~ "tt"
  #                           Day_of_treatment > 0 & Day_of_treatment <= 5 ~ "t1",
  #                           Day_of_treatment > 6 & Day_of_treatment <= 10 ~ "t2",
  #                           Day_of_treatment > 10 & Day_of_treatment <= 16 ~ "t3",
  #                           TRUE ~ "tunknown")) %>%
  group_by(Reactor_Treatment_Dose, OTU, period) %>% 
  summarise(mean = base::mean(Abundance, na.rm = TRUE)) %>% 
  group_by(Reactor_Treatment_Dose, OTU) %>% 
  pivot_wider(names_from = period, 
              values_from = mean) %>% 
  summarise(category = 
              case_when(pret == 0 & (t1 >0 & t2>0 & t3 >0) ~ "new",
                         (t1 > pret  |
                           t2 > pret |
                           t3 > pret) & (t3 > t2 | t3 > t1 | t3 > pret) & (t3/pret > 4 & t2/pret > 3) ~ "prol",
                        t1 < pret  &
                          t3 < t1 ~ "dis",
                        t1 < pret &
                          t3 > t1 ~ "recovering",
                        t1 > pret &
                          t3 < t1 ~ "mid",
                        # pret == 0 & t1>0 &t2>0 &t3>0 ~ "new",
                        TRUE ~ "normal")) %>% 
  arrange(Reactor_Treatment_Dose,category) -> classif

classif$category %>%  unique()
```

```{r}
ps_filt %>% 
  speedyseq::psmelt() %>%
  # filter(Reactor_Treatment_Dose %in% c("TR6_VAN90","TR4_CCUG59168", "TR5_HV292.1", "TR1_CTX+HV292.120")) %>% 
  filter(Day_of_Treatment < 15) %>% 
  select(OTU, Abundance, Day_of_Treatment, Reactor_Treatment_Dose) %>% 
  left_join(classif,
            by = c("Reactor_Treatment_Dose", "OTU")) %>% 
  # filter(category %in% c("normal"),
  # Reactor_Treatment_Dose %in% ("TR4_CCUG59168")) %>% 
  arrange(Reactor_Treatment_Dose) %>% 
  ggplot(data = ., aes_string(color = "OTU", fill = "OTU")) +
  geom_area(aes_string(x = "Day_of_Treatment", y = "Abundance")) + 
  facet_grid(category ~  Reactor_Treatment_Dose, scales = "free") + 
  theme(legend.position = "none") -> p_cat

p_cat

p_cat %>%
  ggsave(file="~/Desktop/16S-chick.svg", plot=., width=10, height=8)
```


same has Hanna but with long df with wide time abundance:
```{r}
chick2
var = "Reactor_Treatment_Dose"
transform = "log"
rank_glom = "Genus"
# ranks = "Genus"
min_prev = 0.25
adj_pval_cut = 0.05
padj_group = "rank"
sub_test = TRUE

#######------------------

require(tidyverse)

#######------------------

chick2 %>% 
  # prune_samples(get_variable(chick2, var) %in% group,
  #               .)  %>%
  filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>% 
  physeq_glom_rename(taxrank = rank_glom, rename_ASV = rank_glom) -> ps_filt
```

```{r}
#15 for f1 13 for f2:

ps_filt %>% 
  speedyseq::psmelt() %>% 
  select(OTU, Abundance, Reactor_Treatment_Dose ,Day_of_Treatment) %>% 
  arrange(Day_of_Treatment) %>% 
  # filter(Abundance > 1) %>%
  pivot_wider(names_from = Day_of_Treatment, 
              values_from = Abundance) %>% 
  rowwise() %>%
  mutate(`-3` = ifelse( -3 %in% names(.), `-3`, NA),
         `-2` = ifelse(-2 %in% names(.), `-2`, NA),
         `-1` = ifelse(-1  %in% names(.), `-1`, NA),
         `0` = ifelse(0 %in% names(.), `0`, NA),
         `1` = ifelse(1 %in% names(.), `1`, NA),
         `2` = ifelse(2 %in% names(.), `2`, NA),
         `3` = ifelse(3 %in% names(.), `3`, NA),
         `4` = ifelse(4 %in% names(.), `4`, NA),
         `5` = ifelse(5 %in% names(.), `5`, NA),
         `6` = ifelse(6 %in% names(.), `6`, NA),
         `7` = ifelse(7 %in% names(.), `7`, NA),
         `8` = ifelse(8 %in% names(.), `8`, NA)) -> df_wide_ready 
# select(OTU, Abundance, Reactor_Treatment_Dose ,Day_of_Treatment) %>% 
# group_by(Reactor_Treatment_Dose, OTU) %>% 
# pivot_wider(names_from = Day_of_Treatment, 
#             values_from = Abundance) %>%

df_wide_ready
```
```{r}
df_wide_ready %>% 
  mutate(OTU_Reactor_Treatment_Dose = paste0(OTU,"_",Reactor_Treatment_Dose)) %>% 
  group_by(Reactor_Treatment_Dose, OTU) %>% 
  # left_join(otu_in_reactor_to_consider,
  #           by = c("Reactor_Treatment_Dose" = "Reactor_Treatment_Dose")) %>% 
  # filter(Reactor_Treatment_Dose == "TR1_CTX+HV292.120" & OTU == "[Eubacterium] brachy group")
  # summarise(category = case_when(-any_of(c("OTU", "Reactor_Treatment_Dose", "-1", "-2","-3","-4") > 0.001))  ~ "appearing")
  summarise(category = 
              case_when(
                # proliferating: increases over time 
                min(`0`, `-1`, `-2`, `-3`) > 0 &
                  min(`3`, `4`,`5`, `6`, na.rm = TRUE) > max(`0`, `-1`, `-2`, `-3`, na.rm = TRUE)  &
                  min(`12`, `13`, `14`, `15`, na.rm = TRUE) > max(`8`, `9`, `10`, `11`, na.rm = TRUE)  ~ "prol",
                min(`0`, `-1`, `-2`, `-3`, na.rm = TRUE) > 0.1 & 
                  max(`1`, `2`, `3` , `4`, `5`, na.rm = TRUE) < 0.50*min(`0`, `-1`, `-2`, `-3`, na.rm = TRUE) &  
                  min( `6`, `7`, `8`,`9`, na.rm = TRUE) > max(`1`, `2`, `3` , `4`, `5`, na.rm = TRUE) ~ "recovering-V",
                min(`0`, `-1`, `-2`, na.rm = TRUE) > 0 & 
                  max(`1`, `2` , na.rm = TRUE) <= min(`0`, `-1`, `-2`, na.rm = TRUE) &
                  max(`3`, `4`, na.rm = TRUE)  <=  min(`1`,`2`, na.rm = TRUE) ~ "dis",
                min(`0`, `-1`, `-2`, na.rm = TRUE) > 0 & 
                  min(`5`, `6`,`7` ,`8`, na.rm = TRUE) <=  min(`1`, `2`, `3`, `4`, na.rm = TRUE) & min(`6`, `7`, `8`,`9`, na.rm = TRUE) ~ "dis",
                min(`0`, `-1`, `-2`, na.rm = TRUE) > 0 & 
                  (min(`1`, `2`, `3` , `4`, `5`, na.rm = TRUE) > 1.25*max(`0`, `-1`, `-2`, na.rm = TRUE) &  
                     max( `6`, `7`, `8`,`9`, na.rm = TRUE)  <  0.5*min(`1`, `2`, `3` , `4`, `5`, na.rm = TRUE)) ~ "inV",
                TRUE ~ "normal")) %>% 
  arrange(Reactor_Treatment_Dose,category) -> classif
```



```{r}
ps_filt %>% 
  speedyseq::psmelt() %>%
  # filter(Reactor_Treatment_Dose %in% c("TR6_VAN90","TR4_CCUG59168", "TR5_HV292.1", "TR1_CTX+HV292.120")) %>% 
  filter(Day_of_Treatment < 15) %>% 
  select(OTU, Abundance, Day_of_Treatment, Reactor_Treatment_Dose) %>% #!!rank_glom,
  left_join(classif,
            by = c("Reactor_Treatment_Dose", "OTU")) %>% 
  arrange(Reactor_Treatment_Dose) %>% 
  ggplot(data = ., aes_string(color = "OTU", fill = "OTU")) +
  geom_area(aes_string(x = "Day_of_Treatment", y = "Abundance")) + 
  facet_grid(category ~  Reactor_Treatment_Dose, scales = "free") + 
  theme(legend.position = "none") -> p_cat

p_cat
# p_cat %>% 
#   ggsave(file="~/Desktop/16S-chick.svg", plot=., width=10, height=8)
```




```{r}

#     # define as disappearing
#   } else if (abund_per_period['pret',asv] > 0 & 
#              abund_per_period['t1',asv] >= abund_per_period['t2',asv] &
#              abund_per_period['t2',asv] >= abund_per_period['t3',asv] &
#              abund_per_period['t3',asv] >= abund_per_period['t4',asv] &
#              abund_per_period['t4',asv] >= abund_per_period['t5',asv] &
#              abund_per_period['t5',asv] == 0){
#     
#     class_table[asv,] = "disappeared"     
#     
#   }  


# Define as recovering
#   } else if (abund_per_period['pret',asv] >= 0.01 &
#              10*abund_per_period['t1',asv] < abund_per_period['pret',asv] &
#              # # Case 1: Recovery only at t5
#              # ((abund_per_period['t2',asv] <= abund_per_period['t1',asv] &
#              #  abund_per_period['t3',asv] <= abund_per_period['t2',asv] &
#              #  abund_per_period['t4',asv] <= abund_per_period['t3',asv])|
#              #  # Case 2: Recovery at t4
#              # (abund_per_period['t2',asv] <= abund_per_period['t1',asv] &
#              #  abund_per_period['t3',asv] <= abund_per_period['t2',asv] &
#              #  abund_per_period['t4',asv] > abund_per_period['t3',asv]) |
#              # # Case 3: Recovery at t3
#              # (abund_per_period['t2',asv] <= abund_per_period['t1',asv] &
#              #  abund_per_period['t3',asv] > abund_per_period['t2',asv]  &
#              #  abund_per_period['t4',asv] > abund_per_period['t3',asv]) |
#              # # Case 4: Recovery at t2
#              # (abund_per_period['t2',asv] > abund_per_period['t1',asv] &
#              #  abund_per_period['t3',asv] > abund_per_period['t2',asv] &
#              #  abund_per_period['t4',asv] > abund_per_period['t3',asv])) &
#              abund_per_period['t5',asv] >= 0.01
#   ) {
#     
#     class_table[asv,] = "recovering"
#     
#     # define as having temporal advantage
#     
#   } else if (abund_per_period['pret',asv] == 0 &
#              abund_per_period['t1',asv] > 0.01 &
#              abund_per_period['t5',asv] < 0.001 &
#              abund_per_period['t5',asv] <= abund_per_period['t1',asv] # not sure about this one
#   ) {
#     class_table[asv,] = "temp adv"
#     
#   } else if (abund_per_period['pret',asv] > 0 &
#              abund_per_period['t1',asv] > 5* abund_per_period['pret',asv] &
#              5*abund_per_period['t5',asv] <= abund_per_period['t1',asv]
#   ) {
#     class_table[asv,] = "temp adv"
#     

#   
#   else {
#     
#     class_table[asv,] = "normal"
#   }
#   
```







```{r}
# check per recator how many non NA we have:
ps_filt %>% 
  speedyseq::psmelt() %>% 
  select(OTU, Abundance, Reactor_Treatment_Dose ,Day_of_Treatment) %>% 
  arrange(Day_of_Treatment) %>% 
  # filter(Abundance > 1) %>% 
  group_by(OTU, Reactor_Treatment_Dose) %>% 
  count(Abundance) %>% 
  distinct(OTU, Reactor_Treatment_Dose, n) %>% 
  filter(n > 8) %>% 
  mutate(OTU_Reactor_Treatment_Dose = paste0(OTU,"_",Reactor_Treatment_Dose)) -> otu_in_reactor_to_consider


otu_in_reactor_to_consider
```

```{r}
df_wide_ready %>% 
  mutate(OTU_Reactor_Treatment_Dose = paste0(OTU,"_",Reactor_Treatment_Dose)) %>% 
  group_by(Reactor_Treatment_Dose, OTU) %>% 
  left_join(otu_in_reactor_to_consider,
            by = c("Reactor_Treatment_Dose" = "Reactor_Treatment_Dose")) %>% 
  # filter(Reactor_Treatment_Dose == "TR2_CTX20" & OTU == "CAG-56")
  # summarise(category = case_when(-any_of(c("OTU", "Reactor_Treatment_Dose", "-1", "-2","-3","-4") > 0.001))  ~ "appearing")
  summarise(category = 
              case_when(
                # proliferating: increases over time 
                min(`0`, `-1`, `-2`, `-3`, na.rm = TRUE) > 0 &
                  base::mean(`3`, `4`,`5`, `6`, na.rm = TRUE) > base::mean(`0`, `-1`, `-2`, `-3`, na.rm = TRUE)  &
                  # base::mean( `7`, `8`, na.rm = TRUE) > base::mean(`1`, `2`, `3`,`4`, na.rm = TRUE) &
                  base::mean(`8`, `9`, `10`, `11`, `12`, `13`, `14`, na.rm = TRUE) > base::mean(`5`, `6`, `7`, `8`, na.rm = TRUE)  ~ "prol",
                min(`0`, `-1`, `-2`, `-3`, na.rm = TRUE) > 0.1 & 
                  max(`1`, `2`, `3` , `4`, `5`, na.rm = TRUE) < 0.50*min(`0`, `-1`, `-2`, `-3`, na.rm = TRUE) &  
                  min( `6`, `7`, `8`,`9`, na.rm = TRUE) > max(`1`, `2`, `3` , `4`, `5`, na.rm = TRUE) ~ "recovering-V",
                min(`0`, `-1`, `-2`, na.rm = TRUE) > 0 & 
                  max(`1`, `2` , na.rm = TRUE) <= min(`0`, `-1`, `-2`, na.rm = TRUE) &
                  max(`3`, `4`, na.rm = TRUE)  <=  min(`1`,`2`, na.rm = TRUE) ~ "dis",
                min(`0`, `-1`, `-2`, na.rm = TRUE) > 0 & 
                  mean(`1`, `2` , na.rm = TRUE) <= mean(`0`, `-1`, `-2`, na.rm = TRUE) &
                  mean(`5`, `6`,`7` ,`8`, na.rm = TRUE) <=  mean(`1`, `2`, `3`, `4`, na.rm = TRUE) ~ "dis",
                min(`0`, `-1`, `-2`, na.rm = TRUE) > 0 & 
                  (min(`1`, `2`, `3` , `4`, `5`, na.rm = TRUE) > 1.25*max(`0`, `-1`, `-2`, na.rm = TRUE) &  
                     max( `6`, `7`, `8`,`9`, na.rm = TRUE)  <  0.5*min(`1`, `2`, `3` , `4`, `5`, na.rm = TRUE)) ~ "inV",
                TRUE ~ "normal")) %>% 
  arrange(Reactor_Treatment_Dose,category) -> classif
```


```{r}
classif %>% 
  group_by(Reactor_Treatment_Dose, category) %>% 
  tally() %>% 
  arrange(category) %>% 
  DT::datatable()
```

```{r}
ps_filt %>% 
  speedyseq::psmelt() %>%
  # filter(Reactor_Treatment_Dose %in% c("TR6_VAN90","TR4_CCUG59168", "TR5_HV292.1", "TR1_CTX+HV292.120")) %>% 
  filter(Day_of_Treatment < 10) %>% 
  select(OTU, Abundance, Day_of_Treatment, Reactor_Treatment_Dose) %>% #!!rank_glom,
  left_join(classif,
            by = c("Reactor_Treatment_Dose", "OTU")) %>% 
  arrange(Reactor_Treatment_Dose) %>% 
  ggplot(data = ., aes_string(color = "OTU", fill = "OTU")) +
  geom_area(aes_string(x = "Day_of_Treatment", y = "Abundance")) + 
  facet_grid(category ~  Reactor_Treatment_Dose, scales = "free") + 
  theme(legend.position = "none") -> p_cat


p_cat %>% 
  ggsave(file="~/Desktop/16S-chick.svg", plot=., width=10, height=8)
```

```{r}
ps_filt %>% 
  speedyseq::psmelt() %>%
  # filter(Reactor_Treatment_Dose %in% c("TR6_VAN90","TR4_CCUG59168", "TR5_HV292.1", "TR1_CTX+HV292.120")) %>% 
  filter(Day_of_Treatment < 10) %>% 
  select(OTU, Model2, Reactor_Treatment_Dose) %>% #!!rank_glom,
  left_join(classif,
            by = c("Reactor_Treatment_Dose", "OTU")) %>% 
  arrange(OTU) %>% 
  distinct(OTU, Reactor_Treatment_Dose, .keep_all = TRUE) %>% 
  pivot_wider(names_from = Reactor_Treatment_Dose, values_from = category) %>% 
  data.frame() %>% 
  write_tsv("~/Desktop/classification.tsv")
```

# Hanna's classification:


<https://github.com/haegih/MSc_Thesis-NRP72/blob/main/scripts/ASV_classification/chicken_asv-classification.md>

Species were categorized into three groups based on the following criteria: 

<https://www.tandfonline.com/doi/full/10.1080/19490976.2021.1900995>

1. Disappeared: the species undetected in all post-treatment samples (T4 to T6); 
2. New: the species undetected in the baseline, but persistently abun- dant (relative abundance > 0.1%) in all post-treatment samples (T4 to T6); 3. Proliferating: the species meeting any of the two criteria: 
1) a species with a relative abundance < 0.1% in baseline and reached 5% in T2 or T3, or 
2) a species with a relative abundance ≥ 0.1% in baseline and increased by at least 10-fold in the relative abundance in T2 or T3.

Challenges: 
- thresholds
- treatment phases vs days

Define function to make transform absolute into relative abundance:

```{r}

rel_abundance_transformation <- function(list){
  list %>%
    data.frame() %>%
    dplyr::filter(!is.na(Period)) %>% 
    select(-c(Model, Treatment)) %>%
    column_to_rownames("Period") %>%
    data.matrix() %>%
    funrar::make_relative() %>%
    data.frame() -> rel_abund_df
  
  
  return(rel_abund_df)
}
```

Define function for asv classification:

```{r}
asv_classification <- function(abund_per_period, class_table){
  
  
  for (asv in rownames(class_table)){
    
    # define as new: 
    # TODO: lower threshold?
    
    # define as proliferating
    if (abund_per_period['pret',asv] == 0 & 
        (abund_per_period['t1',asv] > 0.001 | 
         abund_per_period['t2',asv] > 0.001 |
         abund_per_period['t3',asv] > 0.001 |
         abund_per_period['t4',asv] > 0.001 |
         abund_per_period['t5',asv] > 0.001)){
      
      class_table[asv,] = "new"
      
    } else if (abund_per_period['pret',asv] < 0.001 &
               (abund_per_period['t1',asv] > 0.05 |
                abund_per_period['t2',asv] > 0.05) &
               abund_per_period['t3',asv] > max(abund_per_period['t1',asv], abund_per_period['t2',asv]) &
               abund_per_period['t4',asv] > abund_per_period['t3',asv] &
               abund_per_period['t5',asv] > abund_per_period['t4',asv]
    ){
      class_table[asv,] = "prol"
      
    } else if (abund_per_period['pret',asv] < 0.001 & #including all the options increases amount of classified "proliferating"
               (abund_per_period['t1',asv] > 0.05 | 
                abund_per_period['t2',asv] > 0.05 |
                abund_per_period['t3',asv] > 0.05 |
                abund_per_period['t4',asv] > 0.05 | 
                abund_per_period['t5',asv] > 0.05)
    ){
      
      class_table[asv,] = "prol"
      
    } else if (abund_per_period['pret',asv] > 0 & ##changed
               (abund_per_period['t1',asv] >= 5*abund_per_period['pret',asv] &
                abund_per_period['t2',asv] >= abund_per_period['t1',asv] &
                abund_per_period['t3',asv] >= abund_per_period['t2',asv] &
                abund_per_period['t4',asv] >= abund_per_period['t3',asv] &
                abund_per_period['t5',asv] >= abund_per_period['t4',asv])
    ) {
      
      class_table[asv,] = "prol"
      
    } else if (abund_per_period['pret',asv] >= 0.001 &
               (abund_per_period['t1',asv] >= 5*abund_per_period['pret',asv] |
                abund_per_period['t2',asv] >= 5*abund_per_period['pret',asv] |
                abund_per_period['t3',asv] >= 5*abund_per_period['pret',asv] |
                abund_per_period['t4',asv] >= 5*abund_per_period['pret',asv] |
                abund_per_period['t5',asv] >= 5*abund_per_period['pret',asv])
    ) {
      
      class_table[asv,] = "prol"
      
      # Define as recovering
    } else if (abund_per_period['pret',asv] >= 0.01 &
               10*abund_per_period['t1',asv] < abund_per_period['pret',asv] &
               # # Case 1: Recovery only at t5
               # ((abund_per_period['t2',asv] <= abund_per_period['t1',asv] &
               #  abund_per_period['t3',asv] <= abund_per_period['t2',asv] &
               #  abund_per_period['t4',asv] <= abund_per_period['t3',asv])|
               #  # Case 2: Recovery at t4
               # (abund_per_period['t2',asv] <= abund_per_period['t1',asv] &
               #  abund_per_period['t3',asv] <= abund_per_period['t2',asv] &
               #  abund_per_period['t4',asv] > abund_per_period['t3',asv]) |
               # # Case 3: Recovery at t3
               # (abund_per_period['t2',asv] <= abund_per_period['t1',asv] &
               #  abund_per_period['t3',asv] > abund_per_period['t2',asv]  &
               #  abund_per_period['t4',asv] > abund_per_period['t3',asv]) |
               # # Case 4: Recovery at t2
               # (abund_per_period['t2',asv] > abund_per_period['t1',asv] &
               #  abund_per_period['t3',asv] > abund_per_period['t2',asv] &
               #  abund_per_period['t4',asv] > abund_per_period['t3',asv])) &
               abund_per_period['t5',asv] >= 0.01
    ) {
      
      class_table[asv,] = "recovering"
      
      # define as having temporal advantage
      
    } else if (abund_per_period['pret',asv] == 0 &
               abund_per_period['t1',asv] > 0.01 &
               abund_per_period['t5',asv] < 0.001 &
               abund_per_period['t5',asv] <= abund_per_period['t1',asv] # not sure about this one
    ) {
      class_table[asv,] = "temp adv"
      
    } else if (abund_per_period['pret',asv] > 0 &
               abund_per_period['t1',asv] > 5* abund_per_period['pret',asv] &
               5*abund_per_period['t5',asv] <= abund_per_period['t1',asv]
    ) {
      class_table[asv,] = "temp adv"
      
      # define as disappearing
    } else if (abund_per_period['pret',asv] > 0 & 
               abund_per_period['t1',asv] >= abund_per_period['t2',asv] &
               abund_per_period['t2',asv] >= abund_per_period['t3',asv] &
               abund_per_period['t3',asv] >= abund_per_period['t4',asv] &
               abund_per_period['t4',asv] >= abund_per_period['t5',asv] &
               abund_per_period['t5',asv] == 0){
      
      class_table[asv,] = "disappeared"     
      
    }  
    
    else {
      
      class_table[asv,] = "normal"
    }
    
  }
  return(class_table)
}
```



# from https://www.biorxiv.org/content/10.1101/2022.10.19.512887v1.full.pdf

SVD can decompose a matrix in two separated matrices of eigenvectors and a vector of eigenvalues. When applied to gene abundance (or expression) data over time, one of the matrices is interpreted as the set of temporal patterns underlying the data and the other as the “loadings” (i.e. how much each individual gene is contributing to each pattern). The seasonal version of the forecasting method AutoRegressive Integrated Moving Average (ARIMA) computes cyclical (seasonality), autoregressive (temporal self-dependence), differencing (difference between consecutive time points) and moving-average (averaging of consecutive time points) components of a time-series.

We employed SVD to extract relevant temporal patterns, which were then clustered into 17 fundamental signals.

Those were integrated with collected environmental parameters to build an ARIMA model, augmented with seasonal components, which could explain the observed signals.

Figure 2. Eigengene modelling using ARIMA augmented with environmental parameters and Fourier terms. a. The signals S1-17 encapsulate the time-dependent dynamics underlying the microbial community. The scale of the y-axis is dimensionless as the eigenvectors. b. The S1-17 are explained as ARIMA processes under the influence of the environmental variables. The five blocks of explanatory variables are: Model (ARIMA components), Manual (manually collected environmental variables, directly on the sampling location), Inflow (inflow stream of wastewater in the plant), V1 (first anaerobic tank in the plant), and V2 (second anaerobic tank in the plant). Every circle represents a significant
variable (Benjamini-Hochberg adjusted p<0.05) for the corresponding signals among S1-17, the size represents the value of the coefficient, the ring colour its sign and the fill colour the log10(p-value). c. The signals are connected by a temporal transfer of information, suggesting a succession of ecological events.





```{r}
require(microbiomeutilities)

chick2 %>% 
  microbiome::transform(., "compositional") %>% 
  plot_area(., xvar="Day_of_Treatment", 
            level = "Family",
            facet.by = "Reactor_Treatment_Dose",
            abund.thres = 0.1,
            prev.thres=0.1,
            fill.colors=RColorBrewer::brewer.pal(12,"Paired"),
            ncol=3,
            nrow=4) -> p

p
```


```{r}
chick2 %>% 
  microbiome::transform(., "clr") -> chick2_clr

as(otu_table(chick2_clr), "matrix") %>% 
  as.data.frame() -> otu

otu %>% 
  as.matrix() %>% 
  t() -> otu_asmat

otu_asmat %>%
  svd(.) -> svd_corr



transform_left_basis <- function(vec, U_inv, s){
  return(U_inv %*% vec / s)
}

transform_left_basis_serial <- function(dd, U, s){
  U_inv <- MASS::ginv(U)
  dd_right <- apply(dd, 2, transform_left_basis, U_inv=U_inv, s=s)
  return(t(dd_right))
}

mat.test.svd.corr <- transform_left_basis_serial(otu_asmat, svd_corr$u, svd_corr$d)

mat.test.svd.corr
```

```{r}
education.by.readership = matrix(c(5, 18, 19, 12, 3, 7, 46, 29, 40, 7, 2, 20, 39, 49, 16), 
                                 nrow = 5,
                                 dimnames = list(
                                   "Level of education" = c("Some primary", "Primary completed", "Some secondary", "Secondary completed", "Some tertiary"),
                                   "Category of readership" = c("Glance", "Fairly thorough", "Very thorough"))) 
O = education.by.readership / sum(education.by.readership)
E = rowSums(O) %o% colSums(O)
Z = (O - E) / sqrt(E)

SVD = svd(Z)

SVD
```


# multivariate time series:
- https://www.youtube.com/watch?v=EB7i0Da5NC0
- https://www.youtube.com/watch?v=jVxa-UQyRSU

```{r}
require(WDI)

gdp <- WDI(country = c("US", "CA", "GB", "CN", "JP", "SG"),  indicator = c("NY.GDP.PCAP.CD", "NY.GDP.MKTP.CD"), start = 1960, end = 2011)
```

```{r}
names(gdp)
```

```{r}
gdpcast <- reshape2::dcast(year ~ country, data = gdp[, c("country", "year", "NY.GDP.PCAP.CD")], value.var = "NY.GDP.PCAP.CD")


gdpcast
```


```{r}
gdpTS <- ts(data = gdpcast[,2], start = min(gdpcast$year), end = max(gdpcast$year))

plot(gdpTS, type = "single")
```


```{r}
numDiffs <- forecast::ndiffs(gdpTS)
```

```{r}
gdpDiffed <- diff(gdpTS, differences = numDiffs)

plot(gdpDiffed, plot.type = "single")
```

```{r}
gdpvar <- vars::VAR(gdpDiffed, lag.max = 12)

head(coef ( gdpvar$varresult$Canada))
```

```{r}
stats::arima(gdpDiffed) -> arima

arima
```


```{r}
require(coefplot)

coefplot(gdpvar$varresult$Japan)
```


-   compare donor microbiota and pre-treatment / CR - heatmap: v
-   evaluate effect treatment: -- alpha (shanon, simpson --\> hill d1, d2) -- beta (achinson, log -\> bray)
-   describe effect on taxa (Family, Genus, ASV): v -- heatmap v -- alluvial plot v

--> Ezequiel: process - co-selection of AMR.


```{r}
# ps_rare %>%  
#   sample_data() %>%
#   # dplyr::select(Model2)
#   # dplyr::filter(model2 == "Chicken2") %>%
#   data.frame()
```
## Before Treatment:

Obj: compare donor microbiota and pre-treatment / CR - heatmap

```{r}
ps_rare %>%
  subset_samples(Day_of_Treatment > -4 & Day_of_Treatment <= 0 | Reactor %in% c("DONOR", "CR")) -> before_treat # %>% 
# subset_samples(Reactor %!in%c("IR")) %>% 
# subset_samples() -> before_treat
```

```{r}
ps_rare %>% 
  sample_data() %>% 
  data.frame() %>% 
  filter(Reactor == "DONOR") 
```

```{r}
# before_treat %>% 
#   # microViz::ps_mutate(tp_var = )
#   physeq_most_abundant(group_var = "Model2",
#                        ntax = 24,
#                        tax_level = "Strain") -> top_taxa_per_group

before_treat %>%
  physeq_most_abundant(group_var = "Reactor_Treatment",
                       ntax = 10,
                       tax_level = "Strain") -> top_taxa_per_group

top_taxa_per_group
```

```{r}
before_treat %>%
  rarefy_even_depth(sample.size = min_sample_size, rngseed = 123) %>%
  # subset_samples(Reactor == "DONOR") %>%  sample_data() %>%  data.frame()
  # speedyseq::tax_glom(taxrank = "Family") %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  subset_taxa(Strain %in% top_taxa_per_group) %>% # extract only the taxa to display - after percentage normalisation
  phyloseq_ampvis_heatmap(physeq = ., tax_aggregate = "Species",
                          transform = FALSE,
                          plot_values = FALSE,
                          facet_by = c("Model2", "Fermentation","Reactor_Treatment_Dose"),
                          group_by = "Day_of_Treatment",#"sample_name",
                          ntax = Inf
  ) -> pre_speces

pre_speces$data %>% 
  mutate(Abundance = na_if(Abundance, 0)) %>% 
  mutate(Fermentation = Fermentation %>%  as.character() %>%  replace_na(., "NA")) %>% 
  mutate(Fermentation = factor(Fermentation, levels = c("NA", 1, 2))) -> pre_speces$data 

pre_speces + 
  facet_grid(. ~  Model2 + Fermentation + Reactor_Treatment_Dose, scales = "free", space = "free") +
  scale_color_manual(values = c("#31a354", "orange", 
                                "#f03b20"), drop = FALSE, na.value = 'transparent') -> pre_speces
# scale_fill_viridis_c(breaks = c(0,  0.01, 1, 10, 25, 50, 75, 100),
# labels = c(0,  0.01, 1, 10, 25,  50, 75, 100),
# trans = scales::pseudo_log_trans(sigma = 0.9),
# na.value = 'transparent') -> pre_speces


#pre_speces

```

```{r}
before_treat %>%
  rarefy_even_depth(sample.size = min_sample_size, rngseed = 123) %>%
  # subset_samples(Reactor == "DONOR") %>%  sample_data() %>%  data.frame()
  # speedyseq::tax_glom(taxrank = "Family") %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  phyloseq_ampvis_heatmap(physeq = ., tax_aggregate = "Family",
                          transform = FALSE,
                          plot_values = FALSE,
                          facet_by = c("Model2", "Fermentation","Reactor_Treatment_Dose"),
                          group_by = "Day_of_Treatment",#"sample_name",
                          ntax = 8
  ) -> pre_Family

pre_Family$data %>% 
  mutate(Abundance = na_if(Abundance, 0)) %>% 
  mutate(Fermentation = Fermentation %>%  as.character() %>%  replace_na(., "NA")) %>% 
  mutate(Fermentation = factor(Fermentation, levels = c("NA", 1, 2))) -> pre_Family$data 


pre_Family + 
  facet_grid(. ~  Model2 + Fermentation + Reactor_Treatment_Dose, scales = "free", space = "free") +
  scale_color_manual(values = c("#31a354", "orange", 
                                "#f03b20"), drop = FALSE, na.value = 'transparent') -> pre_Family


# pre_Family
```

```{r}
# cowplot::plot_grid(pre_speces,
#           pre_Family, 
#            ncol = 1,
#           align = "v", axis = "b",
#           rel_heights = c(3, 1)) -> pre_plots
# 
# pre_plots
```

```{r}

ggpubr::ggarrange(pre_Family + theme(legend.position = "null") + 
                    theme(axis.title.x=element_blank(),
                          axis.text.x=element_blank(),
                          axis.ticks.x=element_blank()),
                  pre_speces + theme(
                    strip.background = element_blank(),
                    strip.text.x = element_blank()
                  )  + theme(legend.position = "null"),
                  align = "v",
                  ncol = 1, 
                  heights = c(1, 2.40),
                  common.legend = FALSE) -> pre_plots

pre_plots

pre_plots %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 6, 
                    height = 0.618 * 317.48031496 * 8 , paper = "A4",  scaling = 2,
                    file = out_pptx)

pre_Family %>% 
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 1 , paper = "A4",  scaling = 2,
                    file = out_pptx)


pre_Family$data %>% 
  select(Display, Sample, Abundance,Reactor_Treatment_Dose) %>% 
  pivot_wider(names_from = Display, values_from = Abundance) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "pre_family",
                   showNA = TRUE)
pre_speces$data %>% 
  select(Display, Sample, Abundance,Reactor_Treatment_Dose) %>% 
  pivot_wider(names_from = Display, values_from = Abundance) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "pre_ASV",
                   showNA = TRUE)
```
## Hannah's classification:

<https://github.com/haegih/MSc_Thesis-NRP72/blob/main/scripts/ASV_classification/chicken_asv-classification.md>

Species were categorized into three groups based on the following criteria: 

<https://www.tandfonline.com/doi/full/10.1080/19490976.2021.1900995>

1. Disappeared: the species undetected in all post-treatment samples (T4 to T6); 
2. New: the species undetected in the baseline, but persistently abun- dant (relative abundance > 0.1%) in all post-treatment samples (T4 to T6); 3. Proliferating: the species meeting any of the two criteria: 
1) a species with a relative abundance < 0.1% in baseline and reached 5% in T2 or T3, or 
2) a species with a relative abundance ≥ 0.1% in baseline and increased by at least 10-fold in the relative abundance in T2 or T3.

Challenges: 
- thresholds
- treatment phases vs days

Define function to make transform absolute into relative abundance:

```{r}

rel_abundance_transformation <- function(list){
  list %>%
    data.frame() %>%
    dplyr::filter(!is.na(Period)) %>% 
    select(-c(Model, Treatment)) %>%
    column_to_rownames("Period") %>%
    data.matrix() %>%
    funrar::make_relative() %>%
    data.frame() -> rel_abund_df
  
  
  return(rel_abund_df)
}
```

Define function for asv classification:

```{r}
asv_classification <- function(abund_per_period, class_table){
  
  
  for (asv in rownames(class_table)){
    
    # define as new: 
    # TODO: lower threshold?
    
    # define as proliferating
    if (abund_per_period['pret',asv] == 0 & 
        (abund_per_period['t1',asv] > 0.001 | 
         abund_per_period['t2',asv] > 0.001 |
         abund_per_period['t3',asv] > 0.001 |
         abund_per_period['t4',asv] > 0.001 |
         abund_per_period['t5',asv] > 0.001)){
      
      class_table[asv,] = "new"
      
    } else if (abund_per_period['pret',asv] < 0.001 &
               (abund_per_period['t1',asv] > 0.05 |
                abund_per_period['t2',asv] > 0.05) &
               abund_per_period['t3',asv] > max(abund_per_period['t1',asv], abund_per_period['t2',asv]) &
               abund_per_period['t4',asv] > abund_per_period['t3',asv] &
               abund_per_period['t5',asv] > abund_per_period['t4',asv]
    ){
      class_table[asv,] = "prol"
      
    } else if (abund_per_period['pret',asv] < 0.001 & #including all the options increases amount of classified "proliferating"
               (abund_per_period['t1',asv] > 0.05 | 
                abund_per_period['t2',asv] > 0.05 |
                abund_per_period['t3',asv] > 0.05 |
                abund_per_period['t4',asv] > 0.05 | 
                abund_per_period['t5',asv] > 0.05)
    ){
      
      class_table[asv,] = "prol"
      
    } else if (abund_per_period['pret',asv] > 0 & ##changed
               (abund_per_period['t1',asv] >= 5*abund_per_period['pret',asv] &
                abund_per_period['t2',asv] >= abund_per_period['t1',asv] &
                abund_per_period['t3',asv] >= abund_per_period['t2',asv] &
                abund_per_period['t4',asv] >= abund_per_period['t3',asv] &
                abund_per_period['t5',asv] >= abund_per_period['t4',asv])
    ) {
      
      class_table[asv,] = "prol"
      
    } else if (abund_per_period['pret',asv] >= 0.001 &
               (abund_per_period['t1',asv] >= 5*abund_per_period['pret',asv] |
                abund_per_period['t2',asv] >= 5*abund_per_period['pret',asv] |
                abund_per_period['t3',asv] >= 5*abund_per_period['pret',asv] |
                abund_per_period['t4',asv] >= 5*abund_per_period['pret',asv] |
                abund_per_period['t5',asv] >= 5*abund_per_period['pret',asv])
    ) {
      
      class_table[asv,] = "prol"
      
      # Define as recovering
    } else if (abund_per_period['pret',asv] >= 0.01 &
               10*abund_per_period['t1',asv] < abund_per_period['pret',asv] &
               # # Case 1: Recovery only at t5
               # ((abund_per_period['t2',asv] <= abund_per_period['t1',asv] &
               #  abund_per_period['t3',asv] <= abund_per_period['t2',asv] &
               #  abund_per_period['t4',asv] <= abund_per_period['t3',asv])|
               #  # Case 2: Recovery at t4
               # (abund_per_period['t2',asv] <= abund_per_period['t1',asv] &
               #  abund_per_period['t3',asv] <= abund_per_period['t2',asv] &
               #  abund_per_period['t4',asv] > abund_per_period['t3',asv]) |
               # # Case 3: Recovery at t3
               # (abund_per_period['t2',asv] <= abund_per_period['t1',asv] &
               #  abund_per_period['t3',asv] > abund_per_period['t2',asv]  &
               #  abund_per_period['t4',asv] > abund_per_period['t3',asv]) |
               # # Case 4: Recovery at t2
               # (abund_per_period['t2',asv] > abund_per_period['t1',asv] &
               #  abund_per_period['t3',asv] > abund_per_period['t2',asv] &
               #  abund_per_period['t4',asv] > abund_per_period['t3',asv])) &
               abund_per_period['t5',asv] >= 0.01
    ) {
      
      class_table[asv,] = "recovering"
      
      # define as having temporal advantage
      
    } else if (abund_per_period['pret',asv] == 0 &
               abund_per_period['t1',asv] > 0.01 &
               abund_per_period['t5',asv] < 0.001 &
               abund_per_period['t5',asv] <= abund_per_period['t1',asv] # not sure about this one
    ) {
      class_table[asv,] = "temp adv"
      
    } else if (abund_per_period['pret',asv] > 0 &
               abund_per_period['t1',asv] > 5* abund_per_period['pret',asv] &
               5*abund_per_period['t5',asv] <= abund_per_period['t1',asv]
    ) {
      class_table[asv,] = "temp adv"
      
      # define as disappearing
    } else if (abund_per_period['pret',asv] > 0 & 
               abund_per_period['t1',asv] >= abund_per_period['t2',asv] &
               abund_per_period['t2',asv] >= abund_per_period['t3',asv] &
               abund_per_period['t3',asv] >= abund_per_period['t4',asv] &
               abund_per_period['t4',asv] >= abund_per_period['t5',asv] &
               abund_per_period['t5',asv] == 0){
      
      class_table[asv,] = "disappeared"     
      
    }  
    
    else {
      
      class_table[asv,] = "normal"
    }
    
  }
  return(class_table)
}
```

Define function to plot the figures of each subcategory per model:

```{r}
get_plot_data <- function(list, cols){
  
  fig_title <- unique(list$Treatment)
  
  print(paste0("working on ", fig_title))
  # transform absolute into relative abundance
  rel_abund_df <- rel_abundance_transformation(list)
  
  # define category:
  data.frame(asv = colnames(rel_abund_df)) %>%
    mutate(cat = NA) %>%
    column_to_rownames('asv')-> asv_cat
  
  asv_cat <- asv_classification(rel_abund_df, asv_cat)
  
  # make df longer, such that every asv + period combination has its own entry
  rel_abund_df %>%
    # dplyr::filter(!is.na(Period)) %>% 
    rownames_to_column(var = "period") %>%
    pivot_longer(-period, names_to = "ASV", values_to = "rel_abundance") %>%
    filter(rel_abundance > 0) -> df_longer
  
  
  # combine df_longer with classification categories
  asv_cat %>%
    rownames_to_column(var = "ASV") %>%
    full_join(df_longer, by = "ASV") -> df
  
  # Generate plotting-ready data
  df %>%
    select(cat, period, rel_abundance) %>%
    group_by(period, cat) %>%
    # summarise(sum_abund = sum(rel_abundance)) %>%
    filter(!is.na(period)) -> plot_data
  
  # plot the data
  ggplot(plot_data, aes(fill = cat, y = rel_abundance, x = period)) +
    geom_bar(position = "stack", stat = "identity") +
    ggtitle(fig_title) +
    scale_fill_manual(name = "ASV category", values = cols,
                      na.value = "grey") +
    theme(strip.text.x = element_text(size=20),
          axis.title = element_text(size = 22),
          axis.text.x = element_text(size = 20),
          axis.text.y = element_text(size = 20),
          legend.position = "bottom",
          legend.text = element_text(size=20),
          legend.title = element_text(size=22),
          plot.title = element_text(size=26)) +
    ylab("Relative abundance") -> p1
  
  
  
  #print(p1)
  
  return(list(p1, df))
  
}
```


Define astethics:

```{r}
cat_col <- c(new='#3cb44b', prol="#6495ED", recovering='#FFB6C1', disappeared='#e6194B', `temp adv`='#F28C28', normal='#a9a9a9')

```

In a next step, the data is split into 6 tables: for each model (human and chicken) there are 3 tables for each antibioticum (CTX, untreated, VAN). Then, ASVs need to be classified and finally, the data is plotted:

chicken 1

```{r}
ps_tmp <- ps_rare

# #add Day_of_Treatment groups.
# ps_tmp %>%
#   sample_data() %>%
#   data.frame() %>%
# #   # mutate(Treatment = base::replace(Treatment, Experiment == "Cecum", "DONOR")) %>%
# #   # filter(Day_of_Treatment != -1) %>% # Excluded due to instability
#     dplyr::mutate(Period = ifelse(Day_of_Treatment < 0, "pret", Period))  -> sample_data(ps_tmp)
#   # filter(Model2 == "Chicken1") %>%
# filter(Treatment != "DONOR") -> sample_data(ps_tmp)


# combine phyloseq tables and calculate relativ abundance across periods
ps_tmp %>%
  subset_samples(Model2 == "Chicken1") %>%
  subset_samples(Treatment != "DONOR") %>%
  psmelt() %>%
  select(OTU, Period, Abundance, Model, Treatment) %>%
  group_by(Model, Treatment, OTU, Period) %>%
  summarize(cum_abund = mean(Abundance)) %>%
  pivot_wider(names_from = OTU, values_from = cum_abund) -> ps_merged


# ps_merged$Treatment <- factor(ps_merged$Treatment, ordered = TRUE, levels = c("UNTREATED", "CTX+HV292.1", "CTX", "HV292.1", "VAN+CCUG59168", "VAN", "CCUG59168"))
# split lists
chicken_split <- split(ps_merged, ps_merged$Treatment)

# Get plot data
plot_data <- lapply(1:length(unique(ps_merged$Treatment)),
                    function(i) get_plot_data(chicken_split[[i]], cat_col))


plots <- lapply(plot_data, `[[`, 1)

# plots <- lapply(plot_data, `[[`, 2)

data_classification <- lapply(plot_data, `[[`, 2)
names(data_classification) <- names(chicken_split)

# plot the data
# ggpubr::ggarrange(plotlist = plots, ncol=7, nrow=1, common.legend = TRUE, legend="bottom") -> combined_plot

# combined_plot 
# 
# combined_plot %>% 
#     export::graph2ppt(append = TRUE, 
#                     width = 317.48031496 * 6, 
#                     height = 0.618 * 317.48031496 * 4 , paper = "A4",  scaling = 2,
#                     file = out_pptx)
```

export ASV level classification:

```{r}
data_classification %>% 
  bind_rows(.id = "reactor") %>%
  mutate("model2" = "Chicken1") -> classif_chicken1 
```

chicken 2

```{r}
ps_tmp <- ps_rare

# add Day_of_Treatment groups.
# ps_tmp %>% 
#   subset_samples(Model2 == "Chicken2") %>%
#   subset_samples(Treatment != "DONOR") %>%
#   # subset_samples(Treatment == "VAN+CCUG59168") %>%
#   sample_data() %>%
#   data.frame() %>% #select(Day_of_Treatment, Period)
#   # mutate(Treatment = base::replace(Treatment, Experiment == "Cecum", "DONOR")) %>%
#   # filter(Day_of_Treatment != -1) %>% # Excluded due to instability
#   # dplyr::mutate(Period = ifelse(Day_of_Treatment<= -1, "Pret", Period)) %>%
#   select(Treatment,Day_of_Treatment, Period) %>%  
#   group_by(Treatment, Period) %>% count(Period)#-> sample_data(ps_tmp)
# mutate(case_when())

# combine phyloseq tables and calculate relativ abundance across periods
ps_tmp %>%
  subset_samples(Model2 == "Chicken2") %>%
  subset_samples(Treatment != "DONOR") %>% # subset_samples(Treatment == "UNTREATED") %>%  sample_data() %>%  data.frame()
  # subset_samples(Treatment != "UNTREATED") %>% 
  psmelt() %>%
  select(OTU, Period, Abundance, Model, Treatment) %>%
  group_by(Model, Treatment, OTU, Period) %>%
  summarize(cum_abund = mean(Abundance)) %>%
  pivot_wider(names_from = OTU, values_from = cum_abund) -> ps_merged
#filter(Treatment == "VAN+CCUG59168")

# ps_merged$Treatment <- factor(ps_merged$Treatment, ordered = TRUE, levels = c("UNTREATED", "CTX+HV292.1", "CTX", "HV292.1", "VAN+CCUG59168", "VAN", "CCUG59168"))
# split lists
chicken_split <- split(ps_merged, ps_merged$Treatment)


chicken_split$`VAN+CCUG59168` %>%
  # data.frame() %>% 
  mutate(Period = replace_na(Period,"pret")) -> chicken_split$`VAN+CCUG59168`


# Get plot data
plot_data <- lapply(1:length(unique(ps_merged$Treatment)),
                    function(i) get_plot_data(chicken_split[[i]], cat_col))


plots <- lapply(plot_data, `[[`, 1)

data_classification <- lapply(plot_data, `[[`, 2)
names(data_classification) <- names(chicken_split)

# plot the data
# ggpubr::ggarrange(plotlist = plots, ncol=7, nrow=1, common.legend = TRUE, legend="bottom")
```

export ASV level classification:

```{r}
data_classification %>% 
  bind_rows(.id = "reactor") %>%
  mutate("model2" = "Chicken2") -> classif_chicken2

```

```{r}
rbind(classif_chicken1,
      classif_chicken2) %>% 
  distinct(model2, reactor, ASV, .keep_all = TRUE) %>% 
  dplyr::filter(!is.na(period)) %>% 
  left_join(ps_tmp %>%
              tax_table() %>%
              data.frame() %>%
              rownames_to_column("ASV"),
            by = c("ASV" = "ASV")) %>% 
  dplyr::select(model2, reactor, ASV, cat, everything(), -Kingdom) %>% 
  dplyr::select(-period, -rel_abundance) %>%
  xlsx::write.xlsx(file = "~/Desktop/16S-chick.xlsx",
                   append = TRUE,
                   sheetName = "ASV-classification",
                   showNA = TRUE)
```


```{r}
rbind(classif_chicken1,
      classif_chicken2) %>% 
  dplyr::filter(!is.na(period)) %>% 
  left_join(ps_tmp %>%
              tax_table() %>%
              data.frame() %>%
              rownames_to_column("ASV"),
            by = c("ASV" = "ASV")) %>% 
  mutate(reactor = fct_relevel(reactor,c("UNTREATED", "CTX", "CTX+HV292.1", "HV292.1", "VAN", "VAN+CCUG59168", "CCUG59168"))) %>%  # plot the data
  ggplot(., aes(fill = cat, y = rel_abundance, x = period)) +
  geom_bar(position = "stack", stat = "identity") +
  # ggtitle(fig_title) +
  scale_fill_manual(name = "ASV category", values = cat_col,
                    na.value = "grey") +
  # theme(strip.text.x = element_text(size=20),
  #     axis.title = element_text(size = 22),
  #     axis.text.x = element_text(size = 20),
  #     axis.text.y = element_text(size = 20),
  #     legend.position = "bottom",
  #     legend.text = element_text(size=20),
  #     legend.title = element_text(size=22),
  #     plot.title = element_text(size=26)) +
  ylab("Relative abundance") + 
  facet_grid( model2 ~ reactor) -> ASV_classification_plot

ASV_classification_plot
```

```{r}
ASV_classification_plot %>%
  export::graph2ppt(append = TRUE,
                    width = 317.48031496 * 1.25,
                    height = 0.618 * 317.48031496 * 1 , paper = "A4",  scaling = 2,
                    file = out_pptx)
```


```{r}
ASV_classification_plot$data %>% 
  xlsx::write.xlsx(file = "~/Desktop/16S-chick.xlsx",
                   append = TRUE,
                   sheetName = "ASV-classification2",
                   showNA = TRUE)
```


```{r}
ASV_classification_plot$data %>% 
  ggplot(., aes(fill = Phylum, y = rel_abundance, x = cat)) +
  geom_bar(position = "fill", stat = "identity") +
  # ggtitle(fig_title) +
  # scale_fill_manual(name = "ASV category", values = cat_col,
  #                   na.value = "grey") +
  # theme(strip.text.x = element_text(size=20),
  #     axis.title = element_text(size = 22),
  #     axis.text.x = element_text(size = 20),
  #     axis.text.y = element_text(size = 20),
  #     legend.position = "bottom",
  #     legend.text = element_text(size=20),
  #     legend.title = element_text(size=22),
  #     plot.title = element_text(size=26)) +
ylab("Relative abundance") + 
  facet_grid( model2 ~ reactor) -> ASV_classification_plot_2

ASV_classification_plot_2

ASV_classification_plot_2 %>%   
  export::graph2ppt(append = TRUE,
                    width = 317.48031496 * 1.25,
                    height = 0.618 * 317.48031496 * 1 , paper = "A4",  scaling = 2,
                    file = out_pptx)
```

```{r}
ASV_classification_plot$data %>%
  ggplot(., aes(fill = Family, y = rel_abundance, x = cat)) +
  geom_bar(position = "fill", stat = "identity") +
  # ggtitle(fig_title) +
  scale_fill_manual(values = cat_col,
                    na.value = "grey") +
  # theme(strip.text.x = element_text(size=20),
  #     axis.title = element_text(size = 22),
  #     axis.text.x = element_text(size = 20),
  #     axis.text.y = element_text(size = 20),
  #     legend.position = "bottom",
  #     legend.text = element_text(size=20),
  #     legend.title = element_text(size=22),
  #     plot.title = element_text(size=26)) +
  ylab("Relative abundance") + 
  facet_grid( model2 ~ reactor) -> ASV_classification_plot_3

ASV_classification_plot_3

# ASV_classification_plot_2 %>%   
#       export::graph2ppt(append = TRUE,
#                     width = 317.48031496 * 1.25,
#                     height = 0.618 * 317.48031496 * 1 , paper = "A4",  scaling = 2,
#                     file = out_pptx)
```



other plot:

<https://github.com/haegih/MSc_Thesis-NRP72/blob/main/scripts/ASV_classification/human_taxanomy-with-classification.md>

```{r}
include_classification <- function(tax_df, asv_class_df){
  
  tax_df %>%
    left_join(asv_class_df %>% select(ASV, cat), by = "ASV") -> df
  
  return(df)
}


make_heatmap <- function(tax_df, asv_class_df, tax_level = Species){
  
  full_df <- include_classification(tax_df, asv_class_df)
  
  
  full_df %>% 
    unique %>%
    drop_na(Abundance) %>%
    ggplot(mapping = aes(x = Period,
                         y = {{tax_level}},
                         fill = Abundance,
    )) +
    facet_grid(cat ~ Treatment_Dose, scales = "free", space = "free", drop = TRUE,labeller = labeller(cat = label_wrap_gen(width = 15))) +
    #we adjust the vertical labels such that it is printed correctly. 
    geom_tile() +
    scale_fill_viridis_c(name = "Abundance",
                         na.value = "transparent", trans = scales::pseudo_log_trans(sigma = 0.1),
                         # trans =  scales::pseudo_log_trans(sigma = 1),
                         breaks = c(1, 50, 400, 10000), labels = c(1, 50, 400, 10000),
                         limits = c(0, 22500)) +
    theme_bw() -> p1
  
  return(p1)
}
```


```{r}
# combine phyloseq tables and calculate relativ abundance across periods
ps_rare %>%
  subset_samples(Model == "Chicken") %>% 
  psmelt() %>%
  select(OTU, Period, Abundance, Model2, Treatment_Dose, Kingdom:Strain) -> ps_melted



ps_melted %>%
  group_by(Model2, Treatment_Dose, OTU, Period) %>%
  summarize(cum_abund = sum(Abundance)) %>%
  pivot_wider(names_from = OTU, values_from = cum_abund) -> ps_merged_abund

ps_merged_abund %>%
  pivot_longer(cols = ASV0001:ASV0100, names_to = "ASV", values_to = "Abundance") %>%
  left_join(ps_melted %>% select(OTU, Kingdom:Strain), by = c("ASV" = "OTU")) %>%
  unique() %>%
  mutate(Abundance = na_if(Abundance, 0)) %>%
  mutate(Period = ordered(Period, levels = c("pret", factor(1:7))))-> ps_with_tax

ps_with_tax.split <- split(ps_with_tax, ps_with_tax$Treatment_Dose)


```

```{r}
abundance_lists <- list()
i <- 0
for (treat in names(ps_with_tax.split)){
  i = i + 1
  full_df <- include_classification(ps_with_tax.split[[treat]], asv_class[[treat]]) %>% filter(!is.na(cat))
  df <- full_df %>% select(Model, ASV, Treatment_Dose, cat, Kingdom:Strain) %>% group_by(Model, Treatment_Dose, cat) %>% count() %>% mutate(RelAbund = n/nrow(full_df)) %>% data.frame()
  abundance_lists[[i]] <- df
  
}

abundance_table <- bind_rows(abundance_lists) 

abundance_table %>% data.frame() %>% arrange(RelAbund) 
```



```{r}
tax_list <- list()

for (treat in names(ps_with_tax.split)){
  i = i + 1
  full_df <- include_classification(ps_with_tax.split[[treat]], asv_class[[treat]]) %>% filter(!is.na(cat))
  tax_list[[i]] <- full_df 
}

tax_cat_table <- bind_rows(tax_list) 



qual_col_pals = brewer.pal.info[brewer.pal.info$category == 'qual',]
col = unlist(mapply(brewer.pal, qual_col_pals$maxcolors, rownames(qual_col_pals)))
phyla_col <- col
names(phyla_col) <- tax_cat_table$Phylum %>% unique()


tax_cat_table %>%
  ggplot(aes(fill = Phylum, y = Abundance, x=cat)) +
  geom_bar(position="fill", stat="identity") +
  scale_color_manual(name = "Phylum", values = phyla_col,
                     na.value = "grey") +
  scale_fill_manual(name = "Phylum", values = phyla_col,
                    na.value = "grey") +
  theme(strip.text.x = element_text(size=18),
        axis.title = element_text(size = 18),
        axis.text.x = element_text(angle = 90,  size = 14, vjust = 0.5, hjust=1),
        axis.text.y = element_text(size = 18),
        legend.position = "bottom",
        legend.text = element_text(size=14),
        legend.title = element_text(size=14)) +
  facet_grid(~Treatment_Dose) +
  xlab("ASV category") + 
  ylab("Relative abundance")
```

human1 :


```{r}
# "data/processed/16S/16S_working_phyloseq.RDS" %>% 
#   here::here() %>% 
#   readRDS()  %>% 
#   subset_samples(Enrichment == "NotEnriched") %>%
#       subset_samples(Experiment %in% c("Continuous", "Cecum")) %>%
#    subset_samples(Day_of_Treatment > -6 & Day_of_Treatment <=30 | Reactor == "DONOR") %>% 
#   subset_samples(Model == "Human") %>% 
#    # subset_samples(Reactor != "CR" & Model2 == "Chicken1" | Model2 == "Chicken2" & Reactor %in% c("TR2", "TR3", "TR4", "TR5", "TR6", "TR7", "CR", "DONOR")) %>% 
#    subset_samples(Reactor != "IR") %>%
#   rarefy_even_depth(sample.size = 3738, rngseed = 123) -> ps_tmp
#   
# # add Day_of_Treatment groups.
# ps_tmp %>% 
#   sample_data() %>% 
#   data.frame() %>% 
#   # mutate(Treatment = base::replace(Treatment, Experiment == "Cecum", "DONOR")) %>%
#   filter(Day_of_Treatment != -1) %>% # Excluded due to instability
#   filter(Model2 == "Human1") %>%
#     filter(Treatment != "DONOR") %>% 
#   filter(!Treatment %in%  c("HV292.1", "CCUG59168")) -> sample_data(ps_tmp)
# 
# # combine phyloseq tables and calculate relativ abundance across periods
# ps_tmp %>%
#   psmelt() %>%
#   select(OTU, Period, Abundance, Model, Treatment) %>%
#   group_by(Model, Treatment, OTU, Period) %>%
#   summarize(cum_abund = mean(Abundance)) %>%
#   pivot_wider(names_from = OTU, values_from = cum_abund) -> ps_merged
# 
# 
# # ps_merged$Treatment <- factor(ps_merged$Treatment, ordered = TRUE, levels = c("UNTREATED", "CTX+HV292.1", "CTX", "HV292.1", "VAN+CCUG59168", "VAN", "CCUG59168"))
# # split lists
# chicken_split <- split(ps_merged, ps_merged$Treatment)
# 
# # Get plot data
# plot_data <- lapply(1:length(unique(ps_merged$Treatment)),
#                     function(i) get_plot_data(chicken_split[[i]], cat_col))
# 
# 
# plots <- lapply(plot_data, `[[`, 1)
# 
# data_classification <- lapply(plot_data, `[[`, 2)
# names(data_classification) <- names(chicken_split)
# 
# # plot the data
# ggpubr::ggarrange(plotlist = plots, ncol=7, nrow=1, common.legend = TRUE, legend="bottom")
```

human2 :


```{r}
# "data/processed/16S/16S_working_phyloseq.RDS" %>% 
#   here::here() %>% 
#   readRDS()  %>% 
#   subset_samples(Enrichment == "NotEnriched") %>%
#       subset_samples(Experiment %in% c("Continuous", "Cecum")) %>%
#    subset_samples(Day_of_Treatment > -6 & Day_of_Treatment <=30 | Reactor == "DONOR") %>% 
#   subset_samples(Model == "Human") %>% 
#    # subset_samples(Reactor != "CR" & Model2 == "Chicken1" | Model2 == "Chicken2" & Reactor %in% c("TR2", "TR3", "TR4", "TR5", "TR6", "TR7", "CR", "DONOR")) %>% 
#    subset_samples(Reactor != "IR") %>%
#   rarefy_even_depth(sample.size = 3738, rngseed = 123) -> ps_tmp
#   
# # add Day_of_Treatment groups.
# ps_tmp %>% 
#   sample_data() %>% 
#   data.frame() %>% 
#   # mutate(Treatment = base::replace(Treatment, Experiment == "Cecum", "DONOR")) %>%
#   filter(Day_of_Treatment != -1) %>% # Excluded due to instability
#   filter(Model2 == "Human2") %>%
#     filter(Treatment != "DONOR") %>% 
#   filter(!Treatment %in%  c("CTX+HV292.1")) -> sample_data(ps_tmp)
# 
# # combine phyloseq tables and calculate relativ abundance across periods
# ps_tmp %>%
#   psmelt() %>%
#   select(OTU, Period, Abundance, Model, Treatment) %>%
#   group_by(Model, Treatment, OTU, Period) %>%
#   summarize(cum_abund = mean(Abundance)) %>%
#   pivot_wider(names_from = OTU, values_from = cum_abund) -> ps_merged
# 
# 
# # ps_merged$Treatment <- factor(ps_merged$Treatment, ordered = TRUE, levels = c("UNTREATED", "CTX+HV292.1", "CTX", "HV292.1", "VAN+CCUG59168", "VAN", "CCUG59168"))
# # split lists
# chicken_split <- split(ps_merged, ps_merged$Treatment)
# 
# # Get plot data
# plot_data <- lapply(1:length(unique(ps_merged$Treatment)),
#                     function(i) get_plot_data(chicken_split[[i]], cat_col))
# 
# 
# plots <- lapply(plot_data, `[[`, 1)
# 
# data_classification <- lapply(plot_data, `[[`, 2)
# names(data_classification) <- names(chicken_split)
# 
# # plot the data
# ggpubr::ggarrange(plotlist = plots, ncol=7, nrow=1, common.legend = TRUE, legend="bottom")
```


## Taxa over time:

### Heatmap:

#### chicken1:

```{r}
ps_rare %>% 
  subset_samples(Model2 == "Chicken1" & Treatment != "DONOR" & Day_of_Treatment >= 1) %>% 
  physeq_most_abundant(group_var = "Reactor_Treatment_Dose",
                       ntax = 12,
                       tax_level = "Strain") -> top_taxa_per_group_h1
```

```{r}
ps_rare %>%
  subset_samples(Model2 == "Chicken1" & Treatment != "DONOR" & Day_of_Treatment >= -1) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  subset_taxa(Strain %in% top_taxa_per_group_h1) %>% # extract only the taxa to display - after percentage normalisation
  phyloseq_ampvis_heatmap(physeq = ., tax_aggregate = "Species",
                          transform = FALSE,
                          plot_values = FALSE,
                          facet_by = c("Model2","Reactor_Treatment_Dose"),
                          group_by = "Day_of_Treatment",#"sample_name",
                          ntax = Inf
  ) -> treat_speces

treat_speces$data %>% 
  mutate(Abundance = na_if(Abundance, 0)) -> treat_speces$data 

treat_speces + 
  facet_grid(Model2 ~  Reactor_Treatment_Dose , scales = "free", space = "free") +
  scale_color_manual(values = c("#31a354", "orange", 
                                "#f03b20"), drop = FALSE, na.value = 'transparent') -> treat_speces
# scale_fill_viridis_c(breaks = c(0,  0.01, 1, 10, 25, 50, 75, 100),
# labels = c(0,  0.01, 1, 10, 25,  50, 75, 100),
# trans = scales::pseudo_log_trans(sigma = 0.9),
# na.value = 'transparent') -> pre_speces


treat_speces

# pre_speces %>%
#   export::graph2ppt(append = TRUE, width = 317.48031496 * 4  , height = 0.618 * 317.48031496 * 4 , paper = "A4",  scaling = 2,
#                     file = out_pptx)

```

```{r}
ps_rare %>%
  subset_samples(Model2 == "Chicken1" & Treatment != "DONOR" & Day_of_Treatment >= -1) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  phyloseq_ampvis_heatmap(physeq = ., tax_aggregate = "Family",
                          transform = FALSE,
                          plot_values = FALSE,
                          facet_by = c("Model2","Reactor_Treatment_Dose"),
                          group_by = "Day_of_Treatment",#"sample_name",
                          ntax = 10
  ) -> treat_fam

treat_fam$data %>% 
  mutate(Abundance = na_if(Abundance, 0)) -> treat_fam$data 

treat_fam + 
  facet_grid(Model2 ~  Reactor_Treatment_Dose , scales = "free", space = "free") +
  scale_color_manual(values = c("#31a354", "orange", 
                                "#f03b20"), drop = FALSE, na.value = 'transparent') -> treat_fam
# scale_fill_viridis_c(breaks = c(0,  0.01, 1, 10, 25, 50, 75, 100),
# labels = c(0,  0.01, 1, 10, 25,  50, 75, 100),
# trans = scales::pseudo_log_trans(sigma = 0.9),
# na.value = 'transparent') -> pre_speces


treat_fam


# pre_speces %>%
#   export::graph2ppt(append = TRUE, width = 317.48031496 * 4  , height = 0.618 * 317.48031496 * 4 , paper = "A4",  scaling = 2,
#                     file = out_pptx)

```

```{r}



ggpubr::ggarrange(treat_fam + theme(legend.position = "null") + 
                    theme(axis.title.x=element_blank(),
                          axis.text.x=element_blank(),
                          axis.ticks.x=element_blank()),
                  treat_speces + theme(
                    strip.background = element_blank(),
                    strip.text.x = element_blank()
                  )  + theme(legend.position = "null"),
                  align = "v",
                  ncol = 1, 
                  heights = c(1, 2.60),
                  common.legend = FALSE) -> human1_heat

human1_heat

human1_heat %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 4.75, 
                    height = 0.618 * 317.48031496 * 6.5 , paper = "A4",  scaling = 2,
                    file = out_pptx)



treat_fam$data %>% 
  select(Display, Sample, Abundance,Reactor_Treatment_Dose) %>% 
  pivot_wider(names_from = Display, values_from = Abundance) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "treat_family",
                   showNA = TRUE)
treat_speces$data %>% 
  select(Display, Sample, Abundance,Reactor_Treatment_Dose) %>% 
  pivot_wider(names_from = Display, values_from = Abundance) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "treat_ASV",
                   showNA = TRUE)
```

#### Chicken2:

```{r}
ps_rare %>% 
  subset_samples(Model2 == "Chicken2" & Treatment != "DONOR" & Day_of_Treatment >= -1) %>% 
  physeq_most_abundant(group_var = "Reactor_Treatment_Dose",
                       ntax = 14,
                       tax_level = "Strain") -> top_taxa_per_group_h2
```

```{r}
ps_rare %>%
  subset_samples(Model2 == "Chicken2" & Treatment != "DONOR" & Day_of_Treatment >= -1) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  subset_taxa(Strain %in% top_taxa_per_group_h2) %>% # extract only the taxa to display - after percentage normalisation
  phyloseq_ampvis_heatmap(physeq = ., tax_aggregate = "Species",
                          transform = FALSE,
                          plot_values = FALSE,
                          facet_by = c("Model2","Reactor_Treatment_Dose"),
                          group_by = "Day_of_Treatment",#"sample_name",
                          ntax = Inf
  ) -> treat_speces

treat_speces$data %>% 
  mutate(Abundance = na_if(Abundance, 0)) -> treat_speces$data 

treat_speces + 
  facet_grid(Model2 ~  Reactor_Treatment_Dose , scales = "free", space = "free") +
  scale_color_manual(values = c("#31a354", "orange", 
                                "#f03b20"), drop = FALSE, na.value = 'transparent') -> treat_speces
# scale_fill_viridis_c(breaks = c(0,  0.01, 1, 10, 25, 50, 75, 100),
# labels = c(0,  0.01, 1, 10, 25,  50, 75, 100),
# trans = scales::pseudo_log_trans(sigma = 0.9),
# na.value = 'transparent') -> pre_speces

treat_speces

# pre_speces %>%
#   export::graph2ppt(append = TRUE, width = 317.48031496 * 4  , height = 0.618 * 317.48031496 * 4 , paper = "A4",  scaling = 2,
#                     file = out_pptx)

```

```{r}
ps_rare %>%
  subset_samples(Model2 == "Chicken2" & Treatment != "DONOR" & Day_of_Treatment >= -1) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  phyloseq_ampvis_heatmap(physeq = ., tax_aggregate = "Family",
                          transform = FALSE,
                          plot_values = FALSE,
                          facet_by = c("Model2","Reactor_Treatment_Dose"),
                          group_by = "Day_of_Treatment",#"sample_name",
                          ntax = 10
  ) -> treat_fam

treat_fam$data %>% 
  mutate(Abundance = na_if(Abundance, 0)) -> treat_fam$data 

treat_fam + 
  facet_grid(Model2 ~  Reactor_Treatment_Dose , scales = "free", space = "free") +
  scale_color_manual(values = c("#31a354", "orange", 
                                "#f03b20"), drop = FALSE, na.value = 'transparent') -> treat_fam
# scale_fill_viridis_c(breaks = c(0,  0.01, 1, 10, 25, 50, 75, 100),
# labels = c(0,  0.01, 1, 10, 25,  50, 75, 100),
# trans = scales::pseudo_log_trans(sigma = 0.9),
# na.value = 'transparent') -> pre_speces

treat_fam

# pre_speces %>%
#   export::graph2ppt(append = TRUE, width = 317.48031496 * 4  , height = 0.618 * 317.48031496 * 4 , paper = "A4",  scaling = 2,
#                     file = out_pptx)

```

```{r}
ggpubr::ggarrange(treat_fam + theme(legend.position = "null") + 
                    theme(axis.title.x=element_blank(),
                          axis.text.x=element_blank(),
                          axis.ticks.x=element_blank()),
                  treat_speces + theme(
                    strip.background = element_blank(),
                    strip.text.x = element_blank()
                  )  + theme(legend.position = "null"),
                  align = "v",
                  ncol = 1, 
                  heights = c(1, 2.60),
                  common.legend = FALSE) -> human2_heat


human2_heat

human2_heat %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 4.75, 
                    height = 0.618 * 317.48031496 * 5.5 , paper = "A4",  scaling = 2,
                    file = out_pptx)




treat_fam$data %>% 
  select(Display, Sample, Abundance,Reactor_Treatment_Dose) %>% 
  pivot_wider(names_from = Display, values_from = Abundance) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "treat_family_h2",
                   showNA = TRUE)

treat_speces$data %>% 
  select(Display, Sample, Abundance,Reactor_Treatment_Dose) %>% 
  pivot_wider(names_from = Display, values_from = Abundance) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "treat_ASV_h2",
                   showNA = TRUE)

```

Could be improved:

-clustering taxa based on correlation -\> response (pheatmap?) <https://btep.ccr.cancer.gov/docs/data-visualization-with-r/Lesson5_intro_to_ggplot/#make-this-plot-compatible-with-ggplot2-using-the-package-ggplotify>

<https://jcoliver.github.io/learn-r/008-ggplot-dendrograms-and-heatmaps.html>

-additional layer depicting the % of the community covered by the taxa displayed.

### barplot:

#### chicken1:

```{r}
ps_rare %>% 
  subset_samples(Model2 == "Chicken1" & Treatment != "DONOR" & Day_of_Treatment >= -1) %>% 
  microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  subset_taxa(Strain %in% top_taxa_per_group_h1) %>% # extract only the taxa to display - after percentage normalisation
  microViz::phyloseq_validate() %>%
  microViz::tax_fix() %>% 
  microViz::ps_arrange(Day_of_Treatment) %>% 
  microViz::comp_barplot(
    tax_transform_for_plot = "identity",
    tax_level = "Strain",
    label = "Day_of_Treatment", # name an alternative variable to label axes
    n_taxa = length(top_taxa_per_group_h1), # give more taxa unique colours
    merge_other = FALSE, # split the "other" category to display alpha diversity
    bar_width = 0.9, # reduce the bar width to 70% of one row
    bar_outline_colour = "grey5",
    sample_order = "default") +
  ylab("Proportion - %") + xlab( " Day (Treatment) ") -> p_hist

p_hist + facet_grid(Phylum ~ Reactor_Treatment_Dose, scales = "free", space = "free_x") -> p_hist_human1_strain

p_hist_human1_strain  +  theme_light() + theme(legend.position = "none") -> p_hist_human1_strain_noleg

p_hist_human1_strain_noleg

p_hist_human1_strain %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_hist_human1_strain_leg
```

```{r}
ps_rare %>% 
  subset_samples(Model2 == "Chicken1" & Treatment != "DONOR" & Day_of_Treatment >= -1) %>% 
  microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  microViz::phyloseq_validate() %>%
  microViz::tax_fix() %>% 
  microViz::ps_arrange(Day_of_Treatment) %>% 
  microViz::comp_barplot(
    tax_transform_for_plot = "identity",
    tax_level = "Family",
    label = "Day_of_Treatment", # name an alternative variable to label axes
    n_taxa = 10, # give more taxa unique colours
    palette = microViz::distinct_palette(10, pal = "kelly"),
    merge_other = FALSE, # split the "other" category to display alpha diversity
    bar_width = 0.9, # reduce the bar width to 70% of one row
    bar_outline_colour = "grey5",
    sample_order = "default") +
  ylab("Proportion - %") + xlab( " Day (Treatment) ") -> p_hist

p_hist + facet_grid(Kingdom ~ Reactor_Treatment_Dose, scales = "free", space = "free_x", drop = TRUE) -> p_hist_human1_fam

p_hist_human1_fam  +  theme_light() + theme(legend.position = "none") -> p_hist_human1_fam_no_leg

p_hist_human1_fam_no_leg

p_hist_human1_fam %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_hist_human1_fam_leg
```

#### Chicken2:

```{r}
ps_rare %>% 
  subset_samples(Model2 == "Chicken2" & Treatment != "DONOR" & Day_of_Treatment >= -1) %>% 
  microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  subset_taxa(Strain %in% top_taxa_per_group_h2) %>% # extract only the taxa to display - after percentage normalisation
  microViz::phyloseq_validate() %>%
  microViz::tax_fix() %>% 
  microViz::ps_arrange(Day_of_Treatment) %>% 
  microViz::comp_barplot(
    tax_transform_for_plot = "identity",
    tax_level = "Strain",
    label = "Day_of_Treatment", # name an alternative variable to label axes
    n_taxa = length(top_taxa_per_group_h2), # give more taxa unique colours
    merge_other = FALSE, # split the "other" category to display alpha diversity
    bar_width = 0.9, # reduce the bar width to 70% of one row
    bar_outline_colour = "grey5",
    sample_order = "default") +
  ylab("Proportion - %") + xlab( " Day (Treatment) ") -> p_hist

p_hist + facet_grid(Phylum ~ Reactor_Treatment_Dose, scales = "free", space = "free_x") -> p_hist_human2_strain

p_hist_human2_strain +  theme_light()  + theme(legend.position = "none") -> p_hist_human2_strain_noleg

p_hist_human2_strain_noleg



p_hist_human2_strain %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_hist_human2_strain_leg
```

```{r}
ps_rare %>% 
  subset_samples(Model2 == "Chicken2" & Treatment != "DONOR" & Day_of_Treatment >= -1) %>% 
  microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  # subset_taxa(Strain %in% top_taxa_per_group_h2) %>% # extract only the taxa to display - after percentage normalisation
  microViz::phyloseq_validate() %>%
  microViz::tax_fix() %>% 
  microViz::ps_arrange(Day_of_Treatment) %>% 
  microViz::comp_barplot(
    tax_transform_for_plot = "identity",
    tax_level = "Family",
    label = "Day_of_Treatment", # name an alternative variable to label axes
    palette = microViz::distinct_palette(10, pal = "kelly"),
    n_taxa = 10, # give more taxa unique colours
    merge_other = FALSE, # split the "other" category to display alpha diversity
    bar_width = 0.9, # reduce the bar width to 70% of one row
    bar_outline_colour = "grey5",
    sample_order = "default") +
  ylab("Proportion - %") + xlab( " Day (Treatment) ") -> p_hist

p_hist + facet_grid(Kingdom ~ Reactor_Treatment_Dose, scales = "free", space = "free_x", drop = TRUE) -> p_hist_human2_fam

p_hist_human2_fam +  theme_light()  + theme(legend.position = "none") -> p_hist_human2_fam_no_leg

p_hist_human2_fam_no_leg

p_hist_human2_fam %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_hist_human2_fam_leg


# g <- ggplot_build(p_hist_human2_fam)
# 
# data.frame(colours = unique(g$data[[1]]["fill"]), 
#               label = levels(g$plot$data[, g$plot$labels$unique]))
```

Unify colors:

```{r}
c(p_hist_human2_fam$data$Family, p_hist_human1_fam$data$Family) %>% 
  unique()  -> families
```

```{r}
# families %>%
#   ggpubr::get_palette(k = ., palette = "kelly") -> col

families[!families %in% c("other")] %>% 
  length() %>% 
  microViz::distinct_palette(n = ., pal = "kelly", add = NA) -> col_fam

names(col_fam) <- families[!families %in% c("other")]

c(col_fam, "lightgrey") -> col_fam

names(col_fam) = NULL

names(col_fam) <- c(families[!families %in% c("other")], families[families %in% c("other")])

scales::show_col(col_fam)
```

```{r}
p_hist_human1_fam  +  theme_light() + scale_fill_manual(values = col_fam) ->  p_hist_human1_fam

p_hist_human1_fam + theme(legend.position = "none") -> p_hist_human1_fam_no_leg

p_hist_human1_fam_no_leg
```

```{r}
p_hist_human1_fam %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_hist_human1_fam_leg

p_hist_human1_fam_leg
```

```{r}
c(p_hist_human2_strain$data$Strain, p_hist_human1_strain$data$Strain) %>% 
  unique() -> ASVs

# ASVs
```

```{r}
# families %>%
#   ggpubr::get_palette(k = ., palette = "kelly") -> col
set.seed(1234)

ASVs[!ASVs %in% c("other")] %>% 
  length() %>% 
  randomcoloR::distinctColorPalette(k = ., altCol = FALSE, runTsne = TRUE) -> col_ASVs

names(col_ASVs) <- ASVs[!ASVs %in% c("other")]

c(col_ASVs, "lightgrey") -> col_ASVs

names(col_ASVs) = NULL

names(col_ASVs) <- c(ASVs[!ASVs %in% c("other")], ASVs[ASVs %in% c("other")])

scales::show_col(col_ASVs)
```

```{r}
p_hist_human1_strain + scale_fill_manual(values = col_ASVs) ->  p_hist_human1_strain

p_hist_human1_strain + theme_light() +
  theme(legend.position = "none") -> p_hist_human1_strain_no_leg

p_hist_human1_strain_no_leg
```

```{r}
p_hist_human1_strain %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_hist_human1_strain_leg

p_hist_human1_strain_leg
```

```{r}
p_hist_human2_strain + scale_fill_manual(values = col_ASVs) ->  p_hist_human2_strain

p_hist_human2_strain + theme_light() +theme(legend.position = "none") ->p_hist_human2_strain_no_leg

p_hist_human2_strain_no_leg
```

```{r}
p_hist_human2_strain %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_hist_human2_strain_leg

p_hist_human2_strain_leg
```

```{r}
ggpubr::ggarrange(p_hist_human1_fam_no_leg,
                  p_hist_human1_strain_no_leg ,
                  align = "v",
                  ncol = 1, 
                  heights = c(1, 1),
                  common.legend = FALSE) -> p_hist_human1_f1

p_hist_human1_f1

p_hist_human1_f1 %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 4.75, 
                    height = 0.618 * 317.48031496 * 6.5 , paper = "A4",  scaling = 2,
                    file = out_pptx)

p_hist_human1_fam_leg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 1 , paper = "A4",  scaling = 2,
                    file = out_pptx)

p_hist_human1_strain_leg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 5, 
                    height = 0.618 * 317.48031496 * 2.5 , paper = "A3",  scaling = 2,
                    file = out_pptx)
```

```{r}
ggpubr::ggarrange(p_hist_human2_fam_no_leg,
                  p_hist_human2_strain_noleg,
                  align = "v",
                  ncol = 1, 
                  heights = c(1, 1),
                  common.legend = FALSE) -> p_hist_human2

p_hist_human2

p_hist_human2 %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 4.75, 
                    height = 0.618 * 317.48031496 * 6.5 , paper = "A4",  scaling = 2,
                    file = out_pptx)

p_hist_human2_fam_leg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 1 , paper = "A4",  scaling = 2,
                    file = out_pptx)



p_hist_human2_strain_leg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 6, 
                    height = 0.618 * 317.48031496 * 2.5 , paper = "A3",  scaling = 2,
                    file = out_pptx)
```

### Alluvial:

```{r}
# p_hist_human2_fam$data %>%
#   filter(Model2 == "Human2") %>%
#   # microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
#   # mutate(combo = paste0(Model2, "_" , Reactor_F1_F2)) %>%
#   # mutate(combo = as.factor(combo)) %>%
#   select(SAMPLE, top,OTU, Abundance, Day_of_Treatment, Reactor_Treatment_Dose) %>%  distinct(top)
```

```{r}
# ggpubr::ggarrange(p_hist_human2_fam_no_leg,
#                   p_hist_human2_strain_noleg ,
#                   align = "v",
#                   ncol = 1, 
#                   heights = c(1, 1),
#                   common.legend = FALSE) -> p_hist_human2
# 
# p_hist_human2
# 
# p_hist_human2 %>% 
#   export::graph2ppt(append = TRUE, 
#                     width = 317.48031496 * 4.75, 
#                     height = 0.618 * 317.48031496 * 6.5 , paper = "A4",  scaling = 2,
#                     file = out_pptx)
```

#### human1:

```{r}
require(ggalluvial)
```

##### Family

```{r}
p_hist_human1_fam$data %>%
  # filter(Model2 == "Human2") %>%
  # microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  # mutate(combo = paste0(Model2, "_" , Reactor_F1_F2)) %>%
  # mutate(combo = as.factor(combo)) %>%
  select(SAMPLE, Family,OTU, Abundance, Day_of_Treatment, Reactor_Treatment_Dose) %>% 
  ggplot(.,
         aes(x = Day_of_Treatment, 
             y = Abundance,
             stratum = Family, 
             alluvium = OTU,
             fill = Family,
             label = Family))+
  facet_grid(Reactor_Treatment_Dose  ~ .)+
  # scale_fill_manual(values = c("red", "yellow", "green3"))+
  geom_flow(stat = "alluvium", lode.guidance = "frontback", color = "darkgray")+
  geom_stratum() +
  theme_light() +
  # geom_text(stat = "alluvium", aes(label = top), lode.guidance = "frontback") +
  scale_fill_manual(values = col_fam) +
  # scale_color_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  # scale_fill_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  ylab("Proportion") -> p_allu_human1

p_allu_human1 + theme(legend.position = "none") -> p_allu_human1_noleg 

p_allu_human1_noleg

p_allu_human1 %>% 
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() ->  p_allu_human1_leg
```

```{r}
p_allu_human1_noleg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 2 , paper = "A4",  scaling = 2,
                    file = out_pptx)

p_allu_human1_leg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 2 , paper = "A4",  scaling = 2,
                    file = out_pptx)
```

##### ASV:

```{r}
p_hist_human1_strain$data %>%
  # filter(Model2 == "Human2") %>%
  # microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  # mutate(combo = paste0(Model2, "_" , Reactor_F1_F2)) %>%
  # mutate(combo = as.factor(combo)) %>%
  select(SAMPLE, Strain,OTU, Abundance, Day_of_Treatment, Reactor_Treatment_Dose) %>% 
  ggplot(.,
         aes(x = Day_of_Treatment, 
             y = Abundance,
             stratum = Strain, 
             alluvium = OTU,
             fill = Strain,
             label = Strain))+
  facet_grid(Reactor_Treatment_Dose  ~ .)+
  # scale_fill_manual(values = c("red", "yellow", "green3"))+
  geom_flow(stat = "alluvium", lode.guidance = "frontback", color = "darkgray")+
  geom_stratum() +
  theme_light() +
  # geom_text(stat = "alluvium", aes(label = top), lode.guidance = "frontback") +
  scale_fill_manual(values = col_ASVs) +
  # scale_color_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  # scale_fill_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  ylab("Proportion") -> p_allu_human1_ASV

p_allu_human1_ASV + theme(legend.position = "none") -> p_allu_human1_ASV_noleg 

p_allu_human1_ASV_noleg

p_allu_human1_ASV %>% 
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() ->  p_allu_human1_ASV_leg
```

```{r}
p_allu_human1_ASV_noleg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 2 , paper = "A4",  scaling = 2,
                    file = out_pptx)
p_allu_human1_ASV_leg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 4, 
                    height = 0.618 * 317.48031496 * 2 , paper = "A3",  scaling = 2,
                    file = out_pptx)
```

#### human2

##### Family

```{r}

p_hist_human2_fam$data %>%
  # filter(Model2 == "Human2") %>%
  # microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  # mutate(combo = paste0(Model2, "_" , Reactor_F1_F2)) %>%
  # mutate(combo = as.factor(combo)) %>%
  select(SAMPLE, Family,OTU, Abundance, Day_of_Treatment, Reactor_Treatment_Dose) %>% 
  ggplot(.,
         aes(x = Day_of_Treatment, 
             y = Abundance,
             stratum = Family, 
             alluvium = OTU,
             fill = Family,
             label = Family))+
  facet_grid(Reactor_Treatment_Dose  ~ .)+
  # scale_fill_manual(values = c("red", "yellow", "green3"))+
  geom_flow(stat = "alluvium", lode.guidance = "frontback", color = "darkgray")+
  geom_stratum() +
  theme_light() +
  # geom_text(stat = "alluvium", aes(label = top), lode.guidance = "frontback") +
  scale_fill_manual(values = col_fam) +
  # scale_color_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  # scale_fill_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  ylab("Proportion") -> p_allu_human2

p_allu_human2 + theme(legend.position = "none") -> p_allu_human2_noleg 

p_allu_human2_noleg

```

```{r}
p_allu_human2 %>% 
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() ->  p_allu_human2_leg

p_allu_human2_leg
```

```{r}
p_allu_human2_noleg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 2 , paper = "A4",  scaling = 2,
                    file = out_pptx)

p_allu_human2_leg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 2 , paper = "A4",  scaling = 2,
                    file = out_pptx)
```

##### ASV:

```{r}
p_hist_human2_strain$data %>%
  # filter(Model2 == "Human2") %>%
  # microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  # mutate(combo = paste0(Model2, "_" , Reactor_F1_F2)) %>%
  # mutate(combo = as.factor(combo)) %>%
  select(SAMPLE, Strain,OTU, Abundance, Day_of_Treatment, Reactor_Treatment_Dose) %>% 
  ggplot(.,
         aes(x = Day_of_Treatment, 
             y = Abundance,
             stratum = Strain, 
             alluvium = OTU,
             fill = Strain,
             label = Strain))+
  facet_grid(Reactor_Treatment_Dose  ~ .)+
  # scale_fill_manual(values = c("red", "yellow", "green3"))+
  geom_flow(stat = "alluvium", lode.guidance = "frontback", color = "darkgray")+
  geom_stratum() +
  theme_light() +
  # geom_text(stat = "alluvium", aes(label = top), lode.guidance = "frontback") +
  scale_fill_manual(values = col_ASVs) +
  # scale_color_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  # scale_fill_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  ylab("Proportion") -> p_allu_human2_ASV

p_allu_human2_ASV + theme(legend.position = "none") -> p_allu_human2_ASV_noleg 

p_allu_human2_ASV_noleg

p_allu_human2_ASV %>% 
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() ->  p_allu_human2_ASV_leg
```

```{r}
p_allu_human2_ASV_noleg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 2 , paper = "A4",  scaling = 2,
                    file = out_pptx)

p_allu_human2_ASV_leg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 4, 
                    height = 0.618 * 317.48031496 * 2 , paper = "A3",  scaling = 2,
                    file = out_pptx)
```

## gram plots:

histogram of gram+ gram- & intrinsicly resistant or not an annotation color in a heatmap would be cool as well.

```{r}
"data/processed/16S/16S_based_gram.RDS" %>% 
  here::here() %>% 
  readRDS() -> gram

gram %>% 
  summarise_all(~ sum(is.na(.)))
```

```{r}
dim(gram)

intersect(gram$ASV,
          tax_table(ps_filtered)[,"Strain"] %>%  as.character() ) %>%  length() 
```

Let's take the Class info here.

## intrinsic:

```{r}
"data/processed/16S/16S_based_intrinsinc_AMR.RDS" %>% 
  here::here() %>% 
  readRDS() -> intrinsic_AMR
```

Wich level to chose? the one with less NA:

```{r}
intrinsic_AMR %>% 
  select(starts_with("intrinsic")) %>% 
  summary()
```

Let's take the Strain info here.

```{r}
ps_rare_gram_int <- ps_rare

# We want sample, reactor..., day_0f_treatment 
ps_rare_gram_int %>% 
  tax_table() %>% 
  data.frame() %>% 
  rownames_to_column("ASVid") %>% 
  left_join(gram %>% dplyr::rename("Strain" = "ASV") %>%  
              select(Strain , gram_neg_class)
  ) %>%  #,
  # by = c("Strain" = "ASV"),
  # suffix = c("", ".y")) %>% 
  select(ASVid:Strain, gram_neg_class) %>% 
  dplyr::rename('gram_stain' = 'gram_neg_class') %>% 
  mutate(gram_stain = replace_na(gram_stain, "unknown-gram")) %>% 
  left_join(., intrinsic_AMR,
            suffix = c("", ".y"),
            by = c("Strain" = "ASV")) %>%
  mutate(intrinsic = ifelse(intrinsic_strain_van == TRUE & intrinsic_strain_ctx == TRUE, "intrinsic-VAN-CTX",  ifelse(intrinsic_strain_van == TRUE, "intrinsic-VAN", ifelse(intrinsic_strain_ctx == TRUE, "intrinsic-CTX", "No-intrinsic")))) %>% 
  select(ASVid:gram_stain,intrinsic) %>%
  mutate(gram_intrinsic = paste0(gram_stain, 
                                 ifelse(is.na(intrinsic), "", paste0("_",intrinsic)))) %>% 
  # dplyr::rename('intrinsic_CTX' = 'intrinsic_strain_ctx',
  # 'intrinsic_VAN' = 'intrinsic_strain_van') %>% 
  column_to_rownames('ASVid') %>% 
  as.matrix() -> tax_table(ps_rare_gram_int)
```

### chicken2:

#### gram:

```{r}
ps_rare_gram_int %>% 
  subset_samples(Model2 == "Chicken2" & Treatment != "DONOR" & Day_of_Treatment >= -1) %>% 
  microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  physeq_sel_tax_table(c("gram_stain")) %>% 
  # subset_taxa(Strain %in% top_taxa_per_group_h2) %>% # extract only the taxa to display - after percentage normalisation
  microViz::phyloseq_validate() %>%
  # microViz::tax_fix() %>%
  microViz::ps_arrange(Day_of_Treatment) %>% 
  # microViz::tax_mutate(gram = left_join(tax_table(.) %>% data.frame(), gram,
  # by = c("Strain" = "ASV") %>% select(gram_neg_class))) %>% 
  microViz::comp_barplot(
    tax_transform_for_plot = "identity",
    tax_level = "gram_stain",
    label = "Day_of_Treatment", # name an alternative variable to label axes
    palette = microViz::distinct_palette(3, pal = "greenArmytage"),
    n_taxa = 10, # give more taxa unique colours
    merge_other = FALSE, # split the "other" category to display alpha diversity
    bar_width = 0.9, # reduce the bar width to 70% of one row
    bar_outline_colour = "grey5",
    sample_order = "default") +
  ylab("Proportion - %") + xlab( " Day (Treatment) ") -> p_gram_h2

p_gram_h2 + theme_light() + facet_grid(. ~ Reactor_Treatment_Dose, scales = "free", space = "free_x", drop = TRUE) -> p_gram_h2

p_gram_h2 + theme(legend.position = "none") -> p_gram_h2_no_leg

p_gram_h2_no_leg

p_gram_h2 %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_gram_h2_leg

p_gram_h2_leg
```

```{r}
p_gram_h2$data %>%
  # filter(Model2 == "Human2") %>%
  mutate(gram_stain = factor(gram_stain, levels = c("Gram-positive", "Gram-negative", "unknown-gram"))) %>%
  # mutate(combo = paste0(Model2, "_" , Reactor_F1_F2)) %>%
  # mutate(combo = as.factor(combo)) %>%
  select(SAMPLE, gram_stain,OTU, Abundance, Day_of_Treatment, Reactor_Treatment_Dose) %>% 
  ggplot(.,
         aes(x = Day_of_Treatment, 
             y = Abundance,
             stratum = gram_stain, 
             alluvium = OTU,
             fill = gram_stain,
             label = gram_stain))+
  facet_grid(Reactor_Treatment_Dose  ~ .)+
  # scale_fill_manual(values = c("red", "yellow", "green3"))+
  geom_flow(stat = "alluvium", lode.guidance = "frontback", color = "darkgray")+
  geom_stratum() +
  theme_light() +
  # geom_text(stat = "alluvium", aes(label = top), lode.guidance = "frontback") +
  # scale_fill_manual(values = col_ASVs) +
  scale_color_manual(values  =   microViz::distinct_palette(3, pal = "greenArmytage")) +
  scale_fill_manual(values  =   microViz::distinct_palette(3, pal = "greenArmytage")) +
  # scale_color_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  # scale_fill_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  ylab("Proportion - %") -> p_allu_human2_gram

p_allu_human2_gram + theme(legend.position = "none") -> p_allu_human2_gram_noleg 

p_allu_human2_gram_noleg 

p_allu_human2_gram %>% 
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() ->  p_allu_human2_gram_leg

p_allu_human2_gram_leg
```

#### intrinsic:

```{r}
# ps_rare_gram_int %>% 
#   subset_samples(Model2 == "Chicken1" & Treatment != "DONOR" & Day_of_Treatment >= -1) %>% 
#   microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
#   transform_sample_counts(function(x) x/sum(x) * 100) %>%
#   physeq_sel_tax_table(c("intrinsic")) %>% 
#   # microbiome::transform("log") %>% 
#   # speedyseq::tax_glom("intrinsic") %>% 
#   # physeq_sel_tax_table(c("gram_stain","intrinsic")) %>% 
#   # physeq_glom_rename(phyloseq = ., speedyseq = TRUE, rename_ASV = FALSE, taxrank = "gram_stain") %>% 
#   # subset_taxa(Strain %in% top_taxa_per_group_h2) %>% # extract only the taxa to display - after percentage normalisation
#   microViz::phyloseq_validate() %>%
#   microViz::tax_fix() %>%
#   microViz::ps_arrange(Day_of_Treatment) %>% 
#   # microViz::tax_mutate(gram = left_join(tax_table(.) %>% data.frame(), gram,
#   # by = c("Strain" = "ASV") %>% select(gram_neg_class))) %>% 
#   microViz::comp_barplot(
#     tax_transform_for_plot = "identity",
#     tax_level = "intrinsic",
#     label = "Day_of_Treatment", # name an alternative variable to label axes
#     # palette = microViz::distinct_palette(3, pal = "brewerPlus"),
#     n_taxa = 10, # give more taxa unique colours
#     merge_other = FALSE, # split the "other" category to display alpha diversity
#     bar_width = 0.9, # reduce the bar width to 70% of one row
#     bar_outline_colour = "grey5",
#     sample_order = "default") +
#   ylab("Proportion - %") + xlab( " Day (Treatment) ") -> p_int_h2
# 
# p_int_h2 + theme_light() +facet_grid(. ~ Reactor_Treatment_Dose, scales = "free", space = "free_x", drop = TRUE) -> p_int_h2
# 
# p_int_h2 %>% 
#   ggpubr::set_palette(p = ., palette = "jco") -> p_int_h2
# 
# p_int_h2 + theme(legend.position = "none") -> p_int_h2_no_leg
# 
# p_int_h2_no_leg
# 
# p_int_h2 %>%  
#   ggpubr::get_legend() %>% 
#   ggpubr::as_ggplot() -> p_int_h2_leg
# 
# p_int_h2_leg
```

```{r}
# p_int_h2$data %>%
#   # filter(Model2 == "Human2") %>%
#   # microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
#   # mutate(combo = paste0(Model2, "_" , Reactor_F1_F2)) %>%
#   # mutate(combo = as.factor(combo)) %>%
#   select(SAMPLE, top,OTU, Abundance, Day_of_Treatment, Reactor_Treatment_Dose) %>% 
#   ggplot(.,
#          aes(x = Day_of_Treatment, 
#              y = Abundance,
#              stratum = top, 
#              alluvium = OTU,
#              fill = top,
#              label = top))+
#   facet_grid(Reactor_Treatment_Dose  ~ .)+
#   # scale_fill_manual(values = c("red", "yellow", "green3"))+
#   geom_flow(stat = "alluvium", lode.guidance = "frontback", color = "darkgray")+
#   geom_stratum() +
#   theme_light() +
#   # geom_text(stat = "alluvium", aes(label = top), lode.guidance = "frontback") +
#   # scale_fill_manual(values = col_ASVs) +
#   scale_color_manual(values  =  microViz::distinct_palette(3, pal = "greenArmytage")) +
#   # scale_color_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
#   # scale_fill_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
#   ylab("Proportion - %") -> p_allu_human2_int
# 
# p_allu_human2_int %>% 
#   ggpubr::set_palette(p = ., palette = "jco") -> p_allu_human2_int
# 
# p_allu_human2_int + theme(legend.position = "none") -> p_allu_human2_int_noleg 
# 
# p_allu_human2_int_noleg 
# 
# p_allu_human2_int %>% 
#   ggpubr::get_legend() %>% 
#   ggpubr::as_ggplot() ->  p_allu_human2_int_leg
# ```
# ```{r}
# ggpubr::ggarrange(p_allu_human2_noleg + xlab(NULL),
#                   p_allu_human2_ASV_noleg + ylab(NULL) + xlab(NULL),
#                   p_allu_human2_int_noleg +  coord_cartesian(ylim = c(50,100)),
#                   p_allu_human2_int_noleg+ ylab(NULL),
#                   align = "v", 
#                   ncol = 2, nrow = 2,
#                   heights = c(1, 0.6), widths = c(1,1),
#                   common.legend = FALSE) -> p_allu_human2
# 
# p_allu_human2
# 
# p_allu_human2 %>%
#   export::graph2ppt(append = TRUE,
#                     width = 317.48031496 * 1,
#                     height = 0.618 * 317.48031496 * 1.75 , paper = "A4",  scaling = 2,
#                     file = out_pptx)
```


```{r}
# ggpubr::ggarrange(p_gram_h2_no_leg + ylab(NULL) + 
#                     theme(axis.title.x=element_blank(),
#                           axis.text.x=element_blank(),
#                           axis.ticks.x=element_blank()),
#                   p_hist_human2_fam_no_leg + 
#                     theme(
#                       strip.background = element_blank(),
#                       strip.text.x = element_blank()
#                     ) +
#                     theme(axis.title.x=element_blank(),
#                           axis.text.x=element_blank(),
#                           axis.ticks.x=element_blank()),
#                   p_hist_human2_strain_no_leg + 
#                     theme(
#                       strip.background = element_blank(),
#                       strip.text.x = element_blank()
#                     ) + ylab(NULL) + 
#                     theme(axis.title.x=element_blank(),
#                           axis.text.x=element_blank(),
#                           axis.ticks.x=element_blank()),
#                   p_int_h2_no_leg + ylab(NULL) + coord_cartesian(ylim = c(50,100)) +
#                     theme(
#                       strip.background = element_blank(),
#                       strip.text.x = element_blank()
#                     ), 
#                   align = "v",
#                   ncol = 1, 
#                   heights = c(0.15, 0.4, 0.8,0.15),
#                   common.legend = FALSE) -> all_plot_h2
# 
# all_plot_h2
# 
# all_plot_h2 %>% 
#   export::graph2ppt(append = TRUE, 
#                     width = 317.48031496 * 3.5, 
#                     height = 0.618 * 317.48031496 * 5 , paper = "A4",  scaling = 2,
#                     file = out_pptx)
# 
# 
# p_gram_h2_leg  %>% 
#   export::graph2ppt(append = TRUE, 
#                     width = 317.48031496 * 1, 
#                     height = 0.618 * 317.48031496 * 1 , paper = "A4",  scaling = 2,
#                     file = out_pptx)

p_hist_human2_fam_leg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 1 , paper = "A4",  scaling = 2,
                    file = out_pptx)

p_hist_human2_strain_leg  %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 4, 
                    height = 0.618 * 317.48031496 * 2 , paper = "A3",  scaling = 2,
                    file = out_pptx)
# 
# p_int_h2_leg  %>% 
#   export::graph2ppt(append = TRUE, 
#                     width = 317.48031496 * 1, 
#                     height = 0.618 * 317.48031496 * 1 , paper = "A4",  scaling = 2,
#                     file = out_pptx)
# 
# 
# p_gram_h2_no_leg$data %>% 
#   select(OTU, Sample, Abundance, Treatment, Day_of_Treatment) %>% 
#   xlsx::write.xlsx(file = out_xlsx,
#                    append = TRUE,
#                    sheetName = "p_gram_h2",
#                    showNA = TRUE)
# p_hist_human2_fam$data %>% 
#   select(OTU, Sample, Abundance, Treatment, Day_of_Treatment) %>% 
#   xlsx::write.xlsx(file = out_xlsx,
#                    append = TRUE,
#                    sheetName = "p_hist_human2_fam",
#                    showNA = TRUE)
# p_hist_human2_strain$data %>% 
#   select(Sample, top ,Abundance, Treatment, Day_of_Treatment) %>% 
#   pivot_wider(names_from = top, values_from = Abundance) %>% 
#   xlsx::write.xlsx(file = out_xlsx,
#                    append = TRUE,
#                    sheetName = "p_hist_human2_strain",
#                    showNA = TRUE)
# p_int_h2$data %>% 
#   select(OTU, Sample, Abundance, Treatment, Day_of_Treatment) %>% 
#   xlsx::write.xlsx(file = out_xlsx,
#                    append = TRUE,
#                    sheetName = "p_int_h2",
#                    showNA = TRUE)

```


```{r}
# ggpubr::ggarrange(p_gram_h2_no_leg + ylab(NULL) + 
#                     theme(axis.title.x=element_blank(),
#                           axis.text.x=element_blank(),
#                           axis.ticks.x=element_blank()),
#                   p_hist_human2_fam_no_leg + 
#                     theme(
#                       strip.background = element_blank(),
#                       strip.text.x = element_blank()
#                     ) +
#                     theme(axis.title.x=element_blank(),
#                           axis.text.x=element_blank(),
#                           axis.ticks.x=element_blank()),
#                   treat_speces + 
#                     theme(
#                       strip.background = element_blank(),
#                       strip.text.x = element_blank()
#                     ) + ylab(NULL) + 
#                     theme(axis.title.x=element_blank(),
#                           axis.text.x=element_blank(),
#                           axis.ticks.x=element_blank()),
#                   p_int_h2_no_leg + ylab(NULL) + coord_cartesian(ylim = c(50,100)) +
#                     theme(
#                       strip.background = element_blank(),
#                       strip.text.x = element_blank()
#                     ), 
#                   align = "hv",
#                   ncol = 1, 
#                   heights = c(0.15, 0.4, 0.8,0.15),
#                   common.legend = FALSE) -> all_plot_h2
# 
# all_plot_h2
# 
# all_plot_h2 %>% 
#   export::graph2ppt(append = TRUE, 
#                     width = 317.48031496 * 3.5, 
#                     height = 0.618 * 317.48031496 * 5 , paper = "A4",  scaling = 2,
#                     file = out_pptx)

```



### chicken1 :

#### gram:

```{r}
ps_rare_gram_int %>% 
  subset_samples(Model2 == "Chicken1" & Treatment != "DONOR" & Day_of_Treatment >= -1) %>% 
  # microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  physeq_sel_tax_table(c("gram_stain")) %>% 
  # subset_taxa(Strain %in% top_taxa_per_group_h2) %>% # extract only the taxa to display - after percentage normalisation
  microViz::phyloseq_validate() %>%
  microViz::tax_fix() %>%
  microViz::ps_arrange(Day_of_Treatment) %>% 
  # microViz::tax_mutate(gram = left_join(tax_table(.) %>% data.frame(), gram,
  # by = c("Strain" = "ASV") %>% select(gram_neg_class))) %>% 
  microViz::comp_barplot(
    tax_transform_for_plot = "identity",
    tax_level = "gram_stain",
    label = "Day_of_Treatment", # name an alternative variable to label axes
    palette = microViz::distinct_palette(2, pal = "greenArmytage"),
    n_taxa = 10, # give more taxa unique colours
    merge_other = FALSE, # split the "other" category to display alpha diversity
    bar_width = 0.9, # reduce the bar width to 70% of one row
    bar_outline_colour = "grey5",
    sample_order = "default") +
  ylab("Proportion - %") + xlab( " Day (Treatment) ") -> p_gram_h1

p_gram_h1 + facet_grid(. ~ Reactor_Treatment_Dose, scales = "free", space = "free_x", drop = TRUE) -> p_gram_h1

p_gram_h1 + theme_light() +theme(legend.position = "none") -> p_gram_h1_no_leg

p_gram_h1_no_leg

p_gram_h1 %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_gram_h1_leg

```


```{r}
p_gram_h1_f1$data %>%
  # filter(Model2 == "Human2") %>%
  mutate(gram_stain = factor(gram_stain, levels = c("Gram-positive", "Gram-negative", "unknown-gram"))) %>%
  # mutate(combo = paste0(Model2, "_" , Reactor_F1_F2)) %>%
  # mutate(combo = as.factor(combo)) %>%
  select(SAMPLE, gram_stain,OTU, Abundance, Day_of_Treatment, Reactor_Treatment_Dose) %>% 
  ggplot(.,
         aes(x = Day_of_Treatment, 
             y = Abundance,
             stratum = gram_stain, 
             alluvium = OTU,
             fill = gram_stain,
             label = gram_stain))+
  facet_grid(Reactor_Treatment_Dose  ~ .)+
  # scale_fill_manual(values = c("red", "yellow", "green3"))+
  geom_flow(stat = "alluvium", lode.guidance = "frontback", color = "darkgray")+
  geom_stratum() +
  theme_light() +
  # geom_text(stat = "alluvium", aes(label = top), lode.guidance = "frontback") +
  # scale_fill_manual(values = col_ASVs) +
  scale_color_manual(values  =   microViz::distinct_palette(3, pal = "greenArmytage")) +
  scale_fill_manual(values  =   microViz::distinct_palette(3, pal = "greenArmytage")) +
  # scale_color_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  # scale_fill_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  ylab("Proportion - %") -> p_allu_h1_gram 

p_allu_h1_gram + theme(legend.position = "none") -> p_allu_h1_gram_noleg 

p_allu_h1_gram_noleg 

p_allu_h1_gram %>% 
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() ->  p_allu_h1_gram_leg

p_allu_h1_gram_leg
```



#### intrinsic:

```{r}
# ps_rare_gram_int %>% 
#   subset_samples(Model2 == "Chicken1" & Treatment != "DONOR" & Day_of_Treatment >= -1) %>% 
#   microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
#   transform_sample_counts(function(x) x/sum(x) * 100) %>%
#   physeq_sel_tax_table(c( "intrinsic")) %>% 
#   # microbiome::transform("log") %>% 
#   # speedyseq::tax_glom("intrinsic") %>% 
#   # physeq_sel_tax_table(c("gram_stain","intrinsic")) %>% 
#   # physeq_glom_rename(phyloseq = ., speedyseq = TRUE, rename_ASV = FALSE, taxrank = "gram_stain") %>% 
#   # subset_taxa(Strain %in% top_taxa_per_group_h2) %>% # extract only the taxa to display - after percentage normalisation
#   microViz::phyloseq_validate() %>%
#   microViz::tax_fix() %>%
#   microViz::ps_arrange(Day_of_Treatment) %>% 
#   # microViz::tax_mutate(gram = left_join(tax_table(.) %>% data.frame(), gram,
#   # by = c("Strain" = "ASV") %>% select(gram_neg_class))) %>% 
#   microViz::comp_barplot(
#     tax_transform_for_plot = "identity",
#     tax_level = "intrinsic",
#     label = "Day_of_Treatment", # name an alternative variable to label axes
#     # palette = microViz::distinct_palette(3, pal = "brewerPlus"),
#     n_taxa = 10, # give more taxa unique colours
#     merge_other = FALSE, # split the "other" category to display alpha diversity
#     bar_width = 0.9, # reduce the bar width to 70% of one row
#     bar_outline_colour = "grey5",
#     sample_order = "default") +
#   ylab("Proportion - %") + xlab( " Day (Treatment) ") -> p_int_h1_f1
# 
# p_int_h1_f1 + facet_grid(Kingdom ~ Reactor_Treatment_Dose, scales = "free", space = "free_x", drop = TRUE) -> p_int_h1_f1
# 
# p_int_h1_f1 %>% 
#   ggpubr::set_palette(p = ., palette = "jco") -> p_int_h1_f1
# 
# p_int_h1_f1 + theme_light() +theme(legend.position = "none") -> p_int_h1_f1_no_leg
# 
# p_int_h1_f1_no_leg
# 
# p_int_h1_f1 %>%  
#   ggpubr::get_legend() %>% 
#   ggpubr::as_ggplot() -> p_int_h1_f1_leg
# 
# p_int_h1_f1_leg
```


```{r}
# p_int_h1_f1$data %>%
#   # filter(Model2 == "Human2") %>%
#   # microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
#   # mutate(combo = paste0(Model2, "_" , Reactor_F1_F2)) %>%
#   # mutate(combo = as.factor(combo)) %>%
#   select(SAMPLE, top,OTU, Abundance, Day_of_Treatment, Reactor_Treatment_Dose) %>% 
#   ggplot(.,
#          aes(x = Day_of_Treatment, 
#              y = Abundance,
#              stratum = top, 
#              alluvium = OTU,
#              fill = top,
#              label = top))+
#   facet_grid(Reactor_Treatment_Dose  ~ .)+
#   # scale_fill_manual(values = c("red", "yellow", "green3"))+
#   geom_flow(stat = "alluvium", lode.guidance = "frontback", color = "darkgray")+
#   geom_stratum() +
#   theme_light() +
#   # geom_text(stat = "alluvium", aes(label = top), lode.guidance = "frontback") +
#   # scale_fill_manual(values = col_ASVs) +
#   scale_color_manual(values  =  microViz::distinct_palette(3, pal = "greenArmytage")) +
#   # scale_color_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
#   # scale_fill_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
#   ylab("Proportion - %") -> p_allu_h1_f1_int
# 
# p_allu_h1_f1_int %>% 
#   ggpubr::set_palette(p = ., palette = "jco") -> p_allu_h1_f1_int
# 
# p_allu_h1_f1_int + theme(legend.position = "none") -> p_allu_h1_f1_int_noleg 
# 
# p_allu_h1_f1_int_noleg 
# 
# p_allu_h1_f1_int %>% 
#   ggpubr::get_legend() %>% 
#   ggpubr::as_ggplot() -> p_allu_h1_f1_int_leg
```

```{r}
ggpubr::ggarrange(p_allu_human1_f1_noleg + xlab(NULL),
                  p_allu_human1_ASV_f1_noleg + ylab(NULL) + xlab(NULL),
                  p_allu_h1_f1_int_noleg +  coord_cartesian(ylim = c(50,100)),
                  p_allu_h1_f1_gram_noleg + ylab(NULL),
                  align = "v", 
                  ncol = 2, nrow = 2,
                  heights = c(1, 0.6), widths = c(1,1),
                  common.legend = FALSE) -> p_allu_human1_f1

p_allu_human1_f1

p_allu_human1_f1 %>%
  export::graph2ppt(append = TRUE,
                    width = 317.48031496 * 1,
                    height = 0.618 * 317.48031496 * 1.75 , paper = "A4",  scaling = 2,
                    file = out_pptx)
```

```{r}
ggpubr::ggarrange(p_gram_h1_f1_no_leg + ylab(NULL) + 
                    theme(axis.title.x=element_blank(),
                          axis.text.x=element_blank(),
                          axis.ticks.x=element_blank()),
                  p_hist_human1_f1_fam_no_leg + 
                    theme(
                      strip.background = element_blank(),
                      strip.text.x = element_blank()
                    ) +
                    theme(axis.title.x=element_blank(),
                          axis.text.x=element_blank(),
                          axis.ticks.x=element_blank()),
                  p_hist_human1_f1_strain_no_leg + 
                    theme(
                      strip.background = element_blank(),
                      strip.text.x = element_blank()
                    ) + ylab(NULL) + 
                    theme(axis.title.x=element_blank(),
                          axis.text.x=element_blank(),
                          axis.ticks.x=element_blank()),
                  p_int_h1_f1_no_leg + ylab(NULL) + coord_cartesian(ylim = c(50,100)) +
                    theme(
                      strip.background = element_blank(),
                      strip.text.x = element_blank()
                    ), 
                  align = "v",
                  ncol = 1, 
                  heights = c(0.15, 0.4, 0.8,0.15),
                  common.legend = FALSE) -> all_plot_h1_f1

all_plot_h1_f1

all_plot_h1_f1 %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 3.5, 
                    height = 0.618 * 317.48031496 * 5 , paper = "A4",  scaling = 2,
                    file = out_pptx)


p_gram_h1_f1_leg  %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 1 , paper = "A4",  scaling = 2,
                    file = out_pptx)

p_hist_human1_f1_fam_leg  %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 1 , paper = "A4",  scaling = 2,
                    file = out_pptx)

p_hist_human1_f1_strain_leg  %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 4, 
                    height = 0.618 * 317.48031496 * 2 , paper = "A3",  scaling = 2,
                    file = out_pptx)

p_int_h1_f1_leg  %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 1 , paper = "A4",  scaling = 2,
                    file = out_pptx)



p_gram_h1_f1$data %>% 
  select(OTU, Sample, Abundance, Treatment, Day_of_Treatment) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "p_gram_h1_f1",
                   showNA = TRUE)
p_hist_human1_f1_fam$data %>% 
  select(OTU, Sample, Abundance, Treatment, Day_of_Treatment) %>% 
  pivot_wider(names_from = OTU, values_from = Abundance) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "p_hist_human1_f1_fam",
                   showNA = TRUE)
p_hist_human1_f1_strain$data %>% 
  select(Sample, top ,Abundance, Treatment, Day_of_Treatment) %>% 
  pivot_wider(names_from = top, values_from = Abundance) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "p_hist_human1_f1_strain",
                   showNA = TRUE)
p_int_h1_f1$data %>% 
  select(OTU, Sample, Abundance, Treatment, Day_of_Treatment) %>% 
  pivot_wider(names_from = OTU, values_from = Abundance) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "p_int_h1_f1",
                   showNA = TRUE)

```


-   explore stability metrics: <https://microbiome.github.io/tutorials/Stability.html> to detect taxa impacted. or filter by coefficient variation. or Hannah's classification

Bimodal population distribution across individuals is often associated with instable intermediate abundances within individuals:

```{r,  eval = FALSE}
# Use relative abundances
pseq <- microbiome::transform(ps_rare %>%  subset_samples(Model2 == "Human2"), "compositional")

# Merge rare taxa to speed up examples
pseq <- microbiome::aggregate_rare(ps_rare, level = "Strain", detection = .1/100, prevalence = 10/100)

# For cross-sectional analysis, include
# only the treated points.

pseq0 <- subset_samples(pseq, Day_of_Treatment > 1)

# intermediate_stability: within subjects:
pseq %>% 
  microViz::ps_mutate(subject = Reactor,
                      time = Day_of_Treatment) %>% 
  intermediate_stability(., #output = "scores", 
                         output = "scores") -> intermediate.stability

intermediate.stability %>% 
  head()
```

```{r, message = FALSE, warning = FALSE, eval = FALSE}
# Bimodality is better estimated from abundances at log scale (such as CLR)
pseq0.clr <- microbiome::transform(pseq0, "clr")

set.seed(4433)

# bimodality: between subjects
bimodality.score <- bimodality(pseq0.clr, method = "potential_analysis",
                               bs.iter = 999, peak.threshold = 5,
                               min.density = 5)

bimodality.score %>% 
  head()
```

```{r, eval=FALSE}
taxa <- taxa(pseq0)
df <- data.frame(group = taxa,
                 intermediate.stability = intermediate.stability[taxa],
                 bimodality = bimodality.score[taxa])
theme_set(theme_bw(20))

library(ggrepel)

p <- ggplot(df,
            aes(x = intermediate.stability, y = bimodality, label = '')) + #Doesn't add labels to points
  geom_text_repel() +
  geom_point()

#Labels only those that have bimodality > 0.4 and intermediate stability < 0, adjusts the placement of labels
p <- p + geom_text(aes(label = ifelse(bimodality > 0.75 | intermediate.stability < -0.25, group, '')), vjust = "inward", hjust = "inward")

print(p)
```

```{r, eval=FALSE}
p$data
```

```{r, eval=FALSE}
# Log10 abundance for a selected taxonomic group
# Pick the most bimodal taxa as an example
tax  <- names(which.max(bimodality.score))

# Detect tipping points at log10 abundances
x <- abundances(microbiome::transform(pseq, "clr"))[tax,]

# Bootstrapped potential analysis to identify potential minima
# in practice, use more bootstrap iterations
potential.minima <- potential_analysis(x, bs.iter = 10)$minima

# Same with earlywarnings package (without bootstrap ie. less robust)
# library(earlywarnings)
# res <- livpotential_ews(x)$min.points

# Identify the potential minimum locations as tipping point candidates
tipping.samples <- sapply(potential.minima, function (m) {names(which.min(abs(sort(x) - m)))})
tipping.point <- abundances(pseq)[tax, tipping.samples]
print(tipping.point)
```


```{r, eval=FALSE}
# Illustrate the detected tipping point
plot(density(x), main = tax)
abline(v = x[tipping.samples])
```


```{r}
# # Bimodality hotplot:
# # Consider a unique sample from each subject: the baseline time point 
# p <- microbiome::hotplot(pseq0, tax, tipping.point = 0.005)
# print(p)
# 
# # Visualize bimodality
# pv <- plot_tipping(pseq, tax, tipping.point = 0.005)
# print(pv)
```

```{r}
sessionInfo()
```
