---
title: "NRP72 -metadata - humann - chiken "
author: "Florentin Constancias"
date: "`r format(Sys.time(), '%B %d, %Y')`"
output: 
  html_document: 
    toc: yes
    keep_md: yes

---

```{r setup, include=FALSE}
rm(list = ls())
knitr::opts_chunk$set(echo = TRUE)
```

#### Load required packages

```{r,packages,message=FALSE}
library(tidyverse)
library(phyloseq)
library(microbiome)
library(here)
options(getClass.msg=FALSE) # https://github.com/epurdom/clusterExperiment/issues/66
#this fixes an error message that pops up because the class 'Annotated' is defined in two different packages
source("https://raw.githubusercontent.com/fconstancias/metabarcodingRpipeline/dev/scripts/functions_export_simplified.R")
'%!in%' <- function(x,y)!('%in%'(x,y))

```


# Import phyloseq object

```{r}
ps = "data/processed/16S/1/ps_silva_dada2_human_chicken_meta.RDS"


ps %>% 
  here::here() %>%
  readRDS()  -> physeq

# physeq$physeq -> physeq

physeq

```

```{r}
physeq %>% 
  sample_names() %>% 
  sort() %>% 
  head()

```


```{r}
# here::here("data/raw/metabarcoding/merged_chicken_human_04.08.2021.tsv") %>%
here::here("data/raw/25.11.2021_metadata_updated.tsv") %>%
  readr::read_tsv() %>% 
  pull("sample") %>% 
  sort() %>% 
  head()

```

```{r, eval=TRUE}

physeq %>%
  physeq_add_metadata(physeq = .,
                      metadata = here::here("data/raw/25.11.2021_metadata_updated.tsv") %>%
                        readr::read_tsv(), sample_column = "sample") -> physeq_meta




physeq;physeq_meta

```



```{r, eval=TRUE}

physeq %>%
  physeq_add_metadata(physeq = .,
                      metadata = here::here("data/raw/25.11.2021_metadata_updated.tsv") %>%
                        readr::read_tsv() %>% 
                        mutate(Reactor_Treatment_Dose = ifelse(!is.na(`Antibiotic_mg/mL`),
                                                               paste0(Reactor_Treatment, `Antibiotic_mg/mL`), Reactor_Treatment),
                               Treatment_Dose = ifelse(!is.na(`Antibiotic_mg/mL`),
                                                       paste0(Treatment, `Antibiotic_mg/mL`), Reactor_Treatment))
                      , sample_column = "sample") -> physeq_meta


# 
# meta %>% 
#     mutate(Reactor_Treatment_Dose = ifelse(!is.na(`Antibiotic_mg/mL`), 
#        paste0(Reactor_Treatment, `Antibiotic_mg/mL`), Reactor_Treatment)
#     Treatment_Dose = ifelse(!is.na(`Antibiotic_mg/mL`), 
#        paste0(Treatment, `Antibiotic_mg/mL`), Reactor_Treatment))

# ifelse(!is.na(sample_data(physeq_meta)$Antibiotic_mg.mL), 
#        sample_data(physeq_meta)$Antibiotic_mg.mL, "") %>% 
# mutate(gradebook, Pass.Fail = ifelse(grade > 60, "Pass", "Fail"))



physeq;physeq_meta
```



600 samples initially and 471 with metadata ????

```{r}
intersect(
  sample_names(physeq_meta),
  sample_names(physeq)) %>% 
  length()
```

```{r}
difference <- function(x, y) {
  c(setdiff(x, y), setdiff(y, x))
}

difference(
  sample_names(physeq),
  sample_names(physeq_meta))
```

Above missing from the metadata but present in the initiall phyloseq object.

Update factor levels:

```{r}
physeq_meta %>% 
  sample_data() %>% 
  data.frame() %>% 
  select(-input:-length_filtered_pc,-I7_Index_ID:-index2) %>% 
  mutate(Treatment2 = factor(Treatment2, levels = c("DONOR", "UNTREATED", "AB+E. coli", "AB", "E. coli", "AB+E. faecium", "E. faecium")),
         Reactor_Treatment = base::replace(Reactor_Treatment, Reactor_Treatment == "TR_2CTX", "TR2_CTX"),
         Phase = factor(Phase, levels = c("Stab", "Treat")),
         Model = factor(Model, levels = c("Chicken", "Human")),
         Fermentation = factor(Fermentation, levels = c("1", "2")),
         Period = case_when( # if chicken we have period if human
           Day_of_Treatment == -2 | Day_of_Treatment == -3 | Day_of_Treatment == -6 | Day_of_Treatment == -7 ~ "pret",
           Day_of_Treatment < 10 & Day_of_Treatment >= 0 ~ "t1",
           Day_of_Treatment < 20 & Day_of_Treatment >= 10 ~ "t2",
           Day_of_Treatment < 30 & Day_of_Treatment >= 20 ~ "t3",
           Day_of_Treatment < 40 & Day_of_Treatment >= 30 ~ "t4",
           Day_of_Treatment < 50 & Day_of_Treatment >= 40 ~ "t5"),
         Period = factor(Period, levels = c("pret", "t1", "t2", "t3", "t4", "t5")),
         Reactor_Treatment_Dose = if_else(!is.na(`Antibiotic_mg.mL`), paste0(Reactor_Treatment, `Antibiotic_mg.mL`), Reactor_Treatment),
         Treatment_Dose = if_else(!is.na(`Antibiotic_mg.mL`), paste0(Treatment, `Antibiotic_mg.mL`), Treatment),
         Treatment = base::replace(Treatment, Experiment == "Cecum", "DONOR"),
         Treatment = factor(Treatment, levels = c("negative","positive","STRAIN", "DONOR", "UNTREATED", "CTX", "CTX+HV292.1", "HV292.1", "VAN", "VAN+CCUG59168", "CCUG59168")),
         # Reactor = factor(Reactor, levels = c("negative","positive","STRAIN", "DONOR", "IR1", "IR2", "CR", "TR1", "TR2", "TR3", "TR4", "TR5", "TR6","Batch3","Batch4")),
         Experiment = factor(Experiment, levels = c( "NTC","Mock", "HV292.1", "CCUG59168", "Cecum","Continuous","Batch")),
         
         # mutate( Treatment  = case_when(Experiment == "Cecum" ~ "DONOR",
         #                               TRUE ~ Treatment)) %>% 
         # mutate(Treatment = ifelse(Experiment == "Cecum", "Donor", Treatment)) %>% 
         
  ) -> sample_data(physeq_meta)
# mutate(Reactor_Treatment = base::replace(Reactor_Treatment, Reactor_Treatment == "TR_2CTX", "TR2_CTX")) %>%
#   mutate(Reactor_Treatment_Dose = if_else(!is.na(`Antibiotic_mg.mL`), paste0(Reactor_Treatment, `Antibiotic_mg.mL`), Reactor_Treatment)) %>% 
#   mutate(Treatment_Dose = if_else(!is.na(`Antibiotic_mg.mL`), paste0(Treatment, `Antibiotic_mg.mL`), Treatment)) %>%
#   mutate(
#     Period = case_when(
#       Day_of_Treatment <= 0  ~ "pret",
#       Day_of_Treatment > 0 ~ as.character(Day_of_Treatment)
#     )) %>%
#   mutate(Period = base::replace(Period, Treatment == "DONOR", "pret")) %>%
#   mutate(Day_of_Treatment = addNA(Day_of_Treatment)) %>%
```

```{r}
# physeq_meta %>% 
#   add_phylogeny_to_phyloseq(export = FALSE) %>% 
#   saveRDS(here::here("data/raw/metabarcoding/ps_silva_dada2_humanonly_phylo_meta.RDS"))
```

```{r}
physeq_meta %>%
  saveRDS(here::here("data/processed/16S/1/ps_silva_dada2_human_chicken_meta_fact.RDS"))
```

```{r}
sessionInfo()
```

