---
title: "NTP72 - 16S - taxa compostiion - human draft "
author: "Florentin Constancias"
date: "`r format(Sys.time(), '%B %d, %Y')`"
output: 
  html_document: 
    toc: yes
    keep_md: yes
---

```{r setup, include=FALSE}
rm(list = ls())
gc()
options(java.parameters = "-Xmx80000m")
```

```{r setup2, include=FALSE}

##knitr::opts_chunk$set(cache=TRUE)
##knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(dev = "ragg_png") # https://ragg.r-lib.org/index.html
##knitr::opts_chunk$set(fig.width = 6) #https://r4ds.had.co.nz/graphics-for-communication.html#figure-sizing
##knitr::opts_chunk$set(fig.asp = 0.618)
##knitr::opts_chunk$set(fig.show = "hold")
##knitr::opts_chunk$set(fig.show = "70%")
##knitr::opts_chunk$set(fig.align = "center")
```

```{r packages, results=FALSE, warning=FALSE}
#install.packages("tidyverse")
require(tidyverse); packageVersion("tidyverse")
require(phyloseq); packageVersion("phyloseq")
```

```{r sourcing, warning=FALSE, results=FALSE , message= FALSE}
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_taxa_tests.R") 
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_alpha.R") 
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_beta.R") 
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_heatmap.R")
source("https://raw.githubusercontent.com/fconstancias/metabaRpipe-source/master/Rscripts/functions.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_normalisation.R") 
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_varia.R")


'%!in%' <- function(x,y)!('%in%'(x,y))

```


```{r}
out_pptx = "~/Desktop/16S-human.pptx"
out_xlsx = "~/Desktop/16S-human.xlsx"
```

```{r}
load(url("https://github.com/fconstancias/NRP72-FBT/blob/master/Figures/Rastetics.Rdata?raw=true"))
```

## load data:

```{r}
"data/processed/16S/16S_working_phyloseq.RDS" %>% 
  here::here() %>% 
  readRDS() -> ps_filtered
```

```{r}
ps_filtered %>% 
  phyloseq_check_lib_size(data_color = "Reactor", data_facet = "Reactor", first_n = 50, nreads_display = 1000) -> out

out$df %>% 
  # rownames_to_column('sample_id') %>% 
  # filter(Treatment == "DONOR")  %>% 
  select(SampleID, Day_of_Treatment,Treatment, Reactor, Model2, Model, LibrarySize) 
```

## rarefraction

```{r}
min_sample_size = 3820

ps_filtered %>%
  rarefy_even_depth(sample.size = min_sample_size, rngseed = 123)  -> ps_rare

ps_filtered %>%
  prune_samples(sample_sums(.)>= min_sample_size, .) -> ps_fil
```

Objectives:

-   compare donor microbiota and pre-treatment / CR - heatmap: v
-   evaluate effect treatment: -- alpha (shanon, simpson --\> hill d1, d2) -- beta (achinson, log -\> bray)
-   describe effect on taxa (Family, Genus, ASV): v -- heatmap v -- alluvial plot v

--\> Ezequiel: process - co-selection of AMR.

## Before Treatment:

Obj: compare donor microbiota and pre-treatment / CR - heatmap

```{r}
ps_rare %>%
  subset_samples(Day_of_Treatment > -4 & Day_of_Treatment < 0 | Reactor %in% c("DONOR", "CR")) -> before_treat # %>% 
# subset_samples(Reactor %!in%c("IR")) %>% 
# subset_samples() -> before_treat
```

```{r}
# before_treat %>% 
#   subset_samples(Model2 == "Human1") -> before_treat_h1
```

```{r}
before_treat %>% 
  # microViz::ps_mutate(tp_var = )
  physeq_most_abundant(group_var = "Model2",
                       ntax = 24,
                       tax_level = "Strain") -> top_taxa_per_group

# before_treat %>% 
#   physeq_most_abundant(group_var = "Reactor_Treatment_Dose",
#                        ntax = 7,
#                        tax_level = "Strain") -> top_taxa_per_group2
```

```{r}
before_treat %>%
  rarefy_even_depth(sample.size = min_sample_size, rngseed = 123) %>%
  # subset_samples(Reactor == "DONOR") %>%  sample_data() %>%  data.frame()
  # speedyseq::tax_glom(taxrank = "Family") %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  subset_taxa(Strain %in% top_taxa_per_group) %>% # extract only the taxa to display - after percentage normalisation
  phyloseq_ampvis_heatmap(physeq = ., tax_aggregate = "Species",
                          transform = FALSE,
                          plot_values = FALSE,
                          facet_by = c("Model2", "Fermentation","Reactor_Treatment_Dose"),
                          group_by = "Day_of_Treatment",#"sample_name",
                          ntax = Inf
  ) -> pre_speces

pre_speces$data %>% 
  mutate(Abundance = na_if(Abundance, 0)) %>% 
  mutate(Fermentation = Fermentation %>%  as.character() %>%  replace_na(., "NA")) %>% 
  mutate(Fermentation = factor(Fermentation, levels = c("NA", 1, 2))) -> pre_speces$data 

pre_speces + 
  facet_grid(. ~  Model2 + Fermentation + Reactor_Treatment_Dose, scales = "free", space = "free") +
  scale_color_manual(values = c("#31a354", "orange", 
                                "#f03b20"), drop = FALSE, na.value = 'transparent') -> pre_speces
# scale_fill_viridis_c(breaks = c(0,  0.01, 1, 10, 25, 50, 75, 100),
# labels = c(0,  0.01, 1, 10, 25,  50, 75, 100),
# trans = scales::pseudo_log_trans(sigma = 0.9),
# na.value = 'transparent') -> pre_speces


#pre_speces

```

```{r}
before_treat %>%
  rarefy_even_depth(sample.size = min_sample_size, rngseed = 123) %>%
  # subset_samples(Reactor == "DONOR") %>%  sample_data() %>%  data.frame()
  # speedyseq::tax_glom(taxrank = "Family") %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  phyloseq_ampvis_heatmap(physeq = ., tax_aggregate = "Family",
                          transform = FALSE,
                          plot_values = FALSE,
                          facet_by = c("Model2", "Fermentation","Reactor_Treatment_Dose"),
                          group_by = "Day_of_Treatment",#"sample_name",
                          ntax = 10
  ) -> pre_Family

pre_Family$data %>% 
  mutate(Abundance = na_if(Abundance, 0)) %>% 
  mutate(Fermentation = Fermentation %>%  as.character() %>%  replace_na(., "NA")) %>% 
  mutate(Fermentation = factor(Fermentation, levels = c("NA", 1, 2))) -> pre_Family$data 


pre_Family + 
  facet_grid(. ~  Model2 + Fermentation + Reactor_Treatment_Dose, scales = "free", space = "free") +
  scale_color_manual(values = c("#31a354", "orange", 
                                "#f03b20"), drop = FALSE, na.value = 'transparent') -> pre_Family


# pre_Family
```

```{r}
# cowplot::plot_grid(pre_speces,
#           pre_Family, 
#            ncol = 1,
#           align = "v", axis = "b",
#           rel_heights = c(3, 1)) -> pre_plots
# 
# pre_plots
```

```{r}
ggpubr::ggarrange(pre_Family + theme(legend.position = "null") + 
                    theme(axis.title.x=element_blank(),
                          axis.text.x=element_blank(),
                          axis.ticks.x=element_blank()),
                  pre_speces + theme(
                    strip.background = element_blank(),
                    strip.text.x = element_blank()
                  )  + theme(legend.position = "null"),
                  align = "v",
                  ncol = 1, 
                  heights = c(1, 2.40),
                  common.legend = FALSE) -> pre_plots

pre_plots

pre_plots %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 6, 
                    height = 0.618 * 317.48031496 * 8 , paper = "A4",  scaling = 2,
                    file = out_pptx)

pre_Family %>% 
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 1 , paper = "A4",  scaling = 2,
                    file = out_pptx)
```


```{r}
pre_Family$data %>% 
  select(Display, Sample, Abundance,Reactor_Treatment_Dose) %>% 
  pivot_wider(names_from = Display, values_from = Abundance) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "pre_family",
                   showNA = TRUE)
pre_speces$data %>% 
  select(Display, Sample, Abundance,Reactor_Treatment_Dose) %>% 
  pivot_wider(names_from = Display, values_from = Abundance) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "pre_ASV",
                   showNA = TRUE)
```

## Taxa over time:

### Heatmap:

#### Human1:

```{r}
ps_rare %>% 
  subset_samples(Model2 == "Human1" & Treatment != "DONOR" & Day_of_Treatment >= 1) %>% 
  physeq_most_abundant(group_var = "Reactor_Treatment_Dose",
                       ntax = 12,
                       tax_level = "Strain") -> top_taxa_per_group_h1
```

```{r}
ps_rare %>%
  subset_samples(Model2 == "Human1" & Treatment != "DONOR" & Day_of_Treatment >= -1) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  subset_taxa(Strain %in% top_taxa_per_group_h1) %>% # extract only the taxa to display - after percentage normalisation
  phyloseq_ampvis_heatmap(physeq = ., tax_aggregate = "Species",
                          transform = FALSE,
                          plot_values = FALSE,
                          facet_by = c("Model2","Reactor_Treatment_Dose"),
                          group_by = "Day_of_Treatment",#"sample_name",
                          ntax = Inf
  ) -> treat_speces

treat_speces$data %>% 
  mutate(Abundance = na_if(Abundance, 0)) -> treat_speces$data 

treat_speces + 
  facet_grid(Model2 ~  Reactor_Treatment_Dose , scales = "free", space = "free") +
  scale_color_manual(values = c("#31a354", "orange", 
                                "#f03b20"), drop = FALSE, na.value = 'transparent') -> treat_speces
# scale_fill_viridis_c(breaks = c(0,  0.01, 1, 10, 25, 50, 75, 100),
# labels = c(0,  0.01, 1, 10, 25,  50, 75, 100),
# trans = scales::pseudo_log_trans(sigma = 0.9),
# na.value = 'transparent') -> pre_speces


treat_speces


# pre_speces %>%
#   export::graph2ppt(append = TRUE, width = 317.48031496 * 4  , height = 0.618 * 317.48031496 * 4 , paper = "A4",  scaling = 2,
#                     file = out_pptx)

```

```{r}
ps_rare %>%
  subset_samples(Model2 == "Human1" & Treatment != "DONOR" & Day_of_Treatment >= -1) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  phyloseq_ampvis_heatmap(physeq = ., tax_aggregate = "Family",
                          transform = FALSE,
                          plot_values = FALSE,
                          facet_by = c("Model2","Reactor_Treatment_Dose"),
                          group_by = "Day_of_Treatment",#"sample_name",
                          ntax = 10
  ) -> treat_fam

treat_fam$data %>% 
  mutate(Abundance = na_if(Abundance, 0)) -> treat_fam$data 

treat_fam + 
  facet_grid(Model2 ~  Reactor_Treatment_Dose , scales = "free", space = "free") +
  scale_color_manual(values = c("#31a354", "orange", 
                                "#f03b20"), drop = FALSE, na.value = 'transparent') -> treat_fam
# scale_fill_viridis_c(breaks = c(0,  0.01, 1, 10, 25, 50, 75, 100),
# labels = c(0,  0.01, 1, 10, 25,  50, 75, 100),
# trans = scales::pseudo_log_trans(sigma = 0.9),
# na.value = 'transparent') -> pre_speces


treat_fam


# pre_speces %>%
#   export::graph2ppt(append = TRUE, width = 317.48031496 * 4  , height = 0.618 * 317.48031496 * 4 , paper = "A4",  scaling = 2,
#                     file = out_pptx)

```

```{r}



ggpubr::ggarrange(treat_fam + theme(legend.position = "null") + 
                    theme(axis.title.x=element_blank(),
                          axis.text.x=element_blank(),
                          axis.ticks.x=element_blank()),
                  treat_speces + theme(
                    strip.background = element_blank(),
                    strip.text.x = element_blank()
                  )  + theme(legend.position = "null"),
                  align = "v",
                  ncol = 1, 
                  heights = c(1, 2.60),
                  common.legend = FALSE) -> human1_heat

human1_heat

human1_heat %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 4.75, 
                    height = 0.618 * 317.48031496 * 6.5 , paper = "A4",  scaling = 2,
                    file = out_pptx)



treat_fam$data %>% 
  select(Display, Sample, Abundance,Reactor_Treatment_Dose) %>% 
  pivot_wider(names_from = Display, values_from = Abundance) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "treat_family",
                   showNA = TRUE)
treat_speces$data %>% 
  select(Display, Sample, Abundance,Reactor_Treatment_Dose) %>% 
  pivot_wider(names_from = Display, values_from = Abundance) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "treat_ASV",
                   showNA = TRUE)
```

##### splited by Fermentations:

###### fermentation 1:

```{r}
ps_rare %>% 
  subset_samples(Model2 == "Human1" & Treatment != "DONOR" & Day_of_Treatment >= 1 & Fermentation == 1) %>% 
  physeq_most_abundant(group_var = "Reactor_Treatment_Dose",
                       ntax = 12,
                       tax_level = "Strain") -> top_taxa_per_group_h1_f1
```

```{r}
ps_rare %>%
  subset_samples(Model2 == "Human1" & Treatment != "DONOR" & Day_of_Treatment >= -1 & Fermentation == 1) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  subset_taxa(Strain %in% top_taxa_per_group_h1_f1) %>% # extract only the taxa to display - after percentage normalisation
  phyloseq_ampvis_heatmap(physeq = ., tax_aggregate = "Species",
                          transform = FALSE,
                          plot_values = FALSE,
                          facet_by = c("Model2","Reactor_Treatment_Dose"),
                          group_by = "Day_of_Treatment",#"sample_name",
                          ntax = Inf
  ) -> treat_speces

treat_speces$data %>% 
  mutate(Abundance = na_if(Abundance, 0)) -> treat_speces$data 

treat_speces + 
  facet_grid(Model2 ~  Reactor_Treatment_Dose , scales = "free", space = "free") +
  scale_color_manual(values = c("#31a354", "orange", 
                                "#f03b20"), drop = FALSE, na.value = 'transparent') -> treat_speces
# scale_fill_viridis_c(breaks = c(0,  0.01, 1, 10, 25, 50, 75, 100),
# labels = c(0,  0.01, 1, 10, 25,  50, 75, 100),
# trans = scales::pseudo_log_trans(sigma = 0.9),
# na.value = 'transparent') -> pre_speces


treat_speces


# pre_speces %>%
#   export::graph2ppt(append = TRUE, width = 317.48031496 * 4  , height = 0.618 * 317.48031496 * 4 , paper = "A4",  scaling = 2,
#                     file = out_pptx)

```

```{r}
ps_rare %>%
  subset_samples(Model2 == "Human1" & Treatment != "DONOR" & Day_of_Treatment >= -1  & Fermentation == 1) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  phyloseq_ampvis_heatmap(physeq = ., tax_aggregate = "Family",
                          transform = FALSE,
                          plot_values = FALSE,
                          facet_by = c("Model2","Reactor_Treatment_Dose"),
                          group_by = "Day_of_Treatment",#"sample_name",
                          ntax = 10
  ) -> treat_fam

treat_fam$data %>% 
  mutate(Abundance = na_if(Abundance, 0)) -> treat_fam$data 

treat_fam + 
  facet_grid(Model2 ~  Reactor_Treatment_Dose , scales = "free", space = "free") +
  scale_color_manual(values = c("#31a354", "orange", 
                                "#f03b20"), drop = FALSE, na.value = 'transparent') -> treat_fam
# scale_fill_viridis_c(breaks = c(0,  0.01, 1, 10, 25, 50, 75, 100),
# labels = c(0,  0.01, 1, 10, 25,  50, 75, 100),
# trans = scales::pseudo_log_trans(sigma = 0.9),
# na.value = 'transparent') -> pre_speces


treat_fam


# pre_speces %>%
#   export::graph2ppt(append = TRUE, width = 317.48031496 * 4  , height = 0.618 * 317.48031496 * 4 , paper = "A4",  scaling = 2,
#                     file = out_pptx)

```

```{r}


ggpubr::ggarrange(treat_fam + theme(legend.position = "null") + 
                    theme(axis.title.x=element_blank(),
                          axis.text.x=element_blank(),
                          axis.ticks.x=element_blank()),
                  treat_speces + theme(
                    strip.background = element_blank(),
                    strip.text.x = element_blank()
                  )  + theme(legend.position = "null"),
                  align = "v",
                  ncol = 1, 
                  heights = c(1, 2.60),
                  common.legend = FALSE) -> human1_f1_heat

human1_f1_heat

human1_f1_heat %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 4.75, 
                    height = 0.618 * 317.48031496 * 6.5 , paper = "A4",  scaling = 2,
                    file = out_pptx)



treat_fam$data %>% 
  select(Display, Sample, Abundance,Reactor_Treatment_Dose) %>% 
  pivot_wider(names_from = Display, values_from = Abundance) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "treat_family_h1_f1",
                   showNA = TRUE)
treat_speces$data %>% 
  select(Display, Sample, Abundance,Reactor_Treatment_Dose) %>% 
  pivot_wider(names_from = Display, values_from = Abundance) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "treat_ASV_h1_f1",
                   showNA = TRUE)

```

###### fermentation 2:

```{r}
ps_rare %>% 
  subset_samples(Model2 == "Human1" & Treatment != "DONOR" & Day_of_Treatment >= 1 & Fermentation == 2) %>% 
  physeq_most_abundant(group_var = "Reactor_Treatment_Dose",
                       ntax = 20,
                       tax_level = "Strain") -> top_taxa_per_group_h1_f2
```

```{r}
ps_rare %>%
  subset_samples(Model2 == "Human1" & Treatment != "DONOR" & Day_of_Treatment >= -1 & Fermentation == 2) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  subset_taxa(Strain %in% top_taxa_per_group_h1_f2) %>% # extract only the taxa to display - after percentage normalisation
  phyloseq_ampvis_heatmap(physeq = ., tax_aggregate = "Species",
                          transform = FALSE,
                          plot_values = FALSE,
                          facet_by = c("Model2","Reactor_Treatment_Dose"),
                          group_by = "Day_of_Treatment",#"sample_name",
                          ntax = Inf
  ) -> treat_speces

treat_speces$data %>% 
  mutate(Abundance = na_if(Abundance, 0)) -> treat_speces$data 

treat_speces + 
  facet_grid(Model2 ~  Reactor_Treatment_Dose , scales = "free", space = "free") +
  scale_color_manual(values = c("#31a354", "orange", 
                                "#f03b20"), drop = FALSE, na.value = 'transparent') -> treat_speces
# scale_fill_viridis_c(breaks = c(0,  0.01, 1, 10, 25, 50, 75, 100),
# labels = c(0,  0.01, 1, 10, 25,  50, 75, 100),
# trans = scales::pseudo_log_trans(sigma = 0.9),
# na.value = 'transparent') -> pre_speces


treat_speces


# pre_speces %>%
#   export::graph2ppt(append = TRUE, width = 317.48031496 * 4  , height = 0.618 * 317.48031496 * 4 , paper = "A4",  scaling = 2,
#                     file = out_pptx)

```

```{r}
ps_rare %>%
  subset_samples(Model2 == "Human1" & Treatment != "DONOR" & Day_of_Treatment >= -1  & Fermentation == 2) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  phyloseq_ampvis_heatmap(physeq = ., tax_aggregate = "Family",
                          transform = FALSE,
                          plot_values = FALSE,
                          facet_by = c("Model2","Reactor_Treatment_Dose"),
                          group_by = "Day_of_Treatment",#"sample_name",
                          ntax = 10
  ) -> treat_fam

treat_fam$data %>% 
  mutate(Abundance = na_if(Abundance, 0)) -> treat_fam$data 

treat_fam + 
  facet_grid(Model2 ~  Reactor_Treatment_Dose , scales = "free", space = "free") +
  scale_color_manual(values = c("#31a354", "orange", 
                                "#f03b20"), drop = FALSE, na.value = 'transparent') -> treat_fam
# scale_fill_viridis_c(breaks = c(0,  0.01, 1, 10, 25, 50, 75, 100),
# labels = c(0,  0.01, 1, 10, 25,  50, 75, 100),
# trans = scales::pseudo_log_trans(sigma = 0.9),
# na.value = 'transparent') -> pre_speces


treat_fam


# pre_speces %>%
#   export::graph2ppt(append = TRUE, width = 317.48031496 * 4  , height = 0.618 * 317.48031496 * 4 , paper = "A4",  scaling = 2,
#                     file = out_pptx)

```

```{r}


ggpubr::ggarrange(treat_fam + theme(legend.position = "null") + 
                    theme(axis.title.x=element_blank(),
                          axis.text.x=element_blank(),
                          axis.ticks.x=element_blank()),
                  treat_speces + theme(
                    strip.background = element_blank(),
                    strip.text.x = element_blank()
                  )  + theme(legend.position = "null"),
                  align = "v",
                  ncol = 1, 
                  heights = c(1, 2.60),
                  common.legend = FALSE) -> human1_f2_heat

human1_f2_heat

human1_f2_heat %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 4.75, 
                    height = 0.618 * 317.48031496 * 6.5 , paper = "A4",  scaling = 2,
                    file = out_pptx)



treat_fam$data %>% 
  select(Display, Sample, Abundance,Reactor_Treatment_Dose) %>% 
  pivot_wider(names_from = Display, values_from = Abundance) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "treat_family_h1_f2",
                   showNA = TRUE)
treat_speces$data %>% 
  select(Display, Sample, Abundance,Reactor_Treatment_Dose) %>% 
  pivot_wider(names_from = Display, values_from = Abundance) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "treat_ASV_h1_f2",
                   showNA = TRUE)

```

#### Human2:

```{r}
ps_rare %>% 
  subset_samples(Model2 == "Human2" & Treatment != "DONOR" & Day_of_Treatment >= 1) %>% 
  physeq_most_abundant(group_var = "Reactor_Treatment_Dose",
                       ntax = 14,
                       tax_level = "Strain") -> top_taxa_per_group_h2
```

```{r}
ps_rare %>%
  subset_samples(Model2 == "Human2" & Treatment != "DONOR" & Day_of_Treatment >= -1) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  subset_taxa(Strain %in% top_taxa_per_group_h2) %>% # extract only the taxa to display - after percentage normalisation
  phyloseq_ampvis_heatmap(physeq = ., tax_aggregate = "Species",
                          transform = FALSE,
                          plot_values = FALSE,
                          facet_by = c("Model2","Reactor_Treatment_Dose"),
                          group_by = "Day_of_Treatment",#"sample_name",
                          ntax = Inf
  ) -> treat_speces

treat_speces$data %>% 
  mutate(Abundance = na_if(Abundance, 0)) -> treat_speces$data 

treat_speces + 
  facet_grid(Model2 ~  Reactor_Treatment_Dose , scales = "free", space = "free") +
  scale_color_manual(values = c("#31a354", "orange", 
                                "#f03b20"), drop = FALSE, na.value = 'transparent') -> treat_speces
# scale_fill_viridis_c(breaks = c(0,  0.01, 1, 10, 25, 50, 75, 100),
# labels = c(0,  0.01, 1, 10, 25,  50, 75, 100),
# trans = scales::pseudo_log_trans(sigma = 0.9),
# na.value = 'transparent') -> pre_speces

treat_speces

# pre_speces %>%
#   export::graph2ppt(append = TRUE, width = 317.48031496 * 4  , height = 0.618 * 317.48031496 * 4 , paper = "A4",  scaling = 2,
#                     file = out_pptx)

```

```{r}
ps_rare %>%
  subset_samples(Model2 == "Human2" & Treatment != "DONOR" & Day_of_Treatment >= -1) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  phyloseq_ampvis_heatmap(physeq = ., tax_aggregate = "Family",
                          transform = FALSE,
                          plot_values = FALSE,
                          facet_by = c("Model2","Reactor_Treatment_Dose"),
                          group_by = "Day_of_Treatment",#"sample_name",
                          ntax = 10
  ) -> treat_fam

treat_fam$data %>% 
  mutate(Abundance = na_if(Abundance, 0)) -> treat_fam$data 

treat_fam + 
  facet_grid(Model2 ~  Reactor_Treatment_Dose , scales = "free", space = "free") +
  scale_color_manual(values = c("#31a354", "orange", 
                                "#f03b20"), drop = FALSE, na.value = 'transparent') -> treat_fam
# scale_fill_viridis_c(breaks = c(0,  0.01, 1, 10, 25, 50, 75, 100),
# labels = c(0,  0.01, 1, 10, 25,  50, 75, 100),
# trans = scales::pseudo_log_trans(sigma = 0.9),
# na.value = 'transparent') -> pre_speces

treat_fam

# pre_speces %>%
#   export::graph2ppt(append = TRUE, width = 317.48031496 * 4  , height = 0.618 * 317.48031496 * 4 , paper = "A4",  scaling = 2,
#                     file = out_pptx)

```

```{r}
ggpubr::ggarrange(treat_fam + theme(legend.position = "null") + 
                    theme(axis.title.x=element_blank(),
                          axis.text.x=element_blank(),
                          axis.ticks.x=element_blank()),
                  treat_speces + theme(
                    strip.background = element_blank(),
                    strip.text.x = element_blank()
                  )  + theme(legend.position = "null"),
                  align = "v",
                  ncol = 1, 
                  heights = c(1, 2.60),
                  common.legend = FALSE) -> human2_heat


human2_heat

human2_heat %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 4.75, 
                    height = 0.618 * 317.48031496 * 5.5 , paper = "A4",  scaling = 2,
                    file = out_pptx)




treat_fam$data %>% 
  select(Display, Sample, Abundance,Reactor_Treatment_Dose) %>% 
  pivot_wider(names_from = Display, values_from = Abundance) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "treat_family_h2",
                   showNA = TRUE)

treat_speces$data %>% 
  select(Display, Sample, Abundance,Reactor_Treatment_Dose) %>% 
  pivot_wider(names_from = Display, values_from = Abundance) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "treat_ASV_h2",
                   showNA = TRUE)

```

Could be improved:

-clustering taxa based on correlation -\> response (pheatmap?) <https://btep.ccr.cancer.gov/docs/data-visualization-with-r/Lesson5_intro_to_ggplot/#make-this-plot-compatible-with-ggplot2-using-the-package-ggplotify>

<https://jcoliver.github.io/learn-r/008-ggplot-dendrograms-and-heatmaps.html>

-additional layer depicting the % of the community covered by the taxa displayed.

### barplot:

#### Human1:

```{r}
ps_rare %>% 
  subset_samples(Model2 == "Human1" & Treatment != "DONOR" & Day_of_Treatment >= -1) %>% 
  microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  subset_taxa(Strain %in% top_taxa_per_group_h1) %>% # extract only the taxa to display - after percentage normalisation
  microViz::phyloseq_validate() %>%
  microViz::tax_fix() %>% 
  microViz::ps_arrange(Day_of_Treatment) %>% 
  microViz::comp_barplot(
    tax_transform_for_plot = "identity",
    tax_level = "Strain",
    label = "Day_of_Treatment", # name an alternative variable to label axes
    n_taxa = length(top_taxa_per_group_h1), # give more taxa unique colours
    merge_other = FALSE, # split the "other" category to display alpha diversity
    bar_width = 0.9, # reduce the bar width to 70% of one row
    bar_outline_colour = "grey5",
    sample_order = "default") +
  ylab("Proportion - %") + xlab( " Day (Treatment) ") -> p_hist

p_hist + facet_grid(Phylum ~ Reactor_Treatment_Dose, scales = "free", space = "free_x") -> p_hist_human1_strain

p_hist_human1_strain  +  theme_light() + theme(legend.position = "none") -> p_hist_human1_strain_noleg

p_hist_human1_strain_noleg

p_hist_human1_strain %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_hist_human1_strain_leg
```

```{r}
ps_rare %>% 
  subset_samples(Model2 == "Human1" & Treatment != "DONOR" & Day_of_Treatment >= -1) %>% 
  microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  microViz::phyloseq_validate() %>%
  microViz::tax_fix() %>% 
  microViz::ps_arrange(Day_of_Treatment) %>% 
  microViz::comp_barplot(
    tax_transform_for_plot = "identity",
    tax_level = "Family",
    label = "Day_of_Treatment", # name an alternative variable to label axes
    n_taxa = 10, # give more taxa unique colours
    palette = microViz::distinct_palette(10, pal = "kelly"),
    merge_other = FALSE, # split the "other" category to display alpha diversity
    bar_width = 0.9, # reduce the bar width to 70% of one row
    bar_outline_colour = "grey5",
    sample_order = "default") +
  ylab("Proportion - %") + xlab( " Day (Treatment) ") -> p_hist

p_hist + facet_grid(Kingdom ~ Reactor_Treatment_Dose, scales = "free", space = "free_x", drop = TRUE) -> p_hist_human1_fam

p_hist_human1_fam  +  theme_light() + theme(legend.position = "none") -> p_hist_human1_fam_no_leg

p_hist_human1_fam_no_leg

p_hist_human1_fam %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_hist_human1_fam_leg
```

##### splited by Fermentations:

##### Fermentation 1:

```{r}
ps_rare %>% 
  subset_samples(Model2 == "Human1" & Treatment != "DONOR" & Day_of_Treatment >= -1 & Fermentation == 1) %>% 
  # microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  subset_taxa(Strain %in% top_taxa_per_group_h1_f1) %>% # extract only the taxa to display - after percentage normalisation
  microViz::phyloseq_validate() %>%
  microViz::tax_fix() %>% 
  microViz::ps_arrange(Day_of_Treatment) %>% 
  microViz::comp_barplot(
    tax_transform_for_plot = "identity",
    tax_level = "Strain",
    label = "Day_of_Treatment", # name an alternative variable to label axes
    n_taxa = length(top_taxa_per_group_h1_f1), # give more taxa unique colours
    merge_other = FALSE, # split the "other" category to display alpha diversity
    bar_width = 0.9, # reduce the bar width to 70% of one row
    bar_outline_colour = "grey5",
    sample_order = "default") +
  ylab("Proportion - %") + xlab( " Day (Treatment) ") -> p_hist

p_hist + facet_grid(Phylum ~ Reactor_Treatment_Dose, scales = "free", space = "free_x") -> p_hist_human1_f1_strain

p_hist_human1_f1_strain  +  theme_light() + theme(legend.position = "none") -> p_hist_human1_strain_f1_noleg

p_hist_human1_strain_f1_noleg

p_hist_human1_f1_strain %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_hist_human1_f1_strain_leg
```

```{r}
ps_rare %>% 
  subset_samples(Model2 == "Human1" & Treatment != "DONOR" & Day_of_Treatment >= -1 & Fermentation == 1) %>% 
  # microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  microViz::phyloseq_validate() %>%
  microViz::tax_fix() %>% 
  microViz::ps_arrange(Day_of_Treatment) %>% 
  microViz::comp_barplot(
    tax_transform_for_plot = "identity",
    tax_level = "Family",
    label = "Day_of_Treatment", # name an alternative variable to label axes
    n_taxa = 10, # give more taxa unique colours
    palette = microViz::distinct_palette(10, pal = "kelly"),
    merge_other = FALSE, # split the "other" category to display alpha diversity
    bar_width = 0.9, # reduce the bar width to 70% of one row
    bar_outline_colour = "grey5",
    sample_order = "default") +
  ylab("Proportion - %") + xlab( " Day (Treatment) ") -> p_hist

p_hist + facet_grid(Kingdom ~ Reactor_Treatment_Dose, scales = "free", space = "free_x", drop = TRUE) -> p_hist_human1_f1_fam

p_hist_human1_f1_fam  +  theme_light() + theme(legend.position = "none") -> p_hist_human1_f1_fam_no_leg

p_hist_human1_f1_fam_no_leg

p_hist_human1_f1_fam %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_hist_human1_f1_fam_leg
```

##### Fermentation 2:

```{r}
ps_rare %>% 
  subset_samples(Model2 == "Human1" & Treatment != "DONOR" & Day_of_Treatment >= -1 & Fermentation == 2) %>% 
  # microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  subset_taxa(Strain %in% top_taxa_per_group_h1_f1) %>% # extract only the taxa to display - after percentage normalisation
  microViz::phyloseq_validate() %>%
  microViz::tax_fix() %>% 
  microViz::ps_arrange(Day_of_Treatment) %>% 
  microViz::comp_barplot(
    tax_transform_for_plot = "identity",
    tax_level = "Strain",
    label = "Day_of_Treatment", # name an alternative variable to label axes
    n_taxa = length(top_taxa_per_group_h1_f2), # give more taxa unique colours
    merge_other = FALSE, # split the "other" category to display alpha diversity
    bar_width = 0.9, # reduce the bar width to 70% of one row
    bar_outline_colour = "grey5",
    sample_order = "default") +
  ylab("Proportion - %") + xlab( " Day (Treatment) ") -> p_hist

p_hist + facet_grid(Phylum ~ Reactor_Treatment_Dose, scales = "free", space = "free_x") -> p_hist_human1_f2_strain

p_hist_human1_f2_strain  +  theme_light() + theme(legend.position = "none") -> p_hist_human1_f2_strain_noleg

p_hist_human1_f2_strain_noleg

p_hist_human1_f2_strain %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_hist_human1_f2_strain_leg
```

```{r}
ps_rare %>% 
  subset_samples(Model2 == "Human1" & Treatment != "DONOR" & Day_of_Treatment >= -1 & Fermentation == 2) %>% 
  # microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  microViz::phyloseq_validate() %>%
  microViz::tax_fix() %>% 
  microViz::ps_arrange(Day_of_Treatment) %>% 
  microViz::comp_barplot(
    tax_transform_for_plot = "identity",
    tax_level = "Family",
    label = "Day_of_Treatment", # name an alternative variable to label axes
    n_taxa = 10, # give more taxa unique colours
    palette = microViz::distinct_palette(10, pal = "kelly"),
    merge_other = FALSE, # split the "other" category to display alpha diversity
    bar_width = 0.9, # reduce the bar width to 70% of one row
    bar_outline_colour = "grey5",
    sample_order = "default") +
  ylab("Proportion - %") + xlab( " Day (Treatment) ") -> p_hist

p_hist + facet_grid(Kingdom ~ Reactor_Treatment_Dose, scales = "free", space = "free_x", drop = TRUE) -> p_hist_human1_f2_fam

p_hist_human1_f2_fam  +  theme_light() + theme(legend.position = "none") -> p_hist_human1_f2_fam_no_leg

p_hist_human1_f2_fam_no_leg

p_hist_human1_f2_fam %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_hist_human1_f2_fam_leg
```

#### Human2:

```{r}
ps_rare %>% 
  subset_samples(Model2 == "Human2" & Treatment != "DONOR" & Day_of_Treatment >= -1) %>% 
  microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  subset_taxa(Strain %in% top_taxa_per_group_h2) %>% # extract only the taxa to display - after percentage normalisation
  microViz::phyloseq_validate() %>%
  microViz::tax_fix() %>% 
  microViz::ps_arrange(Day_of_Treatment) %>% 
  microViz::comp_barplot(
    tax_transform_for_plot = "identity",
    tax_level = "Strain",
    label = "Day_of_Treatment", # name an alternative variable to label axes
    n_taxa = length(top_taxa_per_group_h2), # give more taxa unique colours
    merge_other = FALSE, # split the "other" category to display alpha diversity
    bar_width = 0.9, # reduce the bar width to 70% of one row
    bar_outline_colour = "grey5",
    sample_order = "default") +
  ylab("Proportion - %") + xlab( " Day (Treatment) ") -> p_hist

p_hist + facet_grid(Phylum ~ Reactor_Treatment_Dose, scales = "free", space = "free_x") -> p_hist_human2_strain

p_hist_human2_strain +  theme_light()  + theme(legend.position = "none") -> p_hist_human2_strain_noleg

p_hist_human2_strain_noleg



p_hist_human2_strain %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_hist_human2_strain_leg
```

```{r}
ps_rare %>% 
  subset_samples(Model2 == "Human2" & Treatment != "DONOR" & Day_of_Treatment >= -1) %>% 
  microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  # subset_taxa(Strain %in% top_taxa_per_group_h2) %>% # extract only the taxa to display - after percentage normalisation
  microViz::phyloseq_validate() %>%
  microViz::tax_fix() %>% 
  microViz::ps_arrange(Day_of_Treatment) %>% 
  microViz::comp_barplot(
    tax_transform_for_plot = "identity",
    tax_level = "Family",
    label = "Day_of_Treatment", # name an alternative variable to label axes
    palette = microViz::distinct_palette(10, pal = "kelly"),
    n_taxa = 10, # give more taxa unique colours
    merge_other = FALSE, # split the "other" category to display alpha diversity
    bar_width = 0.9, # reduce the bar width to 70% of one row
    bar_outline_colour = "grey5",
    sample_order = "default") +
  ylab("Proportion - %") + xlab( " Day (Treatment) ") -> p_hist

p_hist + facet_grid(Kingdom ~ Reactor_Treatment_Dose, scales = "free", space = "free_x", drop = TRUE) -> p_hist_human2_fam

p_hist_human2_fam +  theme_light()  + theme(legend.position = "none") -> p_hist_human2_fam_no_leg

p_hist_human2_fam_no_leg

p_hist_human2_fam %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_hist_human2_fam_leg
```

Unify colors:

```{r}
c(p_hist_human2_fam$data$top, p_hist_human1_f1_fam$data$top, p_hist_human1_f2_fam$data$top) %>% 
  unique() -> families
```

```{r}
# families %>%
#   ggpubr::get_palette(k = ., palette = "kelly") -> col

families[!families %in% c("other")] %>% 
  length() %>% 
  microViz::distinct_palette(n = ., pal = "kelly", add = NA) -> col_fam

names(col_fam) <- families[!families %in% c("other")]

c(col_fam, "lightgrey") -> col_fam

names(col_fam) = NULL

names(col_fam) <- c(families[!families %in% c("other")], families[families %in% c("other")])

scales::show_col(col_fam)

```

```{r}
p_hist_human1_f1_fam  +  theme_light() + scale_fill_manual(values = col_fam) ->  p_hist_human1_f1_fam

p_hist_human1_f1_fam + theme(legend.position = "none") -> p_hist_human1_f1_fam_no_leg

p_hist_human1_f1_fam_no_leg
```

```{r}
p_hist_human1_f1_fam %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_hist_human1_f1_fam_leg

p_hist_human1_f1_fam_leg
```

```{r}
p_hist_human1_f2_fam  +  theme_light() + scale_fill_manual(values = col_fam) ->  p_hist_human1_f2_fam

p_hist_human1_f2_fam + theme(legend.position = "none") -> p_hist_human1_f2_fam_no_leg

p_hist_human1_f2_fam_no_leg
```

```{r}
p_hist_human1_f2_fam %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_hist_human1_f2_fam_leg

p_hist_human1_f2_fam_leg
```

```{r}
p_hist_human2_fam +  theme_light() +  scale_fill_manual(values = col_fam) ->  p_hist_human2_fam

p_hist_human2_fam + theme(legend.position = "none") ->p_hist_human2_fam_no_leg

p_hist_human2_fam_no_leg
```

```{r}
p_hist_human2_fam %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_hist_human2_fam_leg

p_hist_human2_fam_leg
```

```{r}
c(p_hist_human2_strain$data$top, p_hist_human1_f1_strain$data$top, p_hist_human1_f2_strain$data$top) %>% 
  unique() -> ASVs

# ASVs
```

```{r}
# families %>%
#   ggpubr::get_palette(k = ., palette = "kelly") -> col
set.seed(1234)

ASVs[!ASVs %in% c("other")] %>% 
  length() %>% 
  randomcoloR::distinctColorPalette(k = ., altCol = FALSE, runTsne = TRUE) -> col_ASVs

names(col_ASVs) <- ASVs[!ASVs %in% c("other")]

c(col_ASVs, "lightgrey") -> col_ASVs

names(col_ASVs) = NULL

names(col_ASVs) <- c(ASVs[!ASVs %in% c("other")], ASVs[ASVs %in% c("other")])

scales::show_col(col_ASVs)
```

```{r}
p_hist_human1_f1_strain + scale_fill_manual(values = col_ASVs) ->  p_hist_human1_f1_strain

p_hist_human1_f1_strain + theme_light() +
  theme(legend.position = "none") -> p_hist_human1_f1_strain_no_leg

p_hist_human1_f1_strain_no_leg
```

```{r}
p_hist_human1_f1_strain %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_hist_human1_f1_strain_leg

p_hist_human1_f1_strain_leg
```

```{r}
p_hist_human1_f2_strain + scale_fill_manual(values = col_ASVs) ->  p_hist_human1_f2_strain

p_hist_human1_f2_strain + theme_light() + theme(legend.position = "none") -> p_hist_human1_f2_strain_no_leg

p_hist_human1_f2_strain_no_leg
```

```{r}
p_hist_human1_f2_strain %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_hist_human1_f2_strain_leg

p_hist_human1_f2_strain_leg
```

```{r}
p_hist_human2_strain + scale_fill_manual(values = col_ASVs) ->  p_hist_human2_strain

p_hist_human2_strain + theme_light() +theme(legend.position = "none") ->p_hist_human2_strain_no_leg

p_hist_human2_strain_no_leg
```

```{r}
p_hist_human2_strain %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_hist_human2_strain_leg

p_hist_human2_strain_leg
```

```{r}
ggpubr::ggarrange(p_hist_human1_f1_fam_no_leg,
                  p_hist_human1_f1_strain_no_leg ,
                  align = "v",
                  ncol = 1, 
                  heights = c(1, 1),
                  common.legend = FALSE) -> p_hist_human1_f1

p_hist_human1_f1

p_hist_human1_f1 %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 4.75, 
                    height = 0.618 * 317.48031496 * 6.5 , paper = "A4",  scaling = 2,
                    file = out_pptx)

p_hist_human1_f1_fam_leg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 1 , paper = "A4",  scaling = 2,
                    file = out_pptx)

p_hist_human1_f1_strain_leg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 5, 
                    height = 0.618 * 317.48031496 * 2.5 , paper = "A3",  scaling = 2,
                    file = out_pptx)
```

```{r}
ggpubr::ggarrange(p_hist_human1_f2_fam_no_leg,
                  p_hist_human1_f2_strain_no_leg,
                  align = "v",
                  ncol = 1, 
                  heights = c(1, 1),
                  common.legend = FALSE) -> p_hist_human1_f2

p_hist_human1_f2

p_hist_human1_f2 %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 4.75, 
                    height = 0.618 * 317.48031496 * 6.5 , paper = "A4",  scaling = 2,
                    file = out_pptx)

p_hist_human1_f2_fam_leg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 1 , paper = "A4",  scaling = 2,
                    file = out_pptx)

p_hist_human1_f2_strain_leg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 6, 
                    height = 0.618 * 317.48031496 * 2.5 , paper = "A3",  scaling = 2,
                    file = out_pptx)
```

```{r}
ggpubr::ggarrange(p_hist_human2_fam_no_leg,
                  p_hist_human2_strain_noleg,
                  align = "v",
                  ncol = 1, 
                  heights = c(1, 1),
                  common.legend = FALSE) -> p_hist_human2

p_hist_human2

p_hist_human2 %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 4.75, 
                    height = 0.618 * 317.48031496 * 6.5 , paper = "A4",  scaling = 2,
                    file = out_pptx)

p_hist_human2_fam_leg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 1 , paper = "A4",  scaling = 2,
                    file = out_pptx)



p_hist_human2_strain_leg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 6, 
                    height = 0.618 * 317.48031496 * 2.5 , paper = "A3",  scaling = 2,
                    file = out_pptx)
```

### Alluvial:

```{r}
# p_hist_human2_fam$data %>%
#   filter(Model2 == "Human2") %>%
#   # microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
#   # mutate(combo = paste0(Model2, "_" , Reactor_F1_F2)) %>%
#   # mutate(combo = as.factor(combo)) %>%
#   select(SAMPLE, top,OTU, Abundance, Day_of_Treatment, Reactor_Treatment_Dose) %>%  distinct(top)
```

```{r}
# ggpubr::ggarrange(p_hist_human2_fam_no_leg,
#                   p_hist_human2_strain_noleg ,
#                   align = "v",
#                   ncol = 1, 
#                   heights = c(1, 1),
#                   common.legend = FALSE) -> p_hist_human2
# 
# p_hist_human2
# 
# p_hist_human2 %>% 
#   export::graph2ppt(append = TRUE, 
#                     width = 317.48031496 * 4.75, 
#                     height = 0.618 * 317.48031496 * 6.5 , paper = "A4",  scaling = 2,
#                     file = out_pptx)
```

#### human1 - f1

```{r}
require(ggalluvial)
```

##### Family

```{r}
p_hist_human1_f1_fam$data %>%
  # filter(Model2 == "Human2") %>%
  # microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  # mutate(combo = paste0(Model2, "_" , Reactor_F1_F2)) %>%
  # mutate(combo = as.factor(combo)) %>%
  select(SAMPLE, top,OTU, Abundance, Day_of_Treatment, Reactor_Treatment_Dose) %>% 
  ggplot(.,
         aes(x = Day_of_Treatment, 
             y = Abundance,
             stratum = top, 
             alluvium = OTU,
             fill = top,
             label = top))+
  facet_grid(Reactor_Treatment_Dose  ~ .)+
  # scale_fill_manual(values = c("red", "yellow", "green3"))+
  geom_flow(stat = "alluvium", lode.guidance = "frontback", color = "darkgray")+
  geom_stratum() +
  theme_light() +
  # geom_text(stat = "alluvium", aes(label = top), lode.guidance = "frontback") +
  scale_fill_manual(values = col_fam) +
  # scale_color_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  # scale_fill_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  ylab("Proportion") -> p_allu_human1_f1

p_allu_human1_f1 + theme(legend.position = "none") -> p_allu_human1_f1_noleg 

p_allu_human1_f1_noleg

p_allu_human1_f1 %>% 
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() ->  p_allu_human1_f1_leg
```

```{r}
p_allu_human1_f1_noleg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 2 , paper = "A4",  scaling = 2,
                    file = out_pptx)

p_allu_human1_f1_leg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 2 , paper = "A4",  scaling = 2,
                    file = out_pptx)
```

##### ASV:

```{r}
p_hist_human1_f1_strain$data %>%
  # filter(Model2 == "Human2") %>%
  # microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  # mutate(combo = paste0(Model2, "_" , Reactor_F1_F2)) %>%
  # mutate(combo = as.factor(combo)) %>%
  select(SAMPLE, top,OTU, Abundance, Day_of_Treatment, Reactor_Treatment_Dose) %>% 
  ggplot(.,
         aes(x = Day_of_Treatment, 
             y = Abundance,
             stratum = top, 
             alluvium = OTU,
             fill = top,
             label = top))+
  facet_grid(Reactor_Treatment_Dose  ~ .)+
  # scale_fill_manual(values = c("red", "yellow", "green3"))+
  geom_flow(stat = "alluvium", lode.guidance = "frontback", color = "darkgray")+
  geom_stratum() +
  theme_light() +
  # geom_text(stat = "alluvium", aes(label = top), lode.guidance = "frontback") +
  scale_fill_manual(values = col_ASVs) +
  # scale_color_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  # scale_fill_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  ylab("Proportion") -> p_allu_human1_ASV_f1

p_allu_human1_ASV_f1 + theme(legend.position = "none") -> p_allu_human1_ASV_f1_noleg 

p_allu_human1_ASV_f1_noleg

p_allu_human1_ASV_f1 %>% 
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() ->  p_allu_human1_ASV_f1_leg
```

```{r}
p_allu_human1_ASV_f1_noleg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 2 , paper = "A4",  scaling = 2,
                    file = out_pptx)
p_allu_human1_ASV_f1_leg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 4, 
                    height = 0.618 * 317.48031496 * 2 , paper = "A3",  scaling = 2,
                    file = out_pptx)
```

#### human1 - f2

##### Family

```{r}

p_hist_human1_f2_fam$data %>%
  # filter(Model2 == "Human2") %>%
  # microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  # mutate(combo = paste0(Model2, "_" , Reactor_F1_F2)) %>%
  # mutate(combo = as.factor(combo)) %>%
  select(SAMPLE, top,OTU, Abundance, Day_of_Treatment, Reactor_Treatment_Dose) %>% 
  ggplot(.,
         aes(x = Day_of_Treatment, 
             y = Abundance,
             stratum = top, 
             alluvium = OTU,
             fill = top,
             label = top))+
  facet_grid(Reactor_Treatment_Dose  ~ .)+
  # scale_fill_manual(values = c("red", "yellow", "green3"))+
  geom_flow(stat = "alluvium", lode.guidance = "frontback", color = "darkgray")+
  geom_stratum() +
  theme_light() +
  # geom_text(stat = "alluvium", aes(label = top), lode.guidance = "frontback") +
  scale_fill_manual(values = col_fam) +
  # scale_color_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  # scale_fill_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  ylab("Proportion") -> p_allu_human1_f2

p_allu_human1_f2 + theme(legend.position = "none") -> p_allu_human1_f2_noleg 

p_allu_human1_f2_noleg

p_allu_human1_f2 %>% 
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() ->  p_allu_human1_f2_leg
```

```{r}
p_allu_human1_f2_noleg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 2 , paper = "A4",  scaling = 2,
                    file = out_pptx)

p_allu_human1_f2_leg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 2 , paper = "A4",  scaling = 2,
                    file = out_pptx)
```

##### ASV:

```{r}
p_hist_human1_f2_strain$data %>%
  # filter(Model2 == "Human2") %>%
  # microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  # mutate(combo = paste0(Model2, "_" , Reactor_F1_F2)) %>%
  # mutate(combo = as.factor(combo)) %>%
  select(SAMPLE, top,OTU, Abundance, Day_of_Treatment, Reactor_Treatment_Dose) %>% 
  ggplot(.,
         aes(x = Day_of_Treatment, 
             y = Abundance,
             stratum = top, 
             alluvium = OTU,
             fill = top,
             label = top))+
  facet_grid(Reactor_Treatment_Dose  ~ .)+
  # scale_fill_manual(values = c("red", "yellow", "green3"))+
  geom_flow(stat = "alluvium", lode.guidance = "frontback", color = "darkgray")+
  geom_stratum() +
  theme_light() +
  # geom_text(stat = "alluvium", aes(label = top), lode.guidance = "frontback") +
  scale_fill_manual(values = col_ASVs) +
  # scale_color_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  # scale_fill_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  ylab("Proportion") -> p_allu_human1_ASV_f2

p_allu_human1_ASV_f2 + theme(legend.position = "none") -> p_allu_human1_ASV_f2_noleg 

p_allu_human1_ASV_f2_noleg

p_allu_human1_ASV_f1 %>% 
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() ->  p_allu_human1_ASV_f2_leg
```

```{r}
p_allu_human1_ASV_f2_noleg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 2 , paper = "A4",  scaling = 2,
                    file = out_pptx)

p_allu_human1_ASV_f2_leg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 4, 
                    height = 0.618 * 317.48031496 * 2 , paper = "A3",  scaling = 2,
                    file = out_pptx)
```

#### human2

##### Family

```{r}

p_hist_human2_fam$data %>%
  # filter(Model2 == "Human2") %>%
  # microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  # mutate(combo = paste0(Model2, "_" , Reactor_F1_F2)) %>%
  # mutate(combo = as.factor(combo)) %>%
  select(SAMPLE, top,OTU, Abundance, Day_of_Treatment, Reactor_Treatment_Dose) %>% 
  ggplot(.,
         aes(x = Day_of_Treatment, 
             y = Abundance,
             stratum = top, 
             alluvium = OTU,
             fill = top,
             label = top))+
  facet_grid(Reactor_Treatment_Dose  ~ .)+
  # scale_fill_manual(values = c("red", "yellow", "green3"))+
  geom_flow(stat = "alluvium", lode.guidance = "frontback", color = "darkgray")+
  geom_stratum() +
  theme_light() +
  # geom_text(stat = "alluvium", aes(label = top), lode.guidance = "frontback") +
  scale_fill_manual(values = col_fam) +
  # scale_color_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  # scale_fill_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  ylab("Proportion") -> p_allu_human2

p_allu_human2 + theme(legend.position = "none") -> p_allu_human2_noleg 

p_allu_human2_noleg

```

```{r}
p_allu_human2 %>% 
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() ->  p_allu_human2_leg

p_allu_human2_leg
```

```{r}
p_allu_human2_noleg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 2 , paper = "A4",  scaling = 2,
                    file = out_pptx)

p_allu_human2_leg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 2 , paper = "A4",  scaling = 2,
                    file = out_pptx)
```

##### ASV:

```{r}
p_hist_human2_strain$data %>%
  # filter(Model2 == "Human2") %>%
  # microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  # mutate(combo = paste0(Model2, "_" , Reactor_F1_F2)) %>%
  # mutate(combo = as.factor(combo)) %>%
  select(SAMPLE, top,OTU, Abundance, Day_of_Treatment, Reactor_Treatment_Dose) %>% 
  ggplot(.,
         aes(x = Day_of_Treatment, 
             y = Abundance,
             stratum = top, 
             alluvium = OTU,
             fill = top,
             label = top))+
  facet_grid(Reactor_Treatment_Dose  ~ .)+
  # scale_fill_manual(values = c("red", "yellow", "green3"))+
  geom_flow(stat = "alluvium", lode.guidance = "frontback", color = "darkgray")+
  geom_stratum() +
  theme_light() +
  # geom_text(stat = "alluvium", aes(label = top), lode.guidance = "frontback") +
  scale_fill_manual(values = col_ASVs) +
  # scale_color_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  # scale_fill_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  ylab("Proportion") -> p_allu_human2_ASV

p_allu_human2_ASV + theme(legend.position = "none") -> p_allu_human2_ASV_noleg 

p_allu_human2_ASV_noleg

p_allu_human2_ASV %>% 
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() ->  p_allu_human2_ASV_leg
```

```{r}
p_allu_human2_ASV_noleg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 2 , paper = "A4",  scaling = 2,
                    file = out_pptx)

p_allu_human2_ASV_leg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 4, 
                    height = 0.618 * 317.48031496 * 2 , paper = "A3",  scaling = 2,
                    file = out_pptx)
```

## gram plots:

histogram of gram+ gram- & intrinsicly resistant or not an annotation color in a heatmap would be cool as well.

```{r}
"data/processed/16S/16S_based_gram.RDS" %>% 
  here::here() %>% 
  readRDS() -> gram

gram %>% 
  summarise_all(~ sum(is.na(.)))
```

```{r}
dim(gram)

intersect(gram$ASV,
          tax_table(ps_filtered)[,"Strain"] %>%  as.character() ) %>%  length() 
```

Let's take the Class info here.

## intrinsic:

```{r}
"data/processed/16S/16S_based_intrinsinc_AMR.RDS" %>% 
  here::here() %>% 
  readRDS() -> intrinsic_AMR
```

Wich level to chose? the one with less NA:

```{r}
intrinsic_AMR %>% 
  select(starts_with("intrinsic")) %>% 
  summary()
```

Let's take the Strain info here.

```{r}
ps_rare_gram_int <- ps_rare

# We want sample, reactor..., day_0f_treatment 
ps_rare_gram_int %>% 
  tax_table() %>% 
  data.frame() %>% 
  rownames_to_column("ASVid") %>% 
  left_join(gram %>% dplyr::rename("Strain" = "ASV") %>%  
              select(Strain , gram_neg_class)
  ) %>%  #,
  # by = c("Strain" = "ASV"),
  # suffix = c("", ".y")) %>% 
  select(ASVid:Strain, gram_neg_class) %>% 
  dplyr::rename('gram_stain' = 'gram_neg_class') %>% 
  mutate(gram_stain = replace_na(gram_stain, "unknown-gram")) %>% 
  left_join(., intrinsic_AMR,
            suffix = c("", ".y"),
            by = c("Strain" = "ASV")) %>%
  mutate(intrinsic = ifelse(intrinsic_strain_van == TRUE & intrinsic_strain_ctx == TRUE, "intrinsic-VAN-CTX",  ifelse(intrinsic_strain_van == TRUE, "intrinsic-VAN", ifelse(intrinsic_strain_ctx == TRUE, "intrinsic-CTX", "No-intrinsic")))) %>% 
  select(ASVid:gram_stain,intrinsic) %>%
  mutate(gram_intrinsic = paste0(gram_stain, 
                                 ifelse(is.na(intrinsic), "", paste0("_",intrinsic)))) %>% 
  # dplyr::rename('intrinsic_CTX' = 'intrinsic_strain_ctx',
  # 'intrinsic_VAN' = 'intrinsic_strain_van') %>% 
  column_to_rownames('ASVid') %>% 
  as.matrix() -> tax_table(ps_rare_gram_int)
```

### human2:

#### gram:

```{r}
ps_rare_gram_int %>% 
  subset_samples(Model2 == "Human2" & Treatment != "DONOR" & Day_of_Treatment >= -1) %>% 
  microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  physeq_sel_tax_table(c("Kingdom","gram_stain")) %>% 
  # subset_taxa(Strain %in% top_taxa_per_group_h2) %>% # extract only the taxa to display - after percentage normalisation
  microViz::phyloseq_validate() %>%
  microViz::tax_fix() %>%
  microViz::ps_arrange(Day_of_Treatment) %>% 
  # microViz::tax_mutate(gram = left_join(tax_table(.) %>% data.frame(), gram,
  # by = c("Strain" = "ASV") %>% select(gram_neg_class))) %>% 
  microViz::comp_barplot(
    tax_transform_for_plot = "identity",
    tax_level = "gram_stain",
    label = "Day_of_Treatment", # name an alternative variable to label axes
    palette = microViz::distinct_palette(3, pal = "greenArmytage"),
    n_taxa = 10, # give more taxa unique colours
    merge_other = FALSE, # split the "other" category to display alpha diversity
    bar_width = 0.9, # reduce the bar width to 70% of one row
    bar_outline_colour = "grey5",
    sample_order = "default") +
  ylab("Proportion - %") + xlab( " Day (Treatment) ") -> p_gram_h2

p_gram_h2 + theme_light() + facet_grid(Kingdom ~ Reactor_Treatment_Dose, scales = "free", space = "free_x", drop = TRUE) -> p_gram_h2

p_gram_h2 + theme(legend.position = "none") -> p_gram_h2_no_leg

p_gram_h2_no_leg

p_gram_h2 %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_gram_h2_leg

p_gram_h2_leg
```

```{r}
p_gram_h2$data %>%
  # filter(Model2 == "Human2") %>%
  mutate(top = factor(top, levels = c("Gram-positive", "Gram-negative", "unknown-gram"))) %>%
  # mutate(combo = paste0(Model2, "_" , Reactor_F1_F2)) %>%
  # mutate(combo = as.factor(combo)) %>%
  select(SAMPLE, top,OTU, Abundance, Day_of_Treatment, Reactor_Treatment_Dose) %>% 
  ggplot(.,
         aes(x = Day_of_Treatment, 
             y = Abundance,
             stratum = top, 
             alluvium = OTU,
             fill = top,
             label = top))+
  facet_grid(Reactor_Treatment_Dose  ~ .)+
  # scale_fill_manual(values = c("red", "yellow", "green3"))+
  geom_flow(stat = "alluvium", lode.guidance = "frontback", color = "darkgray")+
  geom_stratum() +
  theme_light() +
  # geom_text(stat = "alluvium", aes(label = top), lode.guidance = "frontback") +
  # scale_fill_manual(values = col_ASVs) +
  scale_color_manual(values  =   microViz::distinct_palette(3, pal = "greenArmytage")) +
  scale_fill_manual(values  =   microViz::distinct_palette(3, pal = "greenArmytage")) +
  # scale_color_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  # scale_fill_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  ylab("Proportion - %") -> p_allu_human2_gram

p_allu_human2_gram + theme(legend.position = "none") -> p_allu_human2_gram_noleg 

p_allu_human2_gram_noleg 

p_allu_human2_gram %>% 
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() ->  p_allu_human2_gram_leg

p_allu_human2_gram_leg
```

#### intrinsic:

```{r}
ps_rare_gram_int %>% 
  subset_samples(Model2 == "Human2" & Treatment != "DONOR" & Day_of_Treatment >= -1) %>% 
  microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  physeq_sel_tax_table(c("Kingdom", "intrinsic")) %>% 
  # microbiome::transform("log") %>% 
  # speedyseq::tax_glom("intrinsic") %>% 
  # physeq_sel_tax_table(c("gram_stain","intrinsic")) %>% 
  # physeq_glom_rename(phyloseq = ., speedyseq = TRUE, rename_ASV = FALSE, taxrank = "gram_stain") %>% 
  # subset_taxa(Strain %in% top_taxa_per_group_h2) %>% # extract only the taxa to display - after percentage normalisation
  microViz::phyloseq_validate() %>%
  microViz::tax_fix() %>%
  microViz::ps_arrange(Day_of_Treatment) %>% 
  # microViz::tax_mutate(gram = left_join(tax_table(.) %>% data.frame(), gram,
  # by = c("Strain" = "ASV") %>% select(gram_neg_class))) %>% 
  microViz::comp_barplot(
    tax_transform_for_plot = "identity",
    tax_level = "intrinsic",
    label = "Day_of_Treatment", # name an alternative variable to label axes
    # palette = microViz::distinct_palette(3, pal = "brewerPlus"),
    n_taxa = 10, # give more taxa unique colours
    merge_other = FALSE, # split the "other" category to display alpha diversity
    bar_width = 0.9, # reduce the bar width to 70% of one row
    bar_outline_colour = "grey5",
    sample_order = "default") +
  ylab("Proportion - %") + xlab( " Day (Treatment) ") -> p_int_h2

p_int_h2 + theme_light() +facet_grid(Kingdom ~ Reactor_Treatment_Dose, scales = "free", space = "free_x", drop = TRUE) -> p_int_h2

p_int_h2 %>% 
  ggpubr::set_palette(p = ., palette = "jco") -> p_int_h2

p_int_h2 + theme(legend.position = "none") -> p_int_h2_no_leg

p_int_h2_no_leg

p_int_h2 %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_int_h2_leg

p_int_h2_leg
```

```{r}
p_int_h2$data %>%
  # filter(Model2 == "Human2") %>%
  # microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  # mutate(combo = paste0(Model2, "_" , Reactor_F1_F2)) %>%
  # mutate(combo = as.factor(combo)) %>%
  select(SAMPLE, top,OTU, Abundance, Day_of_Treatment, Reactor_Treatment_Dose) %>% 
  ggplot(.,
         aes(x = Day_of_Treatment, 
             y = Abundance,
             stratum = top, 
             alluvium = OTU,
             fill = top,
             label = top))+
  facet_grid(Reactor_Treatment_Dose  ~ .)+
  # scale_fill_manual(values = c("red", "yellow", "green3"))+
  geom_flow(stat = "alluvium", lode.guidance = "frontback", color = "darkgray")+
  geom_stratum() +
  theme_light() +
  # geom_text(stat = "alluvium", aes(label = top), lode.guidance = "frontback") +
  # scale_fill_manual(values = col_ASVs) +
  scale_color_manual(values  =  microViz::distinct_palette(3, pal = "greenArmytage")) +
  # scale_color_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  # scale_fill_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  ylab("Proportion - %") -> p_allu_human2_int

p_allu_human2_int %>% 
  ggpubr::set_palette(p = ., palette = "jco") -> p_allu_human2_int

p_allu_human2_int + theme(legend.position = "none") -> p_allu_human2_int_noleg 

p_allu_human2_int_noleg 

p_allu_human2_int %>% 
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() ->  p_allu_human2_int_leg
```
```{r}
ggpubr::ggarrange(p_allu_human2_noleg + xlab(NULL),
                  p_allu_human2_ASV_noleg + ylab(NULL) + xlab(NULL),
                  p_allu_human2_int_noleg +  coord_cartesian(ylim = c(50,100)),
                  p_allu_human2_int_noleg+ ylab(NULL),
                  align = "v", 
                  ncol = 2, nrow = 2,
                  heights = c(1, 0.6), widths = c(1,1),
                  common.legend = FALSE) -> p_allu_human2

p_allu_human2

p_allu_human2 %>%
  export::graph2ppt(append = TRUE,
                    width = 317.48031496 * 1,
                    height = 0.618 * 317.48031496 * 1.75 , paper = "A4",  scaling = 2,
                    file = out_pptx)
```


```{r}
ggpubr::ggarrange(p_gram_h2_no_leg + ylab(NULL) + 
                    theme(axis.title.x=element_blank(),
                          axis.text.x=element_blank(),
                          axis.ticks.x=element_blank()),
                  p_hist_human2_fam_no_leg + 
                    theme(
                      strip.background = element_blank(),
                      strip.text.x = element_blank()
                    ) +
                    theme(axis.title.x=element_blank(),
                          axis.text.x=element_blank(),
                          axis.ticks.x=element_blank()),
                  p_hist_human2_strain_no_leg + 
                    theme(
                      strip.background = element_blank(),
                      strip.text.x = element_blank()
                    ) + ylab(NULL) + 
                    theme(axis.title.x=element_blank(),
                          axis.text.x=element_blank(),
                          axis.ticks.x=element_blank()),
                  p_int_h2_no_leg + ylab(NULL) + coord_cartesian(ylim = c(50,100)) +
                    theme(
                      strip.background = element_blank(),
                      strip.text.x = element_blank()
                    ), 
                  align = "v",
                  ncol = 1, 
                  heights = c(0.15, 0.4, 0.8,0.15),
                  common.legend = FALSE) -> all_plot_h2

all_plot_h2

all_plot_h2 %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 3.5, 
                    height = 0.618 * 317.48031496 * 5 , paper = "A4",  scaling = 2,
                    file = out_pptx)


p_gram_h2_leg  %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 1 , paper = "A4",  scaling = 2,
                    file = out_pptx)

p_hist_human2_fam_leg %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 1 , paper = "A4",  scaling = 2,
                    file = out_pptx)

p_hist_human2_strain_leg  %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 4, 
                    height = 0.618 * 317.48031496 * 2 , paper = "A3",  scaling = 2,
                    file = out_pptx)

p_int_h2_leg  %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 1 , paper = "A4",  scaling = 2,
                    file = out_pptx)


p_gram_h2_no_leg$data %>% 
  select(OTU, Sample, Abundance, Treatment, Day_of_Treatment) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "p_gram_h2",
                   showNA = TRUE)
p_hist_human2_fam$data %>% 
  select(OTU, Sample, Abundance, Treatment, Day_of_Treatment) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "p_hist_human2_fam",
                   showNA = TRUE)
p_hist_human2_strain$data %>% 
  select(Sample, top ,Abundance, Treatment, Day_of_Treatment) %>% 
  pivot_wider(names_from = top, values_from = Abundance) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "p_hist_human2_strain",
                   showNA = TRUE)
p_int_h2$data %>% 
  select(OTU, Sample, Abundance, Treatment, Day_of_Treatment) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "p_int_h2",
                   showNA = TRUE)

```


```{r}
# ggpubr::ggarrange(p_gram_h2_no_leg + ylab(NULL) + 
#                     theme(axis.title.x=element_blank(),
#                           axis.text.x=element_blank(),
#                           axis.ticks.x=element_blank()),
#                   p_hist_human2_fam_no_leg + 
#                     theme(
#                       strip.background = element_blank(),
#                       strip.text.x = element_blank()
#                     ) +
#                     theme(axis.title.x=element_blank(),
#                           axis.text.x=element_blank(),
#                           axis.ticks.x=element_blank()),
#                   treat_speces + 
#                     theme(
#                       strip.background = element_blank(),
#                       strip.text.x = element_blank()
#                     ) + ylab(NULL) + 
#                     theme(axis.title.x=element_blank(),
#                           axis.text.x=element_blank(),
#                           axis.ticks.x=element_blank()),
#                   p_int_h2_no_leg + ylab(NULL) + coord_cartesian(ylim = c(50,100)) +
#                     theme(
#                       strip.background = element_blank(),
#                       strip.text.x = element_blank()
#                     ), 
#                   align = "hv",
#                   ncol = 1, 
#                   heights = c(0.15, 0.4, 0.8,0.15),
#                   common.legend = FALSE) -> all_plot_h2
# 
# all_plot_h2
# 
# all_plot_h2 %>% 
#   export::graph2ppt(append = TRUE, 
#                     width = 317.48031496 * 3.5, 
#                     height = 0.618 * 317.48031496 * 5 , paper = "A4",  scaling = 2,
#                     file = out_pptx)

```



### human1 - f1 :

#### gram:

```{r}
ps_rare_gram_int %>% 
  subset_samples(Model2 == "Human1" & Treatment != "DONOR" & Day_of_Treatment >= -1 & Fermentation == 1) %>% 
  # microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  physeq_sel_tax_table(c("Kingdom","gram_stain")) %>% 
  # subset_taxa(Strain %in% top_taxa_per_group_h2) %>% # extract only the taxa to display - after percentage normalisation
  microViz::phyloseq_validate() %>%
  microViz::tax_fix() %>%
  microViz::ps_arrange(Day_of_Treatment) %>% 
  # microViz::tax_mutate(gram = left_join(tax_table(.) %>% data.frame(), gram,
  # by = c("Strain" = "ASV") %>% select(gram_neg_class))) %>% 
  microViz::comp_barplot(
    tax_transform_for_plot = "identity",
    tax_level = "gram_stain",
    label = "Day_of_Treatment", # name an alternative variable to label axes
    palette = microViz::distinct_palette(2, pal = "greenArmytage"),
    n_taxa = 10, # give more taxa unique colours
    merge_other = FALSE, # split the "other" category to display alpha diversity
    bar_width = 0.9, # reduce the bar width to 70% of one row
    bar_outline_colour = "grey5",
    sample_order = "default") +
  ylab("Proportion - %") + xlab( " Day (Treatment) ") -> p_gram_h1_f1

p_gram_h1_f1 + facet_grid(Kingdom ~ Reactor_Treatment_Dose, scales = "free", space = "free_x", drop = TRUE) -> p_gram_h1_f1

p_gram_h1_f1 + theme_light() +theme(legend.position = "none") -> p_gram_h1_f1_no_leg

p_gram_h1_f1_no_leg

p_gram_h1_f1 %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_gram_h1_f1_leg

```


```{r}
p_gram_h1_f1$data %>%
  # filter(Model2 == "Human2") %>%
  mutate(top = factor(top, levels = c("Gram-positive", "Gram-negative", "unknown-gram"))) %>%
  # mutate(combo = paste0(Model2, "_" , Reactor_F1_F2)) %>%
  # mutate(combo = as.factor(combo)) %>%
  select(SAMPLE, top,OTU, Abundance, Day_of_Treatment, Reactor_Treatment_Dose) %>% 
  ggplot(.,
         aes(x = Day_of_Treatment, 
             y = Abundance,
             stratum = top, 
             alluvium = OTU,
             fill = top,
             label = top))+
  facet_grid(Reactor_Treatment_Dose  ~ .)+
  # scale_fill_manual(values = c("red", "yellow", "green3"))+
  geom_flow(stat = "alluvium", lode.guidance = "frontback", color = "darkgray")+
  geom_stratum() +
  theme_light() +
  # geom_text(stat = "alluvium", aes(label = top), lode.guidance = "frontback") +
  # scale_fill_manual(values = col_ASVs) +
  scale_color_manual(values  =   microViz::distinct_palette(3, pal = "greenArmytage")) +
  scale_fill_manual(values  =   microViz::distinct_palette(3, pal = "greenArmytage")) +
  # scale_color_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  # scale_fill_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  ylab("Proportion - %") -> p_allu_h1_f1_gram 

p_allu_h1_f1_gram + theme(legend.position = "none") -> p_allu_h1_f1_gram_noleg 

p_allu_h1_f1_gram_noleg 

p_allu_h1_f1_gram %>% 
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() ->  p_allu_h1_f1_gram_leg

p_allu_h1_f1_gram_leg
```



#### intrinsic:

```{r}
ps_rare_gram_int %>% 
  subset_samples(Model2 == "Human1" & Treatment != "DONOR" & Day_of_Treatment >= -1 & Fermentation == 1) %>% 
  microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  physeq_sel_tax_table(c("Kingdom", "intrinsic")) %>% 
  # microbiome::transform("log") %>% 
  # speedyseq::tax_glom("intrinsic") %>% 
  # physeq_sel_tax_table(c("gram_stain","intrinsic")) %>% 
  # physeq_glom_rename(phyloseq = ., speedyseq = TRUE, rename_ASV = FALSE, taxrank = "gram_stain") %>% 
  # subset_taxa(Strain %in% top_taxa_per_group_h2) %>% # extract only the taxa to display - after percentage normalisation
  microViz::phyloseq_validate() %>%
  microViz::tax_fix() %>%
  microViz::ps_arrange(Day_of_Treatment) %>% 
  # microViz::tax_mutate(gram = left_join(tax_table(.) %>% data.frame(), gram,
  # by = c("Strain" = "ASV") %>% select(gram_neg_class))) %>% 
  microViz::comp_barplot(
    tax_transform_for_plot = "identity",
    tax_level = "intrinsic",
    label = "Day_of_Treatment", # name an alternative variable to label axes
    # palette = microViz::distinct_palette(3, pal = "brewerPlus"),
    n_taxa = 10, # give more taxa unique colours
    merge_other = FALSE, # split the "other" category to display alpha diversity
    bar_width = 0.9, # reduce the bar width to 70% of one row
    bar_outline_colour = "grey5",
    sample_order = "default") +
  ylab("Proportion - %") + xlab( " Day (Treatment) ") -> p_int_h1_f1

p_int_h1_f1 + facet_grid(Kingdom ~ Reactor_Treatment_Dose, scales = "free", space = "free_x", drop = TRUE) -> p_int_h1_f1

p_int_h1_f1 %>% 
  ggpubr::set_palette(p = ., palette = "jco") -> p_int_h1_f1

p_int_h1_f1 + theme_light() +theme(legend.position = "none") -> p_int_h1_f1_no_leg

p_int_h1_f1_no_leg

p_int_h1_f1 %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_int_h1_f1_leg

p_int_h1_f1_leg
```


```{r}
p_int_h1_f1$data %>%
  # filter(Model2 == "Human2") %>%
  # microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  # mutate(combo = paste0(Model2, "_" , Reactor_F1_F2)) %>%
  # mutate(combo = as.factor(combo)) %>%
  select(SAMPLE, top,OTU, Abundance, Day_of_Treatment, Reactor_Treatment_Dose) %>% 
  ggplot(.,
         aes(x = Day_of_Treatment, 
             y = Abundance,
             stratum = top, 
             alluvium = OTU,
             fill = top,
             label = top))+
  facet_grid(Reactor_Treatment_Dose  ~ .)+
  # scale_fill_manual(values = c("red", "yellow", "green3"))+
  geom_flow(stat = "alluvium", lode.guidance = "frontback", color = "darkgray")+
  geom_stratum() +
  theme_light() +
  # geom_text(stat = "alluvium", aes(label = top), lode.guidance = "frontback") +
  # scale_fill_manual(values = col_ASVs) +
  scale_color_manual(values  =  microViz::distinct_palette(3, pal = "greenArmytage")) +
  # scale_color_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  # scale_fill_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  ylab("Proportion - %") -> p_allu_h1_f1_int

p_allu_h1_f1_int %>% 
  ggpubr::set_palette(p = ., palette = "jco") -> p_allu_h1_f1_int

p_allu_h1_f1_int + theme(legend.position = "none") -> p_allu_h1_f1_int_noleg 

p_allu_h1_f1_int_noleg 

p_allu_h1_f1_int %>% 
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_allu_h1_f1_int_leg
```

```{r}
ggpubr::ggarrange(p_allu_human1_f1_noleg + xlab(NULL),
                  p_allu_human1_ASV_f1_noleg + ylab(NULL) + xlab(NULL),
                  p_allu_h1_f1_int_noleg +  coord_cartesian(ylim = c(50,100)),
                  p_allu_h1_f1_gram_noleg + ylab(NULL),
                  align = "v", 
                  ncol = 2, nrow = 2,
                  heights = c(1, 0.6), widths = c(1,1),
                  common.legend = FALSE) -> p_allu_human1_f1

p_allu_human1_f1

p_allu_human1_f1 %>%
  export::graph2ppt(append = TRUE,
                    width = 317.48031496 * 1,
                    height = 0.618 * 317.48031496 * 1.75 , paper = "A4",  scaling = 2,
                    file = out_pptx)
```

```{r}
ggpubr::ggarrange(p_gram_h1_f1_no_leg + ylab(NULL) + 
                    theme(axis.title.x=element_blank(),
                          axis.text.x=element_blank(),
                          axis.ticks.x=element_blank()),
                  p_hist_human1_f1_fam_no_leg + 
                    theme(
                      strip.background = element_blank(),
                      strip.text.x = element_blank()
                    ) +
                    theme(axis.title.x=element_blank(),
                          axis.text.x=element_blank(),
                          axis.ticks.x=element_blank()),
                  p_hist_human1_f1_strain_no_leg + 
                    theme(
                      strip.background = element_blank(),
                      strip.text.x = element_blank()
                    ) + ylab(NULL) + 
                    theme(axis.title.x=element_blank(),
                          axis.text.x=element_blank(),
                          axis.ticks.x=element_blank()),
                  p_int_h1_f1_no_leg + ylab(NULL) + coord_cartesian(ylim = c(50,100)) +
                    theme(
                      strip.background = element_blank(),
                      strip.text.x = element_blank()
                    ), 
                  align = "v",
                  ncol = 1, 
                  heights = c(0.15, 0.4, 0.8,0.15),
                  common.legend = FALSE) -> all_plot_h1_f1

all_plot_h1_f1

all_plot_h1_f1 %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 3.5, 
                    height = 0.618 * 317.48031496 * 5 , paper = "A4",  scaling = 2,
                    file = out_pptx)


p_gram_h1_f1_leg  %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 1 , paper = "A4",  scaling = 2,
                    file = out_pptx)

p_hist_human1_f1_fam_leg  %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 1 , paper = "A4",  scaling = 2,
                    file = out_pptx)

p_hist_human1_f1_strain_leg  %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 4, 
                    height = 0.618 * 317.48031496 * 2 , paper = "A3",  scaling = 2,
                    file = out_pptx)

p_int_h1_f1_leg  %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 1 , paper = "A4",  scaling = 2,
                    file = out_pptx)



p_gram_h1_f1$data %>% 
  select(OTU, Sample, Abundance, Treatment, Day_of_Treatment) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "p_gram_h1_f1",
                   showNA = TRUE)
p_hist_human1_f1_fam$data %>% 
  select(OTU, Sample, Abundance, Treatment, Day_of_Treatment) %>% 
  pivot_wider(names_from = OTU, values_from = Abundance) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "p_hist_human1_f1_fam",
                   showNA = TRUE)
p_hist_human1_f1_strain$data %>% 
  select(Sample, top ,Abundance, Treatment, Day_of_Treatment) %>% 
  pivot_wider(names_from = top, values_from = Abundance) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "p_hist_human1_f1_strain",
                   showNA = TRUE)
p_int_h1_f1$data %>% 
  select(OTU, Sample, Abundance, Treatment, Day_of_Treatment) %>% 
  pivot_wider(names_from = OTU, values_from = Abundance) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "p_int_h1_f1",
                   showNA = TRUE)

```


### human1 - f2 :

#### gram:

```{r}
ps_rare_gram_int %>% 
  subset_samples(Model2 == "Human1" & Treatment != "DONOR" & Day_of_Treatment >= -1 & Fermentation == 2) %>% 
  microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  physeq_sel_tax_table(c("Kingdom","gram_stain")) %>% 
  # subset_taxa(Strain %in% top_taxa_per_group_h2) %>% # extract only the taxa to display - after percentage normalisation
  microViz::phyloseq_validate() %>%
  microViz::tax_fix() %>%
  microViz::ps_arrange(Day_of_Treatment) %>% 
  # microViz::tax_mutate(gram = left_join(tax_table(.) %>% data.frame(), gram,
  # by = c("Strain" = "ASV") %>% select(gram_neg_class))) %>% 
  microViz::comp_barplot(
    tax_transform_for_plot = "identity",
    tax_level = "gram_stain",
    label = "Day_of_Treatment", # name an alternative variable to label axes
    palette = microViz::distinct_palette(2, pal = "greenArmytage"),
    n_taxa = 10, # give more taxa unique colours
    merge_other = FALSE, # split the "other" category to display alpha diversity
    bar_width = 0.9, # reduce the bar width to 70% of one row
    bar_outline_colour = "grey5",
    sample_order = "default") +
  ylab("Proportion - %") + xlab( " Day (Treatment) ") -> p_gram_h1_f2

p_gram_h1_f2 + facet_grid(Kingdom ~ Reactor_Treatment_Dose, scales = "free", space = "free_x", drop = TRUE) -> p_gram_h1_f2

p_gram_h1_f2 + theme_light() + theme(legend.position = "none") -> p_gram_h1_f2_no_leg

p_gram_h1_f2_no_leg

p_gram_h1_f2 %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_gram_h1_f2_leg

```


```{r}
p_gram_h1_f2$data %>%
  # filter(Model2 == "Human2") %>%
  mutate(top = factor(top, levels = c("Gram-positive", "Gram-negative", "unknown-gram"))) %>%
  # mutate(combo = paste0(Model2, "_" , Reactor_F1_F2)) %>%
  # mutate(combo = as.factor(combo)) %>%
  select(SAMPLE, top,OTU, Abundance, Day_of_Treatment, Reactor_Treatment_Dose) %>% 
  ggplot(.,
         aes(x = Day_of_Treatment, 
             y = Abundance,
             stratum = top, 
             alluvium = OTU,
             fill = top,
             label = top))+
  facet_grid(Reactor_Treatment_Dose  ~ .)+
  # scale_fill_manual(values = c("red", "yellow", "green3"))+
  geom_flow(stat = "alluvium", lode.guidance = "frontback", color = "darkgray")+
  geom_stratum() +
  theme_light() +
  # geom_text(stat = "alluvium", aes(label = top), lode.guidance = "frontback") +
  # scale_fill_manual(values = col_ASVs) +
  scale_color_manual(values  =   microViz::distinct_palette(3, pal = "greenArmytage")) +
  scale_fill_manual(values  =   microViz::distinct_palette(3, pal = "greenArmytage")) +
  # scale_color_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  # scale_fill_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  ylab("Proportion - %") -> p_allu_h1_f2_gram 

p_allu_h1_f2_gram + theme(legend.position = "none") -> p_allu_h1_f2_gram_noleg 

p_allu_h1_f2_gram_noleg 

p_allu_h1_f2_gram %>% 
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() ->  p_allu_h1_f2_gram_leg

p_allu_h1_f2_gram_leg
```

#### intrinsic:

```{r}
ps_rare_gram_int %>% 
  subset_samples(Model2 == "Human1" & Treatment != "DONOR" & Day_of_Treatment >= -1 & Fermentation == 2) %>% 
  microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>%
  physeq_sel_tax_table(c("Kingdom", "intrinsic")) %>% 
  # microbiome::transform("log") %>% 
  # speedyseq::tax_glom("intrinsic") %>% 
  # physeq_sel_tax_table(c("gram_stain","intrinsic")) %>% 
  # physeq_glom_rename(phyloseq = ., speedyseq = TRUE, rename_ASV = FALSE, taxrank = "gram_stain") %>% 
  # subset_taxa(Strain %in% top_taxa_per_group_h2) %>% # extract only the taxa to display - after percentage normalisation
  microViz::phyloseq_validate() %>%
  microViz::tax_fix() %>%
  microViz::ps_arrange(Day_of_Treatment) %>% 
  # microViz::tax_mutate(gram = left_join(tax_table(.) %>% data.frame(), gram,
  # by = c("Strain" = "ASV") %>% select(gram_neg_class))) %>% 
  microViz::comp_barplot(
    tax_transform_for_plot = "identity",
    tax_level = "intrinsic",
    label = "Day_of_Treatment", # name an alternative variable to label axes
    # palette = microViz::distinct_palette(3, pal = "brewerPlus"),
    n_taxa = 10, # give more taxa unique colours
    merge_other = FALSE, # split the "other" category to display alpha diversity
    bar_width = 0.9, # reduce the bar width to 70% of one row
    bar_outline_colour = "grey5",
    sample_order = "default") +
  ylab("Proportion - %") + xlab( " Day (Treatment) ") -> p_int_h1_f2

p_int_h1_f2 + facet_grid(Kingdom ~ Reactor_Treatment_Dose, scales = "free", space = "free_x", drop = TRUE) -> p_int_h1_f2

p_int_h1_f2 %>% 
  ggpubr::set_palette(p = ., palette = "jco") -> p_int_h1_f2

p_int_h1_f2 + theme_light() +theme(legend.position = "none") -> p_int_h1_f2_no_leg

p_int_h1_f2_no_leg

p_int_h1_f2 %>%  
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_int_h1_f2_leg

p_int_h1_f2_leg

```


```{r}
p_int_h1_f2$data %>%
  # filter(Model2 == "Human2") %>%
  # microViz::ps_mutate(Reactor_Treatment_Dose = paste0(Reactor_Treatment_Dose, "_", Fermentation)) %>% 
  # mutate(combo = paste0(Model2, "_" , Reactor_F1_F2)) %>%
  # mutate(combo = as.factor(combo)) %>%
  select(SAMPLE, top,OTU, Abundance, Day_of_Treatment, Reactor_Treatment_Dose) %>% 
  ggplot(.,
         aes(x = Day_of_Treatment, 
             y = Abundance,
             stratum = top, 
             alluvium = OTU,
             fill = top,
             label = top))+
  facet_grid(Reactor_Treatment_Dose  ~ .)+
  # scale_fill_manual(values = c("red", "yellow", "green3"))+
  geom_flow(stat = "alluvium", lode.guidance = "frontback", color = "darkgray")+
  geom_stratum() +
  theme_light() +
  # geom_text(stat = "alluvium", aes(label = top), lode.guidance = "frontback") +
  # scale_fill_manual(values = col_ASVs) +
  scale_color_manual(values  =  microViz::distinct_palette(3, pal = "greenArmytage")) +
  # scale_color_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  # scale_fill_manual(values  = microViz::distinct_palette(10, pal = "kelly")) +
  ylab("Proportion - %") -> p_allu_h1_f2_int

p_allu_h1_f2_int %>% 
  ggpubr::set_palette(p = ., palette = "jco") -> p_allu_h1_f2_int

p_allu_h1_f2_int + theme(legend.position = "none") -> p_allu_h1_f2_int_noleg 

p_allu_h1_f2_int_noleg 

p_allu_h1_f2_int %>% 
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p_allu_h1_f2_int_leg
```


```{r}
# ggpubr::ggarrange(p_allu_human2_noleg,
#                   p_allu_human2_ASV_noleg + ylab(NULL),
#                   align = "v",
#                   ncol = 2, 
#                   heights = c(1, 1),
#                   common.legend = FALSE) -> p_allu_human2
# 
# p_allu_human2

# p_allu_human2 %>% 
#   export::graph2ppt(append = TRUE, 
#                     width = 317.48031496 * 1, 
#                     height = 0.618 * 317.48031496 * 1 , paper = "A4",  scaling = 2,
#                     file = out_pptx)
```

```{r}
ggpubr::ggarrange(p_allu_human1_f2_noleg + xlab(NULL),
                  p_allu_human1_ASV_f2_noleg + ylab(NULL) + xlab(NULL),
                  p_allu_h1_f2_int_noleg +  coord_cartesian(ylim = c(50,100)),
                  p_allu_h1_f2_gram_noleg + ylab(NULL),
                  align = "v", 
                  ncol = 2, nrow = 2,
                  heights = c(1, 0.6), widths = c(1,1),
                  common.legend = FALSE) -> p_allu_human1_f2

p_allu_human1_f2

p_allu_human1_f2 %>%
  export::graph2ppt(append = TRUE,
                    width = 317.48031496 * 1,
                    height = 0.618 * 317.48031496 * 1.75 , paper = "A4",  scaling = 2,
                    file = out_pptx)
```

```{r}
ggpubr::ggarrange(p_allu_human1_f2_noleg,
                  p_allu_human1_ASV_f2_noleg + ylab(NULL),
                  
                  align = "v",
                  ncol = 2, 
                  heights = c(1, 1),
                  common.legend = FALSE) -> p_allu_human1_f2

p_allu_human1_f2

p_allu_human1_f2 %>%
  export::graph2ppt(append = TRUE,
                    width = 317.48031496 * 1,
                    height = 0.618 * 317.48031496 * 1 , paper = "A4",  scaling = 2,
                    file = out_pptx)
```



```{r}
ggpubr::ggarrange(p_gram_h1_f2_no_leg + ylab(NULL) + 
                    theme(axis.title.x=element_blank(),
                          axis.text.x=element_blank(),
                          axis.ticks.x=element_blank()),
                  p_hist_human1_f2_fam_no_leg + 
                    theme(
                      strip.background = element_blank(),
                      strip.text.x = element_blank()
                    ) +
                    theme(axis.title.x=element_blank(),
                          axis.text.x=element_blank(),
                          axis.ticks.x=element_blank()),
                  p_hist_human1_f2_strain_no_leg + 
                    theme(
                      strip.background = element_blank(),
                      strip.text.x = element_blank()
                    ) + ylab(NULL) + 
                    theme(axis.title.x=element_blank(),
                          axis.text.x=element_blank(),
                          axis.ticks.x=element_blank()),
                  p_int_h1_f2_no_leg + ylab(NULL) + coord_cartesian(ylim = c(50,100)) +
                    theme(
                      strip.background = element_blank(),
                      strip.text.x = element_blank()
                    ), 
                  align = "v",
                  ncol = 1, 
                  heights = c(0.15, 0.4, 0.8,0.15),
                  common.legend = FALSE) -> all_plot_h1_f2

all_plot_h1_f2

all_plot_h1_f2 %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 3.5, 
                    height = 0.618 * 317.48031496 * 5 , paper = "A4",  scaling = 2,
                    file = out_pptx)

p_gram_h1_f2_leg  %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 1 , paper = "A4",  scaling = 2,
                    file = out_pptx)

p_hist_human1_f2_fam_leg  %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 1 , paper = "A4",  scaling = 2,
                    file = out_pptx)

p_hist_human1_f2_strain_leg  %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 4, 
                    height = 0.618 * 317.48031496 * 2 , paper = "A3",  scaling = 2,
                    file = out_pptx)
p_int_h1_f2_leg  %>% 
  export::graph2ppt(append = TRUE, 
                    width = 317.48031496 * 1, 
                    height = 0.618 * 317.48031496 * 1 , paper = "A4",  scaling = 2,
                    file = out_pptx)

p_gram_h1_f2$data %>% 
  select(OTU, Sample, Abundance, Treatment, Day_of_Treatment) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "p_gram_h1_f2",
                   showNA = TRUE)

p_hist_human1_f2_fam$data %>% 
  select(OTU, Sample, Abundance, Treatment, Day_of_Treatment) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "p_hist_human1_f2_fam",
                   showNA = TRUE)

p_hist_human1_f2_strain$data %>% 
  select(Sample, top ,Abundance, Treatment, Day_of_Treatment) %>% 
  pivot_wider(names_from = top, values_from = Abundance) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "p_hist_human1_f2_strain",
                   showNA = TRUE)

p_int_h1_f2$data %>% 
  select(OTU, Sample, Abundance, Treatment, Day_of_Treatment) %>% 
  xlsx::write.xlsx(file = out_xlsx,
                   append = TRUE,
                   sheetName = "p_int_h1_f2",
                   showNA = TRUE)

# htmlwidgets::saveWidget(plotly::as_widget(plotly::ggplotly(p_int_h1_f2)),
#                         "~/Desktop/index.html")
```

-   explore stability metrics: <https://microbiome.github.io/tutorials/Stability.html> to detect taxa impacted. or filter by coefficient variation. or Hannah's classification

Bimodal population distribution across individuals is often associated with instable intermediate abundances within individuals:

```{r,  eval = FALSE}
# Use relative abundances
pseq <- microbiome::transform(ps_rare %>%  subset_samples(Model2 == "Human2"), "compositional")

# Merge rare taxa to speed up examples
pseq <- microbiome::aggregate_rare(ps_rare, level = "Strain", detection = .1/100, prevalence = 10/100)

# For cross-sectional analysis, include
# only the treated points.

pseq0 <- subset_samples(pseq, Day_of_Treatment > 1)

# intermediate_stability: within subjects:
pseq %>% 
  microViz::ps_mutate(subject = Reactor,
                      time = Day_of_Treatment) %>% 
  intermediate_stability(., #output = "scores", 
                         output = "scores") -> intermediate.stability

intermediate.stability %>% 
  head()
```

```{r, message = FALSE, warning = FALSE, eval = FALSE}
# Bimodality is better estimated from abundances at log scale (such as CLR)
pseq0.clr <- microbiome::transform(pseq0, "clr")

set.seed(4433)

# bimodality: between subjects
bimodality.score <- bimodality(pseq0.clr, method = "potential_analysis",
                               bs.iter = 999, peak.threshold = 5,
                               min.density = 5)

bimodality.score %>% 
  head()
```

```{r, eval=FALSE}
taxa <- taxa(pseq0)
df <- data.frame(group = taxa,
                 intermediate.stability = intermediate.stability[taxa],
                 bimodality = bimodality.score[taxa])
theme_set(theme_bw(20))

library(ggrepel)

p <- ggplot(df,
            aes(x = intermediate.stability, y = bimodality, label = '')) + #Doesn't add labels to points
  geom_text_repel() +
  geom_point()

#Labels only those that have bimodality > 0.4 and intermediate stability < 0, adjusts the placement of labels
p <- p + geom_text(aes(label = ifelse(bimodality > 0.75 | intermediate.stability < -0.25, group, '')), vjust = "inward", hjust = "inward")

print(p)
```

```{r, eval=FALSE}
p$data
```

```{r, eval=FALSE}
# Log10 abundance for a selected taxonomic group
# Pick the most bimodal taxa as an example
tax  <- names(which.max(bimodality.score))

# Detect tipping points at log10 abundances
x <- abundances(microbiome::transform(pseq, "clr"))[tax,]

# Bootstrapped potential analysis to identify potential minima
# in practice, use more bootstrap iterations
potential.minima <- potential_analysis(x, bs.iter = 10)$minima

# Same with earlywarnings package (without bootstrap ie. less robust)
# library(earlywarnings)
# res <- livpotential_ews(x)$min.points

# Identify the potential minimum locations as tipping point candidates
tipping.samples <- sapply(potential.minima, function (m) {names(which.min(abs(sort(x) - m)))})
tipping.point <- abundances(pseq)[tax, tipping.samples]
print(tipping.point)
```


```{r, eval=FALSE}
# Illustrate the detected tipping point
plot(density(x), main = tax)
abline(v = x[tipping.samples])
```


```{r}
# # Bimodality hotplot:
# # Consider a unique sample from each subject: the baseline time point 
# p <- microbiome::hotplot(pseq0, tax, tipping.point = 0.005)
# print(p)
# 
# # Visualize bimodality
# pv <- plot_tipping(pseq, tax, tipping.point = 0.005)
# print(pv)
```

```{r}
sessionInfo()
```
