---
title: "Resistome data - chicken 1 & 2"
author: "Florentin Constancias "
date: " `r format(Sys.time(), '%B %d, %Y')` "
output: 
  html_document: 
    toc: yes
    keep_md: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
# knitr::opts_chunk$set(dev = "ragg_png") # https://ragg.r-lib.org/index.html
knitr::opts_chunk$set(fig.width = 6) #https://r4ds.had.co.nz/graphics-for-communication.html#figure-sizing
knitr::opts_chunk$set(fig.asp = 0.618)
knitr::opts_chunk$set(fig.show = "hold")
knitr::opts_chunk$set(fig.show = "70%")
knitr::opts_chunk$set(fig.align = "center")
```

```{r}
rm(list= ls())

gc()

require("tidyverse")
require('speedyseq')
require("microViz")

'%!in%' <- function(x,y)!('%in%'(x,y))

source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_taxa_tests.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_normalisation.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_alpha.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_beta.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_heatmap.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_varia.R")
source("https://raw.githubusercontent.com/fconstancias/metabaRpipe-source/master/Rscripts/functions.R")
```


```{r}
export_ppt_width = NULL
export_ppt_height = NULL

out_xlsx = "~/Desktop/plots_manuscript_chickens.xlsx"
out_pptx = "~/Desktop/plots_manuscript_chickens.pptx"

```

# Load data:

## Aestethics:

```{r}
load( here::here("Figures/Rastetics.Rdata"))
```


## 16S:

```{r}
# "data/processed/16S/ps_silva_dada2_human_chicken_all_fct_polyferms.RDS" %>% 
#   here::here() %>% 
#   readRDS() %>% 
#   filter_taxa(function(x) sum(x > 0) > 0, TRUE) -> ps_16S
```

## resistome:

```{r}
# "data/processed/chicken1-2_full_gene_catalog_phyloseq.RDS" %>% 
"data/processed/chicken1-2_anvio-bin-full_gene_catalog_phyloseq.RDS" %>% 
  here::here() %>% 
  readRDS() -> ps

ps

# %>% 
#  .[["Num_Gi_pc"]] 
```

```{r}
ps$Num_Gi_pc %>% 
  sample_names()
```



```{r}
ps$Num_Gi_pc  %>% 
  microViz::ps_mutate(Day_of_Treatment = as.double(Day_of_Treatment)) %>% 
  subset_samples(Day_of_Treatment < 34 | is.na(Day_of_Treatment)) -> ps1

subset_samples(ps1, sample_names(ps1) %!in% c("D3", "D19")) %>% 
  filter_taxa(function(x) sum(x > 0) > 0, TRUE) -> ps2

ps2
```

```{r}
# ps2 %>% 
#   sample_names()
```

```{r}
# ps2 %>% 
#   sample_data() %>% 
#   data.frame()
```

```{r}
as(tax_table(ps2), "matrix") %>%
  data.frame() %>% 
  rownames_to_column("feature") %>%
  distinct(MGE_prediction)
```

```{r}
as(tax_table(ps2), "matrix") %>%
  data.frame() %>% 
  rownames_to_column("feature") %>%
  distinct(bin_name)
```



```{r}
ps2 %>% 
  microViz::tax_mutate(Resistance_mechanism_multi = ifelse(grepl(";", Resistance_mechanism), "multi-mech", Resistance_mechanism)) %>%
  microViz::tax_mutate(AMR_sub_class_multi = ifelse(grepl(";", AMR_sub_class), "multidrug", AMR_sub_class)) %>%
  microViz::tax_mutate(bin_name_clean = case_when(MGE_prediction %in% c("plasmid", "phage") ~  NA_character_ , # could be fine tuned with phage and phage taxonomy...
                                                  TRUE ~ bin_name)) %>% 
  microViz::tax_mutate(MGE_prediction_clean = case_when(!is.na(bin_name_clean) ~ "MAG",
                                                        is.na(MGE_prediction) ~ "unclassified", # to test...
                                                        TRUE ~ MGE_prediction)) %>%
  microViz::tax_mutate(MGE_prediction_clean = case_when(grepl("ambiguous", MGE_prediction_clean) ~ "ambiguous", TRUE ~ MGE_prediction_clean)) -> ps2_clean # %>%
# microViz::tax_mutate(MGE_prediction = fct_relevel(MGE_prediction, c("MAG", "chromosome", "plasmid", "phage", "ambiguous", "unclassified"))) -> ps_AMRclean
```

```{r}
as(tax_table(ps2_clean), "matrix") %>%
  data.frame() %>% 
  rownames_to_column("feature") -> tmp

# tmp %>% 
#   write_tsv("~/Desktop/ps2_clean_tax.tsv")

tmp %>% 
  distinct(MGE_prediction_clean)
```


export new collection without plasmids - could be fine tuned with phage and phage taxonomy later:


```{r}
as(tax_table(ps2_clean), "matrix") %>%
  data.frame() %>% 
  filter(!is.na(bin_name_clean)) %>% 
  rownames_to_column("feature") %>% 
  distinct(Contig.ID, .keep_all = TRUE) %>% 
  select(Contig.ID, bin_name_clean) %>% 
  arrange(bin_name_clean) -> tax_df_tmp
```

c1:

```{r}
"/Users/fconstan/Documents/GitHub/NRP72-FBT/data/processed/anvio/chicken1/final_collection_chick1.tsv" %>% 
  read_tsv(show_col_types = FALSE, col_names = c("ContigID", "bin_name")) %>% 
  separate(ContigID, into = c("megahit", "cont_id", "split", "split_id"),
           remove = FALSE) %>% 
  dplyr::rename(SplitID = ContigID) %>% # when we wanted to keep split info as the same time but doesn't make sense here
  mutate(`Contig ID` = paste0(megahit, "_", cont_id)) %>% 
  select(SplitID, bin_name, `Contig ID`) %>% 
  mutate(`Contig ID` = paste0("c1_", `Contig ID`),
         bin_name = paste0("c1_", bin_name)) -> c1_initil_collection


tax_df_tmp %>% 
  filter(grepl("c1_", bin_name_clean)) %>% 
  pull(Contig.ID) %>% 
  unique() -> c1_contigs


c1_initil_collection %>% 
  filter(`Contig ID` %in% c1_contigs) -> c1_collection_noplas_nophage


c1_initil_collection %>% 
  select(-`Contig ID`) %>% 
  write_tsv("/Users/fconstan/Documents/GitHub/NRP72-FBT/data/processed/anvio/chicken1/final_collection_chick1_plas_phage_out.tsv" ,col_names = FALSE)
```


```{bash, engine.opts='-l', eval=FALSE}
conda activate  /Users/fconstan/miniconda3/envs/anvio-7.1

CONT_DB=CONTIGS.db
PROF=PROFILE.db

cd /Users/fconstan/Documents/GitHub/NRP72-FBT/data/processed/anvio/chicken1


anvi-import-collection  -c ${CONT_DB} -p ${PROF} -C final_chick1_noplas_nophage final_collection_chick1_plas_phage_out.tsv

anvi-estimate-genome-completeness \
-p ${PROF}  \
-c ${CONT_DB}  \
-C final_chick1_noplas_nophage \
-o final_chick1_noplas_nophage_completeness.tsv

anvi-estimate-scg-taxonomy \
-p ${PROF}  \
-c ${CONT_DB}  \
-C final_chick1_noplas_nophage \
-o final_chick1_noplas_nophage_scg_tax.tsv

anvi-summarize -c ${CONT_DB} -p ${PROF} -C final_chick1_noplas_nophage -o SUMMARY/final_chick1_noplas_nophage


```

Then GTDB tk and METACHIP and WAFFLE


c2:

```{r}
"/Users/fconstan/Documents/GitHub/NRP72-FBT/data/processed/anvio/chicken2/final_collection_chick2.tsv" %>% 
  read_tsv(show_col_types = FALSE, col_names = c("ContigID", "bin_name")) %>% 
  separate(ContigID, into = c("megahit", "cont_id", "split", "split_id"),
           remove = FALSE) %>% 
  dplyr::rename(SplitID = ContigID) %>% # when we wanted to keep split info as the same time but doesn't make sense here
  mutate(`Contig ID` = paste0(megahit, "_", cont_id)) %>% 
  select(SplitID, bin_name, `Contig ID`) %>% 
  mutate(`Contig ID` = paste0("c2_", `Contig ID`),
         bin_name = paste0("c2_", bin_name)) -> c2_initil_collection


tax_df_tmp %>% 
  filter(grepl("c2_", bin_name_clean)) %>% 
  pull(Contig.ID) %>% 
  unique() -> c2_contigs


c2_initil_collection %>% 
  filter(`Contig ID` %in% c2_contigs) -> c2_collection_noplas_nophage

c2_collection_noplas_nophage %>% 
  select(-`Contig ID`) %>% 
  write_tsv("/Users/fconstan/Documents/GitHub/NRP72-FBT/data/processed/anvio/chicken2/final_collection_chick2_plas_phage_out.tsv" ,col_names = FALSE)
```

```{bash, engine.opts='-l', eval=FALSE}
conda activate  /Users/fconstan/miniconda3/envs/anvio-7.1

CONT_DB=CONTIGS.db
PROF=PROFILE.db

cd /Users/fconstan/Documents/GitHub/NRP72-FBT/data/processed/anvio/chicken2

anvi-import-collection  -c ${CONT_DB} -p ${PROF} -C final_chick2_noplas_nophage final_collection_chick2_plas_phage_out.tsv

anvi-estimate-genome-completeness \
-p ${PROF}  \
-c ${CONT_DB}  \
-C final_chick2_noplas_nophage \
-o final_chick2_noplas_nophage_completeness.tsv

anvi-estimate-scg-taxonomy \
-p ${PROF}  \
-c ${CONT_DB}  \
-C final_chick2_noplas_nophage \
-o final_chick2_noplas_nophage_scg_tax.tsv

anvi-summarize -c ${CONT_DB} -p ${PROF} -C final_chick2_noplas_nophage -o SUMMARY/final_chick2_noplas_nophage


```


Then GTDB tk and METACHIP and WAFFLE on single.



```{r}
# tax_df_tmp %>% 
#   filter(grepl("c1_", feature)) %>% 
#   distinct(SplitID, .keep_all = TRUE) %>% 
#   select(-feature) %>% 
#   arrange(bin_name_clean) %>% 
#   write_tsv("/Users/fconstan/Documents/GitHub/NRP72-FBT/data/processed/anvio/chicken1/final_collection_chick1_plas_out.tsv" ,col_names = FALSE)
# 
# tax_df_tmp %>% 
#   filter(grepl("c2_", feature)) %>% 
#   distinct(SplitID, .keep_all = TRUE) %>% 
#   select(-feature) %>% 
#   arrange(bin_name_clean) %>% 
#   write_tsv("/Users/fconstan/Documents/GitHub/NRP72-FBT/data/processed/anvio/chicken2/final_collection_chick2_plas_out.tsv" ,col_names = FALSE)
```


Working with clean collection (no plas no phage or no plas) led to quite some changes:

e.g. for chicken2

Bin_19_24	BACTERIA	1	94.37	2.82	149	2477661

-> c2_Bin_19_24	BLANK	1	0	0	64	814458 

--> 85 split removed
--> 1663203 nt removed


problem with filtering data handling or the plasmids are very likely contigs from the MAGs...

```{r}
# as(tax_table(ps2), "matrix") %>%
#   data.frame() -> tmp_df
```


```{r}
# tmp_df %>% 
#   distinct(bin_name, .keep_all = TRUE) %>% 
#   select(bin_name, ncontigs, nsplit, num_splits, total.length, t_family) %>% 
#   arrange(bin_name) %>% filter(bin_name == "c2_Bin_19_24")
```

```{r}
# tmp_df %>% 
#   # filter(bin_name == "c2_Bin_19_24") %>% 
#   # distinct(SplitID, .keep_all = TRUE) %>% 
#   group_by(bin_name) %>% 
#   summarise(ncontigs2 = n_distinct(Contig.ID),
#             nsplit2 =  n_distinct(SplitID)) %>% 
#   arrange(bin_name)  %>% filter(bin_name == "c2_Bin_19_24")

```
```{r}
# "/Users/fconstan/Documents/GitHub/NRP72-FBT/data/processed/anvio/chicken2/final_collection_chick2_plas_out.tsv" %>% 
#   read_tsv(col_names = FALSE) %>% 
#   group_by(X2) %>% 
#   summarise(nsplit3 =  n_distinct(X1)) %>% 
#   arrange(X2)  %>% filter(X2 == "c2_Bin_19_24")

```

TO conclude ->  discrepency between contigs -> split (= contigs instead) -> export

> 72-64
[1] 8 -> 8 contigs plasmids ?


```{r}
# tmp_df %>% 
#   select(Contig.ID, bin_name, MGE_prediction, ARG, ncontigs, nsplit, num_splits, total.length, t_family, ) %>% 
#   filter(bin_name == "c2_Bin_19_24") %>% 
#   distinct(Contig.ID, .keep_all = TRUE) %>% 
#   arrange(MGE_prediction)
```

***yes ... :-) *** 

but we have to check if the missing splits are the one identified from contigs -> plasmids or not... - other script where I combined the data.

***yes ... :-) *** 


so instead we will take the post collection split the split into contig and select the contigs...


<!-- ```{bash, engine.opts='-l', eval=FALSE} -->
<!-- conda activate  /Users/fconstan/miniconda3/envs/anvio-7.1 -->

<!-- CONT_DB=CONTIGS.db -->
<!-- PROF=PROFILE.db -->

<!-- cd /Users/fconstan/Documents/GitHub/NRP72-FBT/data/processed/anvio/chicken2 -->

<!-- anvi-import-collection  -c ${CONT_DB} -p ${PROF} -C final_chick2_noplas final_collection_chick2_plas_out.tsv -->

<!-- anvi-estimate-genome-completeness \ -->
<!-- -p ${PROF}  \ -->
<!-- -c ${CONT_DB}  \ -->
<!-- -C final_chick2_noplas \ -->
<!-- -o final_chick2_noplas_completeness.tsv -->

<!-- anvi-estimate-scg-taxonomy \ -->
<!-- -p ${PROF}  \ -->
<!-- -c ${CONT_DB}  \ -->
<!-- -C final_chick2_noplas \ -->
<!-- -o final_chick2_noplas_scg_tax.tsv -->

<!-- #anvi-import-collection  -c ${CONT_DB} -p ${PROF} -C final_chick2_noplas_phage final_collection_chick2_plas_phage_out.tsv -->


<!-- #anvi-estimate-genome-completeness \ -->
<!-- #       -p ${PROF}  \ -->
<!-- #       -c ${CONT_DB}  \ -->
<!-- #       -C final_chick2_noplas_phage \ -->
<!-- #       -o final_chick2_noplas_phage_completeness.tsv -->

<!-- #anvi-estimate-scg-taxonomy \ -->
<!-- #       -p ${PROF}  \ -->
<!-- #       -c ${CONT_DB}  \ -->
<!-- #       -C final_chick2_noplas_phage \ -->
<!-- #       -o final_chick2_noplas_phage_scg_tax.tsv -->

<!-- #anvi-summarize -c ${CONT_DB} -p ${PROF} -C final_chick2_noplas_phage -o SUMMARY/final_chick2_noplas_phage -->


<!-- cd /Users/fconstan/Documents/GitHub/NRP72-FBT/data/processed/anvio/chicken1 -->


<!-- anvi-import-collection  -c ${CONT_DB} -p ${PROF} -C final_chick1_noplas final_collection_chick1_plas_out.tsv -->

<!-- anvi-estimate-genome-completeness \ -->
<!-- -p ${PROF}  \ -->
<!-- -c ${CONT_DB}  \ -->
<!-- -C final_chick1_noplas \ -->
<!-- -o final_chick1_noplas_completeness.tsv -->

<!-- anvi-estimate-scg-taxonomy \ -->
<!-- -p ${PROF}  \ -->
<!-- -c ${CONT_DB}  \ -->
<!-- -C final_chick1_noplas \ -->
<!-- -o final_chick1_noplas_scg_tax.tsv -->

<!-- #anvi-import-collection  -c ${CONT_DB} -p ${PROF} -C final_chick1_noplas_phage final_collection_chick1_plas_phage_out.tsv -->


<!-- #anvi-estimate-genome-completeness \ -->
<!-- #       -p ${PROF}  \ -->
<!-- #       -c ${CONT_DB}  \ -->
<!-- #       -C final_chick1_noplas_phage \ -->
<!-- #       -o final_chick1_noplas_phage_completeness.tsv -->

<!-- #anvi-estimate-scg-taxonomy \ -->
<!-- #       -p ${PROF}  \ -->
<!-- #       -c ${CONT_DB}  \ -->
<!-- #       -C final_chick1_noplas_phage \ -->
<!-- #       -o final_chick1_noplas_phage_scg_tax.tsv -->

<!-- #anvi-summarize -c ${CONT_DB} -p ${PROF} -C final_chick1_noplas_phage -o SUMMARY/final_chick1_noplas_phage -->


<!-- ``` -->


```{r}
# rank_names(ps)

ps2_clean %>% 
  subset_taxa(!is.na(ARG) ) -> ps_AMR
# | !is.na(rgi_CARD_ARO) | !is.na(PathoFact_Tox_Toxin_classification)) %>% 

ps_AMR
```

```{r}

# as(tax_table(ps_AMRclean), "matrix") %>%
#   as.data.frame() %>%
#   rownames_to_column('ASV') -> tax_table
# 
# as(otu_table(ps_AMRclean), "matrix") %>%
#   as.data.frame() %>%
#   rownames_to_column('ASV') -> otu_table
# 
# otu_table %>%
#   left_join(tax_table, by = c("ASV" = "ASV")) %>%
#   write_tsv("~/Desktop/ps_AMR_chicks.tsv")

```


```{r}
ps_AMR %>%
  physeq_sel_tax_table(c("Contig.ID","ARG","AMR_sub_class", "AMR_sub_class_multi",
                         "Resistance_mechanism","Resistance_mechanism_multi",
                         "MGE_prediction", "MGE_prediction_clean","bin_name", "bin_name_clean",
                         "plasmid_score", "topology_plas","plasmids_mobilog_pc","virus_score","topology_phage", "taxonomy", "Bacteriophages_mobilog_pc",
                         "IntegrativeEle_mobilog_pc")) -> ps_AMRclean

ps_AMRclean
```


```{r}
# 
# as(tax_table(ps_AMR), "matrix") %>%
#   as.data.frame() %>%
#   rownames_to_column('ASV') -> tax_table
# 
# as(otu_table(ps_AMR), "matrix") %>%
#   as.data.frame() %>%
#   rownames_to_column('ASV') -> otu_table
# 
# otu_table %>%
#   left_join(tax_table, by = c("ASV" = "ASV")) %>%
#   write_tsv("~/Desktop/ps_AMR_chicks.tsv")

```



save phyloseq objects:

```{r}
ps_AMR %>% 
  saveRDS(here::here("data/processed/chicken1-2_anvio-bin-clean_AMR_phyloseq.RDS"))


ps2_clean %>% 
  saveRDS(here::here("data/processed/chicken1-2_anvio-bin-clean_phyloseq.RDS"))

```



```{r}
devtools::session_info()
```
