---
title: "Resistome data exploration"
author: "Florentin Constancias "
date: " `r format(Sys.time(), '%B %d, %Y')` "
output: 
  html_document: 
    toc: yes
    keep_md: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
# knitr::opts_chunk$set(dev = "ragg_png") # https://ragg.r-lib.org/index.html
knitr::opts_chunk$set(fig.width = 6) #https://r4ds.had.co.nz/graphics-for-communication.html#figure-sizing
knitr::opts_chunk$set(fig.asp = 0.618)
knitr::opts_chunk$set(fig.show = "hold")
knitr::opts_chunk$set(fig.show = "70%")
knitr::opts_chunk$set(fig.align = "center")
```

```{r}
rm(list= ls())
# gc()
require("tidyverse")
require('speedyseq')

'%!in%' <- function(x,y)!('%in%'(x,y))

source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_taxa_tests.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_normalisation.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_alpha.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_beta.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_heatmap.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_varia.R")
```


```{r}
export_ppt_width = NULL
export_ppt_height = NULL
out_pptx = here::here("output/plots_2.pptx")
```

# Load data:

## 16S:

```{r}
"data/processed/16S/ps_silva_dada2_human_chicken_all_fct_polyferms.RDS" %>% 
  here::here() %>% 
  readRDS() %>% 
  filter_taxa(function(x) sum(x > 0) > 0, TRUE) -> ps_16S
```

## resistome:

```{r}
"data/processed/exp1_ps_AMR.RDS" %>% 
  here::here() %>% 
  readRDS() -> ps_AMR
```

## Aestetics:

```{r}
load( here::here("Figures/Rastetics.Rdata"))
```

```{r}
ps_AMR %>% 
  tax_table() %>% 
  data.frame()
```
## Others:
```{r}
ps_AMR %>% 
  tax_table() %>% 
  data.frame() %>% 
  rownames_to_column("ORF_ID") -> tax_mapping
```


```{r}
# ps_AMR %>% 
#   sample_data() %>% 
#   data.frame() %>% 
#   rownames_to_column("sample_id") %>% 
#     # filter(Model == "Chicken") %>% 
#   mutate(Day_of_Treatment_num = as.double(Day_of_Treatment)) %>% 
#   # mutate(Day_of_Treatment = fct_relevel(Day_of_Treatment, c(NA, "-3", "-1", "1" , "3", "4", "5","7", "30", "47", "48")))  %>% 
#   mutate(Antibiotic_mg.L = factor(Antibiotic_mg.mL, levels = c(NA, 20, 200,90, 600))) -> metadata_mapping

# metadata_mapping %>% 
#   write_tsv("~/Desktop/metadata_mapping.tsv")

"data/raw/metadata_mapping_AMR.tsv" %>% 
  here::here() %>% 
  read_tsv() %>% 
  mutate(Antibiotic_mg.L = factor(Antibiotic_mg.mL, levels = c(NA, 20, 200,90, 600))) -> metadata_mapping
```


```{r}
tax_mapping %>% 
  select(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi",
           "PathoFact_AMR_ARG")) %>% 
  distinct(PathoFact_AMR_ARG, .keep_all = TRUE) %>% 
  add_count(PathoFact_AMR_ARG) %>% 
  arrange(-n) %>% 
  column_to_rownames('PathoFact_AMR_ARG')

```


# Understand how to glom the data:

The tax_glom() function requires to have the order of the columns of the tax_table() organised. If MGE colum element is on the left of PathoFact_AMR_ARG then it will check unique MGE & PathoFact_AMR_ARG features.

`physeq_sel_tax_table()` allows to select, reorganize the columns of the tax_table().

How many AMR genes did we detected? 

```{r}
ps_AMR %>% 
  physeq_sel_tax_table(c("PathoFact_AMR_AMR_category", "PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_Resistance_mechanism", "PathoFact_AMR_ARG")) %>% 
  speedyseq::tax_glom(., taxrank = "PathoFact_AMR_ARG") -> ps_AMR_tmp

ps_AMR_tmp
```


```{r}
ps_AMR_tmp %>% 
  physeq_glom_rename(taxrank = "PathoFact_AMR_ARG", speedyseq = TRUE)
```

```{r}
ps_AMR_tmp %>% 
  physeq_glom_rename(taxrank = "PathoFact_AMR_ARG", speedyseq = TRUE) %>% 
  tax_table() %>% 
  head()
```


# Generate subset of data:

## AMR gene dataset

```{r}
ps_AMR %>% 
  physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi",
                         "PathoFact_AMR_Resistance_mechanism_multi",
                         "PathoFact_AMR_ARG")) %>% 
  physeq_glom_rename(taxrank = "PathoFact_AMR_ARG", speedyseq = TRUE) -> ps_AMR_gene

ps_AMR_gene
```

## non agglomarated dataset including: all the rows but only a selection of info: + MGE, bin, bin taxonomy

```{r}
ps_AMR %>% 
  physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi",
                         "PathoFact_AMR_Resistance_mechanism","PathoFact_AMR_Resistance_mechanism_multi",
                         "PathoFact_AMR_ARG")) %>% 
  physeq_glom_rename(taxrank = "PathoFact_AMR_ARG", speedyseq = TRUE) -> ps_AMR_gene

ps_AMR_gene
```

No need to melt for this one since every otu is a gene with AMR hit and MGE / MAG ...

```{r}
ps_AMR %>%
  physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi",
                         "PathoFact_AMR_Resistance_mechanism","PathoFact_AMR_Resistance_mechanism_multi",
                         "PathoFact_AMR_ARG", "PathoFact_AMR_Database", "PathoFact_AMR_MGE_prediction_clean_2",
                         "viralverify_Prediction", "isescan_type", "isescan_count_IS_contig" , "ANVIO_CONCOCT_HQ_clean",
                         "GTDB_tax_CONCOCT_clean", "PathoFact_AMR_MGE_prediction_clean_2")) -> ps_AMR_all

ps_AMR_all
```



```{r}
# ps_AMR %>% 
#   physeq_sel_tax_table(c("Contig_ID","ANVIO_CONCOCT_HQ_clean", "PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi", 
#                          "GTDB_tax_CONCOCT_clean", 
#                          "PathoFact_AMR_MGE_prediction_clean", "rgi_CARD_Best_Hit_ARO",
#                          "PathoFact_AMR_ARG"))  %>% 
#   # speedyseq::tax_glom(., taxrank = "PathoFact_AMR_MGE_prediction_clean")  %>% tax_table()
#   # speedyseq::tax_glom(., taxrank = "PathoFact_AMR_Resistance_mechanism_multi") %>%   #%>% tax_table()
#   speedyseq::tax_glom(., taxrank = "PathoFact_AMR_ARG") -> phyloseq_AMR_MAG_MGE
# 
# phyloseq_AMR_MAG_MGE
```

Looks good, let's go.


## Reproducing https://www.nature.com/articles/s41467-022-29919-9.pdf


<file:///Users/fconstan/Documents/GitHub/NRP72-FBT/Figures/Figures_beta.html>

### Fig. 2

#### Fig 2a:

```{r}
ps_AMR_all %>% 
  # physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi", "PathoFact_AMR_ARG"))  %>% 
  physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi", "PathoFact_AMR_ARG"))  %>% 
  speedyseq::tax_glom(., taxrank = "PathoFact_AMR_ARG")  %>% 
  filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>% 
  speedyseq::psmelt() %>% 
  select(-Lactose_mM:-Total_SCFA_mM) %>% 
  filter(Day_of_Treatment < 34 &
           !Sample %in% c("D3", "D19")) %>% 
  filter(Reactor != "CR" & Model == "Chicken" | Model == "Human" & Reactor %in% c("TR1", "TR2", "TR3", "TR4", "TR5", "TR6", "CR"))  %>% 
  group_by(Sample) %>% 
  summarise(sum_abundance = sum(Abundance), .groups = "drop") %>% 
  mutate(sum_abundance = na_if(sum_abundance, 0)) %>%
  left_join(metadata_mapping,
            by = c("Sample" = "sample_id")) %>% 
  # # filter(Model == "Chicken") %>% 
  # mutate(Day_of_Treatment_num = as.double(Day_of_Treatment)) %>% 
  # # mutate(Day_of_Treatment = fct_relevel(Day_of_Treatment, c(NA, "-3", "-1", "1" , "3", "4", "5","7", "30", "47", "48")))  %>% 
  # mutate(Antibiotic_mg.mL = factor(Antibiotic_mg.mL, levels = c(NA, 20, 200,90, 600))) %>%
  ggplot(aes_string(x = "Day_of_Treatment", y = "sum_abundance", color = "Treatment", fill  = "Treatment", shape = "Antibiotic_mg.L")) + #shape = Fermentation
  geom_point(size = 2.5) + 
  geom_line(linetype = 2,  size = 0.5,  aes_string(group = "interaction(Model,  Reactor_Treatment_Dose)")) +
  labs(y = "sum Rnum_Gi", x = "Day_of_Treatment") +
  scale_color_manual(name = "", values = treat_col,
                     na.value = "black") +
  scale_fill_manual(name = "", values = treat_col,
                    na.value = "black") +
  scale_shape_manual(name = "" ,values = c(15,16,18,19), na.value =  17) +
  # scale_alpha_continuous(name = "", range=c(0.6, 1), na.value =  1) +
  geom_vline(xintercept = 0, size = 0.5, linetype = 1, color = "grey60") +
  theme_light() + 
  scale_y_continuous(labels=percent) -> p1
# scale_y_continuous(labels = function(x) format(x, scientific = TRUE)) -> p1
```


```{r, fig.asp = 0.618 }
p1 + facet_grid(as.formula(paste0("Antibiotic_mg.mL ~  Model  ")), scales = "free", space = "fixed", drop = TRUE) -> ptmp

ptmp

# ptmp %>%
#   export::graph2ppt(append = TRUE, width = 8, height = 4,
#                     file = out_pptx)
```


```{r, fig.asp = 0.618}

p1 + facet_grid(as.formula(paste0("Antibiotic  ~  Model  ")), scales = "free", space = "fixed", drop = TRUE) -> ptmp

ptmp


ptmp %>% 
  ggsave(plot = .,bg = "transparent",scale = 2,
         units = "mm",
         dpi = 600,
         height = 0.618 * 84,
         width = 84,
         filename = "~/Desktop/plot_1.eps")

ptmp %>%
  export::graph2ppt(append = TRUE, width = 317.48031496 , height = 0.618 * 317.48031496, paper = "A4",  scaling = 2,
                    file = "~/Desktop/plots.pptx")
```


https://stackoverflow.com/questions/52581469/dynamically-normalize-all-rows-with-first-element-within-a-group

```{r}
ps_AMR_all %>% 
  physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi", "PathoFact_AMR_ARG"))  %>% 
  speedyseq::tax_glom(., taxrank = "PathoFact_AMR_ARG")  %>% 
  filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>% #subset_samples(Treatment == "VAN+CCUG59168") %>%  subset_samples(Day_of_Treatment < 34 )
  speedyseq::psmelt() %>% 
  select(-Lactose_mM:-Total_SCFA_mM) %>% 
  filter(Day_of_Treatment < 34 &
           !Sample %in% c("D3", "D19")) %>% 
  filter(Reactor != "CR" & Model == "Chicken" | Model == "Human" & Reactor %in% c("TR1", "TR2", "TR3", "TR4", "TR5", "TR6", "CR"))  %>% 
  group_by(Sample) %>% 
  summarise(sum_abundance = sum(Abundance), .groups = "drop") %>% 
  mutate(sum_abundance = na_if(sum_abundance, 0)) %>%
  left_join(metadata_mapping,
            by = c("Sample" = "sample_id")) %>% 
  ungroup() %>% 
  # mutate(Reactor_Treatment_Dose = paste0(Period, Reactor_Treatment_Dose)) %>% 
  group_by(Model, Reactor_Treatment_Dose) %>% 
  arrange(Day_of_Treatment) %>% 
  mutate(per_cent_change = sum_abundance / first(sum_abundance)) %>% 
  ggplot(aes_string(x = "Day_of_Treatment", y = "per_cent_change", color = "Treatment", fill  = "Treatment", shape = "Antibiotic_mg.L")) + #shape = Fermentation
  geom_point(size = 2.5) + 
  geom_line(linetype = 2,  size = 0.5,  aes_string(group = "interaction(Model, Reactor_Treatment_Dose)")) +
  labs(y = "sum Rnum_Gi / baseline", x = "Day_of_Treatment") +
  scale_color_manual(name = "", values = treat_col,
                     na.value = "black") +
  scale_fill_manual(name = "", values = treat_col,
                    na.value = "black") +
  scale_shape_manual(name = "" ,values = c(15,16,18,19), na.value =  17) +
  # scale_alpha_continuous(name = "", range=c(0.6, 1), na.value =  1) +
  geom_vline(xintercept = 0, size = 0.5, linetype = 1, color = "grey60") +
  geom_hline(yintercept = 1, size = 0.5, linetype = 2, color = "grey60") +
  theme_light() + facet_grid(as.formula(paste0("Antibiotic ~  Model  ")), scales = "free_x", space = "fixed", drop = TRUE) -> p1

p1

p1 %>% 
  ggsave(plot = .,bg = "transparent",scale = 2,
         units = "mm",
         dpi = 600,
         height = 0.618 * 84,
         width = 84,
         filename = "~/Desktop/plot.eps")

p1 %>%
  export::graph2ppt(append = TRUE, width = 317.48031496 , height = 0.618 * 317.48031496, paper = "A4",  scaling = 2,
                    file = "~/Desktop/plots.pptx")
```

<https://blog.exploratory.io/introducing-time-series-analysis-with-dplyr-60683587cf8a>

```{r}
# plot_univ_NRP72 <- function(df, x = "Day_of_Treatment", y = "value",  point_size = 2.5, color = "Treatment", fill  = "Treatment", shape = "Antibiotic", alpha = "Antibiotic_mg.mL", facet_formula = paste0("alphadiversiy ~  Model "), ylab = "Resistome alpha-diversity", xlab = "Days (Treatment)", measures = c("Observed"), path_group = "interaction(Model, Fermentation, Reactor_Treatment)"){
#   
#   df %>% 
#     pivot_longer(cols = all_of(measures), values_to = "value", names_to = 'alphadiversiy', values_drop_na  = TRUE) %>%
#     mutate(alphadiversiy = fct_relevel(alphadiversiy, measures)) -> df_ready
#   
#   df_ready %>% 
#     ggplot(aes_string(x = x, y = y, color = color, fill = fill, shape = shape, alpha = alpha)) + #shape = Fermentation
#     geom_point(size = point_size) + 
#     geom_line(linetype = 2,  size = 0.5,  aes_string(group = path_group)) +
#     labs(y = ylab, x = xlab) +
#     scale_color_manual(name = "", values = treat_col,
#                        na.value = "black") +
#     scale_fill_manual(name = "", values = treat_col,
#                       na.value = "black") +
#     facet_grid(as.formula(facet_formula), scales = "free", space = "fixed", drop = TRUE) +
#     scale_shape_manual(name = "" ,values = antibio_shape, na.value =  17) +
#     scale_alpha_discrete(name = "", range=c(0.6, 1), na.value =  0.6) + 
#     geom_vline(xintercept = 0, size = 0.5, linetype = 1, color = "grey60") -> p_alpha_both
#   
#   return(p_alpha_both)
# }
```

### Heatmap:

file:///Users/fconstan/Documents/GitHub/NRP72-FBT/tmp/human_resistome_visualization.html

```{r}
# phyloseq_AMR_MAG_MGE %>% 
#   # physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi", 
#   #                        "PathoFact_AMR_MGE_prediction_clean", "rgi_CARD_Best_Hit_ARO",
#   #                        "PathoFact_AMR_ARG"))  %>% 
#   physeq_sel_tax_table(c("PathoFact_AMR_ARG"))  %>% 
#   # physeq_sel_tax_table(c("PathoFact_AMR_ARG"))  %>% 
#   speedyseq::tax_glom(., taxrank = "PathoFact_AMR_ARG") %>% 
#   speedyseq::psmelt() %>% 
#   # left_join(tax_mapping,
#   #           by = c("Best_Hit_ARO" = "Best_Hit_ARO"),
#   #           suffix = c("_x", "")) %>% 
#   # mutate(Sample = fct_reorder(Sample, Model,Treatment,Fermentation,Antibiotic_mg.mL, Day_of_Treatment )) %>%
#   mutate(Abundance = na_if(Abundance, 0)) %>%
#   mutate(logTPM = log10(Abundance + 1)) -> pheatmap_df
# 
# 
# pheatmap_df %>% 
#   # replace(is.na(.), 0)  %>% 
#   # select(Sample, Best_Hit_ARO, logTPM) %>% 
#   group_by(Sample, PathoFact_AMR_ARG) %>%
#   pivot_wider(names_from =  Sample, PathoFact_AMR_ARG,
#               values_from = logTPM) %>% 
#   column_to_rownames("PathoFact_AMR_ARG") -> pheatmap_samples
# 
# pheatmap_samples %>% 
#   replace(is.na(.), 0)  %>% 
#   t() %>% 
#   vegan::vegdist(na.rm = TRUE) %>% 
#   hclust() -> clust_t
# 
# # pheatmap_samples %>% 
# #     replace(is.na(.), 0)  %>%
# #     vegdist(na.rm = TRUE) %>% 
# #   # cor() %>% 
# #   hclust() -> clust
# 
# pheatmap_samples %>% 
#   pheatmap::pheatmap(cluster_rows = FALSE ,
#                      cluster_cols = clust_t,
#                      na_col = "transparent",
#                      annotation_row = tax_mapping %>% rownames_to_column('tmp_col') %>% 
#                        filter(PathoFact_AMR_ARG %in% rownames(pheatmap_samples)) %>% 
#                        distinct(PathoFact_AMR_ARG, .keep_all = TRUE) %>% 
#                        column_to_rownames("PathoFact_AMR_ARG") %>% 
#                        select(PathoFact_AMR_Resistance_mechanism_multi, PathoFact_AMR_AMR_sub_class_multi),
#                      # annotation_colors = list("PathoFact_AMR_AMR_sub_class_multi" = col,
#                      #                          "Treatment" = treat_col),
#                      annotation_col = pheatmap_df %>% 
#                        select(Model,Treatment,Fermentation,Antibiotic_mg.mL, Day_of_Treatment)) -> p_pheat
# 
# 
# p_pheat %>% 
# print()
```

Histogram:

AMR drug classes

```{r}
# https://newbedev.com/how-to-control-ordering-of-stacked-bar-chart-using-identity-on-ggplot2
source("https://raw.githubusercontent.com/fconstancias/metabaRpipe-source/master/Rscripts/functions.R")

ps_AMR_all %>% 
  physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi", "PathoFact_AMR_ARG"))  %>% 
  speedyseq::tax_glom(., taxrank = "PathoFact_AMR_ARG")  %>% 
  filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>% 
  physeq_add_metadata(metadata_mapping,
                      sample_column = "sample_id") %>% 
  speedyseq::psmelt() %>% 
  select(-Lactose_mM:-Total_SCFA_mM) %>% 
  filter(Day_of_Treatment < 34 &
           !Sample %in% c("D3", "D19")) %>% 
  filter(Reactor != "CR" & Model == "Chicken" | Model == "Human" & Reactor %in% c("TR1", "TR2", "TR3", "TR4", "TR5", "TR6", "CR"))  %>% 
  # left_join(tax_mapping,
  #         by = c("PathoFact_AMR_ARG" = "PathoFact_AMR_ARG"),
  #         suffix = c("_x", "")) %>%
  mutate(Abundance = na_if(Abundance, 0)) %>%
  mutate(logTPM = log10(Abundance + 1))  %>% 
  # physeq_AMRgn %>%
  #   microbiome::transform(transform = "log10p") %>% 
  #   speedyseq::psmelt() %>% 
  #   left_join(tax_mapping,
  #             by = c("Best_Hit_ARO" = "Best_Hit_ARO"),
  #             suffix = c("_x", "")) %>%
  #   mutate(Abundance = na_if(Abundance, 0)) %>%
  #   mutate(logTPM = log10(Abundance + 1))  %>% 
  # separate_rows(., Resistance_Mechanism,sep = '; ',convert = TRUE) %>% 
  ggplot(aes(x = as.factor(Day_of_Treatment), y = Abundance, fill = reorder(PathoFact_AMR_AMR_sub_class_multi, Abundance, FUN=median, na.rm = TRUE))) +   
  geom_bar( stat = "identity", position = "fill", colour="black", size=0.0125) +
  facet_grid(Model ~ Antibiotic + Antibiotic_mg.mL , scales = "free", space = "free", drop = TRUE) +
  ggtitle("ARG Type proportion per Sample") +
  xlab("Day (Treatment)")  +
  ylab("Proportion") + 
  theme_light() +
  scale_y_continuous(labels=percent) +
  ggpubr::rotate_x_text(60) -> p_mec

p_mec %>%
  ggpubr::set_palette(drug_classe_multi)-> p_mec



p_mec + facet_null() + facet_wrap(Model ~ Antibiotic + Antibiotic_mg.mL + Treatment2, scales = "free", drop = TRUE) -> p_mec 

p_mec

# p_mec %>%
#   export::graph2ppt(append = TRUE, width = 12, height = 8,
#                     file = out_pptx)
```

get samples_names:

```{r}
p_mec$data %>% distinct(Sample)  %>% pull %>%  as.character() -> samples
```

```{r}
require(microViz); require(fantaxtic)

ps_AMR_all %>% 
  physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi", "PathoFact_AMR_ARG"))  %>%
  speedyseq::tax_glom(., taxrank = "PathoFact_AMR_AMR_sub_class_multi")  %>% 
  physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class_multi")) %>% 
  prune_samples(samples, .) %>% 
  physeq_add_metadata(metadata_mapping,
                      sample_column = "sample_id") %>% 
  # ps_mutate(Day_of_Treatment = as.numeric(Day_of_Treatment)) %>% 
  ps_arrange(-Day_of_Treatment) %>% 
  comp_barplot(
    tax_level = "PathoFact_AMR_AMR_sub_class_multi",
    label = "Day_of_Treatment", # name an alternative variable to label axes
    n_taxa = 20, # give more taxa unique colours
    merge_other = FALSE, # split the "other" category to display alpha diversity
    bar_width = 0.9, # reduce the bar width to 70% of one row
    bar_outline_colour = "grey5",
    sample_order = "default") +#,# is the default (use NA to remove outlines)
  # facet_by = c("Model", "Antibiotic", "Antibiotic_mg.mL", "Treatment2")) +
  # group_by = c("Model", "Antibiotic", "Antibiotic_mg.mL", "Treatment2"))
  coord_flip() +  theme_light() +
  scale_y_continuous(labels=percent) + 
  facet_wrap(Model ~ Antibiotic + Antibiotic_mg.mL + Treatment2, scales = "free", ncol = 6, nrow = 4) -> p2
# 
# # Plot them side by side with the patchwork package.
# patch <- patchwork::wrap_plots(plot_list, nrow = 1, guides = "collect")
# patch & coord_flip() # make all plots horizontal (note: use & instead of +)
#   

p2


p2 %>% 
  ggsave(plot = .,bg = "transparent",scale = 2,
         units = "mm",
         dpi = 600,
         height = 0.618 * 84 * 2,
         width = 84 * 2,
         filename = "~/Desktop/plot2.eps")

p2 %>%
  export::graph2ppt(append = TRUE, width = 317.48031496 * 2  , height = 0.618 * 317.48031496 * 2 , paper = "A4",  scaling = 2,
                    file = "~/Desktop/plots.pptx")
```


```{r}
# p_mec$data %>%  select(PathoFact_AMR_AMR_sub_class_multi, PathoFact_AMR_AMR_sub_class, PathoFact_AMR_ARG) %>%  distinct(PathoFact_AMR_ARG,.keep_all = TRUE) %>% 
#   write_tsv("~/Desktop/ARG_gn_check.tsv")
```

```{r}
# p2$data %>% 
#   filter(Model == "Human" & 
#            Antibiotic == "VAN" &
#            Antibiotic_mg.mL == 90) %>% 
#   distinct(Sample, .keep_all = TRUE) %>% 
#   DT::datatable()
#   
```

```{r}
# filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>%
#   speedyseq::psmelt() %>% 
#   mutate(PathoFact_AMR_AMR_sub_class_multi = ifelse(Abundance < 0.000001, "Other", PathoFact_AMR_AMR_sub_class_multi)) %>% 
#   group_by(PathoFact_AMR_AMR_sub_class_multi, Sample) %>% 
#   # slice_max()
#   # top_frac(0.10) %>% 
#   select(-Lactose_mM:-Total_SCFA_mM) %>% 
#     filter(Day_of_Treatment < 34 &
#          !Sample %in% c("D3", "D19")) %>% 
#       filter(Reactor != "CR" & Model == "Chicken" | Model == "Human" & Reactor %in% c("TR1", "TR2", "TR3", "TR4", "TR5", "TR6", "CR"))  %>% 
#     # left_join(tax_mapping,
#   #         by = c("PathoFact_AMR_ARG" = "PathoFact_AMR_ARG"),
#   #         suffix = c("_x", "")) %>%
#   mutate(Abundance = na_if(Abundance, 0)) %>%
#   mutate(logTPM = log10(Abundance + 1))  %>% 
#   # physeq_AMRgn %>%
#   #   microbiome::transform(transform = "log10p") %>% 
#   #   speedyseq::psmelt() %>% 
#   #   left_join(tax_mapping,
#   #             by = c("Best_Hit_ARO" = "Best_Hit_ARO"),
#   #             suffix = c("_x", "")) %>%
#   #   mutate(Abundance = na_if(Abundance, 0)) %>%
#   #   mutate(logTPM = log10(Abundance + 1))  %>% 
#   # separate_rows(., Resistance_Mechanism,sep = '; ',convert = TRUE) %>% 
#   ggplot(aes(x = as.factor(Day_of_Treatment), y = Abundance, fill = reorder(PathoFact_AMR_AMR_sub_class_multi, Abundance, FUN=median, na.rm = TRUE))) +   
#   geom_bar( stat = "identity", position = "fill", colour="black", size=0.0125) +
#   facet_grid(Model ~ Antibiotic + Antibiotic_mg.mL , scales = "free", space = "free", drop = TRUE) +
#   ggtitle("ARG Type proportion per Sample") +
#   xlab("Day (Treatment)")  +
#   ylab("Proportion") + 
#   theme_light() +
#     scale_y_continuous(labels=percent) +
#   ggpubr::rotate_x_text(60) -> p_mec

# p_mec %>%
#   ggpubr::set_palette(drug_classe_multi)-> p_mec

# 
# 
# p_mec + facet_null() + facet_wrap(Model ~ Antibiotic + Antibiotic_mg.mL, scales = "free", drop = TRUE) -> p_mec 
# 
# p_mec

# p_mec %>%
#   export::graph2ppt(append = TRUE, width = 12, height = 8,
#                     file = out_pptx)
```

```{r}
# https://newbedev.com/how-to-control-ordering-of-stacked-bar-chart-using-identity-on-ggplot2
# 
# phyloseq_AMR_MAG_MGE %>% 
#   # physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi", 
#   #                        "PathoFact_AMR_MGE_prediction_clean", "rgi_CARD_Best_Hit_ARO",
#   #                        "PathoFact_AMR_ARG"))  %>% 
#   physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class_multi","PathoFact_AMR_ARG"))  %>% 
#   # physeq_sel_tax_table(c("PathoFact_AMR_ARG"))  %>% 
#   speedyseq::tax_glom(., taxrank = "PathoFact_AMR_ARG")  %>% 
#   filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>% 
#   speedyseq::psmelt() %>% 
#   select(-Lactose_mM:-Total_SCFA_mM) %>% 
#   # left_join(tax_mapping,
#   #         by = c("PathoFact_AMR_ARG" = "PathoFact_AMR_ARG"),
#   #         suffix = c("_x", "")) %>%
#   mutate(Abundance = na_if(Abundance, 0)) %>%
#   mutate(logTPM = log10(Abundance + 1))  %>% 
#   # physeq_AMRgn %>%
#   #   microbiome::transform(transform = "log10p") %>% 
#   #   speedyseq::psmelt() %>% 
#   #   left_join(tax_mapping,
#   #             by = c("Best_Hit_ARO" = "Best_Hit_ARO"),
#   #             suffix = c("_x", "")) %>%
#   #   mutate(Abundance = na_if(Abundance, 0)) %>%
#   #   mutate(logTPM = log10(Abundance + 1))  %>% 
#   # separate_rows(., Resistance_Mechanism,sep = '; ',convert = TRUE) %>% 
#   ggplot(aes(x = as.factor(Day_of_Treatment), y = Abundance, fill = PathoFact_AMR_AMR_sub_class_multi)) +   
#   geom_bar( stat = "identity", colour="black", size=0.0125) +
#   facet_grid(Model ~ Treatment + Antibiotic_mg.mL , scales = "free", space = "free_x") +
#   ggtitle("ARG Type Abundance per Sample") +
#   xlab("Day (Treatment)")  +
#   ylab("Rnum_Gi") + 
#   theme_light() +
#   ggpubr::rotate_x_text(60) -> p_mec
# 
# p_mec %>%
#   ggpubr::set_palette(drug_classe_multi)-> p_mec
# 
# p_mec
# 
# p_mec %>%
#   export::graph2ppt(append = TRUE, width = 12, height = 8,
#                     file = out_pptx)
```

Beta-div resistome:


```{r}
source("https://raw.githubusercontent.com/fconstancias/metabaRpipe-source/master/Rscripts/functions.R")

ps_AMR_all %>% 
  physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi", "PathoFact_AMR_ARG"))  %>%
  speedyseq::tax_glom(., taxrank = "PathoFact_AMR_ARG")  %>% 
  # physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class_multi")) %>% 
  prune_samples(samples, .) %>% 
  physeq_add_metadata(metadata_mapping,
                      sample_column = "sample_id") -> ps_tmp
```

chick
```{r}
ps_tmp %>% 
  subset_samples(Model == "Chicken") %>% 
  filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>%
  phyloseq_plot_bdiv(dlist = NULL,
                     m = "CoDa",
                     seed = 123,
                     axis1 = 1,
                     axis2 = 2) -> coda_resistome

# 
# coda_resistome

coda_resistome$PCA$layers[[1]] = NULL

coda_resistome$PCA + geom_point(size=2,
                                aes(colour = Treatment ,
                                    shape = Antibiotic_mg.L,
                                    alpha = NULL)) +
  geom_path(data = coda_resistome$PCA$data %>% 
              arrange(Day_of_Treatment) ,
            aes(colour = Treatment, group = interaction(Model, Antibiotic,  Reactor_Treatment_Dose)),            arrow = arrow(
              angle = 30, length = unit(0.15, "inches"),
              ends = "last", type = "open"
            ), linetype = "longdash", size = 0.1) +
  # facet_wrap(Model ~ Antibiotic , scales = "fixed", ncol = 6, nrow = 4) +
  theme_light() +
  scale_color_manual(name = "", values = treat_col,
                     na.value = "black") +
  scale_fill_manual(name = "", values = treat_col,
                    na.value = "black") +
  scale_shape_manual(name = "" ,values = c(15,16,18,19), na.value =  17) +
  theme(legend.position = "bottom") + 
  facet_grid(. ~ Model) -> pca_treat

# pca_treat %>% 
# ggpubr::change_palette(p = ., palette = "lancet") -> pca_treat


pca_treat

# ps_tmp %>% 
#   subset_samples(Model == "Chicken") %>% 
#   phyloseq_add_taxa_vector(phyloseq = .,
#                            dist = coda_resistome$aidist,
#                            join_cbind = "join",
#                             taxrank_glom = "PathoFact_AMR_AMR_sub_class",
#                                      tax_rank_plot = "PathoFact_AMR_AMR_sub_class",
#                            figure_ord = pca_treat, 
#                            fact = 0.25, pval_cutoff = 0.05,
#                            top_r = 10) -> pcoa_vect
# 
# pcoa_vect$signenvfit

```

```{r, warning = FALSE, results='hide'}
phyloseq_add_taxa_vector_fix(dist = coda_resistome$aidist,
                             phyloseq = ps_tmp %>%  physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class_multi"))  %>% subset_samples(Model == "Chicken") %>%   filter_taxa(function(x) sum(x > 0) > 0, TRUE),
                             figure_ord = pca_treat,
                             taxrank_glom = "PathoFact_AMR_AMR_sub_class_multi") -> ppp


ppp$signenvfit %>% 
  DT::datatable()
```

```{r}
ppp$plot + 
    ggrepel::geom_text_repel(cex=2,
                           aes(label= Phase),
                           segment.color = 'black',
                           segment.size = 0.5,
                           # nudge_x =  -4,
                           # nudge_y = 0,
                           ppp$plot$data %>% filter(Phase %in% c("Stab")) %>% unique()
                           ) -> pbeta

pbeta

pbeta %>% 
  ggsave(plot = .,bg = "transparent",scale = 2,
         units = "mm",
         dpi = 600,
         height = 0.618 * 84,
         width = 84,
         filename = "~/Desktop/pbeta_chick.eps")

pbeta %>%
  export::graph2ppt(append = TRUE, width = 317.48031496  , height = 0.618 * 317.48031496  , paper = "A4",  scaling = 2,
                    file = "~/Desktop/plots.pptx")
```

```{r}

source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_beta.R")


ps_AMR_all %>% 
  physeq_sel_tax_table(c("PathoFact_AMR_MGE_prediction_clean_2"))  %>%
  speedyseq::tax_glom(., taxrank = "PathoFact_AMR_MGE_prediction_clean_2") %>% 
  # physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class_multi")) %>% 
  prune_samples(samples, .) %>% 
  subset_samples(Model == "Chicken") %>% 
  filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>% 
  physeq_add_metadata(metadata_mapping,
                      sample_column = "sample_id") -> ps_tmp_envfit


ps_tmp_envfit %>% 
  tax_table() %>% 
  data.frame() %>% 
 mutate(PathoFact_AMR_MGE_prediction_clean_2 = case_when(grepl("ambiguous", PathoFact_AMR_MGE_prediction_clean_2) ~ "ambiguous", TRUE ~ PathoFact_AMR_MGE_prediction_clean_2)) %>% 
 mutate(PathoFact_AMR_MGE_prediction_clean_2 = case_when(PathoFact_AMR_MGE_prediction_clean_2 %in% c("MAG", "chromosome")  ~ "MAG - unbinned chromosome", TRUE ~ PathoFact_AMR_MGE_prediction_clean_2)) %>% 
  as.matrix() -> tax_table(ps_tmp_envfit)

ps_tmp_envfit %>% 
  phyloseq_add_taxa_vector_fix(dist = coda_resistome$aidist,
                             phyloseq = . ,
                            taxnames_rm = c("unclassified"),
                             vector_color = "tomato",
                             figure_ord = pbeta,
                             taxrank_glom = "PathoFact_AMR_MGE_prediction_clean_2") -> ppp



ppp$signenvfit %>% 
  DT::datatable()
```

```{r}
ppp$plot


ppp$plot %>% 
  ggsave(plot = .,bg = "transparent",scale = 2,
         units = "mm",
         dpi = 600,
         height = 0.618 * 84,
         width = 84,
         filename = "~/Desktop/pbeta_chicken_all.eps")

ppp$plot %>%
  export::graph2ppt(append = TRUE, width = 317.48031496  , height = 0.618 * 317.48031496  , paper = "A4",  scaling = 2,
                    file = "~/Desktop/plots.pptx")
```

human
```{r}
ps_tmp %>% 
  subset_samples(Model == "Human") %>%
  filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>%
  phyloseq_plot_bdiv(dlist = NULL,
                     m = "CoDa",
                     seed = 123,
                     axis1 = 1,
                     axis2 = 2) -> coda_resistome

# 
# coda_resistome

coda_resistome$PCA$layers[[1]] = NULL

coda_resistome$PCA + geom_point(size=2,
                                aes(colour = Treatment ,
                                    shape = Antibiotic_mg.L,
                                    alpha = NULL)) +
  geom_path(data = coda_resistome$PCA$data %>% 
              arrange(Day_of_Treatment) ,
            aes(colour = Treatment, group = interaction(Model, Antibiotic,  Reactor_Treatment_Dose)),            arrow = arrow(
              angle = 30, length = unit(0.15, "inches"),
              ends = "last", type = "open"
            ), linetype = "longdash", size = 0.1) +
  # facet_wrap(Model ~ Antibiotic , scales = "fixed", ncol = 6, nrow = 4) +
  theme_light() +
  scale_color_manual(name = "", values = treat_col,
                     na.value = "black") +
  scale_fill_manual(name = "", values = treat_col,
                    na.value = "black") +
  scale_shape_manual(name = "" ,values = c(15,16,18,19), na.value =  17) +
  theme(legend.position = "bottom") + 
  facet_grid(. ~ Model) -> pca_treat

# pca_treat %>% 
# ggpubr::change_palette(p = ., palette = "lancet") -> pca_treat


pca_treat

# ps_tmp %>% 
#   subset_samples(Model == "Chicken") %>% 
#   phyloseq_add_taxa_vector(phyloseq = .,
#                            dist = coda_resistome$aidist,
#                            join_cbind = "join",
#                             taxrank_glom = "PathoFact_AMR_AMR_sub_class",
#                                      tax_rank_plot = "PathoFact_AMR_AMR_sub_class",
#                            figure_ord = pca_treat, 
#                            fact = 0.25, pval_cutoff = 0.05,
#                            top_r = 10) -> pcoa_vect
# 
# pcoa_vect$signenvfit

```

```{r, warning = FALSE, results='hide'}
phyloseq_add_taxa_vector_fix(dist = coda_resistome$aidist,
                             phyloseq = ps_tmp %>%  physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class_multi"))  %>% subset_samples(Model == "Human") %>%   filter_taxa(function(x) sum(x > 0) > 0, TRUE),
                             figure_ord = pca_treat,
                             taxrank_glom = "PathoFact_AMR_AMR_sub_class_multi") -> ppp

ppp$signenvfit %>% 
  DT::datatable()
```

```{r}
ppp$plot + 
    ggrepel::geom_text_repel(cex=2,
                           aes(label= Phase),
                           segment.color = 'black',
                           segment.size = 0.5,
                           # nudge_x =  -4,
                           # nudge_y = 0,
                           ppp$plot$data %>% filter(Phase %in% c("Stab")) %>% unique()
                           ) -> pbeta

pbeta

pbeta %>% 
  ggsave(plot = .,bg = "transparent",scale = 2,
         units = "mm",
         dpi = 600,
         height = 0.618 * 84,
         width = 84,
         filename = "~/Desktop/pbeta_human.eps")

pbeta %>%
  export::graph2ppt(append = TRUE, width = 317.48031496  , height = 0.618 * 317.48031496  , paper = "A4",  scaling = 2,
                    file = "~/Desktop/plots.pptx")

```


```{r}
ps_AMR_all %>% 
  physeq_sel_tax_table(c("PathoFact_AMR_MGE_prediction_clean_2"))  %>%
  speedyseq::tax_glom(., taxrank = "PathoFact_AMR_MGE_prediction_clean_2") %>% 
  # physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class_multi")) %>% 
  prune_samples(samples, .) %>% 
  subset_samples(Model == "Human") %>% 
  filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>% 
  physeq_add_metadata(metadata_mapping,
                      sample_column = "sample_id") -> ps_tmp_envfit


ps_tmp_envfit %>% 
  tax_table() %>% 
  data.frame() %>% 
 mutate(PathoFact_AMR_MGE_prediction_clean_2 = case_when(grepl("ambiguous", PathoFact_AMR_MGE_prediction_clean_2) ~ "ambiguous", TRUE ~ PathoFact_AMR_MGE_prediction_clean_2)) %>% 
 mutate(PathoFact_AMR_MGE_prediction_clean_2 = case_when(PathoFact_AMR_MGE_prediction_clean_2 %in% c("MAG", "chromosome")  ~ "MAG - unbinned chromosome", TRUE ~ PathoFact_AMR_MGE_prediction_clean_2)) %>% 
  as.matrix() -> tax_table(ps_tmp_envfit)

ps_tmp_envfit %>% 
  phyloseq_add_taxa_vector_fix(dist = coda_resistome$aidist,
                             phyloseq = . ,
                            taxnames_rm = c("unclassified"),
                             vector_color = "tomato",
                             figure_ord = pbeta,
                             taxrank_glom = "PathoFact_AMR_MGE_prediction_clean_2") -> ppp



ppp$signenvfit %>% 
  DT::datatable()
```

```{r}
ppp$plot


ppp$plot %>% 
  ggsave(plot = .,bg = "transparent",scale = 2,
         units = "mm",
         dpi = 600,
         height = 0.618 * 84,
         width = 84,
         filename = "~/Desktop/pbeta_human_all.eps")

ppp$plot %>%
  export::graph2ppt(append = TRUE, width = 317.48031496  , height = 0.618 * 317.48031496  , paper = "A4",  scaling = 2,
                    file = "~/Desktop/plots.pptx")
```

alfa-div resistome:

```{r}

sample_data(ps_tmp)$Read_nrom = 1000000

ps_tmp %>% 
  phyloseq_density_normalize(physeq = .,
                             # sample_idx = "Reactor",
                             value_idx = "Read_nrom") %>% 
  # transform_sample_counts(function(x) (x*sum(x)) *100) %>% 
  phyloseq_alphas(phylo = FALSE) -> resistome_alpha


resistome_alpha
```

```{r}
plot_alpha_div_NRP72 <- function(df = resistome_alpha, x = "Day_of_Treatment", y = "value",  point_size = 2.5, color = "Treatment", fill  = "Treatment", shape = "Antibiotic", alpha = "Antibiotic_mg.mL", facet_formula = paste0("alphadiversiy ~  Model "), ylab = "Resistome alpha-diversity", xlab = "Days (Treatment)", measures = c("Observed"), path_group = "interaction(Model, Reactor_Treatment_Dose)"){
  
  df %>% 
    pivot_longer(cols = all_of(measures), values_to = "value", names_to = 'alphadiversiy', values_drop_na  = TRUE) %>%
    mutate(alphadiversiy = fct_relevel(alphadiversiy, measures)) -> df_ready
  
  df_ready %>% 
    ggplot(aes_string(x = x, y = y, color = color, fill = fill, shape = shape, alpha = alpha)) + #shape = Fermentation
    geom_point(size = point_size) + 
    geom_line(linetype = 2,  size = 0.5,  aes_string(group = path_group)) +
    labs(y = ylab, x = xlab) +
    # scale_color_manual(name = "", values = treat_col,
    #                    na.value = "black") +
    # scale_fill_manual(name = "", values = treat_col,
    #                   na.value = "black") +
    facet_grid(as.formula(facet_formula), scales = "free", space = "fixed", drop = TRUE) +
    # scale_shape_manual(name = "" ,values = antibio_shape, na.value =  17) +
    # scale_alpha_discrete(name = "", range=c(0.6, 1), na.value =  0.6) + 
    geom_vline(xintercept = 0, size = 0.5, linetype = 1, color = "grey60") -> p_alpha_both
  
  return(p_alpha_both)
}
```


```{r}
resistome_alpha %>% 
  plot_alpha_div_NRP72( shape = "as.factor(Antibiotic_mg.mL)", alpha = NULL) -> p_alpha 

p_alpha +  scale_color_manual(name = "", values = treat_col,
                              na.value = "black") +
  scale_fill_manual(name = "", values = treat_col,
                    na.value = "black") +
  scale_shape_manual(name = "" ,values = c(15,16,18,19), na.value =  17) + 
  theme_light() + facet_null() + facet_grid(Antibiotic  ~ Model, scales = "free_x", drop = TRUE) -> p_alpha


p_alpha

p_alpha %>% 
  ggsave(plot = .,bg = "transparent",scale = 2,
         units = "mm",
         dpi = 600,
         height = 0.618 * 84,
         width = 84,
         filename = "~/Desktop/palpha.eps")

p_alpha %>%
  export::graph2ppt(append = TRUE, width = 317.48031496  , height = 0.618 * 317.48031496  , paper = "A4",  scaling = 2,
                    file = "~/Desktop/plots.pptx")
```

Same with subtraction from baseline.

```{r}

x = "Day_of_Treatment"
y = "value"
point_size = 2.5
color = "Treatment"
fill  = "Treatment"
shape = "as.factor(Antibiotic_mg.mL)"
alpha = NULL
facet_formula = paste0("alphadiversiy ~  Model ")
ylab = "Resistome baseline delta"
xlab = "Days (Treatment)"
measures = c("Observed", "evenness_pielou")
path_group = "interaction(Model, Reactor_Treatment_Dose)"


resistome_alpha %>% 
  pivot_longer(cols = all_of(measures), values_to = "value", names_to = 'alphadiversiy', values_drop_na  = TRUE) %>%
  group_by(alphadiversiy, Model,Fermentation,  Reactor_Treatment_Dose) %>% 
  arrange(Day_of_Treatment) %>% 
  mutate(delta_change = value - first(value),
         ratio_change =  value / first(value)) %>% 
  ggplot(aes_string(x = x, y = "delta_change", color = color, fill = fill, shape = shape, alpha = alpha)) + #shape = Fermentation
  geom_point(size = point_size) + 
  geom_line(linetype = 2,  size = 0.5,  aes_string(group = path_group)) +
  labs(y = ylab, x = xlab) +
  # scale_color_manual(name = "", values = treat_col,
  #                    na.value = "black") +
  # scale_fill_manual(name = "", values = treat_col,
  #                   na.value = "black") +
  facet_grid(as.formula(facet_formula), scales = "free", space = "fixed", drop = TRUE) +
  # scale_shape_manual(name = "" ,values = antibio_shape, na.value =  17) +
  # scale_alpha_discrete(name = "", range=c(0.6, 1), na.value =  0.6) + 
  geom_vline(xintercept = 0, size = 0.5, linetype = 1, color = "grey60")+
  scale_color_manual(name = "", values = treat_col,
                     na.value = "black") +
  scale_fill_manual(name = "", values = treat_col,
                    na.value = "black") +
  scale_shape_manual(name = "" ,values = c(15,16,18,19), na.value =  17) + 
  theme_light() + facet_null() + facet_grid(alphadiversiy + Antibiotic  ~ Model , scales = "free", drop = TRUE) +
  ggh4x::facetted_pos_scales(
    y = list(
      alphadiversiy == "Observed" ~ scale_y_continuous(limits = c(-125,50)),
      alphadiversiy == "evenness_pielou" ~ scale_y_continuous(limits = c(-0.2, 0.2))
    )
  ) -> p_alpha_delta


p_alpha_delta

p_alpha_delta %>% 
  ggsave(plot = .,bg = "transparent",scale = 2,
         units = "mm",
         dpi = 600,
         height = 0.618 * 84,
         width = 84,
         filename = "~/Desktop/palpha_d.eps")

p_alpha_delta %>%
  export::graph2ppt(append = TRUE, width = 317.48031496  , height = 0.618 * 317.48031496  , paper = "A4",  scaling = 2,
                    file = "~/Desktop/plots.pptx")

```

Same with ratio from baseline.

```{r}

x = "Day_of_Treatment"
y = "value"
point_size = 2.5
color = "Treatment"
fill  = "Treatment"
shape = "as.factor(Antibiotic_mg.mL)"
alpha = NULL
facet_formula = paste0("alphadiversiy ~  Model ")
ylab = "Resistome baseline ratio"
xlab = "Days (Treatment)"
measures = c("Observed", "evenness_pielou")
path_group = "interaction(Model, Reactor_Treatment_Dose)"


resistome_alpha %>% 
  pivot_longer(cols = all_of(measures), values_to = "value", names_to = 'alphadiversiy', values_drop_na  = TRUE) %>%
  group_by(alphadiversiy, Model,Fermentation,  Reactor_Treatment_Dose) %>% 
  arrange(Day_of_Treatment) %>% 
  mutate(delta_change = value - first(value),
         ratio_change =  value / first(value)) %>% 
  ggplot(aes_string(x = x, y = "ratio_change", color = color, fill = fill, shape = shape, alpha = alpha)) + #shape = Fermentation
  geom_point(size = point_size) + 
  geom_line(linetype = 2,  size = 0.5,  aes_string(group = path_group)) +
  labs(y = ylab, x = xlab) +
  # scale_color_manual(name = "", values = treat_col,
  #                    na.value = "black") +
  # scale_fill_manual(name = "", values = treat_col,
  #                   na.value = "black") +
  facet_grid(as.formula(facet_formula), scales = "free", space = "fixed", drop = TRUE) +
  # scale_shape_manual(name = "" ,values = antibio_shape, na.value =  17) +
  # scale_alpha_discrete(name = "", range=c(0.6, 1), na.value =  0.6) + 
  geom_vline(xintercept = 0, size = 0.5, linetype = 1, color = "grey60")+
  scale_color_manual(name = "", values = treat_col,
                     na.value = "black") +
  scale_fill_manual(name = "", values = treat_col,
                    na.value = "black") +
  scale_shape_manual(name = "" ,values = c(15,16,18,19), na.value =  17) + 
  theme_light() + facet_null() + facet_grid(alphadiversiy + Antibiotic  ~ Model , scales = "free", drop = TRUE) +
  ggh4x::facetted_pos_scales(
    y = list(
      alphadiversiy == "Observed" ~ scale_y_continuous(limits = c(0.5,1.2)),
      alphadiversiy == "evenness_pielou" ~ scale_y_continuous(limits = c(0.5, 1.4))
    )
  )-> p_alpha_delta


p_alpha_delta

p_alpha_delta %>% 
  ggsave(plot = .,bg = "transparent",scale = 2,
         units = "mm",
         dpi = 600,
         height = 0.618 * 84,
         width = 84,
         filename = "~/Desktop/palpha_r.eps")

p_alpha_delta %>%
  export::graph2ppt(append = TRUE, width = 317.48031496  , height = 0.618 * 317.48031496  , paper = "A4",  scaling = 2,
                    file = "~/Desktop/plots.pptx")

```

Genetic material

```{r}
# https://newbedev.com/how-to-control-ordering-of-stacked-bar-chart-using-identity-on-ggplot2

ps_AMR_all %>% 
  physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi", "PathoFact_AMR_ARG", "PathoFact_AMR_MGE_prediction_clean_2"))  %>%
  speedyseq::tax_glom(., taxrank = "PathoFact_AMR_MGE_prediction_clean_2")  %>% 
  # physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class_multi")) %>% 
  prune_samples(samples, .) %>% 
  filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>% 
  speedyseq::psmelt() %>% 
  mutate(PathoFact_AMR_MGE_prediction_clean_2 = case_when(grepl("ambiguous", PathoFact_AMR_MGE_prediction_clean_2) ~ "ambiguous", TRUE ~ PathoFact_AMR_MGE_prediction_clean_2)) %>% 
  mutate(PathoFact_AMR_MGE_prediction_clean_2 = fct_relevel(PathoFact_AMR_MGE_prediction_clean_2, c("MAG", "chromosome", "plasmid", "phage", "ambiguous", "unclassified"))) %>% 
  select(-Lactose_mM:-Total_SCFA_mM) %>% 
  # left_join(tax_mapping,
  #         by = c("PathoFact_AMR_ARG" = "PathoFact_AMR_ARG"),
  #         suffix = c("_x", "")) %>%
  mutate(Abundance = na_if(Abundance, 0)) %>%
  mutate(logTPM = log10(Abundance + 1))  %>% 
  # physeq_AMRgn %>%
  #   microbiome::transform(transform = "log10p") %>% 
  #   speedyseq::psmelt() %>% 
  #   left_join(tax_mapping,
  #             by = c("Best_Hit_ARO" = "Best_Hit_ARO"),
  #             suffix = c("_x", "")) %>%
  #   mutate(Abundance = na_if(Abundance, 0)) %>%
  #   mutate(logTPM = log10(Abundance + 1))  %>% 
  # separate_rows(., Resistance_Mechanism,sep = '; ',convert = TRUE) %>% 
  ggplot(aes(x = as.factor(Day_of_Treatment), y = Abundance, fill = PathoFact_AMR_MGE_prediction_clean_2)) +   
  geom_bar( stat = "identity", position = "fill", colour="black", size = 0.0125) +
  facet_grid(Model ~ Treatment + Antibiotic_mg.mL , scales = "free", space = "free_x") +
  ggtitle("ARG Type Abundance per Sample") +
  xlab("Day (Treatment)")  +
  ylab("Proportion") + 
  theme_light() +
  ggpubr::rotate_x_text(60) -> p_mec

p_mec %>%
  ggpubr::set_palette(MGE)-> p_mec

p_mec

# p_mec %>%
#   export::graph2ppt(append = TRUE, width = 12, height = 8,
#                     file = out_pptx)
```


```{r}
ps_AMR_all %>% 
  physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi", "PathoFact_AMR_ARG", "PathoFact_AMR_MGE_prediction_clean_2"))  %>%
  speedyseq::tax_glom(., taxrank = "PathoFact_AMR_MGE_prediction_clean_2")  %>% 
  physeq_sel_tax_table(c("PathoFact_AMR_MGE_prediction_clean_2")) %>% 
  # physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class_multi")) %>% 
  prune_samples(samples, .) %>% 
  filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>% 
  physeq_add_metadata(metadata_mapping,
                      sample_column = "sample_id") -> ps_tmp


ps_tmp %>% 
  tax_table() %>% 
  data.frame() %>% 
 mutate(PathoFact_AMR_MGE_prediction_clean_2 = case_when(grepl("ambiguous", PathoFact_AMR_MGE_prediction_clean_2) ~ "ambiguous", TRUE ~ PathoFact_AMR_MGE_prediction_clean_2)) %>% 
 mutate(PathoFact_AMR_MGE_prediction_clean_2 = case_when(PathoFact_AMR_MGE_prediction_clean_2 %in% c("MAG", "chromosome")  ~ "MAG - unbinned chromosome", TRUE ~ PathoFact_AMR_MGE_prediction_clean_2)) %>% 
  as.matrix() -> tax_table(ps_tmp)
 
ps_tmp %>% 
  ps_arrange(-Day_of_Treatment) %>% 
  comp_barplot(
    tax_level = "PathoFact_AMR_MGE_prediction_clean_2",
    label = "Day_of_Treatment", # name an alternative variable to label axes
    n_taxa = 20, # give more taxa unique colours
    merge_other = FALSE, # split the "other" category to display alpha diversity
    bar_width = 0.9, # reduce the bar width to 70% of one row
    bar_outline_colour = "grey5",
    sample_order = "default") +#,# is the default (use NA to remove outlines)
  # facet_by = c("Model", "Antibiotic", "Antibiotic_mg.mL", "Treatment2")) +
  # group_by = c("Model", "Antibiotic", "Antibiotic_mg.mL", "Treatment2"))
  coord_flip() +  theme_light() +
  scale_y_continuous(labels=percent) + 
  facet_wrap(Model ~ Antibiotic + Antibiotic_mg.mL + Treatment2, scales = "free", ncol = 6, nrow = 4) -> p2
# 
# # Plot them side by side with the patchwork package.
# patch <- patchwork::wrap_plots(plot_list, nrow = 1, guides = "collect")
# patch & coord_flip() # make all plots horizontal (note: use & instead of +)
#   

p2 %>%
  ggpubr::set_palette("npg") -> p2

p2

p2 %>% 
  ggsave(plot = .,bg = "transparent",scale = 2,
         units = "mm",
         dpi = 600,
         height = 0.618 * 84 * 2,
         width = 84 * 2,
         filename = "~/Desktop/AMRcarrier.eps")

p2 %>%
  export::graph2ppt(append = TRUE, width = 317.48031496 * 2  , height = 0.618 * 317.48031496 * 2 , paper = "A4",  scaling = 2,
                    file = "~/Desktop/plots.pptx")
```


```{r}
ps_AMR_all %>% 
  physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi", "PathoFact_AMR_ARG", "PathoFact_AMR_MGE_prediction_clean_2"))  %>%
  speedyseq::tax_glom(., taxrank = "PathoFact_AMR_MGE_prediction_clean_2")  %>% 
  physeq_sel_tax_table(c("PathoFact_AMR_MGE_prediction_clean_2")) %>% 
  # physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class_multi")) %>% 
  prune_samples(samples, .) %>% 
  filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>% 
  physeq_add_metadata(metadata_mapping,
                      sample_column = "sample_id") -> ps_tmp


ps_tmp %>% 
  tax_table() %>% 
  data.frame() %>% 
 mutate(PathoFact_AMR_MGE_prediction_clean_2 = case_when(grepl("ambiguous", PathoFact_AMR_MGE_prediction_clean_2) ~ "ambiguous", TRUE ~ PathoFact_AMR_MGE_prediction_clean_2)) %>% 
 mutate(PathoFact_AMR_MGE_prediction_clean_2 = case_when(PathoFact_AMR_MGE_prediction_clean_2 %in% c("MAG", "chromosome")  ~ "MAG - unbinned chromosome", TRUE ~ PathoFact_AMR_MGE_prediction_clean_2)) %>% 
  as.matrix() -> tax_table(ps_tmp)
 
ps_tmp %>% 
  ps_arrange(-Day_of_Treatment) %>% 
  comp_barplot(
    tax_transform_for_plot = "identity",
    tax_level = "PathoFact_AMR_MGE_prediction_clean_2",
    label = "Day_of_Treatment", # name an alternative variable to label axes
    n_taxa = 20, # give more taxa unique colours
    merge_other = FALSE, # split the "other" category to display alpha diversity
    bar_width = 0.9, # reduce the bar width to 70% of one row
    bar_outline_colour = "grey5",
    sample_order = "default") +#,# is the default (use NA to remove outlines)
  # facet_by = c("Model", "Antibiotic", "Antibiotic_mg.mL", "Treatment2")) +
  # group_by = c("Model", "Antibiotic", "Antibiotic_mg.mL", "Treatment2"))
  coord_flip() +  theme_light() +
    scale_y_continuous(labels=percent,   limits = (c(0,0.01))) + 
  facet_wrap(Model ~ Antibiotic + Antibiotic_mg.mL + Treatment2, scales = "free", ncol = 6, nrow = 4) -> p2
# 
# # Plot them side by side with the patchwork package.
# patch <- patchwork::wrap_plots(plot_list, nrow = 1, guides = "collect")
# patch & coord_flip() # make all plots horizontal (note: use & instead of +)
#   

p2 %>%
  ggpubr::set_palette("npg") -> p2

p2

p2 %>% 
  ggsave(plot = .,bg = "transparent",scale = 2,
         units = "mm",
         dpi = 600,
         height = 0.618 * 84 * 2,
         width = 84 * 2,
         filename = "~/Desktop/AMRcarrier_2.eps")

p2 %>%
  export::graph2ppt(append = TRUE, width = 317.48031496 * 2  , height = 0.618 * 317.48031496 * 2 , paper = "A4",  scaling = 2,
                    file = "~/Desktop/plots.pptx")
```



### Fig. 2b

```{r}
ps_AMR_all %>% 
  physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class_multi",
                         "PathoFact_AMR_ARG"))  %>%
  # physeq_sel_tax_table(c("PathoFact_AMR_ARG"))  %>% 
  speedyseq::tax_glom(., taxrank = "PathoFact_AMR_ARG")  %>% 
    prune_samples(samples, .) %>% 
  physeq_add_metadata(metadata_mapping,
                      sample_column = "sample_id") %>% 
  filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>% 
  speedyseq::psmelt() -> df  #%>% 
# filter(grepl('unclassified', PathoFact_AMR_AMR_sub_class_multi))  -> df

results <- vector("list", length(df$PathoFact_AMR_AMR_sub_class_multi %>%  unique()))
names(results) = c(df$PathoFact_AMR_AMR_sub_class_multi %>%  unique())

for (dd in df$PathoFact_AMR_AMR_sub_class_multi %>%  unique())
{
  print(dd)
  df %>% 
    filter(PathoFact_AMR_AMR_sub_class_multi == !!dd) %>% 
    select(-Lactose_mM:-Total_SCFA_mM) %>% 
    group_by(Sample, PathoFact_AMR_AMR_sub_class_multi) %>% 
    summarise(sum_abundance = sum(Abundance), .groups = "drop") %>% 
    mutate(sum_abundance = na_if(sum_abundance, 0)) %>%
    left_join(metadata_mapping,
              by = c("Sample" = "sample_id")) %>% 
    # filter(Model == "Chicken") %>% 
    # mutate(Day_of_Treatment_num = as.double(Day_of_Treatment)) %>% 
    # mutate(Day_of_Treatment = fct_relevel(Day_of_Treatment, c(NA, "-3", "-1", "1" , "3", "4", "5","7", "30", "47", "48")))  %>% 
    mutate(Antibiotic_mg.mL = factor(Antibiotic_mg.mL, levels = c(NA, 20, 200,90, 600))) %>%
    ggplot(aes_string(x = "Day_of_Treatment", y = "sum_abundance", color = "Treatment", fill  = "Treatment", shape = "Antibiotic_mg.mL")) + #shape = Fermentation
    geom_point(size = 2.5) + 
    geom_line(linetype = 2,  size = 0.5,  aes_string(group = "interaction(Model, Reactor_Treatment_Dose)")) +
    labs(y = paste0("sum Rnum_Gi ") , x = "Day_of_Treatment") +
    scale_color_manual(name = "", values = treat_col,
                       na.value = "black") +
    scale_fill_manual(name = "", values = treat_col,
                      na.value = "black") +
    scale_shape_manual(name = "" ,values = c(15,16,18,19), na.value =  17) +
    # scale_alpha_continuous(name = "", range=c(0.6, 1), na.value =  1) +
    geom_vline(xintercept = 0, size = 0.5, linetype = 1, color = "grey60") +
    ggtitle(dd) +
    theme_light() + 
    facet_grid(Antibiotic  ~ Model , scales = "free", drop = TRUE) +  scale_y_continuous(labels=percent) -> p1
    # scale_y_continuous(labels = function(x) format(x, scientific = TRUE)) -> p1
  
  # p1 + facet_grid(as.formula(paste0("Antibiotic_mg.mL + PathoFact_AMR_AMR_sub_class_multi ~  Model  ")), scales = "fixed", space = "fixed", drop = TRUE)
  # 
  # p1 + facet_wrap(as.formula(paste0("PathoFact_AMR_AMR_sub_class_multi ~  Model  ")),scales = "free"
  #)
  results[[dd]] <- p1
  

results[[dd]]  %>% 
  ggsave(plot = .,bg = "transparent",scale = 2,
         units = "mm",
         dpi = 600,
         height = 0.618 * 84,
         width = 84,
         filename = paste0("~/Desktop/", dd, ".eps"))

results[[dd]]  %>%
  export::graph2ppt(append = TRUE, width = 317.48031496 , height = 0.618 * 317.48031496, paper = "A4",  scaling = 2,
                    file = "~/Desktop/plots.pptx")
  
}
```

Vancomycin is a glycopeptide antibiotic used in the prophylaxis and treatment of infections caused by Gram-positive bacteria. Vancomycin inhibits the synthesis of peptidoglycan, the major component of the cell wall of gram-positive bacteria. Its mechanism of action is unusual in that it acts by binding precursors of peptidoglycan, rather than by interacting with an enzyme. -> glycopeptide antibiotic

```{r}
results$`glycopeptide antibiotic`

# results$`glycopeptide antibiotic` %>% 
#   export::graph2ppt(append = TRUE, width = 8, height = 4,
#                     file = out_pptx)
```

Cefotaxime is a semisynthetic cephalosporin taken parenterally. It is resistant to most beta-lactamases and active against Gram-negative rods and cocci due to its aminothiazoyl and methoximino functional groups. -> cephalosporin

```{r}
results$cephalosporin 

# results$cephalosporin %>% 
#   export::graph2ppt(append = TRUE, width = 8, height = 4,
#                     file = out_pptx)
```

same but normalized by baseline:

```{r}
ps_AMR_all %>% 
  physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class_multi",
                         "PathoFact_AMR_ARG"))  %>%
  # physeq_sel_tax_table(c("PathoFact_AMR_ARG"))  %>% 
  speedyseq::tax_glom(., taxrank = "PathoFact_AMR_ARG")  %>% 
    prune_samples(samples, .) %>% 
  physeq_add_metadata(metadata_mapping,
                      sample_column = "sample_id") %>% 
  filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>% 
  speedyseq::psmelt() -> df  #%>% 
# filter(grepl('unclassified', PathoFact_AMR_AMR_sub_class_multi))  -> df

results <- vector("list", length(df$PathoFact_AMR_AMR_sub_class_multi %>%  unique()))
names(results) = c(df$PathoFact_AMR_AMR_sub_class_multi %>%  unique())

for (dd in df$PathoFact_AMR_AMR_sub_class_multi %>%  unique())
{
  print(dd)
  
  df %>% 
    filter(PathoFact_AMR_AMR_sub_class_multi == !!dd) %>% 
    select(-Lactose_mM:-Total_SCFA_mM) %>% 
    group_by(Sample, PathoFact_AMR_AMR_sub_class_multi) %>% 
    summarise(sum_abundance = sum(Abundance), .groups = "drop") %>% 
    mutate(sum_abundance = na_if(sum_abundance, 0)) %>%
    left_join(metadata_mapping,
              by = c("Sample" = "sample_id")) %>% 
      ungroup() %>% 
  # mutate(Reactor_Treatment_Dose = paste0(Period, Reactor_Treatment_Dose)) %>% 
  group_by(Model, Fermentation, 
           Reactor_Treatment_Dose) %>% 
  arrange(Day_of_Treatment) %>% 
  mutate(per_cent_change = sum_abundance / first(sum_abundance)) %>% 
    # filter(Model == "Chicken") %>% 
    # mutate(Day_of_Treatment_num = as.double(Day_of_Treatment)) %>% 
    # mutate(Day_of_Treatment = fct_relevel(Day_of_Treatment, c(NA, "-3", "-1", "1" , "3", "4", "5","7", "30", "47", "48")))  %>% 
    mutate(Antibiotic_mg.mL = factor(Antibiotic_mg.mL, levels = c(NA, 20, 200,90, 600))) %>%
    ggplot(aes_string(x = "Day_of_Treatment", y = "per_cent_change", color = "Treatment", fill  = "Treatment", shape = "Antibiotic_mg.mL")) + #shape = Fermentation
    geom_point(size = 2.5) + 
    geom_line(linetype = 2,  size = 0.5,  aes_string(group = "interaction(Model, Reactor_Treatment_Dose)")) +
    labs(y = paste0("sum Rnum_Gi / baseline ") , x = "Day_of_Treatment") +
    scale_color_manual(name = "", values = treat_col,
                       na.value = "black") +
    scale_fill_manual(name = "", values = treat_col,
                      na.value = "black") +
    scale_shape_manual(name = "" ,values = c(15,16,18,19), na.value =  17) +
    # scale_alpha_continuous(name = "", range=c(0.6, 1), na.value =  1) +
    geom_vline(xintercept = 0, size = 0.5, linetype = 1, color = "grey60") +
    ggtitle(dd) +
    theme_light() + 
    facet_grid(Antibiotic  ~ Model , scales = "free_x", drop = TRUE) -> p1
    # scale_y_continuous(labels = function(x) format(x, scientific = TRUE)) -> p1
  
  # p1 + facet_grid(as.formula(paste0("Antibiotic_mg.mL + PathoFact_AMR_AMR_sub_class_multi ~  Model  ")), scales = "fixed", space = "fixed", drop = TRUE)
  # 
  # p1 + facet_wrap(as.formula(paste0("PathoFact_AMR_AMR_sub_class_multi ~  Model  ")),scales = "free"
  #)
  results[[dd]] <- p1
  

results[[dd]]  %>% 
  ggsave(plot = .,bg = "transparent",scale = 2,
         units = "mm",
         dpi = 600,
         height = 0.618 * 84,
         width = 84,
         filename = paste0("~/Desktop/pc_", dd, ".eps"))

results[[dd]]  %>%
  export::graph2ppt(append = TRUE, width = 317.48031496 , height = 0.618 * 317.48031496, paper = "A4",  scaling = 2,
                    file = "~/Desktop/plots.pptx")
  
}
```




Vancomycin is a glycopeptide antibiotic used in the prophylaxis and treatment of infections caused by Gram-positive bacteria. Vancomycin inhibits the synthesis of peptidoglycan, the major component of the cell wall of gram-positive bacteria. Its mechanism of action is unusual in that it acts by binding precursors of peptidoglycan, rather than by interacting with an enzyme. -> glycopeptide antibiotic

```{r}
results$`glycopeptide antibiotic`

# results$`glycopeptide antibiotic` %>% 
#   export::graph2ppt(append = TRUE, width = 8, height = 4,
#                     file = out_pptx)
```

Cefotaxime is a semisynthetic cephalosporin taken parenterally. It is resistant to most beta-lactamases and active against Gram-negative rods and cocci due to its aminothiazoyl and methoximino functional groups. -> cephalosporin

```{r}
results$cephalosporin 

# results$cephalosporin %>% 
#   export::graph2ppt(append = TRUE, width = 8, height = 4,
#                     file = out_pptx)
```

######### below exploratory ######### 


<!-- ```{r} -->
<!-- phyloseq_AMR_MAG_MGE %>%  -->
<!--   physeq_sel_tax_table(c("PathoFact_AMR_MGE_prediction_clean", -->
<!--                          # "PathoFact_AMR_AMR_sub_class_multi", -->
<!--                          "PathoFact_AMR_ARG"))  %>% -->
<!--   # physeq_sel_tax_table(c("PathoFact_AMR_ARG"))  %>%  -->
<!--   speedyseq::tax_glom(., taxrank = "PathoFact_AMR_ARG")  %>%  -->
<!--   filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>%  -->
<!--   speedyseq::psmelt() %>%  -->
<!--   mutate(PathoFact_AMR_MGE_prediction_clean_2 = case_when(grepl("ambiguous", PathoFact_AMR_MGE_prediction_clean) ~ "ambiguous", TRUE ~ PathoFact_AMR_MGE_prediction_clean)) %>%  -->
<!--   mutate(PathoFact_AMR_MGE_prediction_clean_2 = fct_relevel(PathoFact_AMR_MGE_prediction_clean_2, c("MAG", "chromosome", "plasmid", "phage", "ambiguous", "unclassified"))) %>%  -->
<!--   select(-Lactose_mM:-Total_SCFA_mM) %>%  -->
<!--   # left_join(tax_mapping, -->
<!--   #         by = c("PathoFact_AMR_ARG" = "PathoFact_AMR_ARG"), -->
<!--   #         suffix = c("_x", "")) %>% -->
<!--   mutate(Abundance = na_if(Abundance, 0)) %>% -->
<!--   mutate(logTPM = log10(Abundance + 1))  %>%  -->
<!--   # physeq_AMRgn %>% -->
<!--   #   microbiome::transform(transform = "log10p") %>%  -->
<!--   #   speedyseq::psmelt() %>%  -->
<!--   #   left_join(tax_mapping, -->
<!--   #             by = c("Best_Hit_ARO" = "Best_Hit_ARO"), -->
<!--   #             suffix = c("_x", "")) %>% -->
<!--   #   mutate(Abundance = na_if(Abundance, 0)) %>% -->
<!--   #   mutate(logTPM = log10(Abundance + 1))  %>%  -->
<!--   # separate_rows(., Resistance_Mechanism,sep = '; ',convert = TRUE) %>%  -->
<!--   ggplot(aes(x = as.factor(Day_of_Treatment), y = Abundance, fill = PathoFact_AMR_MGE_prediction_clean_2)) +    -->
<!--   geom_bar( stat = "identity", colour="black", size = 0.0125) + -->
<!--   facet_grid(Model ~ Treatment + Antibiotic_mg.mL , scales = "free", space = "free_x") + -->
<!--   ggtitle("ARG Type Abundance per Sample") + -->
<!--   xlab("Day (Treatment)")  + -->
<!--   # ylab("Proportion") +  -->
<!--   theme_light() + -->
<!--   ggpubr::rotate_x_text(60) -> p_mec -->

<!-- p_mec %>% -->
<!--   ggpubr::set_palette(MGE)-> p_mec -->

<!-- p_mec -->

<!-- p_mec %>% -->
<!--   export::graph2ppt(append = TRUE, width = 12, height = 8, -->
<!--                     file = out_pptx) -->
<!-- ``` -->

<!-- ```{r} -->
<!-- phyloseq_AMR_MAG_MGE %>%  -->
<!--   physeq_sel_tax_table(c("PathoFact_AMR_ARG", -->
<!--                          # "PathoFact_AMR_AMR_sub_class_multi", -->
<!--                          "GTDB_tax_CONCOCT_clean"))  %>% -->
<!--   subset_taxa(!is.na("GTDB_tax_CONCOCT_clean")) %>%  -->
<!--   # physeq_sel_tax_table(c("PathoFact_AMR_ARG"))  %>%  -->
<!--   speedyseq::tax_glom(., taxrank = "GTDB_tax_CONCOCT_clean")  %>%  -->
<!--   filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>%  -->
<!--   microbiome::transform("compositional") %>%  -->
<!--   speedyseq::psmelt() %>%  -->
<!--   separate(GTDB_tax_CONCOCT_clean, into = c("domain", "phylum", "class", "order", "family", "genus", "species"), sep  = ";" , remove = FALSE) -> df -->
<!-- ``` -->

<!-- ```{r} -->
<!-- df %>%  -->
<!--   # filter(Model == "Human") %>%  -->
<!--   filter(genus %in% c("g__Enterococcus", "g__Escherichia")) %>%  -->
<!--   mutate(Abundance = na_if(Abundance, 0)) %>% -->
<!--   mutate(logTPM = log10(Abundance + 1))  %>%  -->
<!--   # separate_rows(., Resistance_Mechanism,sep = '; ',convert = TRUE) %>%  -->
<!--   # ggplot(aes(x = as.factor(Day_of_Treatment), y = Abundance, fill = fct_reorder(Strain, logTPM))) +    -->
<!--   ggplot(aes(x = as.factor(Day_of_Treatment), y = Abundance, fill = GTDB_tax_CONCOCT_clean)) +#reorder(genus, Abundance, FUN=median, na.rm = TRUE))) + -->
<!--   geom_bar( stat = "identity", colour="black", size=0.025) + -->
<!--   facet_grid(Model  ~ Treatment + Fermentation + Antibiotic_mg.mL , scales = "free", space = "free_x", drop = TRUE) + -->
<!--   ggtitle("ARG Type Abundance per Sample") + -->
<!--   xlab("Day (Treatment)")  + -->
<!--   # ylab("Proportion") +  -->
<!--   theme_light() + -->
<!--   # scale_fill_manual(name = "", values = col_strains) + -->
<!--   theme(legend.position = "bottom") + -->
<!--   ggpubr::rotate_x_text(60) -> p_mec -->

<!-- # p_mec %>%  -->
<!-- #   ggpubr::set_palette("jco") -> p_mec -->

<!-- p_mec -->

<!-- p_mec %>% -->
<!--   export::graph2ppt(append = TRUE, width = 12, height = 8, -->
<!--                     file = out_pptx) -->
<!-- ``` -->



<!-- ### Fig. 2c -->

<!-- ```{r} -->
<!-- phyloseq_AMR_MAG_MGE %>%  -->
<!--   physeq_sel_tax_table(c("GTDB_tax_CONCOCT_clean", -->
<!--                          "PathoFact_AMR_AMR_sub_class_multi", -->
<!--                          "PathoFact_AMR_ARG"))  %>% -->
<!--   # physeq_sel_tax_table(c("PathoFact_AMR_ARG"))  %>%  -->
<!--   speedyseq::tax_glom(., taxrank = "PathoFact_AMR_ARG")  %>%  -->
<!--   filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>%  -->
<!--   speedyseq::psmelt() %>%  -->
<!--   # filter(!is.na(CONCOCT_bin )) %>%  -->
<!--   separate(GTDB_tax_CONCOCT_clean, into = c("domain", "phylum", "class", "order", "family", "genus", "species"), sep  = ";" ) -> df -->

<!-- ``` -->


<!-- ### Fig. 3 -->

<!-- ```{r} -->
<!-- phyloseq_AMR_MAG_MGE %>%  -->
<!--   physeq_sel_tax_table(c("PathoFact_AMR_MGE_prediction_clean", -->
<!--                          "PathoFact_AMR_AMR_sub_class_multi", -->
<!--                          "PathoFact_AMR_ARG"))  %>% -->
<!--   # physeq_sel_tax_table(c("PathoFact_AMR_ARG"))  %>%  -->
<!--   speedyseq::tax_glom(., taxrank = "PathoFact_AMR_ARG")  %>%  -->
<!--   filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>%  -->
<!--   speedyseq::psmelt() %>%  -->
<!--   mutate(PathoFact_AMR_MGE_prediction_clean_2 = case_when(grepl("ambiguous", PathoFact_AMR_MGE_prediction_clean) ~ "ambiguous", TRUE ~ PathoFact_AMR_MGE_prediction_clean)) %>%  -->
<!--   mutate(PathoFact_AMR_MGE_prediction_clean_2 = fct_relevel(PathoFact_AMR_MGE_prediction_clean_2, c("MAG", "chromosome", "plasmid", "phage", "ambiguous", "unclassified"))) -> df  #%>%  -->
<!-- # filter(grepl('unclassified', PathoFact_AMR_AMR_sub_class_multi))  -> df -->

<!-- results <- vector("list", length(df$PathoFact_AMR_MGE_prediction_clean_2 %>%  unique())) -->
<!-- names(results) = c(df$PathoFact_AMR_MGE_prediction_clean_2 %>%  unique()) -->

<!-- for (dd in df$PathoFact_AMR_MGE_prediction_clean_2 %>%  unique()) -->
<!-- { -->
<!--   print(dd) -->
<!--   df %>%  -->
<!--     filter(PathoFact_AMR_MGE_prediction_clean_2 == !!dd) %>%  -->
<!--     select(-Lactose_mM:-Total_SCFA_mM) %>%  -->
<!--     group_by(Sample, PathoFact_AMR_MGE_prediction_clean_2) %>%  -->
<!--     summarise(sum_abundance = sum(Abundance), .groups = "drop") %>%  -->
<!--     mutate(sum_abundance = na_if(sum_abundance, 0)) %>% -->
<!--     left_join(metadata_mapping, -->
<!--               by = c("Sample" = "sample_id")) %>%  -->
<!--     # filter(Model == "Chicken") %>%  -->
<!--     mutate(Day_of_Treatment_num = as.double(Day_of_Treatment)) %>%  -->
<!--     # mutate(Day_of_Treatment = fct_relevel(Day_of_Treatment, c(NA, "-3", "-1", "1" , "3", "4", "5","7", "30", "47", "48")))  %>%  -->
<!--     mutate(Antibiotic_mg.mL = factor(Antibiotic_mg.mL, levels = c(NA, 20, 200,90, 600))) %>% -->
<!--     ggplot(aes_string(x = "Day_of_Treatment", y = "sum_abundance", color = "Treatment", fill  = "Treatment", shape = "Antibiotic_mg.mL")) + #shape = Fermentation -->
<!--     geom_point(size = 2.5) +  -->
<!--     geom_line(linetype = 2,  size = 0.5,  aes_string(group = "interaction(Model, Reactor_Treatment_Dose)")) + -->
<!--     labs(y = paste0("sum Rnum_Gi ") , x = "Day_of_Treatment") + -->
<!--     scale_color_manual(name = "", values = treat_col, -->
<!--                        na.value = "black") + -->
<!--     scale_fill_manual(name = "", values = treat_col, -->
<!--                       na.value = "black") + -->
<!--     scale_shape_manual(name = "" ,values = c(15,16,18,19), na.value =  17) + -->
<!--     # scale_alpha_continuous(name = "", range=c(0.6, 1), na.value =  1) + -->
<!--     geom_vline(xintercept = 0, size = 0.5, linetype = 1, color = "grey60") + -->
<!--     ggtitle( dd ) + -->
<!--     theme_light() -> p1 -->

<!--   results[[dd]] <- p1 + facet_grid(as.formula(paste0("Antibiotic_mg.mL ~  Model  ")), scales = "free", space = "fixed", drop = TRUE) -->


<!--   results[[dd]] %>%  -->
<!--     export::graph2ppt(append = TRUE, width = 8, height = 4, -->
<!--                       file = out_pptx) -->

<!--   print(p1 + facet_grid(as.formula(paste0("Antibiotic_mg.mL  ~  Model  ")), scales = "free", space = "fixed", drop = TRUE)) -->
<!-- } -->
<!-- # print(p1 + facet_grid(as.formula(paste0("PathoFact_AMR_AMR_sub_class_multi ~  Model  ")), scales = "free", space = "free", drop = TRUE)) -->


<!-- ``` -->

<!-- ```{r} -->
<!-- results -->
<!-- ``` -->

<!-- per MGE per class... -->


<!-- ```{r} -->
<!-- phyloseq_AMR_MAG_MGE %>%  -->
<!--   physeq_sel_tax_table(c("PathoFact_AMR_MGE_prediction_clean", -->
<!--                          "PathoFact_AMR_AMR_sub_class_multi", -->
<!--                          "PathoFact_AMR_ARG"))  %>% -->
<!--   # physeq_sel_tax_table(c("PathoFact_AMR_ARG"))  %>%  -->
<!--   speedyseq::tax_glom(., taxrank = "PathoFact_AMR_ARG")  %>%  -->
<!--   filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>%  -->
<!--   speedyseq::psmelt() %>%  -->
<!--   mutate(PathoFact_AMR_MGE_prediction_clean_2 = case_when(grepl("ambiguous", PathoFact_AMR_MGE_prediction_clean) ~ "ambiguous", TRUE ~ PathoFact_AMR_MGE_prediction_clean)) %>%  -->
<!--   mutate(PathoFact_AMR_MGE_prediction_clean_2 = fct_relevel(PathoFact_AMR_MGE_prediction_clean_2, c("MAG", "chromosome", "plasmid", "phage", "ambiguous", "unclassified"))) -> df  #%>% -->
<!-- ``` -->


<!-- ```{r} -->
<!-- for(MGE  in c("MAG", "chromosome", "plasmid", "phage", "ambiguous", "unclassified")) -->
<!-- { -->

<!--   print(MGE) -->
<!--   df %>%  -->
<!--     filter(PathoFact_AMR_MGE_prediction_clean_2 == !!MGE) -> df_tmp -->

<!--   results <- vector("list", length(df_tmp$PathoFact_AMR_AMR_sub_class_multi %>%  unique())) -->

<!--   names(results) = c(df_tmp$PathoFact_AMR_AMR_sub_class_multi %>% unique()) -->




<!--   for (dd in df_tmp$PathoFact_AMR_AMR_sub_class_multi %>%  unique()) -->
<!--   { -->
<!--     print(dd) -->
<!--     df_tmp %>%  -->
<!--       filter(PathoFact_AMR_AMR_sub_class_multi == !!dd) %>%  -->
<!--       select(-Lactose_mM:-Total_SCFA_mM) %>%  -->
<!--       group_by(Sample, PathoFact_AMR_AMR_sub_class_multi) %>%  -->
<!--       summarise(sum_abundance = sum(Abundance), .groups = "drop") %>%  -->
<!--       mutate(sum_abundance = na_if(sum_abundance, 0)) %>% -->
<!--       left_join(metadata_mapping, -->
<!--                 by = c("Sample" = "sample_id")) %>%  -->
<!--       # filter(Model == "Chicken") %>%  -->
<!--       mutate(Day_of_Treatment_num = as.double(Day_of_Treatment)) %>%  -->
<!--       # mutate(Day_of_Treatment = fct_relevel(Day_of_Treatment, c(NA, "-3", "-1", "1" , "3", "4", "5","7", "30", "47", "48")))  %>%  -->
<!--       mutate(Antibiotic_mg.mL = factor(Antibiotic_mg.mL, levels = c(NA, 20, 200,90, 600))) %>% -->
<!--       ggplot(aes_string(x = "Day_of_Treatment", y = "sum_abundance", color = "Treatment", fill  = "Treatment", shape = "Antibiotic_mg.mL")) + #shape = Fermentation -->
<!--       geom_point(size = 2.5) +  -->
<!--       geom_line(linetype = 2,  size = 0.5,  aes_string(group = "interaction(Model, Reactor_Treatment_Dose)")) + -->
<!--       labs(y = paste0("sum Rnum_Gi ") , x = "Day_of_Treatment") + -->
<!--       scale_color_manual(name = "", values = treat_col, -->
<!--                          na.value = "black") + -->
<!--       scale_fill_manual(name = "", values = treat_col, -->
<!--                         na.value = "black") + -->
<!--       scale_shape_manual(name = "" ,values = c(15,16,18,19), na.value =  17) + -->
<!--       # scale_alpha_continuous(name = "", range=c(0.6, 1), na.value =  1) + -->
<!--       geom_vline(xintercept = 0, size = 0.5, linetype = 1, color = "grey60") + -->
<!--       ggtitle(paste0( dd, "- ", MGE )) + -->
<!--       theme_light() -> p1 -->

<!--     # p1 + facet_grid(as.formula(paste0("Antibiotic_mg.mL + PathoFact_AMR_AMR_sub_class_multi ~  Model  ")), scales = "fixed", space = "fixed", drop = TRUE) -->
<!--     #  -->
<!--     # p1 + facet_wrap(as.formula(paste0("PathoFact_AMR_AMR_sub_class_multi ~  Model  ")),scales = "free" -->
<!--     #) -->

<!--     results[[dd]] <- p1 + facet_grid(as.formula(paste0("Antibiotic_mg.mL ~  Model  ")),  scales = "free", space = "fixed", drop = TRUE) -->

<!--     results[[dd]] %>%  -->
<!--       export::graph2ppt(append = TRUE, width = 8, height = 4, -->
<!--                         file = out_pptx) -->

<!--     # print(p1 + facet_grid(as.formula(paste0(" Antibiotic_mg.mL ~  Model  ")), scales = "fixed", space = "fixed", drop = TRUE)) -->
<!--     #  -->
<!--     # print(p1 + facet_grid(as.formula(paste0("PathoFact_AMR_AMR_sub_class_multi ~  Model  ")), scales = "free", space = "free", drop = TRUE) -->
<!--   } -->
<!--   assign(MGE, results) -->
<!-- } -->
<!-- ``` -->


<!-- ```{r} -->
<!-- MAG -->
<!-- ``` -->

<!-- ```{r} -->
<!-- chromosome -->
<!-- ``` -->

<!-- ```{r} -->
<!-- plasmid -->
<!-- ``` -->

<!-- ```{r} -->
<!-- phage -->
<!-- ``` -->



<!-- ```{r} -->
<!-- # phyloseq_AMR_MAG_MGE %>%  -->
<!-- #   physeq_sel_tax_table(c("PathoFact_AMR_MGE_prediction_clean", -->
<!-- #                          "PathoFact_AMR_AMR_sub_class_multi", -->
<!-- #                          "PathoFact_AMR_ARG"))  %>% -->
<!-- #   # physeq_sel_tax_table(c("PathoFact_AMR_ARG"))  %>%  -->
<!-- #   speedyseq::tax_glom(., taxrank = "PathoFact_AMR_ARG")  %>%  -->
<!-- #   filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>%  -->
<!-- #   speedyseq::psmelt() %>%  -->
<!-- #   mutate(PathoFact_AMR_MGE_prediction_clean_2 = case_when(grepl("ambiguous", PathoFact_AMR_MGE_prediction_clean) ~ "ambiguous", TRUE ~ PathoFact_AMR_MGE_prediction_clean)) %>%  -->
<!-- #   mutate(PathoFact_AMR_MGE_prediction_clean_2 = fct_relevel(PathoFact_AMR_MGE_prediction_clean_2, c("MAG", "chromosome", "plasmid", "phage", "ambiguous", "unclassified"))) -> df  #%>%  -->
<!-- # # filter(grepl('unclassified', PathoFact_AMR_AMR_sub_class_multi))  -> df -->
<!-- #  -->
<!-- # results <- vector("list", length(df$PathoFact_AMR_AMR_sub_class_multi %>%  unique())) -->
<!-- # names(results) = c(df$PathoFact_AMR_AMR_sub_class_multi %>%  unique()) -->
<!-- #  -->
<!-- # for (dd in df$PathoFact_AMR_MGE_prediction_clean_2 %>%  unique()) -->
<!-- # { -->
<!-- #   print(dd) -->
<!-- #   df %>%  -->
<!-- #     filter(PathoFact_AMR_AMR_sub_class_multi == !!dd) %>%  -->
<!-- #      -->
<!-- #     results <- vector("list", length(df$PathoFact_AMR_MGE_prediction_clean_2 %>%  unique())) -->
<!-- #   names(results) = c(df$PathoFact_AMR_MGE_prediction_clean_2 %>%  unique()) -->
<!-- #    -->
<!-- #   for (dd in df$PathoFact_AMR_MGE_prediction_clean_2 %>%  unique()) -->
<!-- #   { -->
<!-- #     print(dd) -->
<!-- #     df %>%  -->
<!-- #       filter(PathoFact_AMR_MGE_prediction_clean_2 == !!dd) %>%  -->
<!-- #       select(-Lactose_mM:-Total_SCFA_mM) %>%  -->
<!-- #       group_by(Sample, PathoFact_AMR_MGE_prediction_clean_2) %>%  -->
<!-- #       summarise(sum_abundance = sum(Abundance), .groups = "drop") %>%  -->
<!-- #       mutate(sum_abundance = na_if(sum_abundance, 0)) %>% -->
<!-- #       left_join(metadata_mapping, -->
<!-- #                 by = c("Sample" = "sample_id")) %>%  -->
<!-- #       # filter(Model == "Chicken") %>%  -->
<!-- #       mutate(Day_of_Treatment_num = as.double(Day_of_Treatment)) %>%  -->
<!-- #       mutate(Day_of_Treatment = fct_relevel(Day_of_Treatment, c(NA, "-3", "-1", "1" , "3", "4", "5","7", "30", "47", "48")))  %>%  -->
<!-- #       mutate(Antibiotic_mg.mL = factor(Antibiotic_mg.mL, levels = c(NA, 20, 200,90, 600))) %>% -->
<!-- #       ggplot(aes_string(x = "Day_of_Treatment", y = "sum_abundance", color = "Treatment", fill  = "Treatment", shape = "Antibiotic_mg.mL")) + #shape = Fermentation -->
<!-- #       geom_point(size = 2.5) +  -->
<!-- #       geom_line(linetype = 2,  size = 0.5,  aes_string(group = "interaction(Model, Reactor_Treatment_Dose)")) + -->
<!-- #       labs(y = paste0("sum Rnum_Gi ") , x = "Day_of_Treatment") + -->
<!-- #       scale_color_manual(name = "", values = treat_col, -->
<!-- #                          na.value = "black") + -->
<!-- #       scale_fill_manual(name = "", values = treat_col, -->
<!-- #                         na.value = "black") + -->
<!-- #       scale_shape_manual(name = "" ,values = c(15,16,18,19), na.value =  17) + -->
<!-- #       # scale_alpha_continuous(name = "", range=c(0.6, 1), na.value =  1) + -->
<!-- #       geom_vline(xintercept = 0, size = 0.5, linetype = 1, color = "grey60") + -->
<!-- #       ggtitle( dd ) + -->
<!-- #       theme_light() -> p1 -->
<!-- #      -->
<!-- #     results[[dd]] <- p1 + facet_grid(as.formula(paste0("Antibiotic_mg.mL ~  Model  ")), scales = "fixed", space = "fixed", drop = TRUE) -->
<!-- #      -->
<!-- #     print(p1 + facet_grid(as.formula(paste0("Antibiotic_mg.mL  ~  Model  ")), scales = "free", space = "fixed", drop = TRUE)) -->
<!-- #   } -->
<!-- #   # print(p1 + facet_grid(as.formula(paste0("PathoFact_AMR_AMR_sub_class_multi ~  Model  ")), scales = "free", space = "free", drop = TRUE)) -->
<!-- #    -->
<!-- # } -->
<!-- ``` -->



```{r}
devtools::session_info()
```

