---
title: " Connection with anvio"
author: "Florentin Constancias "
date: " `r format(Sys.time(), '%B %d, %Y')` "
output: 
  html_document: 
    toc: yes
    keep_md: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
# knitr::opts_chunk$set(dev = "ragg_png") # https://ragg.r-lib.org/index.html
knitr::opts_chunk$set(fig.width = 6) #https://r4ds.had.co.nz/graphics-for-communication.html#figure-sizing
knitr::opts_chunk$set(fig.asp = 0.618)
knitr::opts_chunk$set(fig.show = "hold")
knitr::opts_chunk$set(fig.show = "70%")
knitr::opts_chunk$set(fig.align = "center")
```

```{r}
rm(list= ls())

gc()

require("tidyverse")
require('speedyseq')
require(microViz)


'%!in%' <- function(x,y)!('%in%'(x,y))

source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_taxa_tests.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_normalisation.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_alpha.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_beta.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_heatmap.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_varia.R")
source("https://raw.githubusercontent.com/fconstancias/metabaRpipe-source/master/Rscripts/functions.R")

```


```{r}
export_ppt_width = NULL
export_ppt_height = NULL
# out_pptx = here::here("output/plots_ISME.pptx")
out_pptx = "~/Desktop/plots_ISME2.pptx"

```

# Load data:

## 16S:

```{r}
# "data/processed/16S/ps_silva_dada2_human_chicken_all_fct_polyferms.RDS" %>% 
#   here::here() %>% 
#   readRDS() %>% 
#   filter_taxa(function(x) sum(x > 0) > 0, TRUE) -> ps_16S
```

## resistome:

```{r}
"data/processed/exp1_ps_AMR.RDS" %>% 
  here::here() %>% 
  readRDS() -> ps_AMR
```

## Aestetics:

```{r}
load( here::here("Figures/Rastetics.Rdata"))
```

```{r}
ps_AMR %>% 
  tax_table() %>% 
  data.frame()
```

## Others:

```{r}
ps_AMR %>% 
  tax_table() %>% 
  data.frame() %>% 
  rownames_to_column("ORF_ID") -> tax_mapping
```


```{r}
# ps_AMR %>% 
#   sample_data() %>% 
#   data.frame() %>% 
#   rownames_to_column("sample_id") %>% 
#     # filter(Model == "Chicken") %>% 
#   mutate(Day_of_Treatment_num = as.double(Day_of_Treatment)) %>% 
#   # mutate(Day_of_Treatment = fct_relevel(Day_of_Treatment, c(NA, "-3", "-1", "1" , "3", "4", "5","7", "30", "47", "48")))  %>% 
#   mutate(Antibiotic_mg.L = factor(Antibiotic_mg.mL, levels = c(NA, 20, 200,90, 600))) -> metadata_mapping

# metadata_mapping %>% 
#   write_tsv("~/Desktop/metadata_mapping.tsv")

"data/raw/metadata_mapping_AMR.tsv" %>% 
  here::here() %>% 
  read_tsv() %>% 
  mutate(Antibiotic_mg.L = factor(Antibiotic_mg.mL, levels = c(NA, 20, 200,90, 600))) -> metadata_mapping
```


```{r}
tax_mapping %>% 
  select(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi",
           "PathoFact_AMR_ARG")) %>% 
  distinct(PathoFact_AMR_ARG, .keep_all = TRUE) %>% 
  add_count(PathoFact_AMR_ARG) %>% 
  arrange(-n) %>% 
  column_to_rownames('PathoFact_AMR_ARG')

```

# Generate subset of data:

## AMR gene dataset

```{r}
ps_AMR %>% 
  physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi",
                         "PathoFact_AMR_Resistance_mechanism_multi",
                         "PathoFact_AMR_ARG")) %>% 
  physeq_glom_rename(taxrank = "PathoFact_AMR_ARG", speedyseq = TRUE) -> ps_AMR_gene

ps_AMR_gene
```

## non agglomarated dataset including: all the rows but only a selection of info: + MGE, bin, bin taxonomy

```{r}
ps_AMR %>% 
  physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi",
                         "PathoFact_AMR_Resistance_mechanism","PathoFact_AMR_Resistance_mechanism_multi",
                         "PathoFact_AMR_ARG")) %>% 
  physeq_glom_rename(taxrank = "PathoFact_AMR_ARG", speedyseq = TRUE) -> ps_AMR_gene

ps_AMR_gene
```

No need to melt for this one since every otu is a gene with AMR hit and MGE / MAG ...

```{r}
ps_AMR %>%
  physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi",
                         "PathoFact_AMR_Resistance_mechanism","PathoFact_AMR_Resistance_mechanism_multi",
                         "PathoFact_AMR_ARG", "PathoFact_AMR_Database", "PathoFact_AMR_MGE_prediction_clean_2",
                         "viralverify_Prediction", "isescan_type", "isescan_count_IS_contig" , "ANVIO_CONCOCT_HQ_clean", "CONCOCT_bin",
                         "GTDB_tax_CONCOCT_clean", "PathoFact_AMR_MGE_prediction_clean_2")) -> ps_AMR_all

ps_AMR_all
```

Looks good, let's go.

# Total Resistome :

## Abundance - sum Rnum_Gi:

```{r}
ps_AMR_all %>%
  # physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi", "PathoFact_AMR_ARG"))  %>% 
  physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi", "PathoFact_AMR_ARG"))  %>% 
  speedyseq::tax_glom(., taxrank = "PathoFact_AMR_ARG")  %>% 
  filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>% 
  speedyseq::psmelt() %>% 
  select(-Lactose_mM:-Total_SCFA_mM) %>% 
  filter(Day_of_Treatment < 34 &
           !Sample %in% c("D3", "D19")) %>% 
  filter(Reactor != "CR" & Model == "Chicken" | Model == "Human" & Reactor %in% c("TR1", "TR2", "TR3", "TR4", "TR5", "TR6", "CR"))  %>% 
  group_by(Sample) %>% 
  summarise(sum_abundance = sum(Abundance), .groups = "drop") %>% 
  mutate(sum_abundance = na_if(sum_abundance, 0)) %>%
  left_join(metadata_mapping,
            by = c("Sample" = "sample_id")) %>% 
  mutate(Antibiotic_mg.mL = factor(Antibiotic_mg.mL, levels = c(NA, 20, 200,90, 600))) %>%
  ggplot(aes_string(x = "Day_of_Treatment", y = "sum_abundance", color = "Treatment", fill  = "Treatment", shape = "Antibiotic_mg.L")) + #shape = Fermentation
  geom_point(size = 2.5) + 
  geom_line(linetype = 2,  size = 0.5,  aes_string(group = "interaction(Model,  Reactor_Treatment_Dose)")) +
  labs(y = "sum Rnum_Gi", x = "Day_of_Treatment") +
  scale_color_manual(name = "", values = treat_col,
                     na.value = "black") +
  scale_fill_manual(name = "", values = treat_col,
                    na.value = "black") +
  scale_shape_manual(name = "" ,values = c(15,16,18,19), na.value =  17) +
  # scale_alpha_continuous(name = "", range=c(0.6, 1), na.value =  1) +
  geom_vline(xintercept = 0, size = 0.5, linetype = 1, color = "grey60") +
  theme_light() + 
  scale_y_continuous(labels=scientific) -> p1 # scale_y_continuous(labels = function(x) format(x, scientific = TRUE))
```


```{r, fig.asp = 0.618}
p1 + facet_grid(as.formula(paste0("Antibiotic  ~  Model  ")), scales = "free", space = "fixed", drop = TRUE) -> ptmp

ptmp

ptmp %>% 
  ggsave(plot = .,bg = "transparent",scale = 2,
         units = "mm",
         dpi = 600,
         height = 0.618 * 84,
         width = 84,
         filename = "~/Desktop/Resistome_abundance.eps")

ptmp %>%
  export::graph2ppt(append = TRUE, width = 317.48031496 , height = 0.618 * 317.48031496, paper = "A4",  scaling = 2,
                    file = out_pptx)
```

Also retrieve the samples we now keep for the final analysis:

```{r}
p1$data %>% distinct(Sample)  %>% pull %>%  as.character() -> samples
```


```{r}
ps_AMR_all %>%
  physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi", "PathoFact_AMR_MGE_prediction_clean_2", "ANVIO_CONCOCT_HQ_clean", "CONCOCT_bin",
                         "GTDB_tax_CONCOCT_clean" ,"PathoFact_AMR_ARG")) %>% 
  prune_samples(samples, .) %>% 
  physeq_add_metadata(metadata_mapping,
                      sample_column = "sample_id") %>% 
  # subset_taxa(!is.na("GTDB_tax_CONCOCT_clean")) %>% 
  # physeq_sel_tax_table(c("PathoFact_AMR_ARG"))  %>% 
  # speedyseq::tax_glom(., taxrank = "GTDB_tax_CONCOCT_clean")  %>% 
  filter_taxa(function(x) sum(x > 0) > 0, TRUE) -> ps_tmp


ps_tmp %>%
  tax_table() %>% 
  data.frame() %>% 
  rownames_to_column('GN_id') %>%
  mutate(PathoFact_AMR_MGE_prediction_clean_2 = case_when(grepl("ambiguous", PathoFact_AMR_MGE_prediction_clean_2) ~ "ambiguous", TRUE ~ PathoFact_AMR_MGE_prediction_clean_2)) %>% 
  # mutate(PathoFact_AMR_MGE_prediction_clean_2 = case_when(PathoFact_AMR_MGE_prediction_clean_2 %in% c("MAG", "chromosome")  ~ "MAG - chromosome", TRUE ~ PathoFact_AMR_MGE_prediction_clean_2)) %>% 
  # mutate(PathoFact_AMR_MGE_prediction_clean_2 = fct_relevel(PathoFact_AMR_MGE_prediction_clean_2, c("MAG", "chromosome", "plasmid", "phage", "ambiguous", "unclassified"))) %>% 
  mutate(GN = paste(GN_id, PathoFact_AMR_MGE_prediction_clean_2, PathoFact_AMR_ARG, sep  = "_")) %>% 
  separate(GTDB_tax_CONCOCT_clean, into = c("domain", "phylum", "class", "order", "family", "genus", "species"), sep  = ";" , remove = FALSE) %>% 
  mutate(binid_phylum_species = paste(CONCOCT_bin, phylum, species, sep = ";")) %>% 
  mutate(MAG_plamid = case_when(PathoFact_AMR_MGE_prediction_clean_2 %in% c("plasmid") ~ PathoFact_AMR_MGE_prediction_clean_2, TRUE ~ binid_phylum_species)) %>%
  # mutate(MAG_plamid_gn = paste0(GN_id, MAG_plamid)) %>%
  column_to_rownames("GN_id") %>% 
  as.matrix() -> tax_table(ps_tmp)


ps_tmp %>% 
  speedyseq::tax_glom("MAG_plamid") %>%
  physeq_sel_tax_table("MAG_plamid") %>%  
  transform_sample_counts(function(x) x/sum(x)) %>% 
  # subset_taxa(PathoFact_AMR_AMR_sub_class_multi %!in% c("Unclassified", "multidrug (unclassified)")) %>% 
  # ps_mutate(Day_of_Treatment = as.numeric(Day_of_Treatment)) %>% 
  ps_arrange(-Day_of_Treatment) %>% 
  comp_barplot(
    tax_transform_for_plot = "identity",
    tax_level = "MAG_plamid",
    label = "Day_of_Treatment", # name an alternative variable to label axes
    n_taxa = 20, # give more taxa unique colours
    merge_other = FALSE, # split the "other" category to display alpha diversity
    bar_width = 0.9, # reduce the bar width to 70% of one row
    bar_outline_colour = "grey5",
    sample_order = "default") +#,# is the default (use NA to remove outlines)
  # facet_by = c("Model", "Antibiotic", "Antibiotic_mg.mL", "Treatment2")) +
  # group_by = c("Model", "Antibiotic", "Antibiotic_mg.mL", "Treatment2"))
  coord_flip() + theme_light() + 
  scale_y_continuous(labels=percent) + 
  facet_wrap(Model ~ Antibiotic + Antibiotic_mg.mL + Treatment2, scales = "free", ncol = 6, nrow = 4) +
  guides(fill=guide_legend(ncol = 1)) -> p2
```


```{r}
p2 +
  theme(legend.position = "none") -> p2_noleg

p2_noleg

p2_noleg %>%
  export::graph2ppt(append = TRUE, width = 317.48031496 * 2  , height = 0.618 * 317.48031496 * 2 , paper = "A4",  scaling = 2,
                    file = out_pptx)
```

```{r}
p2 %>% 
  ggpubr::get_legend() %>% 
  ggpubr::as_ggplot() -> p2_leg

p2_leg

p2_leg %>%
  export::graph2ppt(append = TRUE, width = 317.48031496 *2  , height = 0.618 * 317.48031496 *2 , paper = "A4",  scaling = 2,
                    file = out_pptx)
```

Looking only at only E coli and E facium MAGs: s_Escherichia coli s_Enterococcus faecalis


```{r}
ps_AMR_all %>%
  physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi", "PathoFact_AMR_MGE_prediction_clean_2", "ANVIO_CONCOCT_HQ_clean", "CONCOCT_bin",
                         "GTDB_tax_CONCOCT_clean" ,"PathoFact_AMR_ARG")) %>% 
  prune_samples(samples, .) %>% 
  physeq_add_metadata(metadata_mapping,
                      sample_column = "sample_id") %>% 
  # subset_taxa(!is.na("GTDB_tax_CONCOCT_clean")) %>% 
  # physeq_sel_tax_table(c("PathoFact_AMR_ARG"))  %>% 
  # speedyseq::tax_glom(., taxrank = "GTDB_tax_CONCOCT_clean")  %>% 
  filter_taxa(function(x) sum(x > 0) > 0, TRUE) -> ps_tmp


ps_tmp %>%
  tax_table() %>% 
  data.frame() %>% 
  rownames_to_column('GN_id') %>%
  mutate(PathoFact_AMR_MGE_prediction_clean_2 = case_when(grepl("ambiguous", PathoFact_AMR_MGE_prediction_clean_2) ~ "ambiguous", TRUE ~ PathoFact_AMR_MGE_prediction_clean_2)) %>% 
  # mutate(PathoFact_AMR_MGE_prediction_clean_2 = case_when(PathoFact_AMR_MGE_prediction_clean_2 %in% c("MAG", "chromosome")  ~ "MAG - chromosome", TRUE ~ PathoFact_AMR_MGE_prediction_clean_2)) %>% 
  # mutate(PathoFact_AMR_MGE_prediction_clean_2 = fct_relevel(PathoFact_AMR_MGE_prediction_clean_2, c("MAG", "chromosome", "plasmid", "phage", "ambiguous", "unclassified"))) %>% 
  mutate(GN = paste(GN_id, PathoFact_AMR_MGE_prediction_clean_2, PathoFact_AMR_ARG, sep  = "_")) %>% 
  separate(GTDB_tax_CONCOCT_clean, into = c("domain", "phylum", "class", "order", "family", "genus", "species"), sep  = ";" , remove = FALSE) %>% 
  mutate(binid_phylum_species = paste(CONCOCT_bin, phylum, species, sep = ";")) %>% 
  mutate(MAG_plamid = case_when(PathoFact_AMR_MGE_prediction_clean_2 %in% c("plasmid") ~ PathoFact_AMR_MGE_prediction_clean_2, TRUE ~ binid_phylum_species)) %>%
  # mutate(MAG_plamid_gn = paste0(GN_id, MAG_plamid)) %>%
  column_to_rownames("GN_id") %>% 
  as.matrix() -> tax_table(ps_tmp)


ps_tmp %>% 
  physeq_sel_tax_table("MAG_plamid") %>%  
  speedyseq::tax_glom("MAG_plamid") %>%
  transform_sample_counts(function(x) x/sum(x)) %>% 
  subset_taxa(grepl("coli|plasmid|Enterococcus_B faecium",  MAG_plamid)) %>%  # %!in% c("Unclassified", "multidrug (unclassified)")) %>%
  # ps_mutate(Day_of_Treatment = as.numeric(Day_of_Treatment)) %>% 
  ps_arrange(-Day_of_Treatment) %>% 
  comp_barplot(
    tax_transform_for_plot = "identity",
    tax_level = "MAG_plamid",
    label = "Day_of_Treatment", # name an alternative variable to label axes
    n_taxa = 30, # give more taxa unique colours
    merge_other = FALSE, # split the "other" category to display alpha diversity
    bar_width = 0.9, # reduce the bar width to 70% of one row
    bar_outline_colour = "grey5",
    sample_order = "default") +#,# is the default (use NA to remove outlines)
  # facet_by = c("Model", "Antibiotic", "Antibiotic_mg.mL", "Treatment2")) +
  # group_by = c("Model", "Antibiotic", "Antibiotic_mg.mL", "Treatment2"))
  coord_flip() + theme_light() + 
  scale_y_continuous(labels=percent) + 
  facet_wrap(Model ~ Antibiotic + Antibiotic_mg.mL + Treatment2, scales = "free_y", ncol = 6, nrow = 4) +
  guides(fill=guide_legend(ncol = 1)) -> p2
```


```{r}
p2

p2 +
  theme(legend.position = "none") -> p2_noleg

p2_noleg

p2_noleg %>%
  export::graph2ppt(append = TRUE, width = 317.48031496 * 2  , height = 0.618 * 317.48031496 * 2 , paper = "A4",  scaling = 2,
                    file = out_pptx)
```

Prepare ANVIO stuff:

We need a couple of things here:


Bin_name
GTDB
phylum
Genera
Species

number AMRG

```{r}
ps_AMR_all %>%
  physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi", "PathoFact_AMR_MGE_prediction_clean_2", "ANVIO_CONCOCT_HQ_clean", "CONCOCT_bin",
                         "GTDB_tax_CONCOCT_clean" ,"PathoFact_AMR_ARG")) %>% 
  prune_samples(samples, .) %>% 
  physeq_add_metadata(metadata_mapping,
                      sample_column = "sample_id") %>% 
  # subset_taxa(!is.na("GTDB_tax_CONCOCT_clean")) %>% 
  # physeq_sel_tax_table(c("PathoFact_AMR_ARG"))  %>% 
  # speedyseq::tax_glom(., taxrank = "GTDB_tax_CONCOCT_clean")  %>% 
  filter_taxa(function(x) sum(x > 0) > 0, TRUE) -> ps_tmp


ps_tmp %>%
  tax_table() %>% 
  data.frame() %>% 
  rownames_to_column('GN_id') %>%
  mutate(PathoFact_AMR_MGE_prediction_clean_2 = case_when(grepl("ambiguous", PathoFact_AMR_MGE_prediction_clean_2) ~ "ambiguous", TRUE ~ PathoFact_AMR_MGE_prediction_clean_2)) %>% 
  # mutate(PathoFact_AMR_MGE_prediction_clean_2 = case_when(PathoFact_AMR_MGE_prediction_clean_2 %in% c("MAG", "chromosome")  ~ "MAG - chromosome", TRUE ~ PathoFact_AMR_MGE_prediction_clean_2)) %>% 
  # mutate(PathoFact_AMR_MGE_prediction_clean_2 = fct_relevel(PathoFact_AMR_MGE_prediction_clean_2, c("MAG", "chromosome", "plasmid", "phage", "ambiguous", "unclassified"))) %>% 
  mutate(GN = paste(GN_id, PathoFact_AMR_MGE_prediction_clean_2, PathoFact_AMR_ARG, sep  = "_")) %>% 
  separate(GTDB_tax_CONCOCT_clean, into = c("domain", "phylum", "class", "order", "family", "genus", "species"), sep  = ";" , remove = FALSE) %>% 
  mutate(binid_phylum_species = paste(CONCOCT_bin, phylum, species, sep = ";")) %>% 
  mutate(MAG_plamid = case_when(PathoFact_AMR_MGE_prediction_clean_2 %in% c("plasmid") ~ PathoFact_AMR_MGE_prediction_clean_2, TRUE ~ binid_phylum_species)) %>%
  # mutate(MAG_plamid_gn = paste0(GN_id, MAG_plamid)) %>%
  column_to_rownames("GN_id") %>% 
  as.matrix() -> tax_table(ps_tmp)


ps_tmp %>%
  tax_table() %>% 
  data.frame() %>% 
  rownames_to_column('GN_id') %>%
  mutate(PathoFact_AMR_MGE_prediction_clean_2 = case_when(grepl("ambiguous", PathoFact_AMR_MGE_prediction_clean_2) ~ "ambiguous", TRUE ~ PathoFact_AMR_MGE_prediction_clean_2)) %>% 
  # mutate(PathoFact_AMR_MGE_prediction_clean_2 = case_when(PathoFact_AMR_MGE_prediction_clean_2 %in% c("MAG", "chromosome")  ~ "MAG - chromosome", TRUE ~ PathoFact_AMR_MGE_prediction_clean_2)) %>% 
  # mutate(PathoFact_AMR_MGE_prediction_clean_2 = fct_relevel(PathoFact_AMR_MGE_prediction_clean_2, c("MAG", "chromosome", "plasmid", "phage", "ambiguous", "unclassified"))) %>% 
  mutate(GN = paste(GN_id, PathoFact_AMR_MGE_prediction_clean_2, PathoFact_AMR_ARG, sep  = "_")) %>% 
  separate(GTDB_tax_CONCOCT_clean, into = c("domain", "phylum", "class", "order", "family", "genus", "species"), sep  = ";" , remove = FALSE) %>% 
  mutate(binid_phylum_species = paste(CONCOCT_bin, phylum, species, sep = ";")) %>% 
  mutate(MAG_plamid = case_when(PathoFact_AMR_MGE_prediction_clean_2 %in% c("plasmid") ~ PathoFact_AMR_MGE_prediction_clean_2, TRUE ~ binid_phylum_species)) %>%
  # mutate(MAG_plamid_gn = paste0(GN_id, MAG_plamid)) %>%
  column_to_rownames("GN_id") %>% 
  as.matrix() -> tax_table(ps_tmp)

tax_table(ps_tmp) %>%
  data.frame() %>%
  rownames_to_column("GN_id") %>%
  write_tsv("~/Desktop/MAG_taxa_ARG_h1_2_work_inprogress_first.tsv")
```

tax_table now contains gene id, AMR annotations, binID,..... The next step is to summarise info per MAG i.e., binid_phylum_species

```{r}
# ps_tmp %>% 
#     physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi", "PathoFact_AMR_MGE_prediction_clean_2", "ANVIO_CONCOCT_HQ_clean", 
#                          "GTDB_tax_CONCOCT_clean", "binid_phylum_species")) %>% 
#   speedyseq::tax_glom(taxrank = "binid_phylum_species") -> ps_tmp
```

Count ARG:

```{r}
# tax_table(ps_tmp) %>% 
#   data.frame() %>% 
#   rownames_to_column('rowname') %>% 
#   filter(! grepl('nclassified', PathoFact_AMR_AMR_sub_class_multi)) %>% 
#   separate(rowname,into = c("model", "mega", "contigid", "start", "stop"), sep = "_") %>% 
#   filter(!is.na(CONCOCT_bin)) %>% 
#   # filter(ANVIO_CONCOCT_HQ_clean == "HQ") %>% 
#   select(GN, model, binid_phylum_species, PathoFact_AMR_ARG, species, genus, phylum, CONCOCT_bin, PathoFact_AMR_AMR_sub_class, PathoFact_AMR_AMR_sub_class_multi, GTDB_tax_CONCOCT_clean) %>% 
#   select(model, CONCOCT_bin, binid_phylum_species, PathoFact_AMR_ARG, GTDB_tax_CONCOCT_clean) %>% 
#   # separate_rows(PathoFact_AMR_AMR_sub_class, sep = ";") %>% 
#   group_by(model, CONCOCT_bin, binid_phylum_species, GTDB_tax_CONCOCT_clean) %>% 
#   summarise(num_ARG = length(PathoFact_AMR_ARG)) -> MAGs_count_ARG
# 
# MAGs_count_ARG
# 
# MAGs_count_ARG %>%  filter(model == "h1") %>%  add_count(CONCOCT_bin) %>%  arrange(desc(n))
# 
# MAGs_count_ARG %>%
#   DT::datatable()
```

Problem here for some bins for same host model concoct bin is not unique ! e.g.,: h1	Bin_0_2_1	Bin_0_2_1;NA;NA & h1	Bin_0_2_1	Bin_0_2_1;p__Firmicutes;s__Lacticaseibacillus paracasei.


Seems better below:

```{r}
tax_table(ps_tmp) %>% 
  data.frame() %>% 
  rownames_to_column('GN_id') %>%
  filter(! grepl('nclassified', PathoFact_AMR_AMR_sub_class_multi)) %>% 
  separate(GN_id,into = c("model", "mega", "contigid", "start", "stop"), sep = "_") %>% 
  filter(!is.na(CONCOCT_bin)) %>%
  select(model, CONCOCT_bin, PathoFact_AMR_ARG) %>% 
  # separate_rows(PathoFact_AMR_AMR_sub_class, sep = ";") %>% 
  group_by(model, CONCOCT_bin) %>% 
  summarise(num_ARG = length(PathoFact_AMR_ARG)) -> MAGs_count_ARG

MAGs_count_ARG

MAGs_count_ARG %>% filter(model == "h1") %>%  add_count(CONCOCT_bin) %>%  arrange(desc(n))
```




drug class:

```{r}
# tax_table(ps_tmp) %>% 
#   data.frame() %>% 
#   rownames_to_column('rowname') %>% 
#   separate(rowname,into = c("model", "mega", "contigid", "start", "stop"), sep = "_") %>% 
#     filter(!is.na(CONCOCT_bin)) %>% 
#   filter(ANVIO_CONCOCT_HQ_clean == "HQ") %>% 
#     filter(! grepl('nclassified', PathoFact_AMR_AMR_sub_class_multi)) %>% 
#   select(GN, model, binid_phylum_species, PathoFact_AMR_ARG, species, genus, phylum, CONCOCT_bin, PathoFact_AMR_AMR_sub_class, PathoFact_AMR_AMR_sub_class_multi, GTDB_tax_CONCOCT_clean) %>% 
#   select(model, CONCOCT_bin, binid_phylum_species, CONCOCT_bin, PathoFact_AMR_AMR_sub_class_multi) %>% 
#   mutate(mode_binid_phylum_species = paste0(model, "_", binid_phylum_species)) %>% 
#   mutate(unic_id = paste0(mode_binid_phylum_species, PathoFact_AMR_AMR_sub_class_multi)) %>% 
#   # separate_rows(PathoFact_AMR_AMR_sub_class, sep = ";") %>% 
#   distinct(unic_id, .keep_all = TRUE) %>% 
#   select(-unic_id) %>% 
#   # group_by(model, CONCOCT_bin, binid_phylum_species, PathoFact_AMR_AMR_sub_class_multi) %>% 
#     # summarise(num_ARG = length(PathoFact_AMR_AMR_sub_class_multi)) %>% 
#     pivot_wider(names_from =  PathoFact_AMR_AMR_sub_class_multi,
#               values_from = mode_binid_phylum_species ) 
```


```{r}
# tax_table(ps_tmp) %>% 
#   data.frame() %>% 
#   rownames_to_column('rowname') %>% 
#   separate(rowname,into = c("model", "mega", "contigid", "start", "stop"), sep = "_") %>% 
#   filter(!is.na(CONCOCT_bin)) %>% 
#   filter(ANVIO_CONCOCT_HQ_clean == "HQ") %>% 
#   filter(! grepl('nclassified', PathoFact_AMR_AMR_sub_class_multi)) %>% 
#   separate_rows(PathoFact_AMR_AMR_sub_class, sep = ";", convert = TRUE) %>%
#   select(GN, model, binid_phylum_species, PathoFact_AMR_ARG, species, genus, phylum, CONCOCT_bin, PathoFact_AMR_AMR_sub_class, PathoFact_AMR_AMR_sub_class_multi, GTDB_tax_CONCOCT_clean) %>% 
#   select(model, CONCOCT_bin, binid_phylum_species, CONCOCT_bin, PathoFact_AMR_AMR_sub_class) %>% 
#   mutate(mode_binid_phylum_species = paste0(model, "_", binid_phylum_species)) %>% 
#   # separate_rows(PathoFact_AMR_AMR_sub_class, sep = ";") %>% 
#   group_by(model, CONCOCT_bin, binid_phylum_species, PathoFact_AMR_AMR_sub_class) %>%
#   summarise(num_ARG = length(PathoFact_AMR_AMR_sub_class)) %>%
#   pivot_wider(names_from =  PathoFact_AMR_AMR_sub_class,
#               values_from = num_ARG ) -> MAGs_count_class
# 
# MAGs_count_class
# 
# MAGs_count_class %>%  filter(model == "h1") %>%  add_count(CONCOCT_bin) %>%  arrange(desc(n))
```


```{r}
tax_table(ps_tmp) %>% 
  data.frame() %>% 
  rownames_to_column('rowname') %>% 
  separate(rowname,into = c("model", "mega", "contigid", "start", "stop"), sep = "_") %>% 
  # filter(!is.na(CONCOCT_bin)) %>%
  # filter(ANVIO_CONCOCT_HQ_clean == "HQ") %>% 
  filter(! grepl('nclassified', PathoFact_AMR_AMR_sub_class_multi)) %>% 
  select(GN, model, binid_phylum_species, PathoFact_AMR_ARG, species, genus, phylum, CONCOCT_bin, PathoFact_AMR_AMR_sub_class_multi, GTDB_tax_CONCOCT_clean) %>% 
  select(model, CONCOCT_bin, binid_phylum_species, CONCOCT_bin, PathoFact_AMR_AMR_sub_class_multi) %>% 
  mutate(mode_binid_phylum_species = paste0(model, "_", binid_phylum_species)) %>% 
  # separate_rows(PathoFact_AMR_AMR_sub_class, sep = ";") %>% 
  group_by(model, CONCOCT_bin, binid_phylum_species, PathoFact_AMR_AMR_sub_class_multi) %>%
  summarise(num_ARG = length(PathoFact_AMR_AMR_sub_class_multi)) %>%
  pivot_wider(names_from =  PathoFact_AMR_AMR_sub_class_multi,
              values_from = num_ARG ) -> MAGs_count_class_multi

MAGs_count_class_multi


MAGs_count_class_multi %>%  filter(model == "h1") %>%  add_count(CONCOCT_bin) %>%  arrange(desc(n))

MAGs_count_class_multi %>%
  DT::datatable()
```

same issue. let's try an alternative strategy.


```{r}
tax_table(ps_tmp) %>% 
  data.frame() %>% 
  rownames_to_column('GN_id') %>%
  filter(! grepl('nclassified', PathoFact_AMR_AMR_sub_class_multi)) %>% 
  separate(GN_id,into = c("model", "mega", "contigid", "start", "stop"), sep = "_") %>% 
  # filter(!is.na(CONCOCT_bin)) %>% 
  # select(model, CONCOCT_bin, PathoFact_AMR_ARG) %>% 
  # separate_rows(PathoFact_AMR_AMR_sub_class, sep = ";") %>% 
  filter(!is.na(CONCOCT_bin)) %>% 
  replace(is.na(.), 0) %>% 
  group_by(model, CONCOCT_bin, PathoFact_AMR_AMR_sub_class_multi) %>%
  summarise(num_ARG = length(PathoFact_AMR_AMR_sub_class_multi)) %>%
  pivot_wider(names_from =  PathoFact_AMR_AMR_sub_class_multi,
              values_from = num_ARG )  -> MAGs_count_class_multi

MAGs_count_class_multi


MAGs_count_class_multi %>%  filter(model == "h1") %>%  add_count(CONCOCT_bin) %>%  arrange(desc(n))

MAGs_count_class_multi %>%
  DT::datatable()

# MAGs_count_class_multi %>% 
#   write_tsv("~/Desktop/ddd.tsv")
```

Combined results:

```{r}

MAGs_count_class_multi %>% 
  mutate(model_bin = paste0(model, "_", CONCOCT_bin)) %>% 
  select(-model) %>% 
  rename_at(vars("aminoglycoside antibiotic":"elfamycin antibiotic"), ~paste0("DrugClass!", .)) -> MAGs_count_class_multi

MAGs_count_class_multi 

# colnames(MAGs_count_class_multi) <- paste0("DrugClass!", colnames(MAGs_count_class_multi))

MAGs_count_ARG %>% 
  mutate(model_bin = paste0(model, "_", CONCOCT_bin)) %>% 
  # left_join(.,
  #           MAGs_count_class,
  #           by = c("binid_phylum_species" = "binid_phylum_species")) %>% 
  left_join(.,
            MAGs_count_class_multi,
            by = c("model_bin" = "model_bin"),
            suffix = c("", "_multi")) %>% 
  # select(model,CONCOCT_bin, binid_phylum_species,num_ARG) %>% 
  # rename_at(vars(everything()), ~str_replace_all(., "\\s+", "")) %>% 
  replace(is.na(.), 0) %>%
  # left_join(ps_tmp %>%
  #             tax_table() %>%
  #             data.frame() %>% 
  #           select(binid_phylum_species, GTDB_tax_CONCOCT_clean, domain:species) %>%  distinct(binid_phylum_species, .keep_all = TRUE),
  #                       by = c("binid_phylum_species" = "binid_phylum_species")) %>%
  left_join(.,
            tax_mapping %>% 
              separate(ORF_ID,into = c("model", "mega", "contigid", "start", "stop"), sep = "_") %>% 
              mutate(model_bin = paste0(model, "_", CONCOCT_bin)),
            by = c("model_bin" = "model_bin"),
            suffix = c("", "_y")) %>% 
  separate(GTDB_tax_CONCOCT_clean, into = c("domain", "phylum", "class", "order", "family", "genus", "species"), sep  = ";" , remove = FALSE) %>% 
  select(model_bin:`DrugClass!elfamycin antibiotic`,num_ARG, GTDB_tax_CONCOCT_clean, domain:species) %>%
  distinct(model_bin, .keep_all = TRUE) %>% 
  filter(!is.na(GTDB_tax_CONCOCT_clean)) %>% 
  mutate(binid_phylum_species = "NA") %>%
  select(CONCOCT_bin_multi, model, binid_phylum_species,GTDB_tax_CONCOCT_clean,  everything(), -model_bin) -> out_tmp

out_tmp %>% 
  filter(model == "h1") %>%
  mutate(species = ifelse(species == "s__", genus, species)) %>% 
  write_tsv("~/Desktop/MAG_taxa_ARG_h1_2_work_inprogress.tsv")


out_tmp %>% 
  filter(model == "c1") %>%
    mutate(species = ifelse(species == "s__", genus, species)) %>% 
  write_tsv("~/Desktop/MAG_taxa_ARG_c1_2_work_inprogress.tsv")
```

sample_data_stuff:

sample carrier stuff:

```{r}
ps_AMR_all %>% 
  physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi", "PathoFact_AMR_ARG", "PathoFact_AMR_MGE_prediction_clean_2"))  %>%
  speedyseq::tax_glom(., taxrank = "PathoFact_AMR_MGE_prediction_clean_2")  %>% 
  physeq_sel_tax_table(c("PathoFact_AMR_MGE_prediction_clean_2")) %>% 
  # physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class_multi")) %>% 
  # prune_samples(samples, .) %>% 
  filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>% 
  physeq_add_metadata(metadata_mapping,
                      sample_column = "sample_id") -> ps_tmp

ps_tmp %>% 
  tax_table() %>% 
  data.frame() %>% 
  mutate(PathoFact_AMR_MGE_prediction_clean_2 = case_when(grepl("ambiguous", PathoFact_AMR_MGE_prediction_clean_2) ~ "ambiguous", TRUE ~ PathoFact_AMR_MGE_prediction_clean_2)) %>% 
  # mutate(PathoFact_AMR_MGE_prediction_clean_2 = case_when(PathoFact_AMR_MGE_prediction_clean_2 %in% c("MAG", "chromosome")  ~ "MAG - chromosome", TRUE ~ PathoFact_AMR_MGE_prediction_clean_2)) %>% 
  mutate(PathoFact_AMR_MGE_prediction_clean_2 = fct_relevel(PathoFact_AMR_MGE_prediction_clean_2, c("MAG", "chromosome", "plasmid", "phage", "ambiguous", "unclassified"))) %>% 
  as.matrix() -> tax_table(ps_tmp)

ps_tmp %>% 
  ps_arrange(-Day_of_Treatment) %>% 
  comp_barplot(
    tax_level = "PathoFact_AMR_MGE_prediction_clean_2",
    label = "Day_of_Treatment", # name an alternative variable to label axes
    n_taxa = 20, # give more taxa unique colours
    merge_other = FALSE, # split the "other" category to display alpha diversity
    bar_width = 0.9, # reduce the bar width to 70% of one row
    bar_outline_colour = "grey5",
    sample_order = "default") +#,# is the default (use NA to remove outlines)
  # facet_by = c("Model", "Antibiotic", "Antibiotic_mg.mL", "Treatment2")) +
  # group_by = c("Model", "Antibiotic", "Antibiotic_mg.mL", "Treatment2"))
  coord_flip() +  theme_light() +
  scale_y_continuous(labels=percent) + 
  facet_wrap(Model ~ Antibiotic + Antibiotic_mg.mL + Treatment2, scales = "free", ncol = 6, nrow = 4) -> p2

p_carrier_all <- p2
# # Plot them side by side with the patchwork package.
# patch <- patchwork::wrap_plots(plot_list, nrow = 1, guides = "collect")
# patch & coord_flip() # make all plots horizontal (note: use & instead of +)
#   

p2 %>%
  ggpubr::set_palette("npg") -> p2

p2

# p2 %>% 
#   ggsave(plot = .,bg = "transparent",scale = 2,
#          units = "mm",
#          dpi = 600,
#          height = 0.618 * 84 * 2,
#          width = 84 * 2,
#          filename = "~/Desktop/AMR_carrier_3.eps")
# 
# p2 %>%
#   export::graph2ppt(append = TRUE, width = 317.48031496 * 2  , height = 0.618 * 317.48031496 * 2 , paper = "A4",  scaling = 2,
#                     file = out_pptx)
```


```{r}
p_carrier_all$data %>% 
  select(OTU, Sample, Abundance) %>% 
  dplyr::group_by(Sample) %>% 
  pivot_wider(values_from = Abundance,
              names_from = OTU) -> sample_carrier_table

```

Class_multi:


```{r}
ps_AMR_all %>% 
  physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi", "PathoFact_AMR_ARG"))  %>%
  speedyseq::tax_glom(., taxrank = "PathoFact_AMR_AMR_sub_class_multi")  %>% 
  physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class_multi")) %>% 
  # prune_samples(samples, .) %>% 
  filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>% 
  physeq_add_metadata(metadata_mapping,
                      sample_column = "sample_id") %>% 
  transform_sample_counts(function(x) x/sum(x) * 100) %>% 
  subset_taxa(PathoFact_AMR_AMR_sub_class_multi %!in% c("Unclassified", "multidrug (unclassified)")) %>% 
  # ps_mutate(Day_of_Treatment = as.numeric(Day_of_Treatment)) %>% 
  ps_arrange(-Day_of_Treatment) %>% 
  comp_barplot(
    tax_level = "PathoFact_AMR_AMR_sub_class_multi",
    label = "Day_of_Treatment", # name an alternative variable to label axes
    n_taxa = 15, # give more taxa unique colours
    merge_other = FALSE, # split the "other" category to display alpha diversity
    bar_width = 0.9, # reduce the bar width to 70% of one row
    bar_outline_colour = "grey5",
    sample_order = "default") +#,# is the default (use NA to remove outlines)
  # facet_by = c("Model", "Antibiotic", "Antibiotic_mg.mL", "Treatment2")) +
  # group_by = c("Model", "Antibiotic", "Antibiotic_mg.mL", "Treatment2"))
  coord_flip() +  theme_light() + 
  scale_y_continuous(labels=percent) + 
  facet_wrap(Model ~ Antibiotic + Antibiotic_mg.mL + Treatment2, scales = "free", ncol = 6, nrow = 4) -> p2

# # Plot them side by side with the patchwork package.
# patch <- patchwork::wrap_plots(plot_list, nrow = 1, guides = "collect")
# patch & coord_flip() # make all plots horizontal (note: use & instead of +)
```


```{r}
plot_AMR_multi <- p2


p2

# 
# p2 %>% 
#   ggsave(plot = .,bg = "transparent",scale = 2,
#          units = "mm",
#          dpi = 600,
#          height = 0.618 * 84 * 2,
#          width = 84 * 2,
#          filename = "~/Desktop/Resistome_sub_class_multi_prop.eps")
# 
# p2 %>%
#   export::graph2ppt(append = TRUE, width = 317.48031496 * 2  , height = 0.618 * 317.48031496 * 2 , paper = "A4",  scaling = 2,
#                     file = out_pptx)
```

```{r}
plot_AMR_multi$data %>% 
  filter(! grepl('nclassified', OTU)) %>% 
  select(OTU, Sample, Abundance) %>% 
  dplyr::group_by(Sample) %>% 
  pivot_wider(values_from = Abundance,
              names_from = OTU) -> sample_AMR_table
```



```{r}
ps_AMR %>% 
  physeq_add_metadata(metadata_mapping,
                      sample_column = "sample_id") %>% 
  sample_data() %>% 
  data.frame() %>% 
  filter(Model == "Human") #%>% 
# write_tsv("~/Desktop/samples_data_h1.tsv")
```
add total antibiotic density:

```{r}
ps_AMR_all %>%
  # physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi", "PathoFact_AMR_ARG"))  %>% 
  physeq_sel_tax_table(c("PathoFact_AMR_AMR_sub_class", "PathoFact_AMR_AMR_sub_class_multi", "PathoFact_AMR_ARG"))  %>% 
  speedyseq::tax_glom(., taxrank = "PathoFact_AMR_ARG")  %>% 
  filter_taxa(function(x) sum(x > 0) > 0, TRUE) %>% 
  speedyseq::psmelt() %>% 
  select(-Lactose_mM:-Total_SCFA_mM) %>% 
  # filter(Day_of_Treatment < 34 &
  #          !Sample %in% c("D3", "D19")) %>% 
  # filter(Reactor != "CR" & Model == "Chicken" | Model == "Human" & Reactor %in% c("TR1", "TR2", "TR3", "TR4", "TR5", "TR6", "CR"))  %>% 
  group_by(Sample) %>% 
  summarise(sum_abundance = sum(Abundance), .groups = "drop") %>% 
  mutate(sum_abundance = na_if(sum_abundance, 0)) %>%
  left_join(metadata_mapping,
            by = c("Sample" = "sample_id")) %>% 
  mutate(Antibiotic_mg.mL = factor(Antibiotic_mg.mL, levels = c(NA, 20, 200,90, 600))) %>%
  ggplot(aes_string(x = "Day_of_Treatment", y = "sum_abundance", color = "Treatment", fill  = "Treatment", shape = "Antibiotic_mg.L")) + #shape = Fermentation
  geom_point(size = 2.5) + 
  geom_line(linetype = 2,  size = 0.5,  aes_string(group = "interaction(Model,  Reactor_Treatment_Dose)")) +
  labs(y = "sum Rnum_Gi", x = "Day_of_Treatment") +
  scale_color_manual(name = "", values = treat_col,
                     na.value = "black") +
  scale_fill_manual(name = "", values = treat_col,
                    na.value = "black") +
  scale_shape_manual(name = "" ,values = c(15,16,18,19), na.value =  17) +
  # scale_alpha_continuous(name = "", range=c(0.6, 1), na.value =  1) +
  geom_vline(xintercept = 0, size = 0.5, linetype = 1, color = "grey60") +
  theme_light() + 
  scale_y_continuous(labels=scientific) -> p_tot_resistome # scale_y_continuous(labels = function(x) format(x, scientific = TRUE))
```

```{r}
p_tot_resistome$data %>%
  # filter(! grepl('nclassified', OTU)) %>%
  select(Sample, sum_abundance) -> sample_resistome_density_table
# dplyr::group_by(Sample) %>%
# pivot_wider(names_from  = Sample,
#             values_from = sum_abundance,
#             ) -> sample_resistome_density_table
```

combine:

```{r}

colnames(sample_AMR_table) <- paste0("DrugClass!", colnames(sample_AMR_table))
colnames(sample_carrier_table) <- paste0("Carrier!", colnames(sample_carrier_table))


ps_AMR %>% 
  physeq_add_metadata(metadata_mapping,
                      sample_column = "sample_id") %>% 
  sample_data() %>% 
  data.frame() %>% 
  left_join(.,
            sample_AMR_table,
            by = c("sample_id" = "DrugClass!Sample")) %>% 
  left_join(sample_carrier_table,
            by = c("sample_id" = "Carrier!Sample")) %>% 
  left_join(sample_resistome_density_table,
            by = c("sample_id" = "Sample")) %>% 
  select(-Description2, -Experiment, -Day_of_Connection, -Day_from_Inoculum, -Phase, -Date:-VAN_Copy_Number_permL,
         -raw_metagenomic_pairs, -Period, -Day_of_Treatment_num) %>% 
  # mutate(Others = sum(`pleuromutilin antibiotic`, `para-aminosalicylic acid`, macrolide, `fusidic acid`, carbapenem, `acridine dye` ,`mupirocin`, na.rm = TRUE)) %>% 
  select(-`DrugClass!pleuromutilin antibiotic`:-`DrugClass!mupirocin`) %>% 
  rename_at(vars(ends_with("_mM")), ~ str_c("metabolites!", .)) %>% 
  rename(Total_SCFA_mM = `metabolites!Total_SCFA_mM`) -> sample_metadata_tmp

sample_metadata_tmp %>% 
  filter(Model == "Human") %>% 
  mutate(sample_id = str_replace(sample_id, "[.]", "_")) %>% 
  select(-Model, -Fermentation, )
  write_tsv("~/Desktop/samples_data_h1_end.tsv")

# then manually added smaple order, day of treatment 0 for donor -> samples_data_h1_end_anvio_ready

sample_metadata_tmp %>% 
  filter(Model == "Chicken") %>% 
    select(-Model, -Fermentation, )
  write_tsv("~/Desktop/samples_data_c1_end.tsv")

# then manually added smaple order, day of treatment 0 for donor -> samples_data_c1_end_anvio_ready

```


```{r}
devtools::session_info()
```



- the top gene -> where line plot facet localisation? heatmap?
- heatmap AMR class + genetic material + dendro
- heatmap AMR GN + dendro


Resistome Proportion
AMR categories
Gene heatmap
Diversity: alpha/beta
Contributing genetic material (MAGs/Chromosomes/plasmids/…)
Number of AMR/ contigs / plasmids – length
Number of unmapped reads / sample
Prop/coverage or whatsoever per gene per mag per sample (rector/time)
Visualise contigs with ARG in anvo -> coverage variability. 

proof of concept with WAFLE METACHIP
proof of concept ANVIO -> population variability

MAG resistome?
donor strain chromosomal resistome?
