---
title: "Chicken Resistome Visualization"
author: "Hannah Li HÃ¤gi & FC"
date: "`r format(Sys.time(), '%B %d, %Y')`"
output: 
  html_document: 
    toc: yes
    keep_md: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#### Load required packages

```{r,packages,message=FALSE}
library(tidyverse)
library(phyloseq)
library(speedyseq)
library(ggrepel)
library(here)
library(gridExtra)
library(readxl)
options(getClass.msg=FALSE) # https://github.com/epurdom/clusterExperiment/issues/66
#this fixes an error message that pops up because the class 'Annotated' is defined in two different packages
```

#### Source required functions

```{r message=FALSE}
rm(list = ls())

'%!in%' <- function(x,y)!('%in%'(x,y))

source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_biobakery_functions.R")
```

# Import data:

```{r}
chick = "~/Projects/ETH/Alessia/mobilome/chicken/chicken_resistome_phyloseq.rds"
human = "~/Projects/ETH/Alessia/mobilome/human/human_resistome_phyloseq.rds"

chick %>% 
  readRDS() -> ps_chick


human %>% 
  readRDS() -> ps_human
```



```{r}
taxa_names(ps_chick) <- paste0("chick_", taxa_names(ps_chick))
taxa_names(ps_human) <- paste0("human_", taxa_names(ps_human))
```




```{r}
as(tax_table(ps_chick), "matrix") %>% 
  as.data.frame() %>%
  rownames_to_column('ASV') -> tax_table_chick

as(otu_table(ps_chick), "matrix") %>% 
  as.data.frame() %>%
  rownames_to_column('ASV') -> otu_table_chick

ps_chick %>% 
  sample_data() %>% 
  data.frame() %>% 
  rownames_to_column('sample_id') ->  meta_chick


as(tax_table(ps_human), "matrix") %>% 
  as.data.frame() %>%
  rownames_to_column('ASV') -> tax_table_human

as(otu_table(ps_human), "matrix") %>% 
  as.data.frame() %>%
  rownames_to_column('ASV') -> otu_table_human

ps_human %>% 
  sample_data() %>% 
  data.frame() %>% 
  rownames_to_column('sample_id') ->  meta_human
```

We will combine everything at this stage so we can later agglomerate at the Best_Hit_ARO:
```{r}
otu_table_human %>%
  # left_join(refseq_df, by = 'ASV') %>%
  left_join(tax_table_human, by = c("ASV" = "ASV")) %>%
  dplyr::select(ASV, everything()) -> merged_human

otu_table_chick %>%
  # left_join(refseq_df, by = 'ASV') %>%
  left_join(tax_table_chick, by = c("ASV" = "ASV")) %>%
  dplyr::select(ASV, everything()) -> merged_chick
```

```{r}

full_join(merged_human,
          merged_chick) -> full_merged

full_merged %>% 
  write_tsv("~/Projects/ETH/Alessia/mobilome/meged.tsv")
```

Seems perfect. LEt's do the same for the metadata.


```{r}
full_join(meta_human,
          meta_chick) -> meta_merged

meta_merged %>% 
  write_tsv("~/Projects/ETH/Alessia/mobilome/meta_meged.tsv")
```

Create merged phyloseq object:

```{r}
full_merged %>% 
  select(ASV, 
         sample_names(ps_chick), sample_names(ps_human)) %>% 
  column_to_rownames("ASV") %>% 
  as.matrix() %>% 
  otu_table(taxa_are_rows = TRUE) -> otu_combined

full_merged %>% 
  select(ASV, 
         colnames(tax_table(ps_chick))) %>% 
  column_to_rownames("ASV") %>% 
  as.matrix() %>% 
  tax_table() -> tax_combined


phyloseq(otu_combined,
         tax_combined,
         meta_merged %>%
           column_to_rownames("sample_id") %>% 
           sample_data()) -> ps_combined


nsamples(ps_chick) + nsamples(ps_human); nsamples(ps_combined); print("looks good !")
```

```{r}
ps_combined
```



```{r}
ps_combined %>% 
  sample_data() %>% 
  data.frame() %>% 
  mutate_at(vars(Day_of_Treatment), ~ replace(., is.na(.), -1)) -> sample_data(ps_combined)
```


```{r}

as(tax_table(ps_combined), "matrix") %>% 
  as.data.frame() %>%
  rownames_to_column('ASV') -> tax_table_ps_combined

as(otu_table(ps_combined), "matrix") %>% 
  as.data.frame() %>%
  rownames_to_column('ASV') -> otu_table_ps_combined

ps_combined %>% 
  sample_data() %>% 
  data.frame() %>% 
  rownames_to_column('sample_id') ->  meta_ps_combined

otu_table_ps_combined %>%
  # left_join(refseq_df, by = 'ASV') %>%
  left_join(tax_table_ps_combined, by = c("ASV" = "ASV")) %>%
  dplyr::select(ASV, everything()) -> merged_ps_combined


merged_ps_combined %>% 
  write_tsv("~/Projects/ETH/Alessia/mobilome/merged_ps_combined.tsv")

meta_ps_combined %>% 
  write_tsv("~/Projects/ETH/Alessia/mobilome/merged_meta_ps_combined.tsv")

ps_combined %>% 
  saveRDS("~/Projects/ETH/Alessia/mobilome/ps_combined.rds")
```

<!-- Create phyloseq object: -->
<!-- ```{r, fig.height=18, fig.width=16} -->
<!-- fil_df %>%  -->
<!--   # select(ORF_ID, Best_Hit_ARO, Model_type, `Drug Class`, `Resistance Mechanism`, `AMR Gene Family`, -->
<!--   #        starts_with("TPM "), Predicted_DNA) %>%  -->
<!--   mutate(`Drug Class` = str_remove_all(`Drug Class`," antibiotic")) %>%  -->
<!--   mutate(Drug_Class_multi = `Drug Class`) %>%  -->

<!--   column_to_rownames("ORF_ID") -> ps_ready -->

<!-- ps_ready$Drug_Class_multi[grepl(";", ps_ready$Drug_Class_multi)] <- "multi-drug" -->

<!-- ps_ready %>%  -->
<!--   select(-starts_with("TPM ")) %>%  -->
<!--   as.matrix() %>%  -->
<!--   tax_table() -> ps_tax -->

<!-- ps_ready %>%  -->
<!--   select(starts_with("TPM ")) %>%  -->
<!--   as.matrix() %>%  -->
<!--   otu_table(taxa_are_rows = TRUE) -> ps_otu -->


<!-- ps_ready$Predicted_DNA -> DNA -->

<!-- names(DNA) <- ps_ready$ORF_ID -->

<!-- physeq = phyloseq(ps_tax,  -->
<!--                   ps_otu) %>%  -->
<!--   clean_phyloseq_sample_names(sub_pat =  "TPM ") %>%  -->
<!--   clean_phyloseq_sample_names(sub_pat =  "-", str_replace = "_") -->

<!-- # physeq@refseq <- Biostrings::DNAStringSet(DNA) -->

<!-- merge_phyloseq(physeq, -->
<!--                meta %>% column_to_rownames("metagenomic_sample_name") %>% sample_data()) -> physeq -->

<!-- colnames(tax_table(physeq)) <- str_replace(colnames(tax_table(physeq)), " ", "_") -->





<!-- sample_data(physeq)$Reactor_Treatment <- fct_relevel(sample_data(physeq)$Reactor_Treatment, "DONOR", "CR_UNTREATED", "TR1_CTX+HV292.1", -->
<!--                                                      "TR2_CTX", "TR3_HV292.1", "TR5_VAN+CCUG59168", "TR4_VAN", "TR6_CCUG59168")  -->

<!-- sample_data(physeq)$Treatment <- fct_relevel(sample_data(physeq)$Treatment, "DONOR", "UNTREATED",  "CTX+HV292.1", "CTX","HV292.1","VAN+CCUG59168", "VAN",  "CCUG59168") -->


<!-- # source("https://raw.githubusercontent.com/fconstancias/metabarcodingRpipeline/dev/scripts/functions_export_simplified.R") -->
<!-- #  -->
<!-- # physeq %>%  -->
<!-- #   phloseq_export_otu_tax() %>%  -->
<!-- #   write_tsv("human_resistome_phyloseq_rds.tsv") -->

<!-- as(tax_table(physeq), "matrix") %>%  -->
<!--   as.data.frame() %>% -->
<!--   rownames_to_column('ASV') -> tax_table -->

<!-- as(otu_table(physeq), "matrix") %>%  -->
<!--   as.data.frame() %>% -->
<!--   rownames_to_column('ASV') -> otu_table -->

<!-- otu_table %>% -->
<!--   # left_join(refseq_df, by = 'ASV') %>% -->
<!--   left_join(tax_table, by = c("ASV" = "ASV")) %>% -->
<!--   dplyr::select(ASV, everything()) %>%  -->
<!--   write_tsv("~/Projects/ETH/Alessia/mobilome/chicken/chicken_resistome_phyloseq_rds.tsv") -->

<!-- saveRDS(physeq, file = "~/Projects/ETH/Alessia/mobilome/chicken/chicken_resistome_phyloseq.rds") -->

<!-- #  -->
<!-- #  -->
<!-- # colnames(ps_ready) <- sub("TPM ", "", colnames(ps_ready)) -->
<!-- #  -->
<!-- # agg_df <- aggregate(fil_df[c(paste0("TPM D", 1:43))], by = fil_df["Best_Hit_ARO"] , sum, drop = F)  -->
<!-- #  -->
<!-- # agg_df %>%  -->
<!-- #   left_join(., fil_df[c("Best_Hit_ARO", "Drug.Class")], by ="Best_Hit_ARO") %>% -->
<!-- #   unique %>% -->
<!-- #   as.data.frame() %>% -->
<!-- #   remove_rownames() %>% -->
<!-- #   column_to_rownames(., "Best_Hit_ARO") -> df  -->
<!-- #  -->
<!-- #  -->
<!-- # colnames(df) <- sub("TPM ", "", colnames(df)) -->
<!-- # df$Drug.Class <- str_remove_all(df$Drug.Class," antibiotic") -->
<!-- #  -->
<!-- # # Rename the entries containing more than 1 drug resistance -->
<!-- # df$Drug.Class[grepl(";", df$Drug.Class)] <- "multi-drug" -->
<!-- #  -->
<!-- # #create tax_table -->
<!-- # df %>% -->
<!-- #   dplyr::select(c("Drug.Class")) %>% -->
<!-- #   as.matrix() %>% -->
<!-- #   tax_table()-> TAX -->
<!-- #  -->
<!-- # #create otu table containing the abundances of each gen -->
<!-- # df %>%  -->
<!-- #   dplyr::select(c(paste0("D", 1:43))) %>%  -->
<!-- #   otu_table(taxa_are_rows = TRUE)-> OTU -->
<!-- #  -->
<!-- # #create sample_info table containing meta information -->
<!-- # meta$Treatment <- factor(meta$Treatment,  -->
<!-- #                          ordered = T,  -->
<!-- #                          levels = c("DONOR", "UNTREATED", "CCUG59168", "VAN",  -->
<!-- #                                     "VAN+CCUG59168", "HV292.1", "CTX", "CTX+HV292.1")) -->
<!-- #  -->
<!-- # meta$Day_of_Treatment <- factor(meta$Day_of_Treatment,  -->
<!-- #                                 levels = c("-1", "1", "3", "5", "30", "48")) -->
<!-- #  -->
<!-- # meta %>% -->
<!-- #   as.data.frame() %>%  -->
<!-- #   dplyr::select(. , raw_metagenomic_pairs, metagenomic_sample_name, -->
<!-- #                 Treatment, Day_of_Treatment, Fermentation, Reactor_Treatment) %>%  -->
<!-- #   column_to_rownames(., "metagenomic_sample_name") %>% -->
<!-- #   mutate(Day_of_Treatment = as.factor(Day_of_Treatment)) %>% -->
<!-- #   sample_data()-> sampleinfo -->
<!-- #  -->
<!-- #  -->
<!-- # physeq = phyloseq(OTU, TAX, sampleinfo) -->
<!-- # saveRDS(physeq, file = "resistome_phyloseq.rds") -->

<!-- ``` -->



```{r}
sessionInfo()
```