---
title: "SQM to phyloseq"
output: html_document
date: '2022-05-19'
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r}
rm(list= ls())
library('dplyr')
library("tidyverse")
library('here')
library("SQMtools")
library("phyloseq")
library("readxl")


'%!in%' <- function(x,y)!('%in%'(x,y))

#source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_taxa_tests.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_normalisation.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_alpha.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_beta.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_heatmap.R")


```

Load dataset:

```{r}

human <- "~/Documents/MSc_Thesis-NRP72/data/human_SQM.rds" %>% readRDS()

"~/Documents/NRP72-FBT/data/raw/25.11.2021_metadata_updated.tsv" %>% 
  read_tsv() %>% 
  data.frame() %>% 
  filter(!is.na(metagenomic_sample_name), Model == "Human") -> human_meta


```

Modify meta data:

```{r}
# create Period column

human_meta %>%
  mutate(Reactor_Treatment = base::replace(Reactor_Treatment, Reactor_Treatment == "TR_2CTX", "TR2_CTX")) %>%
  mutate(Reactor_Treatment_Dose = if_else(!is.na(`Antibiotic_mg.mL`), paste0(Reactor_Treatment, `Antibiotic_mg.mL`), Reactor_Treatment)) %>% 
  mutate(Treatment_Dose = if_else(!is.na(`Antibiotic_mg.mL`), paste0(Treatment, `Antibiotic_mg.mL`), Treatment)) %>%
  mutate(
    Period = case_when(
      Day_of_Treatment <= 0  ~ "pret",
      Day_of_Treatment > 0 ~ as.character(Day_of_Treatment)
    )) %>%
  mutate(Period = base::replace(Period, Treatment == "DONOR", "pret")) %>%
  mutate(Day_of_Treatment = addNA(Day_of_Treatment)) %>%
  column_to_rownames("metagenomic_sample_name") -> human_meta


#human_meta$Treatment_Dose <- factor(human_meta$Treatment_Dose, ordered = TRUE, 
#                                 levels = c("DONOR", "UNTREATED", "CTX+HV292.120", "CTX+HV292.1200",
#                                            "CTX20", "CTX200", "HV292.1", "VAN+CCUG5916890", "VAN+CCUG59168600",
#                                            "VAN90", "VAN600", "CCUG59168"))

```


Extract infos for otu table
```{r}
# TPM
human$orfs$table %>% as.data.frame() %>% select(starts_with("TPM")) -> human_otu_tpm


colnames(human_otu_tpm) <- colnames(human_otu_tpm) %>% str_remove_all("TPM ") %>% str_replace_all("-", "_")
rownames(human_otu_tpm) <- paste0("h_", rownames(human_otu_tpm))

# Coverage
human$orfs$table %>% as.data.frame() %>% select(starts_with("Coverage")) -> human_otu_cov

colnames(human_otu_cov) <- colnames(human_otu_cov) %>% str_remove_all("Coverage ") %>% str_replace_all("-", "_")
rownames(human_otu_cov) <- paste0("h_", rownames(human_otu_cov))

# Raw reads
human$orfs$table %>% as.data.frame() %>% select(starts_with("Raw read count")) -> human_otu_rr

colnames(human_otu_rr) <- colnames(human_otu_rr) %>% str_remove_all("Raw read count ") %>% str_replace_all("-", "_")
rownames(human_otu_rr) <- paste0("h_", rownames(human_otu_rr))
```

Extract information for tax table:
```{r}


# Resistome data:
"~/Documents/MSc_Thesis-NRP72/data/resistome/rgi_contig_human.txt" %>% 
  read.delim() %>%
  select(-ORF_ID) %>%
  filter(Model_type %in% "protein homolog model",
       Cut_Off %in% c("Strict", "Perfect"),
       Nudged %!in% c("TRUE"),
       Best_Identities > 95, # plot distribution of bestID vs percentage length ref and color bitscore and symbol class AMR
       Percentage.Length.of.Reference.Sequence > 80) %>%
  separate(Contig, c("Megahit", "Contig", "Split"), sep = "_", fill = "right") %>%
  mutate(ORF_ID = paste(Megahit, Contig, paste(Start, Stop, sep = "-"), sep = "_")) %>%
  #mutate(Contig = paste(Megahit, Contig, sep = "_")) %>%
  select(-c(Megahit, Split, CARD_Protein_Sequence, Predicted_DNA, Predicted_Protein, Contig)) -> card



# HGT events
hgt <- "~/Documents/MSc_Thesis-NRP72/data/HGT/human.contigs.lgt.tsv" %>% read_tsv() %>% select(CONTIG_NAME, CALL)
no_hgt <- "~/Documents/MSc_Thesis-NRP72/data/HGT/human.contigs.no_lgt.tsv" %>% read_tsv() %>% select(CONTIG_NAME, CALL)
unclassified_hgt <- "~/Documents/MSc_Thesis-NRP72/data/HGT/human.contigs.unclassified.tsv" %>% read_tsv() %>% select(CONTIG_NAME, CALL)

hgt_events <- rbind(hgt, no_hgt, unclassified_hgt)

# Viralverify 
"~/Documents/MSc_Thesis-NRP72/data/mobilome/01.SqueezeHuman_result_table_viralverify.csv" %>%
  read_csv() -> viralverify


# Isescan
"~/Documents/MSc_Thesis-NRP72/data/mobilome/human_isescan.tsv" %>%
  read_tsv() -> isescan

isescan %>%
   group_by(seqID) %>%
   top_n(1, wt = `E-value4copy`) %>%
   top_n(1, wt = isBegin) %>%
   mutate(IS = "yes") %>%
   select(IS, seqID:isBegin, type) -> top_isescan

```

MAG information:

```{r}
# MAGs DAS 
"~/Documents/MSc_Thesis-NRP72/data/anvio/human_DAS.txt" %>% 
  read_tsv(col_names = c("split", "DAS_bin")) %>% 
  separate(split, into = c("cont", "cont_num", "split", "split_id"), sep = "_", remove = FALSE) %>% 
  mutate(cont_id = paste0(cont,"_", cont_num)) %>% 
  distinct(cont_id, .keep_all = TRUE) %>% 
  select(cont_id, DAS_bin) -> DAS

# DAS tax info 
"~/Documents/MSc_Thesis-NRP72/data/anvio/GTDB_all.xlsx" %>%
  read_excel() %>%
  filter(dataset == "human1") %>%
  mutate(user_genome = str_remove_all(user_genome, c("human1_|-contigs|refined_2_"))) %>%
  filter(str_detect(user_genome, c("DAS|concoct"))) %>%
  select(user_genome, classification) %>%
  separate(classification, 
           into = c("domain", "phylum", "class", "order", "family", "genus", "species"), 
           sep = ";", 
           fill = "right") -> tax_info
  
tax_info %>%
  filter(str_detect(user_genome, "DAS")) %>%
  mutate(user_genome = str_remove_all(user_genome, "DAS_")) -> DAS_tax

colnames(DAS_tax) <- paste0("DAS_", colnames(DAS_tax))

# HQ MAGs DAS

"~/Documents/MSc_Thesis-NRP72/data/anvio/chicken_DAS-info.txt" %>% 
  read.delim() -> DAS_info

colnames(DAS_info) <- paste0("DAS_", colnames(DAS_info))

DAS_info %>%
  filter(DAS_percent_completion >= 80,
         DAS_percent_redundancy <= 10) %>% 
  mutate(DAS_bins_HQ = DAS_bins) %>%
  select(DAS_bins_HQ)-> DAS_HQ

# MAGs CONCOCT
"~/Documents/MSc_Thesis-NRP72/data/anvio/human_concoct.txt" %>% 
  read_tsv(col_names = c("split", "CONCOCT_bin")) %>% 
  separate(split, into = c("cont", "cont_num", "split", "split_id"), sep = "_", remove = FALSE) %>% 
  mutate(cont_id = paste0(cont,"_", cont_num)) %>% 
  distinct(cont_id, .keep_all = TRUE) %>% 
  select(cont_id, CONCOCT_bin) -> CONCOCT

# CONCOCT tax info

tax_info %>%
  filter(str_detect(user_genome, "concoct")) %>%
  mutate(user_genome = str_remove_all(user_genome, "concoct_10_")) -> CONCOCT_tax

colnames(CONCOCT_tax) <- paste0("CONCOCT_", colnames(CONCOCT_tax))


# HQ MAGs CONCOCT 
"~/Documents/MSc_Thesis-NRP72/data/anvio/chicken_concoct-info.txt" %>% 
  read.delim() -> CONCOCT_info

colnames(CONCOCT_info) <- paste0("CONCOCT_", colnames(CONCOCT_info))

CONCOCT_info %>%
  filter(CONCOCT_percent_completion >= 80,
         CONCOCT_percent_redundancy <= 10) %>% 
  mutate(CONCOCT_bins_HQ = CONCOCT_bins) %>%
  select(CONCOCT_bins_HQ)-> CONCOCT_HQ

```

# Add resistome, mobilime, MAG and hgt event info to tax table

```{r}
human$orfs$tax %>% 
  as.data.frame() %>% 
  rownames_to_column("ORF_ID") %>%
  separate(ORF_ID, c("Megahit", "Contig", "Range"), sep = "_", fill = "right", remove = FALSE) %>%
  mutate(Contig = as.numeric(Contig)) %>%
  mutate(Contig = paste(Megahit, Contig, sep = "_")) %>%
  left_join(card, by = "ORF_ID") %>%
  left_join(hgt_events, by = c("Contig" = "CONTIG_NAME")) %>%
  left_join(viralverify, by = c("Contig" = "Contig name")) %>%
  left_join(top_isescan, by = c("Contig" = "seqID")) %>%
  left_join(DAS, by = c("Contig" = "cont_id")) %>%
  left_join(DAS_tax, by = c("DAS_bin" ="DAS_user_genome")) %>%
  left_join(DAS_HQ, by = c("DAS_bin" = "DAS_bins_HQ"), keep = TRUE) %>%
  left_join(CONCOCT, by = c("Contig" = "cont_id")) %>%
  left_join(CONCOCT_tax, by = c("CONCOCT_bin" ="CONCOCT_user_genome")) %>%
  left_join(CONCOCT_HQ, by = c("CONCOCT_bin" = "CONCOCT_bins_HQ"), keep = TRUE) %>%
  select(-c(Megahit, Range)) %>%
  column_to_rownames("ORF_ID") %>%
  mutate(Classification_mobilome_DAS = case_when(
    Prediction == "Plasmid" ~ "Plasmid",
    !is.na(DAS_bin) ~ "MAG",
    TRUE ~ Prediction)) %>%
  mutate(Classification_mobilome_DAS_HQ = case_when(
    Prediction == "Plasmid" ~ "Plasmid",
    !is.na(DAS_bins_HQ) ~ "MAG",
    TRUE ~ Prediction)) %>%
  mutate(Classification_mobilome_CONCOCT = case_when(
    Prediction == "Plasmid" ~ "Plasmid",
    !is.na(CONCOCT_bin) ~ "MAG",
    TRUE ~ Prediction)) %>%
  mutate(Classification_mobilome_CONCOCT_HQ = case_when(
    Prediction == "Plasmid" ~ "Plasmid",
    !is.na(CONCOCT_bins_HQ) ~ "MAG",
    TRUE ~ Prediction)) %>%
  as.matrix(rownames = TRUE) -> human_tax


rownames(human_tax) <- paste0("h_",rownames(human_tax))

# Check for hgt events in resistome
human_tax %>% as.data.frame() %>% filter(!is.na(ARO), CALL == "lgt")

```

Create phyloseq objects

```{r}

human_physeq_tpm <- phyloseq(human_otu_tpm %>% otu_table(taxa_are_rows = TRUE),
                   human_tax %>% tax_table(),
                   human_meta %>% sample_data())

human_physeq_tpm %>% saveRDS("~/Documents/MSc_Thesis-NRP72/data/human_full_tpm_orf.rds")

# Coverage

human_physeq_cov <- phyloseq(human_otu_cov %>% otu_table(taxa_are_rows = TRUE),
                   human_tax %>% tax_table(),
                   human_meta %>% sample_data())

human_physeq_cov %>% saveRDS("~/Documents/MSc_Thesis-NRP72/data/human_full_cov_orf.rds")

# Raw reads

human_physeq_rr <- phyloseq(human_otu_rr %>% otu_table(taxa_are_rows = TRUE),
                   human_tax %>% tax_table(),
                   human_meta %>% sample_data())

human_physeq_rr %>% saveRDS("~/Documents/MSc_Thesis-NRP72/data/human_full_rr_orf.rds")
```