Test https://git-r3lab.uni.lu/laura.denies/PathoFact/-/tree/orf

# data preparation:

## What we need:

We need: the following files are generated by PathoFact itself if format in the config.yaml is set to assembly. If format is set to orf the own assembly files can be used and needed to be provided as following:


- *.faa: FASTA file conatining translated gene sequences, i.e. amino acid sequences

no whitespaces in FASTA headers for prediction of toxins, virulence factors and antimicrobial resistance genes


- *.contig: TAB-delimited file containing a mapping from contig ID (1st column) to gene ID (2nd column)

no header, one gene ID per line
contig and gene IDs should be the same as in the FASTA files


## What we have:

From SQM:
- fasta file containing genes: 03.chicken.faa
	- 	header contains contig_id + "_" gene coordinates

## We need to:

-Filter AA fasta file
-Simplify the header
-Generate a data frame contig ID gene ID.

-> R?

functions/prepare_faa_fna_PAthoFact_orf.Rscript.R

 Rscript ~/Documents/GitHub/NRP72-FBT/functions/prepare_faa_fna_PAthoFact_orf.Rscript.R --contig_fasta_file_path  ~/Desktop/01.chicken.fasta --aa_file_path ~/Desktop/03.chicken.faa -l 1000 -s chicken1_l1000_test
 
 conda activate PathoFact
 snakemake -s Snakefile --use-conda --reason --cores 5 -p --configfile orfs_chicken1_Tox/config.yaml
 
 
tar \
 --exclude='contig_splitted' --exclude='renamed' --exclude='splitted' --exclude='*.faa' \
 --exclude='*.fna' --exclude='*.contig' -zcvf PathoFact_chicken1_AMR.tar.gz .